{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ENSG](#toc1_)    \n",
    "    - [Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc1_1_1_)    \n",
    "    - [How many total unique gene records are there](#toc1_1_2_)    \n",
    "    - [Drop rows with NAN in alias_symbol](#toc1_1_3_)    \n",
    "    - [Make each row in alias_symbol a set:](#toc1_1_4_)    \n",
    "    - [Explode the alias sets so that it is one per row](#toc1_1_5_)    \n",
    "    - [How many total unique aliases are there](#toc1_1_6_)    \n",
    "    - [Pull out all the rows that have an alias symbol that can be found elsewhere](#toc1_1_7_)    \n",
    "    - [Sort alias symbols alphabetically](#toc1_1_8_)    \n",
    "    - [Number of records with an alias that is shared](#toc1_1_9_)    \n",
    "    - [Count the number of times each multi-use alias is used](#toc1_1_10_)    \n",
    "      - [Save as csv](#toc1_1_10_1_)    \n",
    "    - [Put columns in different order to ephasize alias symbols instead of gene records](#toc1_1_11_)    \n",
    "    - [Merge rows with matching alias symbols](#toc1_1_12_)    \n",
    "- [HGNC](#toc2_)    \n",
    "    - [Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc2_1_1_)    \n",
    "    - [How many total unique gene records are there](#toc2_1_2_)    \n",
    "    - [Drop rows with NAN in alias_symbol](#toc2_1_3_)    \n",
    "    - [Make each row in alias_symbol a set:](#toc2_1_4_)    \n",
    "    - [Explode the alias sets so that it is one per row](#toc2_1_5_)    \n",
    "    - [How many total unique aliases are there](#toc2_1_6_)    \n",
    "    - [Pull out all the rows that have an alias symbol that can be found elsewhere](#toc2_1_7_)    \n",
    "    - [Sort alias symbols alphabetically](#toc2_1_8_)    \n",
    "    - [Number of records with an alias that is shared](#toc2_1_9_)    \n",
    "    - [Count the number of times each multi-use alias is used](#toc2_1_10_)    \n",
    "      - [Save as csv](#toc2_1_10_1_)    \n",
    "    - [Put columns in different order to ephasize alias symbols instead of gene records](#toc2_1_11_)    \n",
    "    - [Merge rows with matching alias symbols](#toc2_1_12_)    \n",
    "- [NCBI Info](#toc3_)    \n",
    "    - [Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc3_1_1_)    \n",
    "    - [How many total unique gene records are there](#toc3_1_2_)    \n",
    "    - [Drop rows with - in alias_symbol](#toc3_1_3_)    \n",
    "    - [Make each row in alias_symbol a set:](#toc3_1_4_)    \n",
    "    - [Explode the alias sets so that it is one per row](#toc3_1_5_)    \n",
    "    - [How many unique aliases are there](#toc3_1_6_)    \n",
    "    - [Pull out all the rows that have an alias symbol that can be found elsewhere](#toc3_1_7_)    \n",
    "    - [Sort alias symbols alphabetically](#toc3_1_8_)    \n",
    "    - [Number of records with an alias that is shared](#toc3_1_9_)    \n",
    "    - [Count the number of times each multi-use alias is used](#toc3_1_10_)    \n",
    "    - [Put columns in different order to ephasize alias symbols instead of gene records](#toc3_1_11_)    \n",
    "    - [Merge rows with matching alias symbols](#toc3_1_12_)    \n",
    "- [Merge to create Alias Overlap Table 1 - Gene Symbol](#toc4_)    \n",
    "- [Merge to create Alias Overlap Table 2 - Alias Symbol](#toc5_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import plotly.express as px"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ENSG](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df = pd.read_csv(\"../Alias_gene_intersections/Created_files/ensg_alias_df.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: duplicate gene symbols can have different ENSG ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>ref_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>ensg_id</th>\n",
       "      <th>gene_synonyms</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1366</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>495150</td>\n",
       "      <td>5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]</td>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124907156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1367</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>495150</td>\n",
       "      <td>5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]</td>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124907485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1368</td>\n",
       "      <td>rRNA</td>\n",
       "      <td>495150</td>\n",
       "      <td>5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]</td>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124908250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>1386</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2874124</td>\n",
       "      <td>killer cell immunoglobulin like receptor, two ...</td>\n",
       "      <td>ENSG00000276779</td>\n",
       "      <td>103AS, 15.212, CD158D</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1387</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2874124</td>\n",
       "      <td>killer cell immunoglobulin like receptor, two ...</td>\n",
       "      <td>ENSG00000276779</td>\n",
       "      <td>103AS, 15.212, CD158D</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>124900568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52151</th>\n",
       "      <td>76329</td>\n",
       "      <td>snRNA</td>\n",
       "      <td>2949399</td>\n",
       "      <td>RNA, variant U1 small nuclear 29 [Source:HGNC ...</td>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52152</th>\n",
       "      <td>76330</td>\n",
       "      <td>snRNA</td>\n",
       "      <td>2949399</td>\n",
       "      <td>RNA, variant U1 small nuclear 29 [Source:HGNC ...</td>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52153</th>\n",
       "      <td>76331</td>\n",
       "      <td>snRNA</td>\n",
       "      <td>2949399</td>\n",
       "      <td>RNA, variant U1 small nuclear 29 [Source:HGNC ...</td>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52196</th>\n",
       "      <td>76395</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2919059</td>\n",
       "      <td>phosphodiesterase 4D interacting protein [Sour...</td>\n",
       "      <td>ENSG00000178104</td>\n",
       "      <td>CMYA2, KIAA0454, KIAA0477, MMGL</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>9659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52197</th>\n",
       "      <td>76396</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2919059</td>\n",
       "      <td>phosphodiesterase 4D interacting protein [Sour...</td>\n",
       "      <td>ENSG00000178104</td>\n",
       "      <td>CMYA2, KIAA0454, KIAA0477, MMGL</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>124904395</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5990 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0       gene_type   ref_id  \\\n",
       "38           1366            rRNA   495150   \n",
       "39           1367            rRNA   495150   \n",
       "40           1368            rRNA   495150   \n",
       "55           1386  protein_coding  2874124   \n",
       "56           1387  protein_coding  2874124   \n",
       "...           ...             ...      ...   \n",
       "52151       76329           snRNA  2949399   \n",
       "52152       76330           snRNA  2949399   \n",
       "52153       76331           snRNA  2949399   \n",
       "52196       76395  protein_coding  2919059   \n",
       "52197       76396  protein_coding  2919059   \n",
       "\n",
       "                                               gene_name          ensg_id  \\\n",
       "38          5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]  ENSG00000278294   \n",
       "39          5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]  ENSG00000278294   \n",
       "40          5.8S ribosomal RNA [Source:RFAM;Acc:RF00002]  ENSG00000278294   \n",
       "55     killer cell immunoglobulin like receptor, two ...  ENSG00000276779   \n",
       "56     killer cell immunoglobulin like receptor, two ...  ENSG00000276779   \n",
       "...                                                  ...              ...   \n",
       "52151  RNA, variant U1 small nuclear 29 [Source:HGNC ...  ENSG00000273768   \n",
       "52152  RNA, variant U1 small nuclear 29 [Source:HGNC ...  ENSG00000273768   \n",
       "52153  RNA, variant U1 small nuclear 29 [Source:HGNC ...  ENSG00000273768   \n",
       "52196  phosphodiesterase 4D interacting protein [Sour...  ENSG00000178104   \n",
       "52197  phosphodiesterase 4D interacting protein [Sour...  ENSG00000178104   \n",
       "\n",
       "                         gene_synonyms gene_symbol  entrez_id  \n",
       "38                                 NaN   5_8S_rRNA  124907156  \n",
       "39                                 NaN   5_8S_rRNA  124907485  \n",
       "40                                 NaN   5_8S_rRNA  124908250  \n",
       "55               103AS, 15.212, CD158D     KIR2DL4       3805  \n",
       "56               103AS, 15.212, CD158D     KIR2DL4  124900568  \n",
       "...                                ...         ...        ...  \n",
       "52151                              NaN    RNVU1-29  124905574  \n",
       "52152                              NaN    RNVU1-29  124905808  \n",
       "52153                              NaN    RNVU1-29  124905809  \n",
       "52196  CMYA2, KIAA0454, KIAA0477, MMGL     PDE4DIP       9659  \n",
       "52197  CMYA2, KIAA0454, KIAA0477, MMGL     PDE4DIP  124904395  \n",
       "\n",
       "[5990 rows x 8 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicateENSG_ID2 = mini_ensg_df[mini_ensg_df.duplicated('ensg_id', keep=False)]\n",
    "duplicateENSG_ID2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicateENSG_ID2.to_csv('../ensg_duplicateENSG_ID2.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MTTF, trnF</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>12S, MOTS-c, MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MTTV, trnV</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>16S, HN, MTRNR2</td>\n",
       "      <td>MT-RNR2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>MTTL1, TRNL1</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52224</th>\n",
       "      <td>ENSG00000206764</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNU6-152P</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52225</th>\n",
       "      <td>ENSG00000231684</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EIF1P3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52226</th>\n",
       "      <td>ENSG00000264470</td>\n",
       "      <td>hsa-mir-4794</td>\n",
       "      <td>MIR4794</td>\n",
       "      <td>100616338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52227</th>\n",
       "      <td>ENSG00000162437</td>\n",
       "      <td>FLJ10770, KIAA1579</td>\n",
       "      <td>RAVER2</td>\n",
       "      <td>55225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52228</th>\n",
       "      <td>ENSG00000122432</td>\n",
       "      <td>SP-2</td>\n",
       "      <td>SPATA1</td>\n",
       "      <td>100505741</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52229 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID         alias_symbol gene_symbol  entrez_id\n",
       "0      ENSG00000210049           MTTF, trnF       MT-TF          0\n",
       "1      ENSG00000211459  12S, MOTS-c, MTRNR1     MT-RNR1          0\n",
       "2      ENSG00000210077           MTTV, trnV       MT-TV          0\n",
       "3      ENSG00000210082      16S, HN, MTRNR2     MT-RNR2          0\n",
       "4      ENSG00000209082         MTTL1, TRNL1      MT-TL1          0\n",
       "...                ...                  ...         ...        ...\n",
       "52224  ENSG00000206764                  NaN   RNU6-152P          0\n",
       "52225  ENSG00000231684                  NaN      EIF1P3          0\n",
       "52226  ENSG00000264470         hsa-mir-4794     MIR4794  100616338\n",
       "52227  ENSG00000162437   FLJ10770, KIAA1579      RAVER2      55225\n",
       "52228  ENSG00000122432                 SP-2      SPATA1  100505741\n",
       "\n",
       "[52229 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df.drop(['gene_type', 'ref_id', 'gene_name', 'Unnamed: 0'], axis=1)\n",
    "mini_ensg_df = mini_ensg_df.rename(columns = {'gene_synonyms':'alias_symbol', 'ensg_id':'ENSG_ID' })\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: duplicate ENSG ids can have different entrez ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124907156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124907485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ENSG00000278294</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5_8S_rRNA</td>\n",
       "      <td>124908250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ENSG00000276779</td>\n",
       "      <td>103AS, 15.212, CD158D</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>ENSG00000276779</td>\n",
       "      <td>103AS, 15.212, CD158D</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>124900568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52151</th>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52152</th>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52153</th>\n",
       "      <td>ENSG00000273768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RNVU1-29</td>\n",
       "      <td>124905809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52196</th>\n",
       "      <td>ENSG00000178104</td>\n",
       "      <td>CMYA2, KIAA0454, KIAA0477, MMGL</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>9659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52197</th>\n",
       "      <td>ENSG00000178104</td>\n",
       "      <td>CMYA2, KIAA0454, KIAA0477, MMGL</td>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>124904395</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5990 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID                     alias_symbol gene_symbol  entrez_id\n",
       "38     ENSG00000278294                              NaN   5_8S_rRNA  124907156\n",
       "39     ENSG00000278294                              NaN   5_8S_rRNA  124907485\n",
       "40     ENSG00000278294                              NaN   5_8S_rRNA  124908250\n",
       "55     ENSG00000276779            103AS, 15.212, CD158D     KIR2DL4       3805\n",
       "56     ENSG00000276779            103AS, 15.212, CD158D     KIR2DL4  124900568\n",
       "...                ...                              ...         ...        ...\n",
       "52151  ENSG00000273768                              NaN    RNVU1-29  124905574\n",
       "52152  ENSG00000273768                              NaN    RNVU1-29  124905808\n",
       "52153  ENSG00000273768                              NaN    RNVU1-29  124905809\n",
       "52196  ENSG00000178104  CMYA2, KIAA0454, KIAA0477, MMGL     PDE4DIP       9659\n",
       "52197  ENSG00000178104  CMYA2, KIAA0454, KIAA0477, MMGL     PDE4DIP  124904395\n",
       "\n",
       "[5990 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicateENSG_ID = mini_ensg_df[mini_ensg_df.duplicated('ENSG_ID', keep=False)]\n",
    "duplicateENSG_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicateENSG_ID.to_csv('../ensg_duplicateENSG_ID.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[How many total unique gene records are there](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By ENSG ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46806"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_gene_id_set = set(mini_ensg_df['ENSG_ID'])\n",
    "len(ensg_gene_id_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "By gene symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "40353"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_gene_symbol_set = set(mini_ensg_df['gene_symbol'])\n",
    "len(ensg_gene_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using IGHV as an example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "str"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(mini_ensg_df.gene_symbol[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ensg_IGHV_alias_df = mini_ensg_df.loc[mini_ensg_df['gene_symbol'].str.contains(\"IGHV\", case=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_IGHV_alias_df = mini_ensg_df[mini_ensg_df['gene_symbol'].str.contains('IGHV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_IGHV_alias_df.to_csv('../hgnc_IGHV_alias_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_3_'></a>[Drop genes with no aliases](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MTTF, trnF</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>12S, MOTS-c, MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>MTTV, trnV</td>\n",
       "      <td>MT-TV</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>16S, HN, MTRNR2</td>\n",
       "      <td>MT-RNR2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>MTTL1, TRNL1</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52220</th>\n",
       "      <td>ENSG00000198216</td>\n",
       "      <td>BII, CACH6, CACNL1A6, Cav2.3</td>\n",
       "      <td>CACNA1E</td>\n",
       "      <td>777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52221</th>\n",
       "      <td>ENSG00000179930</td>\n",
       "      <td>FLJ46813</td>\n",
       "      <td>ZNF648</td>\n",
       "      <td>127665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52226</th>\n",
       "      <td>ENSG00000264470</td>\n",
       "      <td>hsa-mir-4794</td>\n",
       "      <td>MIR4794</td>\n",
       "      <td>100616338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52227</th>\n",
       "      <td>ENSG00000162437</td>\n",
       "      <td>FLJ10770, KIAA1579</td>\n",
       "      <td>RAVER2</td>\n",
       "      <td>55225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52228</th>\n",
       "      <td>ENSG00000122432</td>\n",
       "      <td>SP-2</td>\n",
       "      <td>SPATA1</td>\n",
       "      <td>100505741</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31260 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID                  alias_symbol gene_symbol  entrez_id\n",
       "0      ENSG00000210049                    MTTF, trnF       MT-TF          0\n",
       "1      ENSG00000211459           12S, MOTS-c, MTRNR1     MT-RNR1          0\n",
       "2      ENSG00000210077                    MTTV, trnV       MT-TV          0\n",
       "3      ENSG00000210082               16S, HN, MTRNR2     MT-RNR2          0\n",
       "4      ENSG00000209082                  MTTL1, TRNL1      MT-TL1          0\n",
       "...                ...                           ...         ...        ...\n",
       "52220  ENSG00000198216  BII, CACH6, CACNL1A6, Cav2.3     CACNA1E        777\n",
       "52221  ENSG00000179930                      FLJ46813      ZNF648     127665\n",
       "52226  ENSG00000264470                  hsa-mir-4794     MIR4794  100616338\n",
       "52227  ENSG00000162437            FLJ10770, KIAA1579      RAVER2      55225\n",
       "52228  ENSG00000122432                          SP-2      SPATA1  100505741\n",
       "\n",
       "[31260 rows x 4 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df[mini_ensg_df[\"alias_symbol\"].str.contains(\"NaN\") == False]\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_4_'></a>[Make each row in alias_symbol a set:](#toc0_)\n",
    "    covert to a list \n",
    "    make a set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1602162191.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ensg_df['alias_symbol'] = mini_ensg_df['alias_symbol'].astype(str)\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1602162191.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ensg_df['alias_symbol'] = [x.split(',') for x in mini_ensg_df.alias_symbol]\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1602162191.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ensg_df['alias_symbol']=np.where(mini_ensg_df.alias_symbol=='','',mini_ensg_df.alias_symbol.map(set))\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>{MTTF,  trnF}</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ENSG_ID   alias_symbol gene_symbol  entrez_id\n",
       "0  ENSG00000210049  {MTTF,  trnF}       MT-TF          0"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df['alias_symbol'] = mini_ensg_df['alias_symbol'].astype(str)\n",
    "mini_ensg_df['alias_symbol'] = [x.split(',') for x in mini_ensg_df.alias_symbol]\n",
    "mini_ensg_df['alias_symbol']=np.where(mini_ensg_df.alias_symbol=='','',mini_ensg_df.alias_symbol.map(set))\n",
    "mini_ensg_df.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_5_'></a>[Explode the alias sets so that it is one per row](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>trnF</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MOTS-c</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52221</th>\n",
       "      <td>ENSG00000179930</td>\n",
       "      <td>FLJ46813</td>\n",
       "      <td>ZNF648</td>\n",
       "      <td>127665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52226</th>\n",
       "      <td>ENSG00000264470</td>\n",
       "      <td>hsa-mir-4794</td>\n",
       "      <td>MIR4794</td>\n",
       "      <td>100616338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52227</th>\n",
       "      <td>ENSG00000162437</td>\n",
       "      <td>KIAA1579</td>\n",
       "      <td>RAVER2</td>\n",
       "      <td>55225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52227</th>\n",
       "      <td>ENSG00000162437</td>\n",
       "      <td>FLJ10770</td>\n",
       "      <td>RAVER2</td>\n",
       "      <td>55225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52228</th>\n",
       "      <td>ENSG00000122432</td>\n",
       "      <td>SP-2</td>\n",
       "      <td>SPATA1</td>\n",
       "      <td>100505741</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70498 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID  alias_symbol gene_symbol  entrez_id\n",
       "0      ENSG00000210049          MTTF       MT-TF          0\n",
       "0      ENSG00000210049          trnF       MT-TF          0\n",
       "1      ENSG00000211459        MTRNR1     MT-RNR1          0\n",
       "1      ENSG00000211459           12S     MT-RNR1          0\n",
       "1      ENSG00000211459        MOTS-c     MT-RNR1          0\n",
       "...                ...           ...         ...        ...\n",
       "52221  ENSG00000179930      FLJ46813      ZNF648     127665\n",
       "52226  ENSG00000264470  hsa-mir-4794     MIR4794  100616338\n",
       "52227  ENSG00000162437      KIAA1579      RAVER2      55225\n",
       "52227  ENSG00000162437      FLJ10770      RAVER2      55225\n",
       "52228  ENSG00000122432          SP-2      SPATA1  100505741\n",
       "\n",
       "[70498 rows x 4 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df.explode('alias_symbol')\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_6_'></a>[How many unique aliases are there](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "54926"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_symbol_set = set(mini_ensg_df['alias_symbol'])\n",
    "ensg_alias_len = len(ensg_alias_symbol_set)\n",
    "ensg_alias_len"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_7_'></a>[Pull out all the rows that have an alias symbol that can be found elsewhere](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>ND1</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>4535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ENSG00000281486</td>\n",
       "      <td>G2SYN</td>\n",
       "      <td>SNTG2</td>\n",
       "      <td>54221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ENSG00000281486</td>\n",
       "      <td>SYN5</td>\n",
       "      <td>SNTG2</td>\n",
       "      <td>54221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ENSG00000262826</td>\n",
       "      <td>FLJ21919</td>\n",
       "      <td>INTS3</td>\n",
       "      <td>65123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>ENSG00000262826</td>\n",
       "      <td>C1orf60</td>\n",
       "      <td>INTS3</td>\n",
       "      <td>65123</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ENSG_ID alias_symbol gene_symbol  entrez_id\n",
       "5   ENSG00000198888          ND1      MT-ND1       4535\n",
       "43  ENSG00000281486        G2SYN       SNTG2      54221\n",
       "43  ENSG00000281486         SYN5       SNTG2      54221\n",
       "44  ENSG00000262826     FLJ21919       INTS3      65123\n",
       "44  ENSG00000262826      C1orf60       INTS3      65123"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df['alias_duplicates'] = mini_ensg_df.duplicated(subset= 'alias_symbol', keep=False)\n",
    "mini_ensg_df = mini_ensg_df[mini_ensg_df['alias_duplicates'] == True]\n",
    "mini_ensg_df = mini_ensg_df.drop(['alias_duplicates'], axis=1)\n",
    "mini_ensg_df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_8_'></a>[Sort alias symbols alphabetically](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>ENSG00000277964</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>124900568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>ENSG00000278430</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>671</th>\n",
       "      <td>ENSG00000277362</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>ENSG00000276979</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>p56</td>\n",
       "      <td>PSMC1</td>\n",
       "      <td>5700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37151</th>\n",
       "      <td>ENSG00000011638</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>387590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>ENSG00000100181</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20879 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID alias_symbol gene_symbol  entrez_id\n",
       "80     ENSG00000278074       15.212     KIR2DL4       3805\n",
       "631    ENSG00000277964       15.212     KIR2DL4  124900568\n",
       "811    ENSG00000278430       15.212     KIR2DL4       3805\n",
       "671    ENSG00000277362       15.212     KIR2DL4       3805\n",
       "1300   ENSG00000276979       15.212     KIR2DL4       3805\n",
       "...                ...          ...         ...        ...\n",
       "8105   ENSG00000100764          p56       PSMC1       5700\n",
       "7724   ENSG00000283997    promethin       LDAF1      57146\n",
       "37151  ENSG00000011638    promethin       LDAF1      57146\n",
       "27801  ENSG00000290418    psiTPTE22      TPTEP1     387590\n",
       "27797  ENSG00000100181    psiTPTE22      TPTEP1          0\n",
       "\n",
       "[20879 rows x 4 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df.sort_values('alias_symbol')\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#ensg_CD158b_alias_count_df.to_csv('../hgnc_CD158b_alias_count_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_9_'></a>[Number of records with an alias that is shared](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3680"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_gene_symbol_set_wsharedalias = set(mini_ensg_df['gene_symbol'])\n",
    "len(ensg_gene_symbol_set_wsharedalias)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>ENSG00000277964</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>124900568</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>ENSG00000278430</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>671</th>\n",
       "      <td>ENSG00000277362</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>ENSG00000276979</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>p56</td>\n",
       "      <td>PSMC1</td>\n",
       "      <td>5700</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37151</th>\n",
       "      <td>ENSG00000011638</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>387590</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>ENSG00000100181</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20879 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID alias_symbol gene_symbol  entrez_id source\n",
       "80     ENSG00000278074       15.212     KIR2DL4       3805   ENSG\n",
       "631    ENSG00000277964       15.212     KIR2DL4  124900568   ENSG\n",
       "811    ENSG00000278430       15.212     KIR2DL4       3805   ENSG\n",
       "671    ENSG00000277362       15.212     KIR2DL4       3805   ENSG\n",
       "1300   ENSG00000276979       15.212     KIR2DL4       3805   ENSG\n",
       "...                ...          ...         ...        ...    ...\n",
       "8105   ENSG00000100764          p56       PSMC1       5700   ENSG\n",
       "7724   ENSG00000283997    promethin       LDAF1      57146   ENSG\n",
       "37151  ENSG00000011638    promethin       LDAF1      57146   ENSG\n",
       "27801  ENSG00000290418    psiTPTE22      TPTEP1     387590   ENSG\n",
       "27797  ENSG00000100181    psiTPTE22      TPTEP1          0   ENSG\n",
       "\n",
       "[20879 rows x 5 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df['source'] = 'ENSG'\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_10_'></a>[Count the number of times each multi-use alias is used](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "alias_symbol\n",
       " 15.212      82\n",
       " 5T4-AG       2\n",
       " A1AT         2\n",
       " A3GALT1      2\n",
       " AAP          2\n",
       "             ..\n",
       "p42           2\n",
       "p55           2\n",
       "p56           2\n",
       "promethin     2\n",
       "psiTPTE22     2\n",
       "Length: 5307, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_dup_alias_count_df = mini_ensg_df.pivot_table(index = ['alias_symbol'], aggfunc ='size')\n",
    "ensg_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.212</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5T4-AG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1AT</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAP</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5302</th>\n",
       "      <td>p42</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5303</th>\n",
       "      <td>p55</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5304</th>\n",
       "      <td>p56</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5305</th>\n",
       "      <td>promethin</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5306</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5307 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol   0\n",
       "0          15.212  82\n",
       "1          5T4-AG   2\n",
       "2            A1AT   2\n",
       "3         A3GALT1   2\n",
       "4             AAP   2\n",
       "...           ...  ..\n",
       "5302          p42   2\n",
       "5303          p55   2\n",
       "5304          p56   2\n",
       "5305    promethin   2\n",
       "5306    psiTPTE22   2\n",
       "\n",
       "[5307 rows x 2 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_dup_alias_count_df = ensg_dup_alias_count_df.reset_index()\n",
    "ensg_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.212</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5T4-AG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1AT</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAP</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5302</th>\n",
       "      <td>p42</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5303</th>\n",
       "      <td>p55</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5304</th>\n",
       "      <td>p56</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5305</th>\n",
       "      <td>promethin</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5306</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5307 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "0          15.212                82\n",
       "1          5T4-AG                 2\n",
       "2            A1AT                 2\n",
       "3         A3GALT1                 2\n",
       "4             AAP                 2\n",
       "...           ...               ...\n",
       "5302          p42                 2\n",
       "5303          p55                 2\n",
       "5304          p56                 2\n",
       "5305    promethin                 2\n",
       "5306    psiTPTE22                 2\n",
       "\n",
       "[5307 rows x 2 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_dup_alias_count_df.rename(columns={0:'num_gene_records'}, inplace=True )\n",
    "ensg_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4853</th>\n",
       "      <td>RN5S3</td>\n",
       "      <td>218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4849</th>\n",
       "      <td>RN5S2</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4844</th>\n",
       "      <td>RN5S16</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4845</th>\n",
       "      <td>RN5S17</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4835</th>\n",
       "      <td>RN5S1</td>\n",
       "      <td>216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4865</th>\n",
       "      <td>RNA5-8N2</td>\n",
       "      <td>211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4837</th>\n",
       "      <td>RN5S11</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4839</th>\n",
       "      <td>RN5S12</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4840</th>\n",
       "      <td>RN5S13</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4842</th>\n",
       "      <td>RN5S14</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4843</th>\n",
       "      <td>RN5S15</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4858</th>\n",
       "      <td>RN5S4</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4860</th>\n",
       "      <td>RN5S5</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4861</th>\n",
       "      <td>RN5S6</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4862</th>\n",
       "      <td>RN5S7</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4863</th>\n",
       "      <td>RN5S8</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4836</th>\n",
       "      <td>RN5S10</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>CD158D</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.212</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3137</th>\n",
       "      <td>103AS</td>\n",
       "      <td>82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3776</th>\n",
       "      <td>CT78</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2986</th>\n",
       "      <td>cl-5</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3628</th>\n",
       "      <td>CD158K</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3053</th>\n",
       "      <td>nkat4</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3054</th>\n",
       "      <td>nkat4a</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3055</th>\n",
       "      <td>nkat4b</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1101</th>\n",
       "      <td>KIRC1</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1099</th>\n",
       "      <td>KIR44</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1097</th>\n",
       "      <td>KIR3DL7</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3629</th>\n",
       "      <td>CD158Z</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "4853        RN5S3               218\n",
       "4849        RN5S2               216\n",
       "4844       RN5S16               216\n",
       "4845       RN5S17               216\n",
       "4835        RN5S1               216\n",
       "4865     RNA5-8N2               211\n",
       "4837       RN5S11               108\n",
       "4839       RN5S12               108\n",
       "4840       RN5S13               108\n",
       "4842       RN5S14               108\n",
       "4843       RN5S15               108\n",
       "4858        RN5S4               108\n",
       "4860        RN5S5               108\n",
       "4861        RN5S6               108\n",
       "4862        RN5S7               108\n",
       "4863        RN5S8               108\n",
       "4836       RN5S10               108\n",
       "138        CD158D                82\n",
       "0          15.212                82\n",
       "3137        103AS                82\n",
       "3776         CT78                53\n",
       "2986         cl-5                43\n",
       "3628       CD158K                43\n",
       "3053        nkat4                43\n",
       "3054       nkat4a                43\n",
       "3055       nkat4b                43\n",
       "1101        KIRC1                42\n",
       "1099        KIR44                42\n",
       "1097      KIR3DL7                42\n",
       "3629       CD158Z                42"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_dup_alias_count_df = ensg_dup_alias_count_df.sort_values('num_gene_records', ascending=False)\n",
    "ensg_dup_alias_count_df.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5307"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_alias_collision_set = set(ensg_dup_alias_count_df['alias_symbol'])\n",
    "len(ensg_alias_alias_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_dup_alias_count_df.to_csv('../ensg_dup_alias_count_df.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "num_gene_records\n",
       "2      3738\n",
       "3       394\n",
       "4       173\n",
       "5       112\n",
       "6       160\n",
       "7       278\n",
       "8       258\n",
       "9        17\n",
       "10       50\n",
       "11        6\n",
       "12        4\n",
       "13       11\n",
       "14        8\n",
       "15        5\n",
       "16       15\n",
       "18        1\n",
       "21        2\n",
       "25        4\n",
       "26        1\n",
       "28        1\n",
       "30        4\n",
       "31        6\n",
       "32       10\n",
       "33        9\n",
       "35        8\n",
       "41        2\n",
       "42        4\n",
       "43        5\n",
       "53        1\n",
       "82        3\n",
       "108      11\n",
       "211       1\n",
       "216       4\n",
       "218       1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_count_histogram_df = ensg_dup_alias_count_df.pivot_table(index = ['num_gene_records'], aggfunc ='size')\n",
    "ensg_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>3738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>31</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>32</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>33</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>41</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>42</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>43</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>108</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>211</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>216</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>218</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records     0\n",
       "0                  2  3738\n",
       "1                  3   394\n",
       "2                  4   173\n",
       "3                  5   112\n",
       "4                  6   160\n",
       "5                  7   278\n",
       "6                  8   258\n",
       "7                  9    17\n",
       "8                 10    50\n",
       "9                 11     6\n",
       "10                12     4\n",
       "11                13    11\n",
       "12                14     8\n",
       "13                15     5\n",
       "14                16    15\n",
       "15                18     1\n",
       "16                21     2\n",
       "17                25     4\n",
       "18                26     1\n",
       "19                28     1\n",
       "20                30     4\n",
       "21                31     6\n",
       "22                32    10\n",
       "23                33     9\n",
       "24                35     8\n",
       "25                41     2\n",
       "26                42     4\n",
       "27                43     5\n",
       "28                53     1\n",
       "29                82     3\n",
       "30               108    11\n",
       "31               211     1\n",
       "32               216     4\n",
       "33               218     1"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_count_histogram_df = ensg_alias_count_histogram_df.reset_index()\n",
    "ensg_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>3738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>31</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>32</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>33</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>41</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>42</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>43</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>108</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>211</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>216</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>218</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  num_alias_symbol\n",
       "0                  2              3738\n",
       "1                  3               394\n",
       "2                  4               173\n",
       "3                  5               112\n",
       "4                  6               160\n",
       "5                  7               278\n",
       "6                  8               258\n",
       "7                  9                17\n",
       "8                 10                50\n",
       "9                 11                 6\n",
       "10                12                 4\n",
       "11                13                11\n",
       "12                14                 8\n",
       "13                15                 5\n",
       "14                16                15\n",
       "15                18                 1\n",
       "16                21                 2\n",
       "17                25                 4\n",
       "18                26                 1\n",
       "19                28                 1\n",
       "20                30                 4\n",
       "21                31                 6\n",
       "22                32                10\n",
       "23                33                 9\n",
       "24                35                 8\n",
       "25                41                 2\n",
       "26                42                 4\n",
       "27                43                 5\n",
       "28                53                 1\n",
       "29                82                 3\n",
       "30               108                11\n",
       "31               211                 1\n",
       "32               216                 4\n",
       "33               218                 1"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_count_histogram_df.rename(columns={0:'num_alias_symbol'}, inplace=True )\n",
    "ensg_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>3738</td>\n",
       "      <td>6.805520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>394</td>\n",
       "      <td>0.717329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>173</td>\n",
       "      <td>0.314969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>112</td>\n",
       "      <td>0.203911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "      <td>0.291301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>278</td>\n",
       "      <td>0.506136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>258</td>\n",
       "      <td>0.469723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>0.030951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>50</td>\n",
       "      <td>0.091032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>0.010924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>0.020027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0.014565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>0.009103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>0.027309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>0.003641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>30</td>\n",
       "      <td>4</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>31</td>\n",
       "      <td>6</td>\n",
       "      <td>0.010924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>32</td>\n",
       "      <td>10</td>\n",
       "      <td>0.018206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>33</td>\n",
       "      <td>9</td>\n",
       "      <td>0.016386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "      <td>0.014565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>41</td>\n",
       "      <td>2</td>\n",
       "      <td>0.003641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>42</td>\n",
       "      <td>4</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>43</td>\n",
       "      <td>5</td>\n",
       "      <td>0.009103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>82</td>\n",
       "      <td>3</td>\n",
       "      <td>0.005462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>108</td>\n",
       "      <td>11</td>\n",
       "      <td>0.020027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>211</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>216</td>\n",
       "      <td>4</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>218</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  num_alias_symbol  percent_alias_symbol\n",
       "0                  2              3738              6.805520\n",
       "1                  3               394              0.717329\n",
       "2                  4               173              0.314969\n",
       "3                  5               112              0.203911\n",
       "4                  6               160              0.291301\n",
       "5                  7               278              0.506136\n",
       "6                  8               258              0.469723\n",
       "7                  9                17              0.030951\n",
       "8                 10                50              0.091032\n",
       "9                 11                 6              0.010924\n",
       "10                12                 4              0.007283\n",
       "11                13                11              0.020027\n",
       "12                14                 8              0.014565\n",
       "13                15                 5              0.009103\n",
       "14                16                15              0.027309\n",
       "15                18                 1              0.001821\n",
       "16                21                 2              0.003641\n",
       "17                25                 4              0.007283\n",
       "18                26                 1              0.001821\n",
       "19                28                 1              0.001821\n",
       "20                30                 4              0.007283\n",
       "21                31                 6              0.010924\n",
       "22                32                10              0.018206\n",
       "23                33                 9              0.016386\n",
       "24                35                 8              0.014565\n",
       "25                41                 2              0.003641\n",
       "26                42                 4              0.007283\n",
       "27                43                 5              0.009103\n",
       "28                53                 1              0.001821\n",
       "29                82                 3              0.005462\n",
       "30               108                11              0.020027\n",
       "31               211                 1              0.001821\n",
       "32               216                 4              0.007283\n",
       "33               218                 1              0.001821"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_count_histogram_df['percent_alias_symbol'] = ((ensg_alias_count_histogram_df['num_alias_symbol'] / ensg_alias_len) * 100)\n",
    "ensg_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>6.805520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>0.717329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>0.314969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>0.203911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>0.291301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>0.506136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>0.469723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>0.030951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>0.091032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>0.010924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>0.020027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>0.014565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>15</td>\n",
       "      <td>0.009103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>16</td>\n",
       "      <td>0.027309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>18</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>21</td>\n",
       "      <td>0.003641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>25</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>26</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>30</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>31</td>\n",
       "      <td>0.010924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>32</td>\n",
       "      <td>0.018206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>33</td>\n",
       "      <td>0.016386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>35</td>\n",
       "      <td>0.014565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>41</td>\n",
       "      <td>0.003641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>42</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>43</td>\n",
       "      <td>0.009103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>53</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>82</td>\n",
       "      <td>0.005462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>108</td>\n",
       "      <td>0.020027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>211</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>216</td>\n",
       "      <td>0.007283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>218</td>\n",
       "      <td>0.001821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  percent_alias_symbol\n",
       "0                  2              6.805520\n",
       "1                  3              0.717329\n",
       "2                  4              0.314969\n",
       "3                  5              0.203911\n",
       "4                  6              0.291301\n",
       "5                  7              0.506136\n",
       "6                  8              0.469723\n",
       "7                  9              0.030951\n",
       "8                 10              0.091032\n",
       "9                 11              0.010924\n",
       "10                12              0.007283\n",
       "11                13              0.020027\n",
       "12                14              0.014565\n",
       "13                15              0.009103\n",
       "14                16              0.027309\n",
       "15                18              0.001821\n",
       "16                21              0.003641\n",
       "17                25              0.007283\n",
       "18                26              0.001821\n",
       "19                28              0.001821\n",
       "20                30              0.007283\n",
       "21                31              0.010924\n",
       "22                32              0.018206\n",
       "23                33              0.016386\n",
       "24                35              0.014565\n",
       "25                41              0.003641\n",
       "26                42              0.007283\n",
       "27                43              0.009103\n",
       "28                53              0.001821\n",
       "29                82              0.005462\n",
       "30               108              0.020027\n",
       "31               211              0.001821\n",
       "32               216              0.007283\n",
       "33               218              0.001821"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_count_histogram_df = ensg_alias_count_histogram_df.drop('num_alias_symbol', axis=1)\n",
    "ensg_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "#px.bar(ensg_alias_count_histogram_df, x='num_gene_records', y='percent_alias_symbol')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_dup_alias_count_df.to_csv('../ensg_alias_overlap_count.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_1_10_1_'></a>[Save as csv](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mini_ensg_df_explode.to_csv('../ensg_alias_overlap.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_11_'></a>[Put columns in different order to ephasize alias symbols instead of gene records](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>ENSG00000277964</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>124900568</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>ENSG00000278430</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>671</th>\n",
       "      <td>ENSG00000277362</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>ENSG00000276979</td>\n",
       "      <td>15.212</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>3805</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>p56</td>\n",
       "      <td>PSMC1</td>\n",
       "      <td>5700</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37151</th>\n",
       "      <td>ENSG00000011638</td>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>57146</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>387590</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>ENSG00000100181</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>0</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20879 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID alias_symbol gene_symbol  entrez_id source\n",
       "80     ENSG00000278074       15.212     KIR2DL4       3805   ENSG\n",
       "631    ENSG00000277964       15.212     KIR2DL4  124900568   ENSG\n",
       "811    ENSG00000278430       15.212     KIR2DL4       3805   ENSG\n",
       "671    ENSG00000277362       15.212     KIR2DL4       3805   ENSG\n",
       "1300   ENSG00000276979       15.212     KIR2DL4       3805   ENSG\n",
       "...                ...          ...         ...        ...    ...\n",
       "8105   ENSG00000100764          p56       PSMC1       5700   ENSG\n",
       "7724   ENSG00000283997    promethin       LDAF1      57146   ENSG\n",
       "37151  ENSG00000011638    promethin       LDAF1      57146   ENSG\n",
       "27801  ENSG00000290418    psiTPTE22      TPTEP1     387590   ENSG\n",
       "27797  ENSG00000100181    psiTPTE22      TPTEP1          0   ENSG\n",
       "\n",
       "[20879 rows x 5 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000277964</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000278430</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>671</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000277362</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000276979</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>p56</td>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>PSMC1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37151</th>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG00000011638</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG00000100181</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20879 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol source\n",
       "80          15.212  ENSG00000278074     KIR2DL4   ENSG\n",
       "631         15.212  ENSG00000277964     KIR2DL4   ENSG\n",
       "811         15.212  ENSG00000278430     KIR2DL4   ENSG\n",
       "671         15.212  ENSG00000277362     KIR2DL4   ENSG\n",
       "1300        15.212  ENSG00000276979     KIR2DL4   ENSG\n",
       "...            ...              ...         ...    ...\n",
       "8105           p56  ENSG00000100764       PSMC1   ENSG\n",
       "7724     promethin  ENSG00000283997       LDAF1   ENSG\n",
       "37151    promethin  ENSG00000011638       LDAF1   ENSG\n",
       "27801    psiTPTE22  ENSG00000290418      TPTEP1   ENSG\n",
       "27797    psiTPTE22  ENSG00000100181      TPTEP1   ENSG\n",
       "\n",
       "[20879 rows x 4 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df_2 = mini_ensg_df[['alias_symbol', 'ENSG_ID', 'gene_symbol', 'source']]\n",
    "mini_ensg_df_2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_12_'></a>[Merge rows with matching alias symbols](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df_2 = mini_ensg_df_2.drop_duplicates(subset = [\"alias_symbol\", \"gene_symbol\"], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1088055158.py:1: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  mini_ensg_df_2 = mini_ensg_df_2.applymap(str)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6019</th>\n",
       "      <td>5T4-AG</td>\n",
       "      <td>ENSG00000283085</td>\n",
       "      <td>TPBG</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101</th>\n",
       "      <td>A1AT</td>\n",
       "      <td>ENSG00000197249</td>\n",
       "      <td>SERPINA1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29537</th>\n",
       "      <td>A3GALT1</td>\n",
       "      <td>ENSG00000175164</td>\n",
       "      <td>ABO</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2816</th>\n",
       "      <td>AAP</td>\n",
       "      <td>ENSG00000276838</td>\n",
       "      <td>SERPINF2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50647</th>\n",
       "      <td>p55</td>\n",
       "      <td>ENSG00000197170</td>\n",
       "      <td>PSMD12</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23567</th>\n",
       "      <td>p56</td>\n",
       "      <td>ENSG00000227211</td>\n",
       "      <td>H3P45</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>p56</td>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>PSMC1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6524 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol source\n",
       "80          15.212  ENSG00000278074     KIR2DL4   ENSG\n",
       "6019        5T4-AG  ENSG00000283085        TPBG   ENSG\n",
       "21101         A1AT  ENSG00000197249    SERPINA1   ENSG\n",
       "29537      A3GALT1  ENSG00000175164         ABO   ENSG\n",
       "2816           AAP  ENSG00000276838    SERPINF2   ENSG\n",
       "...            ...              ...         ...    ...\n",
       "50647          p55  ENSG00000197170      PSMD12   ENSG\n",
       "23567          p56  ENSG00000227211       H3P45   ENSG\n",
       "8105           p56  ENSG00000100764       PSMC1   ENSG\n",
       "7724     promethin  ENSG00000283997       LDAF1   ENSG\n",
       "27801    psiTPTE22  ENSG00000290418      TPTEP1   ENSG\n",
       "\n",
       "[6524 rows x 4 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df_2 = mini_ensg_df_2.applymap(str)\n",
    "mini_ensg_df_2\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.212</td>\n",
       "      <td>ENSG00000278074</td>\n",
       "      <td>KIR2DL4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5T4-AG</td>\n",
       "      <td>ENSG00000283085</td>\n",
       "      <td>TPBG</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1AT</td>\n",
       "      <td>ENSG00000197249</td>\n",
       "      <td>SERPINA1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A3GALT1</td>\n",
       "      <td>ENSG00000175164</td>\n",
       "      <td>ABO</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAP</td>\n",
       "      <td>ENSG00000276838</td>\n",
       "      <td>SERPINF2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5302</th>\n",
       "      <td>p42</td>\n",
       "      <td>ENSG00000100519, ENSG00000227443</td>\n",
       "      <td>PSMC6, H3P30</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5303</th>\n",
       "      <td>p55</td>\n",
       "      <td>ENSG00000117461, ENSG00000197170</td>\n",
       "      <td>PIK3R3, PSMD12</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5304</th>\n",
       "      <td>p56</td>\n",
       "      <td>ENSG00000227211, ENSG00000100764</td>\n",
       "      <td>H3P45, PSMC1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5305</th>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5306</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5307 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol                           ENSG_ID     gene_symbol source\n",
       "0          15.212                   ENSG00000278074         KIR2DL4   ENSG\n",
       "1          5T4-AG                   ENSG00000283085            TPBG   ENSG\n",
       "2            A1AT                   ENSG00000197249        SERPINA1   ENSG\n",
       "3         A3GALT1                   ENSG00000175164             ABO   ENSG\n",
       "4             AAP                   ENSG00000276838        SERPINF2   ENSG\n",
       "...           ...                               ...             ...    ...\n",
       "5302          p42  ENSG00000100519, ENSG00000227443    PSMC6, H3P30   ENSG\n",
       "5303          p55  ENSG00000117461, ENSG00000197170  PIK3R3, PSMD12   ENSG\n",
       "5304          p56  ENSG00000227211, ENSG00000100764    H3P45, PSMC1   ENSG\n",
       "5305    promethin                   ENSG00000283997           LDAF1   ENSG\n",
       "5306    psiTPTE22                   ENSG00000290418          TPTEP1   ENSG\n",
       "\n",
       "[5307 rows x 4 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df_2 = mini_ensg_df_2.groupby('alias_symbol').agg({'ENSG_ID': ', '.join, \n",
    "                             'gene_symbol': ', '.join, \n",
    "                             'source':'first' }).reset_index()\n",
    "mini_ensg_df_2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc2_'></a>[HGNC](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.read_csv(\"../hgnc_filtered.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_1_'></a>[Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>FLJ23569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ACF;ASP;ACF64;ACF65;APOBEC1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>FWP007;S863-7;CPAMD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2M-AS1</td>\n",
       "      <td>ENSG00000245105</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43159</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>FLJ13456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43160</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>ENSG00000159840</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43161</th>\n",
       "      <td>ZYXP1</td>\n",
       "      <td>ENSG00000274572</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43162</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>KIAA0399;ZZZ4;FLJ10821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43163</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>DKFZP564I052;ATAC1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43164 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID                   alias_symbol\n",
       "0            A1BG  ENSG00000121410                            NaN\n",
       "1        A1BG-AS1  ENSG00000268895                       FLJ23569\n",
       "2            A1CF  ENSG00000148584  ACF;ASP;ACF64;ACF65;APOBEC1CF\n",
       "3             A2M  ENSG00000175899           FWP007;S863-7;CPAMD5\n",
       "4         A2M-AS1  ENSG00000245105                            NaN\n",
       "...           ...              ...                            ...\n",
       "43159      ZYG11B  ENSG00000162378                       FLJ13456\n",
       "43160         ZYX  ENSG00000159840                            NaN\n",
       "43161       ZYXP1  ENSG00000274572                            NaN\n",
       "43162       ZZEF1  ENSG00000074755         KIAA0399;ZZZ4;FLJ10821\n",
       "43163        ZZZ3  ENSG00000036549             DKFZP564I052;ATAC1\n",
       "\n",
       "[43164 rows x 3 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = df1.drop(['Unnamed: 0', 'hgnc_id', 'locus_type', 'name', 'mane_select', 'locus_group', 'entrez_id', 'agr', 'refseq_accession', 'alias_name', 'ENSEMBLtrans', 'NA', 'unknown'], axis=1)\n",
    "mini_hgnc_df = mini_hgnc_df.rename(columns = {'ensembl_gene_id':'ENSG_ID', 'symbol':'gene_symbol'})\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_2_'></a>[How many total unique gene records are there](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43164"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_gene_symbol_set = set(mini_hgnc_df['gene_symbol'])\n",
    "len(hgnc_gene_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Looking at IGHV as an example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_IGHV_alias_df = mini_hgnc_df[mini_hgnc_df['gene_symbol'].str.contains('IGHV')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "str"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(mini_hgnc_df.gene_symbol[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_IGHV_alias_df.to_csv('../hgnc_IGHV_alias_df.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12539</th>\n",
       "      <td>IGHV1-24</td>\n",
       "      <td>ENSG00000211950</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol\n",
       "12539    IGHV1-24  ENSG00000211950          NaN"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_IGHV_alias_count_df = mini_hgnc_df.copy\n",
    "hgnc_IGHV_alias_count_df = mini_hgnc_df.loc[mini_hgnc_df['gene_symbol'] == \"IGHV1-24\" ]\n",
    "hgnc_IGHV_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12539</th>\n",
       "      <td>IGHV1-24</td>\n",
       "      <td>ENSG00000211950</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol\n",
       "12539    IGHV1-24  ENSG00000211950          NaN"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_IGHV_alias_count_df = mini_hgnc_df.copy\n",
    "hgnc_IGHV_alias_count_df = mini_hgnc_df.loc[mini_hgnc_df['ENSG_ID'] == \"ENSG00000211950\" ]\n",
    "hgnc_IGHV_alias_count_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_3_'></a>[Drop genes with no aliases](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>FLJ23569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ACF;ASP;ACF64;ACF65;APOBEC1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>FWP007;S863-7;CPAMD5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>ENSG00000166535</td>\n",
       "      <td>FLJ25179;p170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>ENSG00000184389</td>\n",
       "      <td>IGBS3S;IGB3S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43156</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>ENSG00000070476</td>\n",
       "      <td>MGC11349;FLJ13861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43157</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>ENSG00000203995</td>\n",
       "      <td>ZYG11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43159</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>FLJ13456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43162</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>KIAA0399;ZZZ4;FLJ10821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43163</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>DKFZP564I052;ATAC1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21967 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID                   alias_symbol\n",
       "1        A1BG-AS1  ENSG00000268895                       FLJ23569\n",
       "2            A1CF  ENSG00000148584  ACF;ASP;ACF64;ACF65;APOBEC1CF\n",
       "3             A2M  ENSG00000175899           FWP007;S863-7;CPAMD5\n",
       "5           A2ML1  ENSG00000166535                  FLJ25179;p170\n",
       "9         A3GALT2  ENSG00000184389                   IGBS3S;IGB3S\n",
       "...           ...              ...                            ...\n",
       "43156        ZXDC  ENSG00000070476              MGC11349;FLJ13861\n",
       "43157      ZYG11A  ENSG00000203995                          ZYG11\n",
       "43159      ZYG11B  ENSG00000162378                       FLJ13456\n",
       "43162       ZZEF1  ENSG00000074755         KIAA0399;ZZZ4;FLJ10821\n",
       "43163        ZZZ3  ENSG00000036549             DKFZP564I052;ATAC1\n",
       "\n",
       "[21967 rows x 3 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = mini_hgnc_df[mini_hgnc_df[\"alias_symbol\"].str.contains(\"NaN\") == False]\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_4_'></a>[Make each row in alias_symbol a set:](#toc0_)\n",
    "    covert to a list \n",
    "    make a set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1085824394.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_hgnc_df['alias_symbol'] = mini_hgnc_df['alias_symbol'].astype(str)\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1085824394.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_hgnc_df['alias_symbol'] = [x.split(';') for x in mini_hgnc_df.alias_symbol]\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/1085824394.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_hgnc_df['alias_symbol']=np.where(mini_hgnc_df.alias_symbol=='','',mini_hgnc_df.alias_symbol.map(set))\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>{FLJ23569}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol          ENSG_ID alias_symbol\n",
       "1    A1BG-AS1  ENSG00000268895   {FLJ23569}"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df['alias_symbol'] = mini_hgnc_df['alias_symbol'].astype(str)\n",
    "mini_hgnc_df['alias_symbol'] = [x.split(';') for x in mini_hgnc_df.alias_symbol]\n",
    "mini_hgnc_df['alias_symbol']=np.where(mini_hgnc_df.alias_symbol=='','',mini_hgnc_df.alias_symbol.map(set))\n",
    "mini_hgnc_df.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_5_'></a>[Explode the alias sets so that it is one per row](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>FLJ23569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ASP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ACF65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ACF64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ACF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol          ENSG_ID alias_symbol\n",
       "1    A1BG-AS1  ENSG00000268895     FLJ23569\n",
       "2        A1CF  ENSG00000148584          ASP\n",
       "2        A1CF  ENSG00000148584        ACF65\n",
       "2        A1CF  ENSG00000148584        ACF64\n",
       "2        A1CF  ENSG00000148584          ACF"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.explode(column=\"alias_symbol\")\n",
    "mini_hgnc_df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Looking at CD158b as an example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene_symbol, ENSG_ID, alias_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df.loc[mini_hgnc_df['gene_symbol'] == \"CD158b\" ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene_symbol, ENSG_ID, alias_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_CD158b_alias_count_df = mini_hgnc_df.copy\n",
    "hgnc_CD158b_alias_count_df = mini_hgnc_df.loc[mini_hgnc_df['alias_symbol'] == \"CD158b\" ]\n",
    "hgnc_CD158b_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_CD158b_alias_count_df.to_csv('../hgnc_CD158b_alias_count_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_6_'></a>[How many total unique aliases are there](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41589"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_symbol_set = set(mini_hgnc_df['alias_symbol'])\n",
    "hgnc_alias_len = len(hgnc_alias_symbol_set)\n",
    "hgnc_alias_len"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_7_'></a>[Pull out all the rows that have an alias symbol that can be found elsewhere](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>ASP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>ENSG00000103591</td>\n",
       "      <td>p34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>ABCB8</td>\n",
       "      <td>ENSG00000197150</td>\n",
       "      <td>M-ABC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>ABCB10</td>\n",
       "      <td>ENSG00000135776</td>\n",
       "      <td>M-ABC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>ABCB10P1</td>\n",
       "      <td>ENSG00000274099</td>\n",
       "      <td>M-ABC2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_symbol          ENSG_ID alias_symbol\n",
       "2         A1CF  ENSG00000148584          ASP\n",
       "22       AAGAB  ENSG00000103591          p34\n",
       "65       ABCB8  ENSG00000197150       M-ABC1\n",
       "67      ABCB10  ENSG00000135776       M-ABC2\n",
       "68    ABCB10P1  ENSG00000274099       M-ABC2"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df['alias_duplicates'] = mini_hgnc_df.duplicated(subset= 'alias_symbol', keep=False)\n",
    "mini_hgnc_df = mini_hgnc_df[mini_hgnc_df['alias_duplicates'] == True]\n",
    "mini_hgnc_df = mini_hgnc_df.drop(['alias_duplicates'], axis=1)\n",
    "mini_hgnc_df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_8_'></a>[Sort alias symbols alphabetically](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35226</th>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>2F1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13978</th>\n",
       "      <td>KLRG1</td>\n",
       "      <td>ENSG00000139187</td>\n",
       "      <td>2F1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33987</th>\n",
       "      <td>S100A8</td>\n",
       "      <td>ENSG00000143546</td>\n",
       "      <td>60B8AG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33988</th>\n",
       "      <td>S100A9</td>\n",
       "      <td>ENSG00000163220</td>\n",
       "      <td>60B8AG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31335</th>\n",
       "      <td>RNU6V</td>\n",
       "      <td>ENSG00000206832</td>\n",
       "      <td>87U6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38078</th>\n",
       "      <td>TEX28P2</td>\n",
       "      <td>ENSG00000277008</td>\n",
       "      <td>pTEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26200</th>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>smk1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26203</th>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>smk1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36661</th>\n",
       "      <td>SPATA2</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>tamo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36662</th>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>tamo</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2348 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol\n",
       "35226     SLC25A5  ENSG00000005022          2F1\n",
       "13978       KLRG1  ENSG00000139187          2F1\n",
       "33987      S100A8  ENSG00000143546       60B8AG\n",
       "33988      S100A9  ENSG00000163220       60B8AG\n",
       "31335       RNU6V  ENSG00000206832         87U6\n",
       "...           ...              ...          ...\n",
       "38078     TEX28P2  ENSG00000277008         pTEX\n",
       "26200     PPP4R3A  ENSG00000100796         smk1\n",
       "26203     PPP4R3C  ENSG00000224960         smk1\n",
       "36661      SPATA2  ENSG00000158480         tamo\n",
       "36662     SPATA2L  ENSG00000158792         tamo\n",
       "\n",
       "[2348 rows x 3 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.sort_values('alias_symbol')\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_9_'></a>[Number of records with an alias that is shared](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2084"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_gene_symbol_set_wsharedalias = set(mini_hgnc_df['gene_symbol'])\n",
    "len(hgnc_gene_symbol_set_wsharedalias)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35226</th>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>2F1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13978</th>\n",
       "      <td>KLRG1</td>\n",
       "      <td>ENSG00000139187</td>\n",
       "      <td>2F1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33987</th>\n",
       "      <td>S100A8</td>\n",
       "      <td>ENSG00000143546</td>\n",
       "      <td>60B8AG</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33988</th>\n",
       "      <td>S100A9</td>\n",
       "      <td>ENSG00000163220</td>\n",
       "      <td>60B8AG</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31335</th>\n",
       "      <td>RNU6V</td>\n",
       "      <td>ENSG00000206832</td>\n",
       "      <td>87U6</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38078</th>\n",
       "      <td>TEX28P2</td>\n",
       "      <td>ENSG00000277008</td>\n",
       "      <td>pTEX</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26200</th>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>smk1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26203</th>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>smk1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36661</th>\n",
       "      <td>SPATA2</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>tamo</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36662</th>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>tamo</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2348 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol source\n",
       "35226     SLC25A5  ENSG00000005022          2F1   HGNC\n",
       "13978       KLRG1  ENSG00000139187          2F1   HGNC\n",
       "33987      S100A8  ENSG00000143546       60B8AG   HGNC\n",
       "33988      S100A9  ENSG00000163220       60B8AG   HGNC\n",
       "31335       RNU6V  ENSG00000206832         87U6   HGNC\n",
       "...           ...              ...          ...    ...\n",
       "38078     TEX28P2  ENSG00000277008         pTEX   HGNC\n",
       "26200     PPP4R3A  ENSG00000100796         smk1   HGNC\n",
       "26203     PPP4R3C  ENSG00000224960         smk1   HGNC\n",
       "36661      SPATA2  ENSG00000158480         tamo   HGNC\n",
       "36662     SPATA2L  ENSG00000158792         tamo   HGNC\n",
       "\n",
       "[2348 rows x 4 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df['source'] = 'HGNC'\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_10_'></a>[Count the number of times each multi-use alias is used](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "alias_symbol\n",
       "2F1       2\n",
       "60B8AG    2\n",
       "87U6      2\n",
       "9G8       2\n",
       "A1        3\n",
       "         ..\n",
       "p97       4\n",
       "pH2A/f    2\n",
       "pTEX      2\n",
       "smk1      2\n",
       "tamo      2\n",
       "Length: 1040, dtype: int64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_dup_alias_count_df = mini_hgnc_df.pivot_table(index = ['alias_symbol'], aggfunc ='size')\n",
    "hgnc_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2F1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>87U6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9G8</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>p97</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>pH2A/f</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1038</th>\n",
       "      <td>smk1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1039</th>\n",
       "      <td>tamo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1040 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  0\n",
       "0             2F1  2\n",
       "1          60B8AG  2\n",
       "2            87U6  2\n",
       "3             9G8  2\n",
       "4              A1  3\n",
       "...           ... ..\n",
       "1035          p97  4\n",
       "1036       pH2A/f  2\n",
       "1037         pTEX  2\n",
       "1038         smk1  2\n",
       "1039         tamo  2\n",
       "\n",
       "[1040 rows x 2 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_dup_alias_count_df = hgnc_dup_alias_count_df.reset_index()\n",
    "hgnc_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['alias_symbol', 0], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(hgnc_dup_alias_count_df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2F1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>87U6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9G8</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>p97</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>pH2A/f</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1038</th>\n",
       "      <td>smk1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1039</th>\n",
       "      <td>tamo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1040 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "0             2F1                 2\n",
       "1          60B8AG                 2\n",
       "2            87U6                 2\n",
       "3             9G8                 2\n",
       "4              A1                 3\n",
       "...           ...               ...\n",
       "1035          p97                 4\n",
       "1036       pH2A/f                 2\n",
       "1037         pTEX                 2\n",
       "1038         smk1                 2\n",
       "1039         tamo                 2\n",
       "\n",
       "[1040 rows x 2 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_dup_alias_count_df.rename(columns={0:'num_gene_records'}, inplace=True )\n",
    "hgnc_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>ASP</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>PAP</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>935</th>\n",
       "      <td>U4</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>CAP</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>MYM</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>F379</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>AIP1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>p40</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ALP</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>571</th>\n",
       "      <td>NAP1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>933</th>\n",
       "      <td>U3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>BTF2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>735</th>\n",
       "      <td>PSP</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>932</th>\n",
       "      <td>U2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>p18</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>p14</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1024</th>\n",
       "      <td>p55</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>PPIase</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>TFIIH</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ACT</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>766</th>\n",
       "      <td>RIP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1034</th>\n",
       "      <td>p85</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>p97</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>GBP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863</th>\n",
       "      <td>T1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375</th>\n",
       "      <td>HCP1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ARP</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>ARP1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>929</th>\n",
       "      <td>U13</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>ILWEQ</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "68            ASP                 7\n",
       "653           PAP                 7\n",
       "935            U4                 7\n",
       "118           CAP                 6\n",
       "568           MYM                 6\n",
       "267          F379                 6\n",
       "27           AIP1                 6\n",
       "1012          p40                 5\n",
       "31            ALP                 5\n",
       "571          NAP1                 5\n",
       "933            U3                 5\n",
       "100          BTF2                 5\n",
       "735           PSP                 5\n",
       "932            U2                 5\n",
       "991           p18                 5\n",
       "986           p14                 5\n",
       "1024          p55                 5\n",
       "718        PPIase                 5\n",
       "887         TFIIH                 5\n",
       "18            ACT                 4\n",
       "766           RIP                 4\n",
       "1034          p85                 4\n",
       "1035          p97                 4\n",
       "308           GBP                 4\n",
       "863            T1                 4\n",
       "375          HCP1                 4\n",
       "55            ARP                 4\n",
       "56           ARP1                 4\n",
       "929           U13                 4\n",
       "425         ILWEQ                 4"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_dup_alias_count_df = hgnc_dup_alias_count_df.sort_values('num_gene_records', ascending=False)\n",
    "hgnc_dup_alias_count_df.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1040"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_alias_collision_set = set(hgnc_dup_alias_count_df['alias_symbol'])\n",
    "len(hgnc_alias_alias_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_dup_alias_count_df.to_csv('../hgnc_dup_alias_count_df.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "num_gene_records\n",
       "2    858\n",
       "3    125\n",
       "4     38\n",
       "5     12\n",
       "6      4\n",
       "7      3\n",
       "dtype: int64"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_count_histogram_df = hgnc_dup_alias_count_df.pivot_table(index = ['num_gene_records'], aggfunc ='size')\n",
    "hgnc_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   num_gene_records    0\n",
       "0                 2  858\n",
       "1                 3  125\n",
       "2                 4   38\n",
       "3                 5   12\n",
       "4                 6    4\n",
       "5                 7    3"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_count_histogram_df = hgnc_alias_count_histogram_df.reset_index()\n",
    "hgnc_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   num_gene_records  num_alias_symbol\n",
       "0                 2               858\n",
       "1                 3               125\n",
       "2                 4                38\n",
       "3                 5                12\n",
       "4                 6                 4\n",
       "5                 7                 3"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_count_histogram_df.rename(columns={0:'num_alias_symbol'}, inplace=True )\n",
    "hgnc_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>858</td>\n",
       "      <td>2.063046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>125</td>\n",
       "      <td>0.300560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>38</td>\n",
       "      <td>0.091370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>0.028854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>0.009618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>0.007213</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   num_gene_records  num_alias_symbol  percent_alias_symbol\n",
       "0                 2               858              2.063046\n",
       "1                 3               125              0.300560\n",
       "2                 4                38              0.091370\n",
       "3                 5                12              0.028854\n",
       "4                 6                 4              0.009618\n",
       "5                 7                 3              0.007213"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_count_histogram_df['percent_alias_symbol'] = ((hgnc_alias_count_histogram_df['num_alias_symbol'] / hgnc_alias_len) * 100)\n",
    "hgnc_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2.063046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>0.300560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>0.091370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>0.028854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>0.009618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>0.007213</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   num_gene_records  percent_alias_symbol\n",
       "0                 2              2.063046\n",
       "1                 3              0.300560\n",
       "2                 4              0.091370\n",
       "3                 5              0.028854\n",
       "4                 6              0.009618\n",
       "5                 7              0.007213"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_count_histogram_df = hgnc_alias_count_histogram_df.drop('num_alias_symbol', axis=1)\n",
    "hgnc_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#px.bar(hgnc_alias_count_histogram_df, x='num_gene_records', y='percent_alias_symbol')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc2_1_10_1_'></a>[Save as csv](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mini_hgnc_df_explode.to_csv('../hgnc_alias_overlap.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_11_'></a>[Put columns in different order to ephasize alias symbols instead of gene records](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_hgnc_df_2 = mini_hgnc_df.drop_duplicates(subset = [\"alias_symbol\", \"gene_symbol\"], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35226</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13978</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000139187</td>\n",
       "      <td>KLRG1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33987</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>ENSG00000143546</td>\n",
       "      <td>S100A8</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33988</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>ENSG00000163220</td>\n",
       "      <td>S100A9</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31335</th>\n",
       "      <td>87U6</td>\n",
       "      <td>ENSG00000206832</td>\n",
       "      <td>RNU6V</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38078</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>ENSG00000277008</td>\n",
       "      <td>TEX28P2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26200</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26203</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36661</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>SPATA2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36662</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2348 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol source\n",
       "35226          2F1  ENSG00000005022     SLC25A5   HGNC\n",
       "13978          2F1  ENSG00000139187       KLRG1   HGNC\n",
       "33987       60B8AG  ENSG00000143546      S100A8   HGNC\n",
       "33988       60B8AG  ENSG00000163220      S100A9   HGNC\n",
       "31335         87U6  ENSG00000206832       RNU6V   HGNC\n",
       "...            ...              ...         ...    ...\n",
       "38078         pTEX  ENSG00000277008     TEX28P2   HGNC\n",
       "26200         smk1  ENSG00000100796     PPP4R3A   HGNC\n",
       "26203         smk1  ENSG00000224960     PPP4R3C   HGNC\n",
       "36661         tamo  ENSG00000158480      SPATA2   HGNC\n",
       "36662         tamo  ENSG00000158792     SPATA2L   HGNC\n",
       "\n",
       "[2348 rows x 4 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df_2 = mini_hgnc_df[['alias_symbol', 'ENSG_ID', 'gene_symbol', 'source']]\n",
    "mini_hgnc_df_2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc2_1_12_'></a>[Merge rows with matching alias symbols](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/2517928805.py:1: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  mini_hgnc_df_2 = mini_hgnc_df_2.applymap(str)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35226</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13978</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000139187</td>\n",
       "      <td>KLRG1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33987</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>ENSG00000143546</td>\n",
       "      <td>S100A8</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33988</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>ENSG00000163220</td>\n",
       "      <td>S100A9</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31335</th>\n",
       "      <td>87U6</td>\n",
       "      <td>ENSG00000206832</td>\n",
       "      <td>RNU6V</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38078</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>ENSG00000277008</td>\n",
       "      <td>TEX28P2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26200</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26203</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36661</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>SPATA2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36662</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2348 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol source\n",
       "35226          2F1  ENSG00000005022     SLC25A5   HGNC\n",
       "13978          2F1  ENSG00000139187       KLRG1   HGNC\n",
       "33987       60B8AG  ENSG00000143546      S100A8   HGNC\n",
       "33988       60B8AG  ENSG00000163220      S100A9   HGNC\n",
       "31335         87U6  ENSG00000206832       RNU6V   HGNC\n",
       "...            ...              ...         ...    ...\n",
       "38078         pTEX  ENSG00000277008     TEX28P2   HGNC\n",
       "26200         smk1  ENSG00000100796     PPP4R3A   HGNC\n",
       "26203         smk1  ENSG00000224960     PPP4R3C   HGNC\n",
       "36661         tamo  ENSG00000158480      SPATA2   HGNC\n",
       "36662         tamo  ENSG00000158792     SPATA2L   HGNC\n",
       "\n",
       "[2348 rows x 4 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df_2 = mini_hgnc_df_2.applymap(str)\n",
    "mini_hgnc_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022, ENSG00000139187</td>\n",
       "      <td>SLC25A5, KLRG1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>ENSG00000143546, ENSG00000163220</td>\n",
       "      <td>S100A8, S100A9</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>87U6</td>\n",
       "      <td>ENSG00000206832, ENSG00000065135</td>\n",
       "      <td>RNU6V, GNAI3</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9G8</td>\n",
       "      <td>ENSG00000164609, ENSG00000115875</td>\n",
       "      <td>SLU7, SRSF7</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1</td>\n",
       "      <td>ENSG00000035928, ENSG00000163918, ENSG00000049541</td>\n",
       "      <td>RFC1, RFC4, RFC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1035</th>\n",
       "      <td>p97</td>\n",
       "      <td>ENSG00000165280, ENSG00000110321, ENSG00000179...</td>\n",
       "      <td>VCP, EIF4G2, GEMIN4, CFDP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1036</th>\n",
       "      <td>pH2A/f</td>\n",
       "      <td>ENSG00000196787, ENSG00000234816</td>\n",
       "      <td>H2AC11, H2AC5P</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1037</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>ENSG00000274962, ENSG00000277008</td>\n",
       "      <td>TEX28P1, TEX28P2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1038</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796, ENSG00000224960</td>\n",
       "      <td>PPP4R3A, PPP4R3C</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1039</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480, ENSG00000158792</td>\n",
       "      <td>SPATA2, SPATA2L</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1040 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol                                            ENSG_ID  \\\n",
       "0             2F1                   ENSG00000005022, ENSG00000139187   \n",
       "1          60B8AG                   ENSG00000143546, ENSG00000163220   \n",
       "2            87U6                   ENSG00000206832, ENSG00000065135   \n",
       "3             9G8                   ENSG00000164609, ENSG00000115875   \n",
       "4              A1  ENSG00000035928, ENSG00000163918, ENSG00000049541   \n",
       "...           ...                                                ...   \n",
       "1035          p97  ENSG00000165280, ENSG00000110321, ENSG00000179...   \n",
       "1036       pH2A/f                   ENSG00000196787, ENSG00000234816   \n",
       "1037         pTEX                   ENSG00000274962, ENSG00000277008   \n",
       "1038         smk1                   ENSG00000100796, ENSG00000224960   \n",
       "1039         tamo                   ENSG00000158480, ENSG00000158792   \n",
       "\n",
       "                     gene_symbol source  \n",
       "0                 SLC25A5, KLRG1   HGNC  \n",
       "1                 S100A8, S100A9   HGNC  \n",
       "2                   RNU6V, GNAI3   HGNC  \n",
       "3                    SLU7, SRSF7   HGNC  \n",
       "4               RFC1, RFC4, RFC2   HGNC  \n",
       "...                          ...    ...  \n",
       "1035  VCP, EIF4G2, GEMIN4, CFDP1   HGNC  \n",
       "1036              H2AC11, H2AC5P   HGNC  \n",
       "1037            TEX28P1, TEX28P2   HGNC  \n",
       "1038            PPP4R3A, PPP4R3C   HGNC  \n",
       "1039             SPATA2, SPATA2L   HGNC  \n",
       "\n",
       "[1040 rows x 4 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df_2 = mini_hgnc_df_2.groupby('alias_symbol').agg({'ENSG_ID': ', '.join, \n",
    "                             'gene_symbol': ', '.join, \n",
    "                             'source':'first' }).reset_index()\n",
    "mini_hgnc_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mini_hgnc_df_2.to_csv('../hgnc_alias_overlap_2.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc3_'></a>[NCBI Info](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/3750511974.py:1: DtypeWarning: Columns (15,16) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df2 = pd.read_csv(\"../ncbi_info_20220719_filtered.csv\")\n"
     ]
    }
   ],
   "source": [
    "df2 = pd.read_csv(\"../ncbi_info_20220719_filtered.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_1_'></a>[Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2MP1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>ENSG00000256069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75495</th>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75496</th>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75497</th>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75498</th>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75499</th>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75500 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol               alias_symbol          ENSG_ID\n",
       "0            A1BG       A1B|ABG|GAB|HYST2477  ENSG00000121410\n",
       "1             A2M  A2MD|CPAMD5|FWP007|S863-7  ENSG00000175899\n",
       "2           A2MP1                       A2MP  ENSG00000256069\n",
       "3            NAT1       AAC1|MNAT|NAT-1|NATI  ENSG00000171428\n",
       "4            NAT2            AAC2|NAT-2|PNAT  ENSG00000156006\n",
       "...           ...                        ...              ...\n",
       "75495        trnD                          -              NAN\n",
       "75496        trnP                          -              NAN\n",
       "75497        trnA                          -              NAN\n",
       "75498        COX1                          -              NAN\n",
       "75499    16S rRNA                          -              NAN\n",
       "\n",
       "[75500 rows x 3 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = df2.drop(['Unnamed: 0', '#tax_id','GeneID', 'dbXrefs', 'description', 'type_of_gene', 'Symbol_from_nomenclature_authority', 'Full_name_from_nomenclature_authority', 'Other_designations', 'MIM', 'HGNC', 'AllianceGenome','MIRbase', 'IMGTgene_db', 'dash', 'unknown'], axis=1)\n",
    "mini_ncbi_df = mini_ncbi_df.rename(columns = {'Symbol':'gene_symbol','Synonyms':'alias_symbol', 'ENSEMBL':'ENSG_ID'})\n",
    "mini_ncbi_df['ENSG_ID'] = mini_ncbi_df['ENSG_ID'].astype(str)\n",
    "mini_ncbi_df['ENSG_ID'] = mini_ncbi_df['ENSG_ID'].apply(str.upper)\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_2_'></a>[How many total unique gene records are there](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "75346"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_gene_symbol_set = set(mini_ncbi_df['gene_symbol'])\n",
    "len(ncbi_gene_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_3_'></a>[Drop genes with no aliases](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2MP1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>ENSG00000256069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75495</th>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75496</th>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75497</th>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75498</th>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75499</th>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75500 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol               alias_symbol          ENSG_ID\n",
       "0            A1BG       A1B|ABG|GAB|HYST2477  ENSG00000121410\n",
       "1             A2M  A2MD|CPAMD5|FWP007|S863-7  ENSG00000175899\n",
       "2           A2MP1                       A2MP  ENSG00000256069\n",
       "3            NAT1       AAC1|MNAT|NAT-1|NATI  ENSG00000171428\n",
       "4            NAT2            AAC2|NAT-2|PNAT  ENSG00000156006\n",
       "...           ...                        ...              ...\n",
       "75495        trnD                        NaN              NAN\n",
       "75496        trnP                        NaN              NAN\n",
       "75497        trnA                        NaN              NAN\n",
       "75498        COX1                        NaN              NAN\n",
       "75499    16S rRNA                        NaN              NAN\n",
       "\n",
       "[75500 rows x 3 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.replace(\"-\", np.nan)\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2MP1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>ENSG00000256069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71686</th>\n",
       "      <td>LOC124906931</td>\n",
       "      <td>WASH</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71738</th>\n",
       "      <td>LOC124906983</td>\n",
       "      <td>WASH</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72857</th>\n",
       "      <td>LOC124908102</td>\n",
       "      <td>WASH</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74876</th>\n",
       "      <td>POLGARF</td>\n",
       "      <td>ORF-Y|POLG</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75424</th>\n",
       "      <td>CDCA3P1</td>\n",
       "      <td>lncCDCA3L</td>\n",
       "      <td>NAN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27316 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene_symbol               alias_symbol          ENSG_ID\n",
       "0              A1BG       A1B|ABG|GAB|HYST2477  ENSG00000121410\n",
       "1               A2M  A2MD|CPAMD5|FWP007|S863-7  ENSG00000175899\n",
       "2             A2MP1                       A2MP  ENSG00000256069\n",
       "3              NAT1       AAC1|MNAT|NAT-1|NATI  ENSG00000171428\n",
       "4              NAT2            AAC2|NAT-2|PNAT  ENSG00000156006\n",
       "...             ...                        ...              ...\n",
       "71686  LOC124906931                       WASH              NAN\n",
       "71738  LOC124906983                       WASH              NAN\n",
       "72857  LOC124908102                       WASH              NAN\n",
       "74876       POLGARF                 ORF-Y|POLG              NAN\n",
       "75424       CDCA3P1                  lncCDCA3L              NAN\n",
       "\n",
       "[27316 rows x 3 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.dropna(subset=['alias_symbol'])\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_4_'></a>[Make each row in alias_symbol a set:](#toc0_)\n",
    "    covert to a list \n",
    "    make a set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/2736731455.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df['alias_symbol'] = mini_ncbi_df['alias_symbol'].astype(str)\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/2736731455.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df['alias_symbol'] = [x.split('|') for x in mini_ncbi_df.alias_symbol]\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/2736731455.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df['alias_symbol']=np.where(mini_ncbi_df.alias_symbol=='','',mini_ncbi_df.alias_symbol.map(set))\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>{HYST2477, A1B, ABG, GAB}</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol               alias_symbol          ENSG_ID\n",
       "0        A1BG  {HYST2477, A1B, ABG, GAB}  ENSG00000121410"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['alias_symbol'] = mini_ncbi_df['alias_symbol'].astype(str)\n",
    "mini_ncbi_df['alias_symbol'] = [x.split('|') for x in mini_ncbi_df.alias_symbol]\n",
    "mini_ncbi_df['alias_symbol']=np.where(mini_ncbi_df.alias_symbol=='','',mini_ncbi_df.alias_symbol.map(set))\n",
    "mini_ncbi_df.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_5_'></a>[Explode the alias sets so that it is one per row](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>S863-7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol alias_symbol          ENSG_ID\n",
       "0        A1BG     HYST2477  ENSG00000121410\n",
       "0        A1BG          A1B  ENSG00000121410\n",
       "0        A1BG          ABG  ENSG00000121410\n",
       "0        A1BG          GAB  ENSG00000121410\n",
       "1         A2M       S863-7  ENSG00000175899"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.explode(column=\"alias_symbol\")\n",
    "mini_ncbi_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "#ncbi_CD158b_alias_count_df.to_csv('../ncbi_CD158b_alias_count_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_6_'></a>[How many unique aliases are there](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "67454"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_symbol_set = set(mini_ncbi_df['alias_symbol'])\n",
    "ncbi_alias_len = len(ncbi_alias_symbol_set)\n",
    "ncbi_alias_len"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_7_'></a>[Pull out all the rows that have an alias symbol that can be found elsewhere](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NAT1</td>\n",
       "      <td>NAT-1</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SERPINA3</td>\n",
       "      <td>ACT</td>\n",
       "      <td>ENSG00000196136</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol alias_symbol          ENSG_ID\n",
       "0        A1BG          A1B  ENSG00000121410\n",
       "3        NAT1         AAC1  ENSG00000171428\n",
       "3        NAT1        NAT-1  ENSG00000171428\n",
       "4        NAT2         AAC2  ENSG00000156006\n",
       "6    SERPINA3          ACT  ENSG00000196136"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['alias_duplicates'] = mini_ncbi_df.duplicated(subset= 'alias_symbol', keep=False)\n",
    "mini_ncbi_df = mini_ncbi_df[mini_ncbi_df['alias_duplicates'] == True]\n",
    "mini_ncbi_df = mini_ncbi_df.drop(['alias_duplicates'], axis=1)\n",
    "mini_ncbi_df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_8_'></a>[Sort alias symbols alphabetically](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4534</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000107779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>ALOX12</td>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>ALOX15</td>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000161905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol alias_symbol          ENSG_ID\n",
       "4534        PTEN     10q23del  ENSG00000171862\n",
       "537       BMPR1A     10q23del  ENSG00000107779\n",
       "199       ALOX12       12-LOX  ENSG00000108839\n",
       "205       ALOX15       12-LOX  ENSG00000161905\n",
       "245      SLC25A5          2F1  ENSG00000005022"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.sort_values('alias_symbol')\n",
    "mini_ncbi_df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_9_'></a>[Number of records with an alias that is shared](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5670"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_gene_symbol_set_wsharedalias = set(mini_ncbi_df['gene_symbol'])\n",
    "len(ncbi_gene_symbol_set_wsharedalias)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4534</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000107779</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>ALOX12</td>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>ALOX15</td>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000161905</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18205</th>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12929</th>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13546</th>\n",
       "      <td>PPP4R3B</td>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000275052</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17549</th>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7647</th>\n",
       "      <td>SPATA2</td>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8247 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol alias_symbol          ENSG_ID     source\n",
       "4534         PTEN     10q23del  ENSG00000171862  NCBI Info\n",
       "537        BMPR1A     10q23del  ENSG00000107779  NCBI Info\n",
       "199        ALOX12       12-LOX  ENSG00000108839  NCBI Info\n",
       "205        ALOX15       12-LOX  ENSG00000161905  NCBI Info\n",
       "245       SLC25A5          2F1  ENSG00000005022  NCBI Info\n",
       "...           ...          ...              ...        ...\n",
       "18205     PPP4R3C         smk1  ENSG00000224960  NCBI Info\n",
       "12929     PPP4R3A         smk1  ENSG00000100796  NCBI Info\n",
       "13546     PPP4R3B         smk1  ENSG00000275052  NCBI Info\n",
       "17549     SPATA2L         tamo  ENSG00000158792  NCBI Info\n",
       "7647       SPATA2         tamo  ENSG00000158480  NCBI Info\n",
       "\n",
       "[8247 rows x 4 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['source'] = 'NCBI Info'\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_10_'></a>[Count the number of times each multi-use alias is used](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "alias_symbol\n",
       "10q23del       2\n",
       "12-LOX         2\n",
       "2F1            2\n",
       "3-alpha-HSD    2\n",
       "35DAG          2\n",
       "              ..\n",
       "pTEX           2\n",
       "polymerase     3\n",
       "rpL7a          2\n",
       "smk1           3\n",
       "tamo           2\n",
       "Length: 3427, dtype: int64"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_dup_alias_count_df = mini_ncbi_df.pivot_table(index = ['alias_symbol'], aggfunc ='size')\n",
    "ncbi_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_dup_alias_count_df.to_csv('../ncbi_alias_overlap_count.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2F1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-alpha-HSD</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>35DAG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3422</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3423</th>\n",
       "      <td>polymerase</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3424</th>\n",
       "      <td>rpL7a</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3425</th>\n",
       "      <td>smk1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>tamo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3427 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  0\n",
       "0        10q23del  2\n",
       "1          12-LOX  2\n",
       "2             2F1  2\n",
       "3     3-alpha-HSD  2\n",
       "4           35DAG  2\n",
       "...           ... ..\n",
       "3422         pTEX  2\n",
       "3423   polymerase  3\n",
       "3424        rpL7a  2\n",
       "3425         smk1  3\n",
       "3426         tamo  2\n",
       "\n",
       "[3427 rows x 2 columns]"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_dup_alias_count_df = ncbi_dup_alias_count_df.reset_index()\n",
    "ncbi_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2F1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-alpha-HSD</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>35DAG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3422</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3423</th>\n",
       "      <td>polymerase</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3424</th>\n",
       "      <td>rpL7a</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3425</th>\n",
       "      <td>smk1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>tamo</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3427 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "0        10q23del                 2\n",
       "1          12-LOX                 2\n",
       "2             2F1                 2\n",
       "3     3-alpha-HSD                 2\n",
       "4           35DAG                 2\n",
       "...           ...               ...\n",
       "3422         pTEX                 2\n",
       "3423   polymerase                 3\n",
       "3424        rpL7a                 2\n",
       "3425         smk1                 3\n",
       "3426         tamo                 2\n",
       "\n",
       "[3427 rows x 2 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_dup_alias_count_df.rename(columns={0:'num_gene_records'}, inplace=True )\n",
    "ncbi_dup_alias_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>num_gene_records</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3260</th>\n",
       "      <td>VH</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>H4-16</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1297</th>\n",
       "      <td>H4C9</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1286</th>\n",
       "      <td>H4C11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1287</th>\n",
       "      <td>H4C12</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1288</th>\n",
       "      <td>H4C13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1296</th>\n",
       "      <td>H4C8</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1289</th>\n",
       "      <td>H4C14</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1290</th>\n",
       "      <td>H4C15</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1291</th>\n",
       "      <td>H4C2</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1292</th>\n",
       "      <td>H4C3</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1293</th>\n",
       "      <td>H4C4</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1294</th>\n",
       "      <td>H4C5</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1295</th>\n",
       "      <td>H4C6</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1285</th>\n",
       "      <td>H4C1</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1958</th>\n",
       "      <td>MT1</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1192</th>\n",
       "      <td>GPCR</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1442</th>\n",
       "      <td>HOX1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2455</th>\n",
       "      <td>PR</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1444</th>\n",
       "      <td>HOX2</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1272</th>\n",
       "      <td>H3C12</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3225</th>\n",
       "      <td>UDPGT</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3233</th>\n",
       "      <td>UGT1</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>H3C1</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2447</th>\n",
       "      <td>PPIase</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1270</th>\n",
       "      <td>H3C10</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1271</th>\n",
       "      <td>H3C11</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1992</th>\n",
       "      <td>NAP1</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1276</th>\n",
       "      <td>H3C2</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>H3C3</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol  num_gene_records\n",
       "3260           VH                36\n",
       "1284        H4-16                14\n",
       "1297         H4C9                13\n",
       "1286        H4C11                13\n",
       "1287        H4C12                13\n",
       "1288        H4C13                13\n",
       "1296         H4C8                13\n",
       "1289        H4C14                13\n",
       "1290        H4C15                13\n",
       "1291         H4C2                13\n",
       "1292         H4C3                13\n",
       "1293         H4C4                13\n",
       "1294         H4C5                13\n",
       "1295         H4C6                13\n",
       "1285         H4C1                13\n",
       "1958          MT1                12\n",
       "1192         GPCR                11\n",
       "1442         HOX1                10\n",
       "2455           PR                10\n",
       "1444         HOX2                 9\n",
       "1272        H3C12                 9\n",
       "3225        UDPGT                 9\n",
       "3233         UGT1                 9\n",
       "1269         H3C1                 9\n",
       "2447       PPIase                 9\n",
       "1270        H3C10                 9\n",
       "1271        H3C11                 9\n",
       "1992         NAP1                 9\n",
       "1276         H3C2                 9\n",
       "1277         H3C3                 9"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_dup_alias_count_df = ncbi_dup_alias_count_df.sort_values('num_gene_records', ascending=False)\n",
    "ncbi_dup_alias_count_df.head(30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3427"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_alias_collision_set = set(ncbi_dup_alias_count_df['alias_symbol'])\n",
    "len(ncbi_alias_alias_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_dup_alias_count_df.to_csv('../ncbi_dup_alias_count_df.csv', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "num_gene_records\n",
       "2     2748\n",
       "3      404\n",
       "4      142\n",
       "5       51\n",
       "6       24\n",
       "7       17\n",
       "8        7\n",
       "9       15\n",
       "10       2\n",
       "11       1\n",
       "12       1\n",
       "13      13\n",
       "14       1\n",
       "36       1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_count_histogram_df = ncbi_dup_alias_count_df.pivot_table(index = ['num_gene_records'], aggfunc ='size')\n",
    "ncbi_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records     0\n",
       "0                  2  2748\n",
       "1                  3   404\n",
       "2                  4   142\n",
       "3                  5    51\n",
       "4                  6    24\n",
       "5                  7    17\n",
       "6                  8     7\n",
       "7                  9    15\n",
       "8                 10     2\n",
       "9                 11     1\n",
       "10                12     1\n",
       "11                13    13\n",
       "12                14     1\n",
       "13                36     1"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_count_histogram_df = ncbi_alias_count_histogram_df.reset_index()\n",
    "ncbi_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  num_alias_symbol\n",
       "0                  2              2748\n",
       "1                  3               404\n",
       "2                  4               142\n",
       "3                  5                51\n",
       "4                  6                24\n",
       "5                  7                17\n",
       "6                  8                 7\n",
       "7                  9                15\n",
       "8                 10                 2\n",
       "9                 11                 1\n",
       "10                12                 1\n",
       "11                13                13\n",
       "12                14                 1\n",
       "13                36                 1"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_count_histogram_df.rename(columns={0:'num_alias_symbol'}, inplace=True )\n",
    "ncbi_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>num_alias_symbol</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>2748</td>\n",
       "      <td>4.073887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>404</td>\n",
       "      <td>0.598927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>142</td>\n",
       "      <td>0.210514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>51</td>\n",
       "      <td>0.075607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>0.035580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>0.025202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>0.010377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>0.022237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>0.002965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>0.019272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  num_alias_symbol  percent_alias_symbol\n",
       "0                  2              2748              4.073887\n",
       "1                  3               404              0.598927\n",
       "2                  4               142              0.210514\n",
       "3                  5                51              0.075607\n",
       "4                  6                24              0.035580\n",
       "5                  7                17              0.025202\n",
       "6                  8                 7              0.010377\n",
       "7                  9                15              0.022237\n",
       "8                 10                 2              0.002965\n",
       "9                 11                 1              0.001482\n",
       "10                12                 1              0.001482\n",
       "11                13                13              0.019272\n",
       "12                14                 1              0.001482\n",
       "13                36                 1              0.001482"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_count_histogram_df['percent_alias_symbol'] = ((ncbi_alias_count_histogram_df['num_alias_symbol'] / ncbi_alias_len) * 100)\n",
    "ncbi_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_gene_records</th>\n",
       "      <th>percent_alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>4.073887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>0.598927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>0.210514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>0.075607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>0.035580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>0.025202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>8</td>\n",
       "      <td>0.010377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>9</td>\n",
       "      <td>0.022237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>0.002965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>11</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13</td>\n",
       "      <td>0.019272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>36</td>\n",
       "      <td>0.001482</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    num_gene_records  percent_alias_symbol\n",
       "0                  2              4.073887\n",
       "1                  3              0.598927\n",
       "2                  4              0.210514\n",
       "3                  5              0.075607\n",
       "4                  6              0.035580\n",
       "5                  7              0.025202\n",
       "6                  8              0.010377\n",
       "7                  9              0.022237\n",
       "8                 10              0.002965\n",
       "9                 11              0.001482\n",
       "10                12              0.001482\n",
       "11                13              0.019272\n",
       "12                14              0.001482\n",
       "13                36              0.001482"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_count_histogram_df = ncbi_alias_count_histogram_df.drop('num_alias_symbol', axis=1)\n",
    "ncbi_alias_count_histogram_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# px.bar(ncbi_alias_count_histogram_df, x='num_gene_records', y='percent_alias_symbol')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_11_'></a>[Put columns in different order to ephasize alias symbols instead of gene records](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df_2 = mini_ncbi_df.drop_duplicates(subset = [\"alias_symbol\", \"gene_symbol\"], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4534</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000107779</td>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839</td>\n",
       "      <td>ALOX12</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000161905</td>\n",
       "      <td>ALOX15</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18205</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12929</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13546</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000275052</td>\n",
       "      <td>PPP4R3B</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17549</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7647</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>SPATA2</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8247 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol     source\n",
       "4534      10q23del  ENSG00000171862        PTEN  NCBI Info\n",
       "537       10q23del  ENSG00000107779      BMPR1A  NCBI Info\n",
       "199         12-LOX  ENSG00000108839      ALOX12  NCBI Info\n",
       "205         12-LOX  ENSG00000161905      ALOX15  NCBI Info\n",
       "245            2F1  ENSG00000005022     SLC25A5  NCBI Info\n",
       "...            ...              ...         ...        ...\n",
       "18205         smk1  ENSG00000224960     PPP4R3C  NCBI Info\n",
       "12929         smk1  ENSG00000100796     PPP4R3A  NCBI Info\n",
       "13546         smk1  ENSG00000275052     PPP4R3B  NCBI Info\n",
       "17549         tamo  ENSG00000158792     SPATA2L  NCBI Info\n",
       "7647          tamo  ENSG00000158480      SPATA2  NCBI Info\n",
       "\n",
       "[8247 rows x 4 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df_2 = mini_ncbi_df[['alias_symbol', 'ENSG_ID', 'gene_symbol', 'source']]\n",
    "mini_ncbi_df_2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc3_1_12_'></a>[Merge rows with matching alias symbols](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_76117/792032257.py:1: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  mini_ncbi_df_2 = mini_ncbi_df_2.applymap(str)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4534</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000107779</td>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839</td>\n",
       "      <td>ALOX12</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000161905</td>\n",
       "      <td>ALOX15</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18205</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12929</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13546</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000275052</td>\n",
       "      <td>PPP4R3B</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17549</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7647</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>SPATA2</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8247 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol     source\n",
       "4534      10q23del  ENSG00000171862        PTEN  NCBI Info\n",
       "537       10q23del  ENSG00000107779      BMPR1A  NCBI Info\n",
       "199         12-LOX  ENSG00000108839      ALOX12  NCBI Info\n",
       "205         12-LOX  ENSG00000161905      ALOX15  NCBI Info\n",
       "245            2F1  ENSG00000005022     SLC25A5  NCBI Info\n",
       "...            ...              ...         ...        ...\n",
       "18205         smk1  ENSG00000224960     PPP4R3C  NCBI Info\n",
       "12929         smk1  ENSG00000100796     PPP4R3A  NCBI Info\n",
       "13546         smk1  ENSG00000275052     PPP4R3B  NCBI Info\n",
       "17549         tamo  ENSG00000158792     SPATA2L  NCBI Info\n",
       "7647          tamo  ENSG00000158480      SPATA2  NCBI Info\n",
       "\n",
       "[8247 rows x 4 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df_2 = mini_ncbi_df_2.applymap(str)\n",
    "mini_ncbi_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4534</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000107779</td>\n",
       "      <td>BMPR1A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839</td>\n",
       "      <td>ALOX12</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000161905</td>\n",
       "      <td>ALOX15</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18205</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960</td>\n",
       "      <td>PPP4R3C</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12929</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000100796</td>\n",
       "      <td>PPP4R3A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13546</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000275052</td>\n",
       "      <td>PPP4R3B</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17549</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792</td>\n",
       "      <td>SPATA2L</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7647</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158480</td>\n",
       "      <td>SPATA2</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8247 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      alias_symbol          ENSG_ID gene_symbol     source\n",
       "4534      10q23del  ENSG00000171862        PTEN  NCBI Info\n",
       "537       10q23del  ENSG00000107779      BMPR1A  NCBI Info\n",
       "199         12-LOX  ENSG00000108839      ALOX12  NCBI Info\n",
       "205         12-LOX  ENSG00000161905      ALOX15  NCBI Info\n",
       "245            2F1  ENSG00000005022     SLC25A5  NCBI Info\n",
       "...            ...              ...         ...        ...\n",
       "18205         smk1  ENSG00000224960     PPP4R3C  NCBI Info\n",
       "12929         smk1  ENSG00000100796     PPP4R3A  NCBI Info\n",
       "13546         smk1  ENSG00000275052     PPP4R3B  NCBI Info\n",
       "17549         tamo  ENSG00000158792     SPATA2L  NCBI Info\n",
       "7647          tamo  ENSG00000158480      SPATA2  NCBI Info\n",
       "\n",
       "[8247 rows x 4 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df_2['ENSG_ID'] = mini_ncbi_df_2['ENSG_ID'].str.replace('NAN','nan')\n",
    "mini_ncbi_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10q23del</td>\n",
       "      <td>ENSG00000171862, ENSG00000107779</td>\n",
       "      <td>PTEN, BMPR1A</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12-LOX</td>\n",
       "      <td>ENSG00000108839, ENSG00000161905</td>\n",
       "      <td>ALOX12, ALOX15</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2F1</td>\n",
       "      <td>ENSG00000005022, ENSG00000139187</td>\n",
       "      <td>SLC25A5, KLRG1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3-alpha-HSD</td>\n",
       "      <td>ENSG00000198610, ENSG00000073737</td>\n",
       "      <td>AKR1C4, DHRS9</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>35DAG</td>\n",
       "      <td>ENSG00000170624, ENSG00000102683</td>\n",
       "      <td>SGCD, SGCG</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3422</th>\n",
       "      <td>pTEX</td>\n",
       "      <td>ENSG00000274962, ENSG00000277008</td>\n",
       "      <td>TEX28P1, TEX28P2</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3423</th>\n",
       "      <td>polymerase</td>\n",
       "      <td>nan, nan, nan</td>\n",
       "      <td>ERVK-9, ERVK-11, ERVK-19</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3424</th>\n",
       "      <td>rpL7a</td>\n",
       "      <td>ENSG00000240522, ENSG00000213272</td>\n",
       "      <td>RPL7AP10, RPL7AP9</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3425</th>\n",
       "      <td>smk1</td>\n",
       "      <td>ENSG00000224960, ENSG00000100796, ENSG00000275052</td>\n",
       "      <td>PPP4R3C, PPP4R3A, PPP4R3B</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3426</th>\n",
       "      <td>tamo</td>\n",
       "      <td>ENSG00000158792, ENSG00000158480</td>\n",
       "      <td>SPATA2L, SPATA2</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3427 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol                                            ENSG_ID  \\\n",
       "0        10q23del                   ENSG00000171862, ENSG00000107779   \n",
       "1          12-LOX                   ENSG00000108839, ENSG00000161905   \n",
       "2             2F1                   ENSG00000005022, ENSG00000139187   \n",
       "3     3-alpha-HSD                   ENSG00000198610, ENSG00000073737   \n",
       "4           35DAG                   ENSG00000170624, ENSG00000102683   \n",
       "...           ...                                                ...   \n",
       "3422         pTEX                   ENSG00000274962, ENSG00000277008   \n",
       "3423   polymerase                                      nan, nan, nan   \n",
       "3424        rpL7a                   ENSG00000240522, ENSG00000213272   \n",
       "3425         smk1  ENSG00000224960, ENSG00000100796, ENSG00000275052   \n",
       "3426         tamo                   ENSG00000158792, ENSG00000158480   \n",
       "\n",
       "                    gene_symbol     source  \n",
       "0                  PTEN, BMPR1A  NCBI Info  \n",
       "1                ALOX12, ALOX15  NCBI Info  \n",
       "2                SLC25A5, KLRG1  NCBI Info  \n",
       "3                 AKR1C4, DHRS9  NCBI Info  \n",
       "4                    SGCD, SGCG  NCBI Info  \n",
       "...                         ...        ...  \n",
       "3422           TEX28P1, TEX28P2  NCBI Info  \n",
       "3423   ERVK-9, ERVK-11, ERVK-19  NCBI Info  \n",
       "3424          RPL7AP10, RPL7AP9  NCBI Info  \n",
       "3425  PPP4R3C, PPP4R3A, PPP4R3B  NCBI Info  \n",
       "3426            SPATA2L, SPATA2  NCBI Info  \n",
       "\n",
       "[3427 rows x 4 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df_2 = mini_ncbi_df_2.groupby('alias_symbol').agg({'ENSG_ID': ', '.join, \n",
    "                             'gene_symbol': ', '.join, \n",
    "                             'source':'first' }).reset_index()\n",
    "mini_ncbi_df_2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc4_'></a>[Merge to create Alias Overlap Table 1 - Gene Symbol](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35226</th>\n",
       "      <td>SLC25A5</td>\n",
       "      <td>ENSG00000005022</td>\n",
       "      <td>2F1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13978</th>\n",
       "      <td>KLRG1</td>\n",
       "      <td>ENSG00000139187</td>\n",
       "      <td>2F1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33987</th>\n",
       "      <td>S100A8</td>\n",
       "      <td>ENSG00000143546</td>\n",
       "      <td>60B8AG</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33988</th>\n",
       "      <td>S100A9</td>\n",
       "      <td>ENSG00000163220</td>\n",
       "      <td>60B8AG</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31335</th>\n",
       "      <td>RNU6V</td>\n",
       "      <td>ENSG00000206832</td>\n",
       "      <td>87U6</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8105</th>\n",
       "      <td>PSMC1</td>\n",
       "      <td>ENSG00000100764</td>\n",
       "      <td>p56</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7724</th>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37151</th>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG00000011638</td>\n",
       "      <td>promethin</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27801</th>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27797</th>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG00000100181</td>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31474 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol source\n",
       "35226     SLC25A5  ENSG00000005022          2F1   HGNC\n",
       "13978       KLRG1  ENSG00000139187          2F1   HGNC\n",
       "33987      S100A8  ENSG00000143546       60B8AG   HGNC\n",
       "33988      S100A9  ENSG00000163220       60B8AG   HGNC\n",
       "31335       RNU6V  ENSG00000206832         87U6   HGNC\n",
       "...           ...              ...          ...    ...\n",
       "8105        PSMC1  ENSG00000100764          p56   ENSG\n",
       "7724        LDAF1  ENSG00000283997    promethin   ENSG\n",
       "37151       LDAF1  ENSG00000011638    promethin   ENSG\n",
       "27801      TPTEP1  ENSG00000290418    psiTPTE22   ENSG\n",
       "27797      TPTEP1  ENSG00000100181    psiTPTE22   ENSG\n",
       "\n",
       "[31474 rows x 4 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_1 = pd.concat([mini_hgnc_df[['gene_symbol', 'ENSG_ID', 'alias_symbol', 'source']],mini_ncbi_df[['gene_symbol', 'ENSG_ID', 'alias_symbol', 'source']], mini_ensg_df[['gene_symbol', 'ENSG_ID', 'alias_symbol', 'source']]])\n",
    "merged_alias_overlap_df_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_overlap_df_1.to_csv('../merged_alias_overlap_df_1.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene_symbol, ENSG_ID, alias_symbol, source]\n",
       "Index: []"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_1.loc[merged_alias_overlap_df_1.gene_symbol == 'NAP1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5768</th>\n",
       "      <td>CXCL8</td>\n",
       "      <td>ENSG00000169429</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21850</th>\n",
       "      <td>NAPSA</td>\n",
       "      <td>ENSG00000131400</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>ACOT8</td>\n",
       "      <td>ENSG00000101473</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2070</th>\n",
       "      <td>AZI2</td>\n",
       "      <td>ENSG00000163512</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21830</th>\n",
       "      <td>NAP1L1</td>\n",
       "      <td>ENSG00000187109</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20405</th>\n",
       "      <td>TAB3</td>\n",
       "      <td>ENSG00000157625</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7804</th>\n",
       "      <td>ACOT8</td>\n",
       "      <td>ENSG00000101473</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11818</th>\n",
       "      <td>TRMO</td>\n",
       "      <td>ENSG00000136932</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3677</th>\n",
       "      <td>NAP1L1</td>\n",
       "      <td>ENSG00000187109</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14180</th>\n",
       "      <td>AZI2</td>\n",
       "      <td>ENSG00000163512</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2893</th>\n",
       "      <td>CXCL8</td>\n",
       "      <td>ENSG00000169429</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15040</th>\n",
       "      <td>NAA25</td>\n",
       "      <td>ENSG00000111300</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7353</th>\n",
       "      <td>NAPSA</td>\n",
       "      <td>ENSG00000131400</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8458</th>\n",
       "      <td>NCKAP1</td>\n",
       "      <td>ENSG00000061676</td>\n",
       "      <td>NAP1</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_symbol          ENSG_ID alias_symbol     source\n",
       "5768        CXCL8  ENSG00000169429         NAP1       HGNC\n",
       "21850       NAPSA  ENSG00000131400         NAP1       HGNC\n",
       "221         ACOT8  ENSG00000101473         NAP1       HGNC\n",
       "2070         AZI2  ENSG00000163512         NAP1       HGNC\n",
       "21830      NAP1L1  ENSG00000187109         NAP1       HGNC\n",
       "20405        TAB3  ENSG00000157625         NAP1  NCBI Info\n",
       "7804        ACOT8  ENSG00000101473         NAP1  NCBI Info\n",
       "11818        TRMO  ENSG00000136932         NAP1  NCBI Info\n",
       "3677       NAP1L1  ENSG00000187109         NAP1  NCBI Info\n",
       "14180        AZI2  ENSG00000163512         NAP1  NCBI Info\n",
       "2893        CXCL8  ENSG00000169429         NAP1  NCBI Info\n",
       "15040       NAA25  ENSG00000111300         NAP1  NCBI Info\n",
       "7353        NAPSA  ENSG00000131400         NAP1  NCBI Info\n",
       "8458       NCKAP1  ENSG00000061676         NAP1  NCBI Info"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_1.loc[merged_alias_overlap_df_1['alias_symbol'] == \"NAP1\" ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "source\n",
       "ENSG         20879\n",
       "NCBI Info     8247\n",
       "HGNC          2348\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_1['source'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc5_'></a>[Merge to create Alias Overlap Table 2 - Alias Symbol](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2F1</td>\n",
       "      <td>SLC25A5, KLRG1</td>\n",
       "      <td>ENSG00000005022, ENSG00000139187</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>S100A8, S100A9</td>\n",
       "      <td>ENSG00000143546, ENSG00000163220</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>87U6</td>\n",
       "      <td>RNU6V, GNAI3</td>\n",
       "      <td>ENSG00000206832, ENSG00000065135</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9G8</td>\n",
       "      <td>SLU7, SRSF7</td>\n",
       "      <td>ENSG00000164609, ENSG00000115875</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1</td>\n",
       "      <td>RFC1, RFC4, RFC2</td>\n",
       "      <td>ENSG00000035928, ENSG00000163918, ENSG00000049541</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5302</th>\n",
       "      <td>p42</td>\n",
       "      <td>PSMC6, H3P30</td>\n",
       "      <td>ENSG00000100519, ENSG00000227443</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5303</th>\n",
       "      <td>p55</td>\n",
       "      <td>PIK3R3, PSMD12</td>\n",
       "      <td>ENSG00000117461, ENSG00000197170</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5304</th>\n",
       "      <td>p56</td>\n",
       "      <td>H3P45, PSMC1</td>\n",
       "      <td>ENSG00000227211, ENSG00000100764</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5305</th>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5306</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9774 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol       gene_symbol  \\\n",
       "0             2F1    SLC25A5, KLRG1   \n",
       "1          60B8AG    S100A8, S100A9   \n",
       "2            87U6      RNU6V, GNAI3   \n",
       "3             9G8       SLU7, SRSF7   \n",
       "4              A1  RFC1, RFC4, RFC2   \n",
       "...           ...               ...   \n",
       "5302          p42      PSMC6, H3P30   \n",
       "5303          p55    PIK3R3, PSMD12   \n",
       "5304          p56      H3P45, PSMC1   \n",
       "5305    promethin             LDAF1   \n",
       "5306    psiTPTE22            TPTEP1   \n",
       "\n",
       "                                                ENSG_ID source  \n",
       "0                      ENSG00000005022, ENSG00000139187   HGNC  \n",
       "1                      ENSG00000143546, ENSG00000163220   HGNC  \n",
       "2                      ENSG00000206832, ENSG00000065135   HGNC  \n",
       "3                      ENSG00000164609, ENSG00000115875   HGNC  \n",
       "4     ENSG00000035928, ENSG00000163918, ENSG00000049541   HGNC  \n",
       "...                                                 ...    ...  \n",
       "5302                   ENSG00000100519, ENSG00000227443   ENSG  \n",
       "5303                   ENSG00000117461, ENSG00000197170   ENSG  \n",
       "5304                   ENSG00000227211, ENSG00000100764   ENSG  \n",
       "5305                                    ENSG00000283997   ENSG  \n",
       "5306                                    ENSG00000290418   ENSG  \n",
       "\n",
       "[9774 rows x 4 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_2 = pd.concat([mini_hgnc_df_2[['alias_symbol', 'gene_symbol', 'ENSG_ID', 'source']],mini_ncbi_df_2[['alias_symbol', 'gene_symbol', 'ENSG_ID', 'source']], mini_ensg_df_2[['alias_symbol', 'gene_symbol', 'ENSG_ID', 'source']]])\n",
    "merged_alias_overlap_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ALP</td>\n",
       "      <td>PDLIM3, ATRNL1, CCL27, ASRGL1, SLPI</td>\n",
       "      <td>ENSG00000154553, ENSG00000107518, ENSG00000213...</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>ALP</td>\n",
       "      <td>CCL27, ATHS, PDLIM3, ATRNL1, ALPP, NAT10, SLPI...</td>\n",
       "      <td>ENSG00000213927, nan, ENSG00000154553, ENSG000...</td>\n",
       "      <td>NCBI Info</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3245</th>\n",
       "      <td>ALP</td>\n",
       "      <td>CCL27, ASRGL1, PDLIM3, ATRNL1</td>\n",
       "      <td>ENSG00000213927, ENSG00000162174, ENSG00000154...</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol                                        gene_symbol  \\\n",
       "31            ALP                PDLIM3, ATRNL1, CCL27, ASRGL1, SLPI   \n",
       "115           ALP  CCL27, ATHS, PDLIM3, ATRNL1, ALPP, NAT10, SLPI...   \n",
       "3245          ALP                      CCL27, ASRGL1, PDLIM3, ATRNL1   \n",
       "\n",
       "                                                ENSG_ID     source  \n",
       "31    ENSG00000154553, ENSG00000107518, ENSG00000213...       HGNC  \n",
       "115   ENSG00000213927, nan, ENSG00000154553, ENSG000...  NCBI Info  \n",
       "3245  ENSG00000213927, ENSG00000162174, ENSG00000154...       ENSG  "
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_2.loc[merged_alias_overlap_df_2['alias_symbol'] == \"ALP\" ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>source</th>\n",
       "      <th>gene_symbol_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2F1</td>\n",
       "      <td>SLC25A5, KLRG1</td>\n",
       "      <td>ENSG00000005022, ENSG00000139187</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>60B8AG</td>\n",
       "      <td>S100A8, S100A9</td>\n",
       "      <td>ENSG00000143546, ENSG00000163220</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>87U6</td>\n",
       "      <td>RNU6V, GNAI3</td>\n",
       "      <td>ENSG00000206832, ENSG00000065135</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9G8</td>\n",
       "      <td>SLU7, SRSF7</td>\n",
       "      <td>ENSG00000164609, ENSG00000115875</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1</td>\n",
       "      <td>RFC1, RFC4, RFC2</td>\n",
       "      <td>ENSG00000035928, ENSG00000163918, ENSG00000049541</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5302</th>\n",
       "      <td>p42</td>\n",
       "      <td>PSMC6, H3P30</td>\n",
       "      <td>ENSG00000100519, ENSG00000227443</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5303</th>\n",
       "      <td>p55</td>\n",
       "      <td>PIK3R3, PSMD12</td>\n",
       "      <td>ENSG00000117461, ENSG00000197170</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5304</th>\n",
       "      <td>p56</td>\n",
       "      <td>H3P45, PSMC1</td>\n",
       "      <td>ENSG00000227211, ENSG00000100764</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5305</th>\n",
       "      <td>promethin</td>\n",
       "      <td>LDAF1</td>\n",
       "      <td>ENSG00000283997</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5306</th>\n",
       "      <td>psiTPTE22</td>\n",
       "      <td>TPTEP1</td>\n",
       "      <td>ENSG00000290418</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9774 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol       gene_symbol  \\\n",
       "0             2F1    SLC25A5, KLRG1   \n",
       "1          60B8AG    S100A8, S100A9   \n",
       "2            87U6      RNU6V, GNAI3   \n",
       "3             9G8       SLU7, SRSF7   \n",
       "4              A1  RFC1, RFC4, RFC2   \n",
       "...           ...               ...   \n",
       "5302          p42      PSMC6, H3P30   \n",
       "5303          p55    PIK3R3, PSMD12   \n",
       "5304          p56      H3P45, PSMC1   \n",
       "5305    promethin             LDAF1   \n",
       "5306    psiTPTE22            TPTEP1   \n",
       "\n",
       "                                                ENSG_ID source  \\\n",
       "0                      ENSG00000005022, ENSG00000139187   HGNC   \n",
       "1                      ENSG00000143546, ENSG00000163220   HGNC   \n",
       "2                      ENSG00000206832, ENSG00000065135   HGNC   \n",
       "3                      ENSG00000164609, ENSG00000115875   HGNC   \n",
       "4     ENSG00000035928, ENSG00000163918, ENSG00000049541   HGNC   \n",
       "...                                                 ...    ...   \n",
       "5302                   ENSG00000100519, ENSG00000227443   ENSG   \n",
       "5303                   ENSG00000117461, ENSG00000197170   ENSG   \n",
       "5304                   ENSG00000227211, ENSG00000100764   ENSG   \n",
       "5305                                    ENSG00000283997   ENSG   \n",
       "5306                                    ENSG00000290418   ENSG   \n",
       "\n",
       "      gene_symbol_count  \n",
       "0                    14  \n",
       "1                    14  \n",
       "2                    12  \n",
       "3                    11  \n",
       "4                    16  \n",
       "...                 ...  \n",
       "5302                 12  \n",
       "5303                 14  \n",
       "5304                 12  \n",
       "5305                  5  \n",
       "5306                  6  \n",
       "\n",
       "[9774 rows x 5 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_2['gene_symbol_count'] = [len(c) for c in merged_alias_overlap_df_2['gene_symbol']]\n",
    "merged_alias_overlap_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>source</th>\n",
       "      <th>gene_symbol_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3260</th>\n",
       "      <td>VH</td>\n",
       "      <td>IGHV4-59, IGHV3-64, IGHV2-70, IGHV3-38, IGHV1-...</td>\n",
       "      <td>ENSG00000224373, ENSG00000223648, ENSG00000274...</td>\n",
       "      <td>NCBI Info</td>\n",
       "      <td>347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>H4-16</td>\n",
       "      <td>H4C6, H4C13, H4C11, H4C14, H4C1, H4C5, H4C12, ...</td>\n",
       "      <td>ENSG00000274618, ENSG00000275126, ENSG00000197...</td>\n",
       "      <td>NCBI Info</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1192</th>\n",
       "      <td>GPCR</td>\n",
       "      <td>GPR166P, VN1R17P, MRGPRX1, GPR151, MRGPRX4, OX...</td>\n",
       "      <td>ENSG00000220349, nan, ENSG00000170255, ENSG000...</td>\n",
       "      <td>NCBI Info</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>DUX10</td>\n",
       "      <td>DUX4L1, LOC107987486, LOC107987485, LOC1249064...</td>\n",
       "      <td>ENSG00000280757, nan, nan, nan, nan, nan, ENSG...</td>\n",
       "      <td>NCBI Info</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>869</th>\n",
       "      <td>DUX4</td>\n",
       "      <td>DUX4L2, DUX4L1, LOC107987486, LOC107987485, LO...</td>\n",
       "      <td>ENSG00000280457, ENSG00000280757, nan, nan, na...</td>\n",
       "      <td>NCBI Info</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3932</th>\n",
       "      <td>DOM3Z</td>\n",
       "      <td>DXO</td>\n",
       "      <td>ENSG00000206346</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1682</th>\n",
       "      <td>Nop1</td>\n",
       "      <td>FBL</td>\n",
       "      <td>ENSG00000105202</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4078</th>\n",
       "      <td>FLJ12614</td>\n",
       "      <td>NXN</td>\n",
       "      <td>ENSG00000281300</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4742</th>\n",
       "      <td>OSC</td>\n",
       "      <td>LSS</td>\n",
       "      <td>ENSG00000281289</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4758</th>\n",
       "      <td>PBDX</td>\n",
       "      <td>XG</td>\n",
       "      <td>ENSG00000124343</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9774 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     alias_symbol                                        gene_symbol  \\\n",
       "3260           VH  IGHV4-59, IGHV3-64, IGHV2-70, IGHV3-38, IGHV1-...   \n",
       "1284        H4-16  H4C6, H4C13, H4C11, H4C14, H4C1, H4C5, H4C12, ...   \n",
       "1192         GPCR  GPR166P, VN1R17P, MRGPRX1, GPR151, MRGPRX4, OX...   \n",
       "868         DUX10  DUX4L1, LOC107987486, LOC107987485, LOC1249064...   \n",
       "869          DUX4  DUX4L2, DUX4L1, LOC107987486, LOC107987485, LO...   \n",
       "...           ...                                                ...   \n",
       "3932        DOM3Z                                                DXO   \n",
       "1682         Nop1                                                FBL   \n",
       "4078     FLJ12614                                                NXN   \n",
       "4742          OSC                                                LSS   \n",
       "4758         PBDX                                                 XG   \n",
       "\n",
       "                                                ENSG_ID     source  \\\n",
       "3260  ENSG00000224373, ENSG00000223648, ENSG00000274...  NCBI Info   \n",
       "1284  ENSG00000274618, ENSG00000275126, ENSG00000197...  NCBI Info   \n",
       "1192  ENSG00000220349, nan, ENSG00000170255, ENSG000...  NCBI Info   \n",
       "868   ENSG00000280757, nan, nan, nan, nan, nan, ENSG...  NCBI Info   \n",
       "869   ENSG00000280457, ENSG00000280757, nan, nan, na...  NCBI Info   \n",
       "...                                                 ...        ...   \n",
       "3932                                    ENSG00000206346       ENSG   \n",
       "1682                                    ENSG00000105202       ENSG   \n",
       "4078                                    ENSG00000281300       ENSG   \n",
       "4742                                    ENSG00000281289       ENSG   \n",
       "4758                                    ENSG00000124343       ENSG   \n",
       "\n",
       "      gene_symbol_count  \n",
       "3260                347  \n",
       "1284                 88  \n",
       "1192                 85  \n",
       "868                  85  \n",
       "869                  84  \n",
       "...                 ...  \n",
       "3932                  3  \n",
       "1682                  3  \n",
       "4078                  3  \n",
       "4742                  3  \n",
       "4758                  2  \n",
       "\n",
       "[9774 rows x 5 columns]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_2 = merged_alias_overlap_df_2.sort_values(by='gene_symbol_count', ascending= False)\n",
    "merged_alias_overlap_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "source\n",
       "ENSG         5307\n",
       "NCBI Info    3427\n",
       "HGNC         1040\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_overlap_df_2['source'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_overlap_df_2.to_csv('../merged_alias_overlap_df_2.csv', index=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DGIdb ambiguous query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MN1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:7180</td>\n",
       "      <td>MN1</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P2RX7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:8537</td>\n",
       "      <td>P2RX7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:1960</td>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAPK8</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6881</td>\n",
       "      <td>MAPK8</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAPK10</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6872</td>\n",
       "      <td>MAPK10</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73884</th>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:6889</td>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73885</th>\n",
       "      <td>PTGER3</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:9595</td>\n",
       "      <td>PTGER3</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73886</th>\n",
       "      <td>CD1D</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:1637</td>\n",
       "      <td>CD1D</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73887</th>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:10993</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>IDG</td>\n",
       "      <td>15-Jul-19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73888</th>\n",
       "      <td>SOCS1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:19383</td>\n",
       "      <td>SOCS1</td>\n",
       "      <td>FoundationOneGenes</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73889 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name nomenclature  concept_id gene_claim_name  \\\n",
       "0           MN1  Gene Symbol   hgnc:7180             MN1   \n",
       "1         P2RX7  Gene Symbol   hgnc:8537           P2RX7   \n",
       "2        CHRNA7  Gene Symbol   hgnc:1960          CHRNA7   \n",
       "3         MAPK8  Gene Symbol   hgnc:6881           MAPK8   \n",
       "4        MAPK10  Gene Symbol   hgnc:6872          MAPK10   \n",
       "...         ...          ...         ...             ...   \n",
       "73884  MAPKAPK5    Gene Name   hgnc:6889        MAPKAPK5   \n",
       "73885    PTGER3    Gene Name   hgnc:9595          PTGER3   \n",
       "73886      CD1D    Gene Name   hgnc:1637            CD1D   \n",
       "73887   SLC26A1  Gene Symbol  hgnc:10993         SLC26A1   \n",
       "73888     SOCS1  Gene Symbol  hgnc:19383           SOCS1   \n",
       "\n",
       "                   source_db_name source_db_version  \n",
       "0      CarisMolecularIntelligence          5-Jun-23  \n",
       "1                          ChEMBL                32  \n",
       "2                          ChEMBL                32  \n",
       "3                          ChEMBL                32  \n",
       "4                          ChEMBL                32  \n",
       "...                           ...               ...  \n",
       "73884                         DTC          5-Jun-23  \n",
       "73885                         DTC          5-Jun-23  \n",
       "73886                         DTC          5-Jun-23  \n",
       "73887                         IDG         15-Jul-19  \n",
       "73888          FoundationOneGenes          5-Jun-23  \n",
       "\n",
       "[73889 rows x 6 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_df = pd.read_csv(\"../dgidb_genes_df.tsv\", sep='\\t')\n",
    "dgidb_gene_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     62221\n",
       "False    11668\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "symbol_col_comparison = dgidb_gene_df['name'] == dgidb_gene_df['gene_claim_name']\n",
    "symbol_col_comparison.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0        True\n",
       "1        True\n",
       "2        True\n",
       "3        True\n",
       "4        True\n",
       "         ... \n",
       "73884    True\n",
       "73885    True\n",
       "73886    True\n",
       "73887    True\n",
       "73888    True\n",
       "Length: 73889, dtype: bool"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "symbol_col_comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "name                 object\n",
       "nomenclature         object\n",
       "concept_id           object\n",
       "gene_claim_name      object\n",
       "source_db_name       object\n",
       "source_db_version    object\n",
       "dtype: object"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HSP90</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:5253</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>RPLP</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>SEPT6</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:15848</td>\n",
       "      <td>SEPTIN6</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>SEPT5</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:9164</td>\n",
       "      <td>SEPTIN5</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>WISP3</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:12771</td>\n",
       "      <td>CCN6</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73631</th>\n",
       "      <td>ENSEMBL:ENSG00000185821</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:31305</td>\n",
       "      <td>OR6C76</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73642</th>\n",
       "      <td>C11ORF30</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:18071</td>\n",
       "      <td>EMSY</td>\n",
       "      <td>FoundationOneGenes</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73727</th>\n",
       "      <td>ENSEMBL:ENSG00000183921</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:35414</td>\n",
       "      <td>SDR42E2</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73872</th>\n",
       "      <td>MR</td>\n",
       "      <td>NCBI Gene Name</td>\n",
       "      <td>hgnc:7979</td>\n",
       "      <td>NR3C2</td>\n",
       "      <td>BaderLab</td>\n",
       "      <td>14-Feb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73878</th>\n",
       "      <td>ENSEMBL:ENSG00000173213</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:24983</td>\n",
       "      <td>TUBB8B</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11668 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          name     nomenclature  concept_id gene_claim_name  \\\n",
       "25                       HSP90      Gene Symbol   hgnc:5253        HSP90AA1   \n",
       "100                       RPLP      Gene Symbol         NaN             NaN   \n",
       "167                      SEPT6      Gene Symbol  hgnc:15848         SEPTIN6   \n",
       "169                      SEPT5      Gene Symbol   hgnc:9164         SEPTIN5   \n",
       "213                      WISP3      Gene Symbol  hgnc:12771            CCN6   \n",
       "...                        ...              ...         ...             ...   \n",
       "73631  ENSEMBL:ENSG00000185821  Ensembl Gene ID  hgnc:31305          OR6C76   \n",
       "73642                 C11ORF30      Gene Symbol  hgnc:18071            EMSY   \n",
       "73727  ENSEMBL:ENSG00000183921  Ensembl Gene ID  hgnc:35414         SDR42E2   \n",
       "73872                       MR   NCBI Gene Name   hgnc:7979           NR3C2   \n",
       "73878  ENSEMBL:ENSG00000173213  Ensembl Gene ID  hgnc:24983          TUBB8B   \n",
       "\n",
       "                   source_db_name source_db_version  \n",
       "25                         ChEMBL                32  \n",
       "100                        ChEMBL                32  \n",
       "167    CarisMolecularIntelligence          5-Jun-23  \n",
       "169    CarisMolecularIntelligence          5-Jun-23  \n",
       "213    CarisMolecularIntelligence          5-Jun-23  \n",
       "...                           ...               ...  \n",
       "73631                  RussLampel         26-Jul-11  \n",
       "73642          FoundationOneGenes          5-Jun-23  \n",
       "73727                  RussLampel         26-Jul-11  \n",
       "73872                    BaderLab            14-Feb  \n",
       "73878                  RussLampel         26-Jul-11  \n",
       "\n",
       "[11668 rows x 6 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_df.query('name != gene_claim_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>RPLP</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>EMBA</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>RPLQ</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>ILES</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>MURA</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26386</th>\n",
       "      <td>CCNK-CDK13_HUMAN</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26387</th>\n",
       "      <td>INO80_HUMAN-1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26388</th>\n",
       "      <td>CCNC-CDK3_HUMAN</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26573</th>\n",
       "      <td>TRYPTASE_B2_HUMAN</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26574</th>\n",
       "      <td>TRYPTASE_B1_HUMAN</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO</td>\n",
       "      <td>5-Jun-23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2159 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    name nomenclature concept_id gene_claim_name  \\\n",
       "100                 RPLP  Gene Symbol        NaN             NaN   \n",
       "353                 EMBA  Gene Symbol        NaN             NaN   \n",
       "355                 RPLQ  Gene Symbol        NaN             NaN   \n",
       "402                 ILES  Gene Symbol        NaN             NaN   \n",
       "439                 MURA  Gene Symbol        NaN             NaN   \n",
       "...                  ...          ...        ...             ...   \n",
       "26386   CCNK-CDK13_HUMAN  Gene Symbol        NaN             NaN   \n",
       "26387      INO80_HUMAN-1  Gene Symbol        NaN             NaN   \n",
       "26388    CCNC-CDK3_HUMAN  Gene Symbol        NaN             NaN   \n",
       "26573  TRYPTASE_B2_HUMAN  Gene Symbol        NaN             NaN   \n",
       "26574  TRYPTASE_B1_HUMAN  Gene Symbol        NaN             NaN   \n",
       "\n",
       "      source_db_name source_db_version  \n",
       "100           ChEMBL                32  \n",
       "353           ChEMBL                32  \n",
       "355           ChEMBL                32  \n",
       "402           ChEMBL                32  \n",
       "439           ChEMBL                32  \n",
       "...              ...               ...  \n",
       "26386             GO          5-Jun-23  \n",
       "26387             GO          5-Jun-23  \n",
       "26388             GO          5-Jun-23  \n",
       "26573             GO          5-Jun-23  \n",
       "26574             GO          5-Jun-23  \n",
       "\n",
       "[2159 rows x 6 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_claim_symbols_df = dgidb_gene_df[dgidb_gene_df['gene_claim_name'].isnull()]\n",
    "no_claim_symbols_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21619"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_name_set = set(dgidb_gene_df['name'])\n",
    "len(dgidb_name_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11287"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_claim_name_set = set(dgidb_gene_df['gene_claim_name'])\n",
    "len(dgidb_gene_claim_name_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10422"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ensg_notmatch = dgidb_name_set.difference(ensg_gene_symbol_set)\n",
    "len(name_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "107"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_notmatch = dgidb_gene_claim_name_set.difference(ensg_gene_symbol_set)\n",
    "len(gene_claim_name_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "107"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_gene_claim_name_ensg_notmatch = [x for x in gene_claim_name_ensg_notmatch if str(x) != 'NaN']\n",
    "len(cleaned_gene_claim_name_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10374"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_hgnc_notmatch = dgidb_name_set.difference(hgnc_gene_symbol_set)\n",
    "len(name_hgnc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_hngc_notmatch_aacollision = name_hgnc_notmatch.intersection(hgnc_alias_alias_collision_set)\n",
    "len(name_hngc_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10374"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_name_hgnc_notmatch = [x for x in name_hgnc_notmatch if str(x) != 'NaN']\n",
    "len(cleaned_name_hgnc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_notmatch = dgidb_gene_claim_name_set.difference(hgnc_gene_symbol_set)\n",
    "len(gene_claim_name_hgnc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10351"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_notmatch = dgidb_name_set.difference(ncbi_gene_symbol_set)\n",
    "len(name_ncbi_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "96"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_notmatch_aacollision = name_ncbi_notmatch.intersection(ncbi_alias_alias_collision_set)\n",
    "len(name_ncbi_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10335"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_hgnc_notmatch = name_ncbi_notmatch.difference(hgnc_gene_symbol_set)\n",
    "len(name_ncbi_hgnc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10331"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_hgnc_ensg_notmatch = name_ncbi_hgnc_notmatch.difference(ensg_gene_symbol_set)\n",
    "len(name_ncbi_hgnc_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "63"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_notmatch = dgidb_gene_claim_name_set.difference(ncbi_gene_symbol_set)\n",
    "len(gene_claim_name_ncbi_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "47"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_hngc_notmatch = gene_claim_name_ncbi_notmatch.difference(hgnc_gene_symbol_set)\n",
    "len(gene_claim_name_ncbi_hngc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_hngc_ensg_notmatch = gene_claim_name_ncbi_hngc_notmatch.difference(ensg_gene_symbol_set)\n",
    "len(gene_claim_name_ncbi_hngc_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11245"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_hgnc_match = dgidb_name_set.intersection(hgnc_gene_symbol_set)\n",
    "len(name_hgnc_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_hgnc_match_aacollision = name_hgnc_match.intersection(hgnc_alias_alias_collision_set)\n",
    "len(name_hgnc_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11197"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ensg_match = dgidb_name_set.intersection(ensg_gene_symbol_set)\n",
    "len(name_ensg_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11268"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_match = dgidb_name_set.intersection(ncbi_gene_symbol_set)\n",
    "len(name_ncbi_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "112"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_match_aacollision = name_ncbi_match.intersection(ncbi_alias_alias_collision_set)\n",
    "len(name_ncbi_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11178"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_ensg_match = name_ncbi_match.intersection(ensg_gene_symbol_set)\n",
    "len(name_ncbi_ensg_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11176"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_ensg_hgnc_match = name_ncbi_ensg_match.intersection(hgnc_gene_symbol_set)\n",
    "len(name_ncbi_ensg_hgnc_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11221"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_match = dgidb_gene_claim_name_set.intersection(hgnc_gene_symbol_set)\n",
    "len(gene_claim_name_hgnc_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11180"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_match = dgidb_gene_claim_name_set.intersection(ensg_gene_symbol_set)\n",
    "len(gene_claim_name_ensg_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_aacollision_match = dgidb_gene_claim_name_set.intersection(ensg_alias_alias_collision_set)\n",
    "len(gene_claim_name_ensg_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_aacollision_match = dgidb_gene_claim_name_set.intersection(hgnc_alias_alias_collision_set)\n",
    "len(gene_claim_name_hgnc_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "113"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_aacollision_match = dgidb_gene_claim_name_set.intersection(ncbi_alias_alias_collision_set)\n",
    "len(gene_claim_name_ncbi_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ensg_aacollision_match = dgidb_name_set.intersection(ensg_alias_alias_collision_set)\n",
    "len(name_ensg_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "58"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_hgnc_aacollision_match = dgidb_name_set.intersection(hgnc_alias_alias_collision_set)\n",
    "len(name_hgnc_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "208"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ncbi_aacollision_match = dgidb_name_set.intersection(ncbi_alias_alias_collision_set)\n",
    "len(name_ncbi_aacollision_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "66"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_notmatch = dgidb_gene_claim_name_set.difference(hgnc_gene_symbol_set)\n",
    "len(gene_claim_name_hgnc_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_notmatch_aacollision = gene_claim_name_hgnc_notmatch.intersection(hgnc_alias_alias_collision_set)\n",
    "len(gene_claim_name_hgnc_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "107"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_notmatch = dgidb_gene_claim_name_set.difference(ensg_gene_symbol_set)\n",
    "len(gene_claim_name_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "63"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_notmatch = dgidb_gene_claim_name_set.difference(ncbi_gene_symbol_set)\n",
    "len(gene_claim_name_ncbi_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_notmatch_aacollision = gene_claim_name_ncbi_notmatch.intersection(ncbi_alias_alias_collision_set)\n",
    "len(gene_claim_name_ncbi_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11221"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_match = dgidb_gene_claim_name_set.intersection(hgnc_gene_symbol_set)\n",
    "len(gene_claim_name_hgnc_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_match_aacollision = gene_claim_name_hgnc_match.intersection(hgnc_alias_alias_collision_set)\n",
    "len(gene_claim_name_hgnc_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11224"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_match = dgidb_gene_claim_name_set.intersection(ncbi_gene_symbol_set)\n",
    "len(gene_claim_name_ncbi_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "112"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_match_aacollision = gene_claim_name_ncbi_match.intersection(ncbi_alias_alias_collision_set)\n",
    "len(gene_claim_name_ncbi_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ensg_match_aacollision = name_ensg_match.intersection(ensg_alias_alias_collision_set)\n",
    "len(name_ensg_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44"
      ]
     },
     "execution_count": 171,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_claim_name_ncbi_hngc_ensg_notmatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10331"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(name_ncbi_hgnc_ensg_notmatch)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21619"
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dgidb_name_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11287"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(dgidb_gene_claim_name_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_ensg_notmatch_aacollision = name_ensg_notmatch.intersection(ensg_alias_alias_collision_set)\n",
    "len(name_ensg_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_match_aacollision = gene_claim_name_ensg_match.intersection(ensg_alias_alias_collision_set)\n",
    "len(gene_claim_name_ensg_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ensg_notmatch_aacollision = gene_claim_name_ensg_notmatch.intersection(ensg_alias_alias_collision_set)\n",
    "len(gene_claim_name_ensg_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_notmatch_aacollision = gene_claim_name_ncbi_notmatch.intersection(ncbi_alias_alias_collision_set)\n",
    "len(gene_claim_name_ncbi_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "112"
      ]
     },
     "execution_count": 179,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_ncbi_match_aacollision = gene_claim_name_ncbi_match.intersection(ncbi_alias_alias_collision_set)\n",
    "len(gene_claim_name_ncbi_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_match_aacollision = gene_claim_name_hgnc_match.intersection(hgnc_alias_alias_collision_set)\n",
    "len(gene_claim_name_hgnc_match_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_claim_name_hgnc_notmatch_aacollision = gene_claim_name_hgnc_notmatch.intersection(hgnc_alias_alias_collision_set)\n",
    "len(gene_claim_name_hgnc_notmatch_aacollision)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pull out instances of claim symbols that match to a primary gene symbol and the corresponding group symbols not matching to a primary gene symbol. Check for patterns of modes of error\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "      <th>hgnc_claim_match_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MN1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:7180</td>\n",
       "      <td>MN1</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P2RX7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:8537</td>\n",
       "      <td>P2RX7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:1960</td>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAPK8</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6881</td>\n",
       "      <td>MAPK8</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAPK10</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6872</td>\n",
       "      <td>MAPK10</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73884</th>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:6889</td>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73885</th>\n",
       "      <td>PTGER3</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:9595</td>\n",
       "      <td>PTGER3</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73886</th>\n",
       "      <td>CD1D</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:1637</td>\n",
       "      <td>CD1D</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73887</th>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:10993</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>IDG</td>\n",
       "      <td>15-Jul-19</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73888</th>\n",
       "      <td>SOCS1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:19383</td>\n",
       "      <td>SOCS1</td>\n",
       "      <td>FoundationOneGenes</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73889 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name nomenclature  concept_id gene_claim_name  \\\n",
       "0           MN1  Gene Symbol   hgnc:7180             MN1   \n",
       "1         P2RX7  Gene Symbol   hgnc:8537           P2RX7   \n",
       "2        CHRNA7  Gene Symbol   hgnc:1960          CHRNA7   \n",
       "3         MAPK8  Gene Symbol   hgnc:6881           MAPK8   \n",
       "4        MAPK10  Gene Symbol   hgnc:6872          MAPK10   \n",
       "...         ...          ...         ...             ...   \n",
       "73884  MAPKAPK5    Gene Name   hgnc:6889        MAPKAPK5   \n",
       "73885    PTGER3    Gene Name   hgnc:9595          PTGER3   \n",
       "73886      CD1D    Gene Name   hgnc:1637            CD1D   \n",
       "73887   SLC26A1  Gene Symbol  hgnc:10993         SLC26A1   \n",
       "73888     SOCS1  Gene Symbol  hgnc:19383           SOCS1   \n",
       "\n",
       "                   source_db_name source_db_version  hgnc_claim_match_status  \n",
       "0      CarisMolecularIntelligence          5-Jun-23                     True  \n",
       "1                          ChEMBL                32                     True  \n",
       "2                          ChEMBL                32                     True  \n",
       "3                          ChEMBL                32                     True  \n",
       "4                          ChEMBL                32                     True  \n",
       "...                           ...               ...                      ...  \n",
       "73884                         DTC          5-Jun-23                     True  \n",
       "73885                         DTC          5-Jun-23                     True  \n",
       "73886                         DTC          5-Jun-23                     True  \n",
       "73887                         IDG         15-Jul-19                     True  \n",
       "73888          FoundationOneGenes          5-Jun-23                     True  \n",
       "\n",
       "[73889 rows x 7 columns]"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_df['hgnc_claim_match_status'] = dgidb_gene_df['gene_claim_name'].isin(hgnc_gene_symbol_set)\n",
    "dgidb_gene_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "      <th>hgnc_claim_match_status</th>\n",
       "      <th>hgnc_name_match_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MN1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:7180</td>\n",
       "      <td>MN1</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P2RX7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:8537</td>\n",
       "      <td>P2RX7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:1960</td>\n",
       "      <td>CHRNA7</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MAPK8</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6881</td>\n",
       "      <td>MAPK8</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MAPK10</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:6872</td>\n",
       "      <td>MAPK10</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73884</th>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:6889</td>\n",
       "      <td>MAPKAPK5</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73885</th>\n",
       "      <td>PTGER3</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:9595</td>\n",
       "      <td>PTGER3</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73886</th>\n",
       "      <td>CD1D</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:1637</td>\n",
       "      <td>CD1D</td>\n",
       "      <td>DTC</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73887</th>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:10993</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>IDG</td>\n",
       "      <td>15-Jul-19</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73888</th>\n",
       "      <td>SOCS1</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:19383</td>\n",
       "      <td>SOCS1</td>\n",
       "      <td>FoundationOneGenes</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73889 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           name nomenclature  concept_id gene_claim_name  \\\n",
       "0           MN1  Gene Symbol   hgnc:7180             MN1   \n",
       "1         P2RX7  Gene Symbol   hgnc:8537           P2RX7   \n",
       "2        CHRNA7  Gene Symbol   hgnc:1960          CHRNA7   \n",
       "3         MAPK8  Gene Symbol   hgnc:6881           MAPK8   \n",
       "4        MAPK10  Gene Symbol   hgnc:6872          MAPK10   \n",
       "...         ...          ...         ...             ...   \n",
       "73884  MAPKAPK5    Gene Name   hgnc:6889        MAPKAPK5   \n",
       "73885    PTGER3    Gene Name   hgnc:9595          PTGER3   \n",
       "73886      CD1D    Gene Name   hgnc:1637            CD1D   \n",
       "73887   SLC26A1  Gene Symbol  hgnc:10993         SLC26A1   \n",
       "73888     SOCS1  Gene Symbol  hgnc:19383           SOCS1   \n",
       "\n",
       "                   source_db_name source_db_version  hgnc_claim_match_status  \\\n",
       "0      CarisMolecularIntelligence          5-Jun-23                     True   \n",
       "1                          ChEMBL                32                     True   \n",
       "2                          ChEMBL                32                     True   \n",
       "3                          ChEMBL                32                     True   \n",
       "4                          ChEMBL                32                     True   \n",
       "...                           ...               ...                      ...   \n",
       "73884                         DTC          5-Jun-23                     True   \n",
       "73885                         DTC          5-Jun-23                     True   \n",
       "73886                         DTC          5-Jun-23                     True   \n",
       "73887                         IDG         15-Jul-19                     True   \n",
       "73888          FoundationOneGenes          5-Jun-23                     True   \n",
       "\n",
       "       hgnc_name_match_status  \n",
       "0                        True  \n",
       "1                        True  \n",
       "2                        True  \n",
       "3                        True  \n",
       "4                        True  \n",
       "...                       ...  \n",
       "73884                    True  \n",
       "73885                    True  \n",
       "73886                    True  \n",
       "73887                    True  \n",
       "73888                    True  \n",
       "\n",
       "[73889 rows x 8 columns]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgidb_gene_df['hgnc_name_match_status'] = dgidb_gene_df['name'].isin(hgnc_gene_symbol_set)\n",
    "dgidb_gene_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>nomenclature</th>\n",
       "      <th>concept_id</th>\n",
       "      <th>gene_claim_name</th>\n",
       "      <th>source_db_name</th>\n",
       "      <th>source_db_version</th>\n",
       "      <th>hgnc_claim_match_status</th>\n",
       "      <th>hgnc_name_match_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>HSP90</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:5253</td>\n",
       "      <td>HSP90AA1</td>\n",
       "      <td>ChEMBL</td>\n",
       "      <td>32</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>SEPT6</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:15848</td>\n",
       "      <td>SEPTIN6</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>SEPT5</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:9164</td>\n",
       "      <td>SEPTIN5</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>WISP3</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:12771</td>\n",
       "      <td>CCN6</td>\n",
       "      <td>CarisMolecularIntelligence</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>MLL2</td>\n",
       "      <td>Gene Name</td>\n",
       "      <td>hgnc:7133</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>CGI</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73631</th>\n",
       "      <td>ENSEMBL:ENSG00000185821</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:31305</td>\n",
       "      <td>OR6C76</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73642</th>\n",
       "      <td>C11ORF30</td>\n",
       "      <td>Gene Symbol</td>\n",
       "      <td>hgnc:18071</td>\n",
       "      <td>EMSY</td>\n",
       "      <td>FoundationOneGenes</td>\n",
       "      <td>5-Jun-23</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73727</th>\n",
       "      <td>ENSEMBL:ENSG00000183921</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:35414</td>\n",
       "      <td>SDR42E2</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73872</th>\n",
       "      <td>MR</td>\n",
       "      <td>NCBI Gene Name</td>\n",
       "      <td>hgnc:7979</td>\n",
       "      <td>NR3C2</td>\n",
       "      <td>BaderLab</td>\n",
       "      <td>14-Feb</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73878</th>\n",
       "      <td>ENSEMBL:ENSG00000173213</td>\n",
       "      <td>Ensembl Gene ID</td>\n",
       "      <td>hgnc:24983</td>\n",
       "      <td>TUBB8B</td>\n",
       "      <td>RussLampel</td>\n",
       "      <td>26-Jul-11</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9378 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          name     nomenclature  concept_id gene_claim_name  \\\n",
       "25                       HSP90      Gene Symbol   hgnc:5253        HSP90AA1   \n",
       "167                      SEPT6      Gene Symbol  hgnc:15848         SEPTIN6   \n",
       "169                      SEPT5      Gene Symbol   hgnc:9164         SEPTIN5   \n",
       "213                      WISP3      Gene Symbol  hgnc:12771            CCN6   \n",
       "294                       MLL2        Gene Name   hgnc:7133           KMT2D   \n",
       "...                        ...              ...         ...             ...   \n",
       "73631  ENSEMBL:ENSG00000185821  Ensembl Gene ID  hgnc:31305          OR6C76   \n",
       "73642                 C11ORF30      Gene Symbol  hgnc:18071            EMSY   \n",
       "73727  ENSEMBL:ENSG00000183921  Ensembl Gene ID  hgnc:35414         SDR42E2   \n",
       "73872                       MR   NCBI Gene Name   hgnc:7979           NR3C2   \n",
       "73878  ENSEMBL:ENSG00000173213  Ensembl Gene ID  hgnc:24983          TUBB8B   \n",
       "\n",
       "                   source_db_name source_db_version  hgnc_claim_match_status  \\\n",
       "25                         ChEMBL                32                     True   \n",
       "167    CarisMolecularIntelligence          5-Jun-23                     True   \n",
       "169    CarisMolecularIntelligence          5-Jun-23                     True   \n",
       "213    CarisMolecularIntelligence          5-Jun-23                     True   \n",
       "294                           CGI          5-Jun-23                     True   \n",
       "...                           ...               ...                      ...   \n",
       "73631                  RussLampel         26-Jul-11                     True   \n",
       "73642          FoundationOneGenes          5-Jun-23                     True   \n",
       "73727                  RussLampel         26-Jul-11                     True   \n",
       "73872                    BaderLab            14-Feb                     True   \n",
       "73878                  RussLampel         26-Jul-11                     True   \n",
       "\n",
       "       hgnc_name_match_status  \n",
       "25                      False  \n",
       "167                     False  \n",
       "169                     False  \n",
       "213                     False  \n",
       "294                     False  \n",
       "...                       ...  \n",
       "73631                   False  \n",
       "73642                   False  \n",
       "73727                   False  \n",
       "73872                   False  \n",
       "73878                   False  \n",
       "\n",
       "[9378 rows x 8 columns]"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "claim_true_name_false_df = dgidb_gene_df.loc[dgidb_gene_df['hgnc_claim_match_status'] & ~dgidb_gene_df['hgnc_name_match_status']]\n",
    "claim_true_name_false_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
