{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Expanding collision acronyms\n",
    "One problem for Aim 1 in Anastasia's dissertation is that once a gene-alias collision pair has been identified, we have to determine what the collision symbol actually represents in the context of the parent symbol. For example, CAP is listed as an alias for BRD4 but this symbol collides with so many other CAP aliases. In the context of BRD4, CAP actually refers to 'chromosome associated protein'. What cap stands for differs across different parent gene symbols. While this can be manually curated for a small set of genes, there exist over 100,000 gene-alias pairs to consider and so a programmatic approach will be needed. Additionally, a separate but related problem will be to programmatically identify the type of collision(?)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Scratch Testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'gene_symbols' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[102], line 5\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mtqdm\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mnotebook\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m tqdm\n\u001b[1;32m      3\u001b[0m i \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m----> 5\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m uniprot \u001b[38;5;129;01min\u001b[39;00m tqdm(\u001b[43mgene_symbols\u001b[49m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUniProt ID\u001b[39m\u001b[38;5;124m'\u001b[39m]):\n\u001b[1;32m      6\u001b[0m     r \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mhttps://rest.uniprot.org/uniprotkb/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00muniprot\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m      7\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m r\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m200\u001b[39m:\n",
      "\u001b[0;31mNameError\u001b[0m: name 'gene_symbols' is not defined"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from tqdm.notebook import tqdm\n",
    "i = 0\n",
    "\n",
    "for uniprot in tqdm(gene_symbols['UniProt ID']):\n",
    "    r = requests.get(f'https://rest.uniprot.org/uniprotkb/{uniprot}')\n",
    "    if r.status_code == 200:\n",
    "        try:\n",
    "            symbol = r.json()['genes'][0]['geneName']['value']\n",
    "        except:\n",
    "            symbol = 'Error'\n",
    "        # print(r.json()['genes'][0]['geneName']['value'])\n",
    "    else:\n",
    "        symbol = None\n",
    "    gene_symbols['symbol'][i] = symbol\n",
    "    i += 1\n",
    "\n",
    "gene_symbols\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'entryType': 'UniProtKB reviewed (Swiss-Prot)',\n",
       " 'primaryAccession': 'Q53GG5',\n",
       " 'secondaryAccessions': ['B2R866',\n",
       "  'O43590',\n",
       "  'O60439',\n",
       "  'O60440',\n",
       "  'Q8N6Y6',\n",
       "  'Q9BVP4'],\n",
       " 'uniProtkbId': 'PDLI3_HUMAN',\n",
       " 'entryAudit': {'firstPublicDate': '2005-12-06',\n",
       "  'lastAnnotationUpdateDate': '2024-07-24',\n",
       "  'lastSequenceUpdateDate': '2005-05-24',\n",
       "  'entryVersion': 164,\n",
       "  'sequenceVersion': 1},\n",
       " 'annotationScore': 5.0,\n",
       " 'organism': {'scientificName': 'Homo sapiens',\n",
       "  'commonName': 'Human',\n",
       "  'taxonId': 9606,\n",
       "  'lineage': ['Eukaryota',\n",
       "   'Metazoa',\n",
       "   'Chordata',\n",
       "   'Craniata',\n",
       "   'Vertebrata',\n",
       "   'Euteleostomi',\n",
       "   'Mammalia',\n",
       "   'Eutheria',\n",
       "   'Euarchontoglires',\n",
       "   'Primates',\n",
       "   'Haplorrhini',\n",
       "   'Catarrhini',\n",
       "   'Hominidae',\n",
       "   'Homo']},\n",
       " 'proteinExistence': '1: Evidence at protein level',\n",
       " 'proteinDescription': {'recommendedName': {'fullName': {'value': 'PDZ and LIM domain protein 3'}},\n",
       "  'alternativeNames': [{'fullName': {'value': 'Actinin-associated LIM protein'}},\n",
       "   {'fullName': {'value': 'Alpha-actinin-2-associated LIM protein'}}]},\n",
       " 'genes': [{'geneName': {'value': 'PDLIM3'}, 'synonyms': [{'value': 'ALP'}]}],\n",
       " 'comments': [{'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250'}],\n",
       "     'value': 'May play a role in the organization of actin filament arrays within muscle cells'}],\n",
       "   'commentType': 'FUNCTION'},\n",
       "  {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000250',\n",
       "       'source': 'UniProtKB',\n",
       "       'id': 'O70209'},\n",
       "      {'evidenceCode': 'ECO:0000250', 'source': 'UniProtKB', 'id': 'Q66HS7'}],\n",
       "     'value': 'Interacts with ACTN2 (By similarity). Forms a heterodimer with PDLIM4 (via LIM domain) (By similarity)'}],\n",
       "   'commentType': 'SUBUNIT'},\n",
       "  {'commentType': 'INTERACTION',\n",
       "   'interactions': [{'interactantOne': {'uniProtKBAccession': 'Q53GG5',\n",
       "      'intActId': 'EBI-5658852'},\n",
       "     'interactantTwo': {'uniProtKBAccession': 'P35609',\n",
       "      'geneName': 'ACTN2',\n",
       "      'intActId': 'EBI-77797'},\n",
       "     'numberOfExperiments': 6,\n",
       "     'organismDiffer': False},\n",
       "    {'interactantOne': {'uniProtKBAccession': 'Q53GG5',\n",
       "      'intActId': 'EBI-5658852'},\n",
       "     'interactantTwo': {'uniProtKBAccession': 'P49792',\n",
       "      'geneName': 'RANBP2',\n",
       "      'intActId': 'EBI-973138'},\n",
       "     'numberOfExperiments': 3,\n",
       "     'organismDiffer': False},\n",
       "    {'interactantOne': {'uniProtKBAccession': 'Q53GG5-2',\n",
       "      'intActId': 'EBI-12702438'},\n",
       "     'interactantTwo': {'uniProtKBAccession': 'P35609',\n",
       "      'geneName': 'ACTN2',\n",
       "      'intActId': 'EBI-77797'},\n",
       "     'numberOfExperiments': 3,\n",
       "     'organismDiffer': False}]},\n",
       "  {'commentType': 'SUBCELLULAR LOCATION',\n",
       "   'note': {'texts': [{'value': 'Localizes to myofiber Z-lines'}]},\n",
       "   'subcellularLocations': [{'location': {'evidences': [{'evidenceCode': 'ECO:0000269',\n",
       "        'source': 'PubMed',\n",
       "        'id': '10063829'}],\n",
       "      'value': 'Cytoplasm, myofibril, sarcomere, Z line',\n",
       "      'id': 'SL-0314'}}]},\n",
       "  {'commentType': 'ALTERNATIVE PRODUCTS',\n",
       "   'events': ['Alternative splicing'],\n",
       "   'isoforms': [{'name': {'value': '1'},\n",
       "     'synonyms': [{'value': 'ALP-SK'}],\n",
       "     'isoformIds': ['Q53GG5-1'],\n",
       "     'isoformSequenceStatus': 'Displayed'},\n",
       "    {'name': {'value': '2'},\n",
       "     'synonyms': [{'value': 'ALP-H'}],\n",
       "     'isoformIds': ['Q53GG5-2'],\n",
       "     'sequenceIds': ['VSP_016501'],\n",
       "     'isoformSequenceStatus': 'Described'},\n",
       "    {'name': {'value': '3'},\n",
       "     'isoformIds': ['Q53GG5-3'],\n",
       "     'sequenceIds': ['VSP_016500', 'VSP_016502'],\n",
       "     'isoformSequenceStatus': 'Described'}]},\n",
       "  {'texts': [{'evidences': [{'evidenceCode': 'ECO:0000269',\n",
       "       'source': 'PubMed',\n",
       "       'id': '9334352'}],\n",
       "     'value': 'Isoform 1 is highly expressed in differentiated skeletal muscle. Isoform 2 is heart-specific'}],\n",
       "   'commentType': 'TISSUE SPECIFICITY'}],\n",
       " 'features': [{'type': 'Chain',\n",
       "   'location': {'start': {'value': 1, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 364, 'modifier': 'EXACT'}},\n",
       "   'description': 'PDZ and LIM domain protein 3',\n",
       "   'featureId': 'PRO_0000075867'},\n",
       "  {'type': 'Domain',\n",
       "   'location': {'start': {'value': 1, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 84, 'modifier': 'EXACT'}},\n",
       "   'description': 'PDZ',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000255',\n",
       "     'source': 'PROSITE-ProRule',\n",
       "     'id': 'PRU00143'}]},\n",
       "  {'type': 'Domain',\n",
       "   'location': {'start': {'value': 292, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 351, 'modifier': 'EXACT'}},\n",
       "   'description': 'LIM zinc-binding',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000255',\n",
       "     'source': 'PROSITE-ProRule',\n",
       "     'id': 'PRU00125'}]},\n",
       "  {'type': 'Modified residue',\n",
       "   'location': {'start': {'value': 18, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 18, 'modifier': 'EXACT'}},\n",
       "   'description': 'Phosphoserine',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000250',\n",
       "     'source': 'UniProtKB',\n",
       "     'id': 'Q66HS7'}]},\n",
       "  {'type': 'Modified residue',\n",
       "   'location': {'start': {'value': 93, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 93, 'modifier': 'EXACT'}},\n",
       "   'description': 'Phosphoserine',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0007744',\n",
       "     'source': 'PubMed',\n",
       "     'id': '24275569'}]},\n",
       "  {'type': 'Modified residue',\n",
       "   'location': {'start': {'value': 264, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 264, 'modifier': 'EXACT'}},\n",
       "   'description': 'Phosphoserine',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000250',\n",
       "     'source': 'UniProtKB',\n",
       "     'id': 'Q66HS7'}]},\n",
       "  {'type': 'Alternative sequence',\n",
       "   'location': {'start': {'value': 111, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 233, 'modifier': 'EXACT'}},\n",
       "   'description': 'in isoform 3',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000303',\n",
       "     'source': 'PubMed',\n",
       "     'id': '15489334'},\n",
       "    {'evidenceCode': 'ECO:0000303', 'source': 'Reference', 'id': 'Ref.4'}],\n",
       "   'featureId': 'VSP_016500',\n",
       "   'alternativeSequence': {'originalSequence': 'DGNYFEHKHNIRPKPFVIPGRSSGCSTPSGIDCGSGRSTPSSVSTVSTICPGDLKVAAKLAPNIPLEMELPGVKIVHAQFNTPMQLYSDDNIMETLQGQVSTALGETPLMSEPTASVPPESDV',\n",
       "    'alternativeSequences': ['EFKPIGTAHNRRAQPFVAAANIDDKRQVVSASYNSPIGLYSTSNIQDALHGQLRGLIPSSPQKTGTTLNTSIIFGPNLS']}},\n",
       "  {'type': 'Alternative sequence',\n",
       "   'location': {'start': {'value': 111, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 224, 'modifier': 'EXACT'}},\n",
       "   'description': 'in isoform 2',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000303',\n",
       "     'source': 'PubMed',\n",
       "     'id': '15489334'},\n",
       "    {'evidenceCode': 'ECO:0000303', 'source': 'PubMed', 'id': '9334352'}],\n",
       "   'featureId': 'VSP_016501',\n",
       "   'alternativeSequence': {'originalSequence': 'DGNYFEHKHNIRPKPFVIPGRSSGCSTPSGIDCGSGRSTPSSVSTVSTICPGDLKVAAKLAPNIPLEMELPGVKIVHAQFNTPMQLYSDDNIMETLQGQVSTALGETPLMSEPT',\n",
       "    'alternativeSequences': ['EFKPIGTAHNRRAQPFVAAANIDDKRQVVSASYNSPIGLYSTSNIQDALHGQLRGLIPSSPQNEPT']}},\n",
       "  {'type': 'Alternative sequence',\n",
       "   'location': {'start': {'value': 234, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 364, 'modifier': 'EXACT'}},\n",
       "   'description': 'in isoform 3',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000303',\n",
       "     'source': 'PubMed',\n",
       "     'id': '15489334'},\n",
       "    {'evidenceCode': 'ECO:0000303', 'source': 'Reference', 'id': 'Ref.4'}],\n",
       "   'featureId': 'VSP_016502',\n",
       "   'alternativeSequence': {}},\n",
       "  {'type': 'Natural variant',\n",
       "   'location': {'start': {'value': 127, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 127, 'modifier': 'EXACT'}},\n",
       "   'description': 'in dbSNP:rs11944325',\n",
       "   'featureCrossReferences': [{'database': 'dbSNP', 'id': 'rs11944325'}],\n",
       "   'featureId': 'VAR_050166',\n",
       "   'alternativeSequence': {'originalSequence': 'V',\n",
       "    'alternativeSequences': ['M']}},\n",
       "  {'type': 'Sequence conflict',\n",
       "   'location': {'start': {'value': 12, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 12, 'modifier': 'EXACT'}},\n",
       "   'description': 'in Ref. 1; AAC16670/AAC16672',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000305'}],\n",
       "   'alternativeSequence': {'originalSequence': 'P',\n",
       "    'alternativeSequences': ['A']}},\n",
       "  {'type': 'Sequence conflict',\n",
       "   'location': {'start': {'value': 40, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 40, 'modifier': 'EXACT'}},\n",
       "   'description': 'in Ref. 1; AAC16670/AAC16672',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000305'}],\n",
       "   'alternativeSequence': {'originalSequence': 'A',\n",
       "    'alternativeSequences': ['G']}},\n",
       "  {'type': 'Sequence conflict',\n",
       "   'location': {'start': {'value': 217, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 217, 'modifier': 'EXACT'}},\n",
       "   'description': 'in Ref. 1; AAC16670',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000305'}],\n",
       "   'alternativeSequence': {'originalSequence': 'T',\n",
       "    'alternativeSequences': ['I']}},\n",
       "  {'type': 'Sequence conflict',\n",
       "   'location': {'start': {'value': 221, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 221, 'modifier': 'EXACT'}},\n",
       "   'description': 'in Ref. 2; AAB96665',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000305'}],\n",
       "   'alternativeSequence': {'originalSequence': 'S',\n",
       "    'alternativeSequences': ['N']}},\n",
       "  {'type': 'Sequence conflict',\n",
       "   'location': {'start': {'value': 353, 'modifier': 'EXACT'},\n",
       "    'end': {'value': 353, 'modifier': 'EXACT'}},\n",
       "   'description': 'in Ref. 1; AAC16670/AAC16672',\n",
       "   'evidences': [{'evidenceCode': 'ECO:0000305'}],\n",
       "   'alternativeSequence': {'originalSequence': 'E',\n",
       "    'alternativeSequences': ['R']}}],\n",
       " 'keywords': [{'id': 'KW-0025',\n",
       "   'category': 'Coding sequence diversity',\n",
       "   'name': 'Alternative splicing'},\n",
       "  {'id': 'KW-0963', 'category': 'Cellular component', 'name': 'Cytoplasm'},\n",
       "  {'id': 'KW-0440', 'category': 'Domain', 'name': 'LIM domain'},\n",
       "  {'id': 'KW-0479', 'category': 'Ligand', 'name': 'Metal-binding'},\n",
       "  {'id': 'KW-0597', 'category': 'PTM', 'name': 'Phosphoprotein'},\n",
       "  {'id': 'KW-1185',\n",
       "   'category': 'Technical term',\n",
       "   'name': 'Reference proteome'},\n",
       "  {'id': 'KW-0862', 'category': 'Ligand', 'name': 'Zinc'}],\n",
       " 'references': [{'referenceNumber': 1,\n",
       "   'citation': {'id': '9334352',\n",
       "    'citationType': 'journal article',\n",
       "    'authors': ['Xia H.',\n",
       "     'Winokur S.T.',\n",
       "     'Kuo W.-L.',\n",
       "     'Altherr M.R.',\n",
       "     'Bredt D.S.'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '9334352'},\n",
       "     {'database': 'DOI', 'id': '10.1083/jcb.139.2.507'}],\n",
       "    'title': 'Actinin-associated LIM protein: identification of a domain interaction between PDZ and spectrin-like repeat motifs.',\n",
       "    'publicationDate': '1997',\n",
       "    'journal': 'J. Cell Biol.',\n",
       "    'firstPage': '507',\n",
       "    'lastPage': '515',\n",
       "    'volume': '139'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [MRNA] (ISOFORMS 1 AND 2)',\n",
       "    'ALTERNATIVE SPLICING',\n",
       "    'TISSUE SPECIFICITY'],\n",
       "   'referenceComments': [{'value': 'Skeletal muscle', 'type': 'TISSUE'}]},\n",
       "  {'referenceNumber': 2,\n",
       "   'citation': {'id': '10063829',\n",
       "    'citationType': 'journal article',\n",
       "    'authors': ['Bouju S.',\n",
       "     'Pietu G.',\n",
       "     'Le Cunff M.',\n",
       "     'Cros N.',\n",
       "     'Malzac P.',\n",
       "     'Pellissier J.-F.',\n",
       "     'Pons F.',\n",
       "     'Leger J.-J.',\n",
       "     'Auffray C.',\n",
       "     'Dechesne C.A.'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '10063829'},\n",
       "     {'database': 'DOI', 'id': '10.1016/s0960-8966(98)00087-x'}],\n",
       "    'title': 'Exclusion of muscle specific actinin-associated LIM protein (ALP) gene from 4q35 facioscapulohumeral muscular dystrophy (FSHD) candidate genes.',\n",
       "    'publicationDate': '1999',\n",
       "    'journal': 'Neuromuscul. Disord.',\n",
       "    'firstPage': '3',\n",
       "    'lastPage': '10',\n",
       "    'volume': '9'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [MRNA] (ISOFORM 1)',\n",
       "    'SUBCELLULAR LOCATION']},\n",
       "  {'referenceNumber': 3,\n",
       "   'citation': {'id': '14702039',\n",
       "    'citationType': 'journal article',\n",
       "    'authors': ['Ota T.',\n",
       "     'Suzuki Y.',\n",
       "     'Nishikawa T.',\n",
       "     'Otsuki T.',\n",
       "     'Sugiyama T.',\n",
       "     'Irie R.',\n",
       "     'Wakamatsu A.',\n",
       "     'Hayashi K.',\n",
       "     'Sato H.',\n",
       "     'Nagai K.',\n",
       "     'Kimura K.',\n",
       "     'Makita H.',\n",
       "     'Sekine M.',\n",
       "     'Obayashi M.',\n",
       "     'Nishi T.',\n",
       "     'Shibahara T.',\n",
       "     'Tanaka T.',\n",
       "     'Ishii S.',\n",
       "     'Yamamoto J.',\n",
       "     'Saito K.',\n",
       "     'Kawai Y.',\n",
       "     'Isono Y.',\n",
       "     'Nakamura Y.',\n",
       "     'Nagahari K.',\n",
       "     'Murakami K.',\n",
       "     'Yasuda T.',\n",
       "     'Iwayanagi T.',\n",
       "     'Wagatsuma M.',\n",
       "     'Shiratori A.',\n",
       "     'Sudo H.',\n",
       "     'Hosoiri T.',\n",
       "     'Kaku Y.',\n",
       "     'Kodaira H.',\n",
       "     'Kondo H.',\n",
       "     'Sugawara M.',\n",
       "     'Takahashi M.',\n",
       "     'Kanda K.',\n",
       "     'Yokoi T.',\n",
       "     'Furuya T.',\n",
       "     'Kikkawa E.',\n",
       "     'Omura Y.',\n",
       "     'Abe K.',\n",
       "     'Kamihara K.',\n",
       "     'Katsuta N.',\n",
       "     'Sato K.',\n",
       "     'Tanikawa M.',\n",
       "     'Yamazaki M.',\n",
       "     'Ninomiya K.',\n",
       "     'Ishibashi T.',\n",
       "     'Yamashita H.',\n",
       "     'Murakawa K.',\n",
       "     'Fujimori K.',\n",
       "     'Tanai H.',\n",
       "     'Kimata M.',\n",
       "     'Watanabe M.',\n",
       "     'Hiraoka S.',\n",
       "     'Chiba Y.',\n",
       "     'Ishida S.',\n",
       "     'Ono Y.',\n",
       "     'Takiguchi S.',\n",
       "     'Watanabe S.',\n",
       "     'Yosida M.',\n",
       "     'Hotuta T.',\n",
       "     'Kusano J.',\n",
       "     'Kanehori K.',\n",
       "     'Takahashi-Fujii A.',\n",
       "     'Hara H.',\n",
       "     'Tanase T.-O.',\n",
       "     'Nomura Y.',\n",
       "     'Togiya S.',\n",
       "     'Komai F.',\n",
       "     'Hara R.',\n",
       "     'Takeuchi K.',\n",
       "     'Arita M.',\n",
       "     'Imose N.',\n",
       "     'Musashino K.',\n",
       "     'Yuuki H.',\n",
       "     'Oshima A.',\n",
       "     'Sasaki N.',\n",
       "     'Aotsuka S.',\n",
       "     'Yoshikawa Y.',\n",
       "     'Matsunawa H.',\n",
       "     'Ichihara T.',\n",
       "     'Shiohata N.',\n",
       "     'Sano S.',\n",
       "     'Moriya S.',\n",
       "     'Momiyama H.',\n",
       "     'Satoh N.',\n",
       "     'Takami S.',\n",
       "     'Terashima Y.',\n",
       "     'Suzuki O.',\n",
       "     'Nakagawa S.',\n",
       "     'Senoh A.',\n",
       "     'Mizoguchi H.',\n",
       "     'Goto Y.',\n",
       "     'Shimizu F.',\n",
       "     'Wakebe H.',\n",
       "     'Hishigaki H.',\n",
       "     'Watanabe T.',\n",
       "     'Sugiyama A.',\n",
       "     'Takemoto M.',\n",
       "     'Kawakami B.',\n",
       "     'Yamazaki M.',\n",
       "     'Watanabe K.',\n",
       "     'Kumagai A.',\n",
       "     'Itakura S.',\n",
       "     'Fukuzumi Y.',\n",
       "     'Fujimori Y.',\n",
       "     'Komiyama M.',\n",
       "     'Tashiro H.',\n",
       "     'Tanigami A.',\n",
       "     'Fujiwara T.',\n",
       "     'Ono T.',\n",
       "     'Yamada K.',\n",
       "     'Fujii Y.',\n",
       "     'Ozaki K.',\n",
       "     'Hirao M.',\n",
       "     'Ohmori Y.',\n",
       "     'Kawabata A.',\n",
       "     'Hikiji T.',\n",
       "     'Kobatake N.',\n",
       "     'Inagaki H.',\n",
       "     'Ikema Y.',\n",
       "     'Okamoto S.',\n",
       "     'Okitani R.',\n",
       "     'Kawakami T.',\n",
       "     'Noguchi S.',\n",
       "     'Itoh T.',\n",
       "     'Shigeta K.',\n",
       "     'Senba T.',\n",
       "     'Matsumura K.',\n",
       "     'Nakajima Y.',\n",
       "     'Mizuno T.',\n",
       "     'Morinaga M.',\n",
       "     'Sasaki M.',\n",
       "     'Togashi T.',\n",
       "     'Oyama M.',\n",
       "     'Hata H.',\n",
       "     'Watanabe M.',\n",
       "     'Komatsu T.',\n",
       "     'Mizushima-Sugano J.',\n",
       "     'Satoh T.',\n",
       "     'Shirai Y.',\n",
       "     'Takahashi Y.',\n",
       "     'Nakagawa K.',\n",
       "     'Okumura K.',\n",
       "     'Nagase T.',\n",
       "     'Nomura N.',\n",
       "     'Kikuchi H.',\n",
       "     'Masuho Y.',\n",
       "     'Yamashita R.',\n",
       "     'Nakai K.',\n",
       "     'Yada T.',\n",
       "     'Nakamura Y.',\n",
       "     'Ohara O.',\n",
       "     'Isogai T.',\n",
       "     'Sugano S.'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '14702039'},\n",
       "     {'database': 'DOI', 'id': '10.1038/ng1285'}],\n",
       "    'title': 'Complete sequencing and characterization of 21,243 full-length human cDNAs.',\n",
       "    'publicationDate': '2004',\n",
       "    'journal': 'Nat. Genet.',\n",
       "    'firstPage': '40',\n",
       "    'lastPage': '45',\n",
       "    'volume': '36'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA]']},\n",
       "  {'referenceNumber': 4,\n",
       "   'citation': {'id': 'CI-4IUTJLP3044AK',\n",
       "    'citationType': 'submission',\n",
       "    'authors': ['Kalnine N.',\n",
       "     'Chen X.',\n",
       "     'Rolfs A.',\n",
       "     'Halleck A.',\n",
       "     'Hines L.',\n",
       "     'Eisenstein S.',\n",
       "     'Koundinya M.',\n",
       "     'Raphael J.',\n",
       "     'Moreira D.',\n",
       "     'Kelley T.',\n",
       "     'LaBaer J.',\n",
       "     'Lin Y.',\n",
       "     'Phelan M.',\n",
       "     'Farmer A.'],\n",
       "    'title': 'Cloning of human full-length CDSs in BD Creator(TM) system donor vector.',\n",
       "    'publicationDate': 'MAY-2003',\n",
       "    'submissionDatabase': 'EMBL/GenBank/DDBJ databases'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORM 3)']},\n",
       "  {'referenceNumber': 5,\n",
       "   'citation': {'id': 'CI-5AVBODINQ93DF',\n",
       "    'citationType': 'submission',\n",
       "    'authors': ['Suzuki Y.',\n",
       "     'Sugano S.',\n",
       "     'Totoki Y.',\n",
       "     'Toyoda A.',\n",
       "     'Takeda T.',\n",
       "     'Sakaki Y.',\n",
       "     'Tanaka A.',\n",
       "     'Yokoyama S.'],\n",
       "    'publicationDate': 'APR-2005',\n",
       "    'submissionDatabase': 'EMBL/GenBank/DDBJ databases'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORM 1)'],\n",
       "   'referenceComments': [{'value': 'Heart', 'type': 'TISSUE'}]},\n",
       "  {'referenceNumber': 6,\n",
       "   'citation': {'id': 'CI-5GBD0VIIJ7C63',\n",
       "    'citationType': 'submission',\n",
       "    'authors': ['Mural R.J.',\n",
       "     'Istrail S.',\n",
       "     'Sutton G.G.',\n",
       "     'Florea L.',\n",
       "     'Halpern A.L.',\n",
       "     'Mobarry C.M.',\n",
       "     'Lippert R.',\n",
       "     'Walenz B.',\n",
       "     'Shatkay H.',\n",
       "     'Dew I.',\n",
       "     'Miller J.R.',\n",
       "     'Flanigan M.J.',\n",
       "     'Edwards N.J.',\n",
       "     'Bolanos R.',\n",
       "     'Fasulo D.',\n",
       "     'Halldorsson B.V.',\n",
       "     'Hannenhalli S.',\n",
       "     'Turner R.',\n",
       "     'Yooseph S.',\n",
       "     'Lu F.',\n",
       "     'Nusskern D.R.',\n",
       "     'Shue B.C.',\n",
       "     'Zheng X.H.',\n",
       "     'Zhong F.',\n",
       "     'Delcher A.L.',\n",
       "     'Huson D.H.',\n",
       "     'Kravitz S.A.',\n",
       "     'Mouchard L.',\n",
       "     'Reinert K.',\n",
       "     'Remington K.A.',\n",
       "     'Clark A.G.',\n",
       "     'Waterman M.S.',\n",
       "     'Eichler E.E.',\n",
       "     'Adams M.D.',\n",
       "     'Hunkapiller M.W.',\n",
       "     'Myers E.W.',\n",
       "     'Venter J.C.'],\n",
       "    'publicationDate': 'SEP-2005',\n",
       "    'submissionDatabase': 'EMBL/GenBank/DDBJ databases'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE GENOMIC DNA]']},\n",
       "  {'referenceNumber': 7,\n",
       "   'citation': {'id': '15489334',\n",
       "    'citationType': 'journal article',\n",
       "    'authoringGroup': ['The MGC Project Team'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '15489334'},\n",
       "     {'database': 'DOI', 'id': '10.1101/gr.2596504'}],\n",
       "    'title': 'The status, quality, and expansion of the NIH full-length cDNA project: the Mammalian Gene Collection (MGC).',\n",
       "    'publicationDate': '2004',\n",
       "    'journal': 'Genome Res.',\n",
       "    'firstPage': '2121',\n",
       "    'lastPage': '2127',\n",
       "    'volume': '14'},\n",
       "   'referencePositions': ['NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA] (ISOFORMS 2 AND 3)'],\n",
       "   'referenceComments': [{'value': 'Lung', 'type': 'TISSUE'},\n",
       "    {'value': 'Skin', 'type': 'TISSUE'}]},\n",
       "  {'referenceNumber': 8,\n",
       "   'citation': {'id': '21269460',\n",
       "    'citationType': 'journal article',\n",
       "    'authors': ['Burkard T.R.',\n",
       "     'Planyavsky M.',\n",
       "     'Kaupe I.',\n",
       "     'Breitwieser F.P.',\n",
       "     'Buerckstuemmer T.',\n",
       "     'Bennett K.L.',\n",
       "     'Superti-Furga G.',\n",
       "     'Colinge J.'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '21269460'},\n",
       "     {'database': 'DOI', 'id': '10.1186/1752-0509-5-17'}],\n",
       "    'title': 'Initial characterization of the human central proteome.',\n",
       "    'publicationDate': '2011',\n",
       "    'journal': 'BMC Syst. Biol.',\n",
       "    'firstPage': '17',\n",
       "    'lastPage': '17',\n",
       "    'volume': '5'},\n",
       "   'referencePositions': ['IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS]']},\n",
       "  {'referenceNumber': 9,\n",
       "   'citation': {'id': '24275569',\n",
       "    'citationType': 'journal article',\n",
       "    'authors': ['Bian Y.',\n",
       "     'Song C.',\n",
       "     'Cheng K.',\n",
       "     'Dong M.',\n",
       "     'Wang F.',\n",
       "     'Huang J.',\n",
       "     'Sun D.',\n",
       "     'Wang L.',\n",
       "     'Ye M.',\n",
       "     'Zou H.'],\n",
       "    'citationCrossReferences': [{'database': 'PubMed', 'id': '24275569'},\n",
       "     {'database': 'DOI', 'id': '10.1016/j.jprot.2013.11.014'}],\n",
       "    'title': 'An enzyme assisted RP-RPLC approach for in-depth analysis of human liver phosphoproteome.',\n",
       "    'publicationDate': '2014',\n",
       "    'journal': 'J. Proteomics',\n",
       "    'firstPage': '253',\n",
       "    'lastPage': '262',\n",
       "    'volume': '96'},\n",
       "   'referencePositions': ['PHOSPHORYLATION [LARGE SCALE ANALYSIS] AT SER-93',\n",
       "    'IDENTIFICATION BY MASS SPECTROMETRY [LARGE SCALE ANALYSIS]'],\n",
       "   'referenceComments': [{'value': 'Liver', 'type': 'TISSUE'}]}],\n",
       " 'uniProtKBCrossReferences': [{'database': 'EMBL',\n",
       "   'id': 'AF002280',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'AAC16670.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'AF002282',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'AAC16672.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'AF039018',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'AAB96665.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'AK313253',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'BAG36063.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'BT007341',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'AAP36005.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'AK222966',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'BAD96686.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'CH471056',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'EAX04641.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'Genomic_DNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'BC001017',\n",
       "   'properties': [{'key': 'ProteinId', 'value': '-'},\n",
       "    {'key': 'Status', 'value': 'NOT_ANNOTATED_CDS'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'EMBL',\n",
       "   'id': 'BC027870',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'AAH27870.1'},\n",
       "    {'key': 'Status', 'value': '-'},\n",
       "    {'key': 'MoleculeType', 'value': 'mRNA'}]},\n",
       "  {'database': 'CCDS',\n",
       "   'id': 'CCDS3844.1',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}],\n",
       "   'isoformId': 'Q53GG5-1'},\n",
       "  {'database': 'CCDS',\n",
       "   'id': 'CCDS47172.1',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}],\n",
       "   'isoformId': 'Q53GG5-2'},\n",
       "  {'database': 'RefSeq',\n",
       "   'id': 'NP_001107579.1',\n",
       "   'properties': [{'key': 'NucleotideSequenceId', 'value': 'NM_001114107.4'}],\n",
       "   'isoformId': 'Q53GG5-2'},\n",
       "  {'database': 'RefSeq',\n",
       "   'id': 'NP_055291.2',\n",
       "   'properties': [{'key': 'NucleotideSequenceId', 'value': 'NM_014476.5'}],\n",
       "   'isoformId': 'Q53GG5-1'},\n",
       "  {'database': 'AlphaFoldDB',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'BMRB',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'SMR',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'BioGRID',\n",
       "   'id': '118119',\n",
       "   'properties': [{'key': 'Interactions', 'value': '27'}]},\n",
       "  {'database': 'IntAct',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Interactions', 'value': '11'}]},\n",
       "  {'database': 'STRING',\n",
       "   'id': '9606.ENSP00000284767',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'iPTMnet',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PhosphoSitePlus',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'BioMuta',\n",
       "   'id': 'PDLIM3',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'DMDM',\n",
       "   'id': '74740479',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'jPOST',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'MassIVE',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PaxDb',\n",
       "   'id': '9606-ENSP00000284770',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PeptideAtlas',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'ProteomicsDB',\n",
       "   'id': '62479',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}],\n",
       "   'isoformId': 'Q53GG5-1'},\n",
       "  {'database': 'ProteomicsDB',\n",
       "   'id': '62480',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}],\n",
       "   'isoformId': 'Q53GG5-2'},\n",
       "  {'database': 'ProteomicsDB',\n",
       "   'id': '62481',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}],\n",
       "   'isoformId': 'Q53GG5-3'},\n",
       "  {'database': 'Pumba',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'Antibodypedia',\n",
       "   'id': '1500',\n",
       "   'properties': [{'key': 'antibodies',\n",
       "     'value': '202 antibodies from 24 providers'}]},\n",
       "  {'database': 'DNASU',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'Ensembl',\n",
       "   'id': 'ENST00000284767.12',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'ENSP00000284767.8'},\n",
       "    {'key': 'GeneId', 'value': 'ENSG00000154553.17'}],\n",
       "   'isoformId': 'Q53GG5-1'},\n",
       "  {'database': 'Ensembl',\n",
       "   'id': 'ENST00000284771.7',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'ENSP00000284771.6'},\n",
       "    {'key': 'GeneId', 'value': 'ENSG00000154553.17'}],\n",
       "   'isoformId': 'Q53GG5-2'},\n",
       "  {'database': 'GeneID',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'KEGG',\n",
       "   'id': 'hsa:27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'MANE-Select',\n",
       "   'id': 'ENST00000284767.12',\n",
       "   'properties': [{'key': 'ProteinId', 'value': 'ENSP00000284767.8'},\n",
       "    {'key': 'RefSeqNucleotideId', 'value': 'NM_014476.6'},\n",
       "    {'key': 'RefSeqProteinId', 'value': 'NP_055291.2'}]},\n",
       "  {'database': 'UCSC',\n",
       "   'id': 'uc003ixw.5',\n",
       "   'properties': [{'key': 'OrganismName', 'value': 'human'}],\n",
       "   'isoformId': 'Q53GG5-1'},\n",
       "  {'database': 'AGR',\n",
       "   'id': 'HGNC:20767',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'CTD',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'DisGeNET',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'GeneCards',\n",
       "   'id': 'PDLIM3',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'HGNC',\n",
       "   'id': 'HGNC:20767',\n",
       "   'properties': [{'key': 'GeneName', 'value': 'PDLIM3'}]},\n",
       "  {'database': 'HPA',\n",
       "   'id': 'ENSG00000154553',\n",
       "   'properties': [{'key': 'ExpressionPatterns',\n",
       "     'value': 'Group enriched (skeletal muscle, tongue)'}]},\n",
       "  {'database': 'MIM',\n",
       "   'id': '605889',\n",
       "   'properties': [{'key': 'Type', 'value': 'gene'}]},\n",
       "  {'database': 'neXtProt',\n",
       "   'id': 'NX_Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'OpenTargets',\n",
       "   'id': 'ENSG00000154553',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PharmGKB',\n",
       "   'id': 'PA134970631',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'VEuPathDB',\n",
       "   'id': 'HostDB:ENSG00000154553',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'eggNOG',\n",
       "   'id': 'KOG1703',\n",
       "   'properties': [{'key': 'ToxonomicScope', 'value': 'Eukaryota'}]},\n",
       "  {'database': 'GeneTree',\n",
       "   'id': 'ENSGT00940000156741',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'InParanoid',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'OMA',\n",
       "   'id': 'IHAQFNT',\n",
       "   'properties': [{'key': 'Fingerprint', 'value': '-'}]},\n",
       "  {'database': 'OrthoDB',\n",
       "   'id': '370973at2759',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PhylomeDB',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'TreeFam',\n",
       "   'id': 'TF106408',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'PathwayCommons',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'SignaLink',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'BioGRID-ORCS',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'hits', 'value': '5 hits in 1141 CRISPR screens'}]},\n",
       "  {'database': 'ChiTaRS',\n",
       "   'id': 'PDLIM3',\n",
       "   'properties': [{'key': 'OrganismName', 'value': 'human'}]},\n",
       "  {'database': 'GeneWiki',\n",
       "   'id': 'PDLIM3',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'GenomeRNAi',\n",
       "   'id': '27295',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'Pharos',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'DevelopmentLevel', 'value': 'Tbio'}]},\n",
       "  {'database': 'PRO',\n",
       "   'id': 'PR:Q53GG5',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'Proteomes',\n",
       "   'id': 'UP000005640',\n",
       "   'properties': [{'key': 'Component', 'value': 'Chromosome 4'}]},\n",
       "  {'database': 'RNAct',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'moleculeType', 'value': 'Protein'}]},\n",
       "  {'database': 'Bgee',\n",
       "   'id': 'ENSG00000154553',\n",
       "   'properties': [{'key': 'ExpressionPatterns',\n",
       "     'value': 'Expressed in skeletal muscle tissue of biceps brachii and 179 other cell types or tissues'}]},\n",
       "  {'database': 'ExpressionAtlas',\n",
       "   'id': 'Q53GG5',\n",
       "   'properties': [{'key': 'ExpressionPatterns',\n",
       "     'value': 'baseline and differential'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0005912',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'C:adherens junction'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0005829',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'C:cytosol'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IDA:HPA'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0031941',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'C:filamentous actin'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0001725',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'C:stress fiber'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0030018',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'C:Z disc'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0003779',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'F:actin binding'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0046872',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'F:metal ion binding'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IEA:UniProtKB-KW'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0051371',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'F:muscle alpha-actinin binding'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0030036',\n",
       "   'properties': [{'key': 'GoTerm',\n",
       "     'value': 'P:actin cytoskeleton organization'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0007507',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'P:heart development'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'GO',\n",
       "   'id': 'GO:0061061',\n",
       "   'properties': [{'key': 'GoTerm', 'value': 'P:muscle structure development'},\n",
       "    {'key': 'GoEvidenceType', 'value': 'IBA:GO_Central'}]},\n",
       "  {'database': 'CDD',\n",
       "   'id': 'cd09450',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'LIM_ALP'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'CDD',\n",
       "   'id': 'cd00992',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ_signaling'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'DisProt',\n",
       "   'id': 'DP01782',\n",
       "   'properties': [{'key': 'Description', 'value': '-'}]},\n",
       "  {'database': 'Gene3D',\n",
       "   'id': '2.30.42.10',\n",
       "   'properties': [{'key': 'EntryName', 'value': '-'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'Gene3D',\n",
       "   'id': '2.10.110.10',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'Cysteine Rich Protein'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR031847',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDLI1-4/Zasp-like_mid'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR001478',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR050604',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ-LIM_domain'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR036034',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ_sf'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR006643',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'Zasp-like_motif'}]},\n",
       "  {'database': 'InterPro',\n",
       "   'id': 'IPR001781',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'Znf_LIM'}]},\n",
       "  {'database': 'PANTHER',\n",
       "   'id': 'PTHR24214:SF7',\n",
       "   'properties': [{'key': 'EntryName',\n",
       "     'value': 'PDZ AND LIM DOMAIN PROTEIN 3'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'PANTHER',\n",
       "   'id': 'PTHR24214',\n",
       "   'properties': [{'key': 'EntryName',\n",
       "     'value': 'PDZ AND LIM DOMAIN PROTEIN ZASP'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'Pfam',\n",
       "   'id': 'PF15936',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'DUF4749'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'Pfam',\n",
       "   'id': 'PF00412',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'LIM'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'Pfam',\n",
       "   'id': 'PF00595',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'SMART',\n",
       "   'id': 'SM00132',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'LIM'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'SMART',\n",
       "   'id': 'SM00228',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'SMART',\n",
       "   'id': 'SM00735',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'ZM'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'SUPFAM',\n",
       "   'id': 'SSF57716',\n",
       "   'properties': [{'key': 'EntryName',\n",
       "     'value': 'Glucocorticoid receptor-like (DNA-binding domain)'},\n",
       "    {'key': 'MatchStatus', 'value': '2'}]},\n",
       "  {'database': 'SUPFAM',\n",
       "   'id': 'SSF50156',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ domain-like'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'PROSITE',\n",
       "   'id': 'PS00478',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'LIM_DOMAIN_1'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'PROSITE',\n",
       "   'id': 'PS50023',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'LIM_DOMAIN_2'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]},\n",
       "  {'database': 'PROSITE',\n",
       "   'id': 'PS50106',\n",
       "   'properties': [{'key': 'EntryName', 'value': 'PDZ'},\n",
       "    {'key': 'MatchStatus', 'value': '1'}]}],\n",
       " 'sequence': {'value': 'MPQTVILPGPAPWGFRLSGGIDFNQPLVITRITPGSKAAAANLCPGDVILAIDGFGTESMTHADAQDRIKAAAHQLCLKIDRGETHLWSPQVSEDGKAHPFKINLESEPQDGNYFEHKHNIRPKPFVIPGRSSGCSTPSGIDCGSGRSTPSSVSTVSTICPGDLKVAAKLAPNIPLEMELPGVKIVHAQFNTPMQLYSDDNIMETLQGQVSTALGETPLMSEPTASVPPESDVYRMLHDNRNEPTQPRQSGSFRVLQGMVDDGSDDRPAGTRSVRAPVTKVHGGSGGAQRMPLCDKCGSGIVGAVVKARDKYRHPECFVCADCNLNLKQKGYFFIEGELYCETHARARTKPPEGYDTVTLYPKA',\n",
       "  'length': 364,\n",
       "  'molWeight': 39232,\n",
       "  'crc64': '083802DF22D7A237',\n",
       "  'md5': 'C05029C8966F75EA99F57DACC183A1B5'},\n",
       " 'extraAttributes': {'countByCommentType': {'FUNCTION': 1,\n",
       "   'SUBUNIT': 1,\n",
       "   'INTERACTION': 3,\n",
       "   'SUBCELLULAR LOCATION': 1,\n",
       "   'ALTERNATIVE PRODUCTS': 3,\n",
       "   'TISSUE SPECIFICITY': 1},\n",
       "  'countByFeatureType': {'Chain': 1,\n",
       "   'Domain': 2,\n",
       "   'Modified residue': 3,\n",
       "   'Alternative sequence': 3,\n",
       "   'Natural variant': 1,\n",
       "   'Sequence conflict': 5},\n",
       "  'uniParcId': 'UPI000013DDB6'}}"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests\n",
    "uniprot = 'Q53GG5'\n",
    "\n",
    "r = requests.get(f'https://rest.uniprot.org/uniprotkb/{uniprot}?format=json')\n",
    "\n",
    "r.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ['references']['id'] Abstract searching for text?\n",
    "# ['alternativeNames']['fullName'] Regex matching text?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For gene symbol:\n",
    "#   Grab Uniprot ID (from gene normalizer?)\n",
    "# Query UniProt\n",
    "#   For AlternativeName: \n",
    "#       Grab FullName\n",
    "#           Regex/Rule Match 1 -- Length of symbol + First letter of terms check\n",
    "#           Regex/Rule Match 2 -- First letter of terms check, but include n-1 examples\n",
    "#           Regex/Rule Match n -- ???\n",
    "#               if potentialMatch:\n",
    "#                   possible_expansions.append(match)\n",
    "#   df['gene symbol'][position] = possible_expansions\n",
    "#\n",
    "#   For References['ID']:\n",
    "#       Query Pubmed\n",
    "#           Grab Abstract\n",
    "#           Create n-length pairs of words (n = length of symbol)\n",
    "#           For n-length pairs of words:\n",
    "#               Regex/Rule Match 1 -- Length of Symbol + First letter of terms check\n",
    "#               Regex/Rule Match 2 -- First letter of terms check, but include n-1 examples\n",
    "#               Regex/Rule Match 3 -- ???\n",
    "#                   if potentialMatch:\n",
    "#                       possible_expansions.append(match)\n",
    "#   df['gene symbol'][position] = possible_expansions\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Grab Uniprot ID from Gene Normalizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'type': 'Extension', 'name': 'symbol_status', 'value': 'approved'},\n",
       " {'type': 'Extension',\n",
       "  'name': 'approved_name',\n",
       "  'value': 'bromodomain containing 4'},\n",
       " {'type': 'Extension',\n",
       "  'name': 'associated_with',\n",
       "  'value': ['ucsc:uc002nar.4',\n",
       "   'vega:OTTHUMG00000183252',\n",
       "   'iuphar:1945',\n",
       "   'orphanet:449868',\n",
       "   'refseq:NM_058243',\n",
       "   'cosmic:BRD4',\n",
       "   'uniprot:O60885',\n",
       "   'ccds:CCDS46004',\n",
       "   'ccds:CCDS82307',\n",
       "   'ena.embl:Y12059',\n",
       "   'omim:608749',\n",
       "   'pubmed:10938129',\n",
       "   'ccds:CCDS12328']},\n",
       " {'type': 'Extension', 'name': 'previous_symbols', 'value': ['LOC90616']},\n",
       " {'type': 'Extension', 'name': 'strand', 'value': '-'},\n",
       " {'type': 'Extension',\n",
       "  'name': 'hgnc_locations',\n",
       "  'value': [{'_id': 'ga4gh:VCL.OnjZ5UPScFNAI_iMhJVyrD9JQ70HmeJ0',\n",
       "    'type': 'ChromosomeLocation',\n",
       "    'species_id': 'taxonomy:9606',\n",
       "    'chr': '19',\n",
       "    'interval': {'type': 'CytobandInterval',\n",
       "     'start': 'p13.12',\n",
       "     'end': 'p13.12'}}]},\n",
       " {'type': 'Extension',\n",
       "  'name': 'ensembl_locations',\n",
       "  'value': [{'_id': 'ga4gh:VSL.fjm4RMn5JCH07iPTJr3iFM-HZ_u9Ww1I',\n",
       "    'type': 'SequenceLocation',\n",
       "    'sequence_id': 'ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl',\n",
       "    'interval': {'type': 'SequenceInterval',\n",
       "     'start': {'type': 'Number', 'value': 15235518},\n",
       "     'end': {'type': 'Number', 'value': 15332545}}}]},\n",
       " {'type': 'Extension',\n",
       "  'name': 'ncbi_locations',\n",
       "  'value': [{'_id': 'ga4gh:VCL.OnjZ5UPScFNAI_iMhJVyrD9JQ70HmeJ0',\n",
       "    'type': 'ChromosomeLocation',\n",
       "    'species_id': 'taxonomy:9606',\n",
       "    'chr': '19',\n",
       "    'interval': {'type': 'CytobandInterval',\n",
       "     'start': 'p13.12',\n",
       "     'end': 'p13.12'}},\n",
       "   {'_id': 'ga4gh:VSL.KcFXQwtHCORDm-EFKktfJrlgRRkpfxsR',\n",
       "    'type': 'SequenceLocation',\n",
       "    'sequence_id': 'ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl',\n",
       "    'interval': {'type': 'SequenceInterval',\n",
       "     'start': {'type': 'Number', 'value': 15235518},\n",
       "     'end': {'type': 'Number', 'value': 15332539}}}]},\n",
       " {'type': 'Extension',\n",
       "  'name': 'hgnc_locus_type',\n",
       "  'value': 'gene with protein product'},\n",
       " {'type': 'Extension', 'name': 'ncbi_gene_type', 'value': 'protein-coding'},\n",
       " {'type': 'Extension', 'name': 'ensembl_biotype', 'value': 'protein_coding'}]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "gene = 'BRD4' # CAP\n",
    "query = f'https://normalize.cancervariants.org/gene/normalize?q={gene}'\n",
    "r = requests.get(query)\n",
    "uniprot_id = r.json()['gene_descriptor']['extensions'][2]['value'][6] # make this a for, check for associated_with\n",
    "uniprot_id = uniprot_id.replace('uniprot:','')\n",
    "uniprot_id\n",
    "\n",
    "r.json()['gene_descriptor']['extensions']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['uniprot:O60885']\n"
     ]
    }
   ],
   "source": [
    "for extension in r.json()['gene_descriptor']['extensions']:\n",
    "        if extension['name'] == 'associated_with':\n",
    "            uniprot_id = [value for value in extension['value'] if value.startswith('uniprot:')]\n",
    "            print(uniprot_id)\n",
    "            pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Query UniProt for Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Check Alternative Names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Protein HUNK1'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r = requests.get(f'https://rest.uniprot.org/uniprotkb/{uniprot_id}?format=json')\n",
    "expanded_name = r.json()['proteinDescription']['alternativeNames'][0]['fullName']['value'] \n",
    "expanded_name # No Match"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Check PMIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['11733348',\n",
       " 'CI-DMQH005K2JO7F',\n",
       " '15057824',\n",
       " 'CI-5GBDQ6B103N1E',\n",
       " '15489334',\n",
       " '12543779',\n",
       " '16109376',\n",
       " '16109377',\n",
       " '17081983',\n",
       " '16940503',\n",
       " '17189189',\n",
       " '18922874',\n",
       " '18669648',\n",
       " '19413330',\n",
       " '19596240',\n",
       " '19103749',\n",
       " '19608861',\n",
       " '20068231',\n",
       " '21269460',\n",
       " '21555454',\n",
       " '21890894',\n",
       " '21406692',\n",
       " '22334664',\n",
       " '23086925',\n",
       " '22509028',\n",
       " '24360279',\n",
       " '23186163',\n",
       " '23317504',\n",
       " '23589332',\n",
       " '23728299',\n",
       " '24275569',\n",
       " '25218447',\n",
       " '25114211',\n",
       " '25772364',\n",
       " '25593309',\n",
       " '25755297',\n",
       " '28112733',\n",
       " '29374058',\n",
       " '31969702',\n",
       " '17344846',\n",
       " '18500820',\n",
       " '20871596',\n",
       " '21068722',\n",
       " '21568322',\n",
       " '21964340',\n",
       " '22137933',\n",
       " '22645123',\n",
       " '22136404',\n",
       " '23095041',\n",
       " '22464331',\n",
       " '23517011',\n",
       " '23530754',\n",
       " '29176719',\n",
       " '32193360',\n",
       " '29379197',\n",
       " '29440723',\n",
       " '31168063',\n",
       " '35470444']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmids_to_check = []\n",
    "for pmid in r.json()['references']:\n",
    "    pmids_to_check.append(pmid['citation']['id'])\n",
    "\n",
    "\n",
    "pmids_to_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Thanks senor GPT\n",
    "def get_pubmed_abstract(pmid, api_key='YOUR_API_KEY'):\n",
    "    \"\"\" Fetch the abstract for a given PMID from PubMed. \"\"\"\n",
    "    url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi\"\n",
    "    params = {\n",
    "        'db': 'pubmed',\n",
    "        'id': pmid,\n",
    "        'retmode': 'xml'\n",
    "        # 'api_key': api_key\n",
    "    }\n",
    "    \n",
    "    response = requests.get(url, params=params)\n",
    "    \n",
    "    if response.status_code == 200:\n",
    "        # Parse the XML response\n",
    "        from xml.etree import ElementTree\n",
    "        root = ElementTree.fromstring(response.text)\n",
    "        \n",
    "        # Find the AbstractText element and return its text\n",
    "        abstract_text = root.find('.//AbstractText')\n",
    "        if abstract_text is not None:\n",
    "            return abstract_text.text\n",
    "        else:\n",
    "            return \"No abstract found.\"\n",
    "    else:\n",
    "        return f\"Error: {response.status_code}\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 58/58 [01:14<00:00,  1.28s/it]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "56"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import time\n",
    "from tqdm import tqdm\n",
    "\n",
    "abstracts_to_check = []\n",
    "for pmid in tqdm(pmids_to_check):\n",
    "    entry = {}\n",
    "    if pmid.startswith('CI'):\n",
    "        continue\n",
    "    else:\n",
    "        time.sleep(1)\n",
    "        abstract = get_pubmed_abstract(pmid)\n",
    "        entry[pmid] = abstract\n",
    "        abstracts_to_check.append(entry)\n",
    "        \n",
    "len(abstracts_to_check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'11733348': 'Translocation t(15;19)(q13;p13.1) defines a lethal midline carcinoma arising adjacent to respiratory tract in young people. To characterize molecular alterations responsible for the distinctly aggressive biological behavior of this cancer, we mapped the chromosome 15 and 19 translocation breakpoints by fluorescence in situ hybridization (FISH) and Southern blotting. To evaluate preliminarily the frequency, anatomical distribution, and histological features of t(15;19) cancer, we developed a FISH assay for paraffin sections. Our findings reveal a novel oncogenic mechanism in which the chromosome 19 translocation breakpoint interrupts the coding sequence of a bromodomain gene, BRD4. These studies implicate BRD4 as a potential partner in a t(15;19)-associated fusion oncogene. In addition, we localized the chromosome 15 breakpoint to a 9-kb region in each of two cases, thereby identifying several candidate oncogenes which might represent the BRD4 fusion partner. FISH evaluation of 13 pediatric carcinomas revealed t(15;19) in one of four sinonasal carcinomas, whereas this translocation was not detected in thymic (n = 3), mucoepidermoid (n = 3), laryngeal (n = 2), or nasopharyngeal (n = 1) carcinomas. Our studies shed light on the oncogenic mechanism underlying t(15;19) and provide further evidence that this highly lethal cancer arises from respiratory mucosa.'},\n",
       " {'15057824': 'Chromosome 19 has the highest gene density of all human chromosomes, more than double the genome-wide average. The large clustered gene families, corresponding high G + C content, CpG islands and density of repetitive DNA indicate a chromosome rich in biological and evolutionary significance. Here we describe 55.8 million base pairs of highly accurate finished sequence representing 99.9% of the euchromatin portion of the chromosome. Manual curation of gene loci reveals 1,461 protein-coding genes and 321 pseudogenes. Among these are genes directly implicated in mendelian disorders, including familial hypercholesterolaemia and insulin-resistant diabetes. Nearly one-quarter of these genes belong to tandemly arranged families, encompassing more than 25% of the chromosome. Comparative analyses show a fascinating picture of conservation and divergence, revealing large blocks of gene orthology with rodents, scattered regions with more recent gene family expansions and deletions, and segments of coding and non-coding conservation with the distant fish species Takifugu.'},\n",
       " {'15489334': \"The National Institutes of Health's Mammalian Gene Collection (MGC) project was designed to generate and sequence a publicly accessible cDNA resource containing a complete open reading frame (ORF) for every human and mouse gene. The project initially used a random strategy to select clones from a large number of cDNA libraries from diverse tissues. Candidate clones were chosen based on 5'-EST sequences, and then fully sequenced to high accuracy and analyzed by algorithms developed for this project. Currently, more than 11,000 human and 10,000 mouse genes are represented in MGC by at least one clone with a full ORF. The random selection approach is now reaching a saturation point, and a transition to protocols targeted at the missing transcripts is now required to complete the mouse and human collections. Comparison of the sequence of the MGC clones to reference genome sequences reveals that most cDNA clones are of very high sequence quality, although it is likely that some cDNAs may carry missense variants as a consequence of experimental artifact, such as PCR, cloning, or reverse transcriptase errors. Recently, a rat cDNA component was added to the project, and ongoing frog (Xenopus) and zebrafish (Danio) cDNA projects were expanded to take advantage of the high-throughput MGC pipeline.\"},\n",
       " {'12543779': 'The poorly differentiated carcinoma with t(15;19)(q13, p13.1) is characterized by its highly aggressive, invariably lethal clinical course. The chromosome 19 translocation breakpoint targets the BRD4 double bromodomain-containing gene, which functions in regulation of cell cycle progression. Herein we demonstrate that BRD4 is fused with nearly the entire transcript of the novel 15q13 gene, NUT (nuclear protein in testis), forming a 6.4-kb fusion oncogene, BRD4-NUT. NUT, like BRD4, is predicted to encode a nuclear protein but, unlike the ubiquitous BRD4 transcript, is expressed only in testis. These findings establish a model to elucidate the oncogenic consequences of unscheduled NUT expression and altered BRD4 function. Very few fusion oncogenes have been identified in epithelial tumors, and BRD4-NUT is the first fusion oncogene mechanism identified in a highly lethal form of carcinoma.'},\n",
       " {'16109376': 'Brd4 is a mammalian bromodomain protein that binds to acetylated chromatin. Proteomic analysis revealed that Brd4 interacts with cyclinT1 and Cdk9 that constitutes core positive transcription elongation factor b (P-TEFb). Brd4 interacted with P-TEFb in the living nucleus through its bromodomain. About half of P-TEFb was bound to the inhibitory subunit and functionally inactive. Brd4 interacted with P-TEFb that was free of the inhibitory subunit. An increase in Brd4 expression led to increased P-TEFb-dependent phosphorylation of RNA polymerase II (RNAPII) CTD and stimulation of transcription from promoters in vivo. Conversely, a reduction in Brd4 expression by siRNA reduced CTD phosphorylation and transcription, revealing that Brd4 is a positive regulatory component of P-TEFb. In chromatin immunoprecipitation (ChIP) assays, the recruitment of P-TEFb to a promoter was dependent on Brd4 and was enhanced by an increase in chromatin acetylation. Together, P-TEFb alternately interacts with Brd4 and the inhibitory subunit to maintain functional equilibrium in the cell.'},\n",
       " {'16109377': 'The cyclinT1/Cdk9 heterodimer that constitutes core P-TEFb is generally presumed to be the transcriptionally active form for stimulating RNA polymerase II elongation. About half of cellular P-TEFb also exists in an inactive complex with the 7SK snRNA and the HEXIM1 protein. Here, we show that the remaining half associates with the bromodomain protein Brd4. In stress-induced cells, the 7SK/HEXIM1-bound P-TEFb is quantitatively converted into the Brd4-associated form. The association with Brd4 is necessary to form the transcriptionally active P-TEFb, recruits P-TEFb to a promoter, and enables P-TEFb to contact the Mediator complex, a potential target for the Brd4-mediated recruitment. Although generally required for transcription, the P-TEFb-recruitment function of Brd4 can be substituted by that of HIV-1 Tat, which recruits P-TEFb directly for activated HIV-1 transcription. Brd4, HEXIM1, and 7SK are all implicated in regulating cell growth, which may result from their dynamic control of the general transcription factor P-TEFb.'},\n",
       " {'17081983': 'Cell signaling mechanisms often transmit information via posttranslational protein modifications, most importantly reversible protein phosphorylation. Here we develop and apply a general mass spectrometric technology for identification and quantitation of phosphorylation sites as a function of stimulus, time, and subcellular location. We have detected 6,600 phosphorylation sites on 2,244 proteins and have determined their temporal dynamics after stimulating HeLa cells with epidermal growth factor (EGF) and recorded them in the Phosida database. Fourteen percent of phosphorylation sites are modulated at least 2-fold by EGF, and these were classified by their temporal profiles. Surprisingly, a majority of proteins contain multiple phosphorylation sites showing different kinetics, suggesting that they serve as platforms for integrating signals. In addition to protein kinase cascades, the targets of reversible phosphorylation include ubiquitin ligases, guanine nucleotide exchange factors, and at least 46 different transcriptional regulators. The dynamic phosphoproteome provides a missing link in a global, integrative view of cellular regulation.'},\n",
       " {'16940503': \"The latency-associated nuclear antigen (LANA) of Kaposi's sarcoma-associated herpesvirus (KSHV) is required for viral episome maintenance in host cells during latent infection. Two regions of the protein have been implicated in tethering LANA/viral episomes to the host mitotic chromosomes, and LANA chromosome-binding sites are subjects of high interest. Because previous studies had identified bromodomain protein Brd4 as the mitotic chromosome anchor for the bovine papillomavirus E2 protein, which tethers the viral episomes to host mitotic chromosomes (J. You, J. L. Croyle, A. Nishimura, K. Ozato, and P. M. Howley, Cell 117:349-360, 2004, and J. You, M. R. Schweiger, and P. M. Howley, J. Virol. 79:14956-14961, 2005), we examined whether KSHV LANA interacts with Brd4. We found that LANA binds Brd4 in vivo and in vitro and that the binding is mediated by a direct protein-protein interaction between the ET (extraterminal) domain of Brd4 and a carboxyl-terminal region of LANA previously implicated in chromosome binding. Brd4 associates with mitotic chromosomes throughout mitosis and demonstrates a strong colocalization with LANA and the KSHV episomes on host mitotic chromosomes. Although another bromodomain protein, RING3/Brd2, binds to LANA in a similar fashion in vitro, it is largely excluded from the mitotic chromosomes in KSHV-uninfected cells and is partially recruited to the chromosomes in KSHV-infected cells. These data identify Brd4 as an interacting protein for the carboxyl terminus of LANA on mitotic chromosomes and suggest distinct functional roles for the two bromodomain proteins RING3/Brd2 and Brd4 in LANA binding. Additionally, because Brd4 has recently been shown to have a role in transcription, we examined whether Brd4 can regulate the CDK2 promoter, which can be transactivated by LANA.\"},\n",
       " {'17189189': 'Autonomously replicating DNA viruses must evade mitotic checkpoints and actively partition their genomes to maintain persistent infection. The E2 protein serves these functions by tethering papillomavirus episomes to mitotic chromosomes; however, the mechanism remains unresolved. We show that E2 binds ChlR1, a DNA helicase that plays a role in sister chromatid cohesion. The E2 mutation W130R fails to bind ChlR1 and correspondingly does not associate with mitotic chromosomes. Viral genomes encoding this E2 mutation are not episomally maintained in cell culture. Notably, E2 W130R binds Brd4, which reportedly acts as a mitotic tether, indicating this interaction is insufficient for E2 association with mitotic chromosomes. RNAi-induced depletion of ChlR1 significantly reduced E2 localization to mitotic chromosomes. These studies provide compelling evidence that ChlR1 association is required for loading the papillomavirus E2 protein onto mitotic chromosomes and represents a kinetochore-independent mechanism for viral genome maintenance and segregation.'},\n",
       " {'18922874': \"The EBNA1 protein of Epstein-Barr virus (EBV) is essential for EBV latent infection in ensuring the replication and stable segregation of the EBV genomes and in activating the transcription of other EBV latency genes. We have tested the ability of four host proteins (Brd2, Brd4, DEK, and MeCP2) implicated in the segregation of papillomavirus and Kaposi's sarcoma-associated herpesvirus to support EBNA1-mediated segregation of EBV-based plasmids in Saccharomyces cerevisiae. We found that Brd4 enabled EBNA1-mediated segregation while Brd2 and MeCP2 had a general stimulatory effect on plasmid maintenance. EBNA1 interacted with Brd4 in both yeast and human cells through N-terminal sequences previously shown to mediate transcriptional activation but not segregation. In keeping with this interaction site, silencing of Brd4 in human cells decreased transcriptional activation by EBNA1 but not the mitotic chromosome attachment of EBNA1 that is required for segregation. In addition, Brd4 was found to be preferentially localized to the FR enhancer element regulated by EBNA1, over other EBV sequences, in latently EBV-infected cells. The results indicate that EBNA1 can functionally interact with Brd4 in native and heterologous systems and that this interaction facilitates transcriptional activation by EBNA1 from the FR element.\"},\n",
       " {'18669648': 'The eukaryotic cell division cycle is characterized by a sequence of orderly and highly regulated events resulting in the duplication and separation of all cellular material into two newly formed daughter cells. Protein phosphorylation by cyclin-dependent kinases (CDKs) drives this cycle. To gain further insight into how phosphorylation regulates the cell cycle, we sought to identify proteins whose phosphorylation is cell cycle regulated. Using stable isotope labeling along with a two-step strategy for phosphopeptide enrichment and high mass accuracy mass spectrometry, we examined protein phosphorylation in a human cell line arrested in the G(1) and mitotic phases of the cell cycle. We report the identification of >14,000 different phosphorylation events, more than half of which, to our knowledge, have not been described in the literature, along with relative quantitative data for the majority of these sites. We observed >1,000 proteins with increased phosphorylation in mitosis including many known cell cycle regulators. The majority of sites on regulated phosphopeptides lie in [S/T]P motifs, the minimum required sequence for CDKs, suggesting that many of the proteins may be CDK substrates. Analysis of non-proline site-containing phosphopeptides identified two unique motifs that suggest there are at least two undiscovered mitotic kinases.'},\n",
       " {'19413330': 'The analysis of proteome-wide phosphorylation events is still a major analytical challenge because of the enormous complexity of protein phosphorylation networks. In this work, we evaluate the complementarity of Lys-N, Lys-C, and trypsin with regard to their ability to contribute to the global analysis of the phosphoproteome. A refined version of low-pH strong cation exchange was used to efficiently separate N-terminally acetylated, phosphorylated, and nonmodified peptides. A total of 5036 nonredundant phosphopeptides could be identified with a false discovery rate of <1% from 1 mg of protein using a combination of the three enzymes. Our data revealed that the overlap between the phosphopeptide data sets generated with different proteases was marginal, whereas the overlap between two similarly generated tryptic data sets was found to be at least 4 times higher. In this way, the parallel use of Lys-N and trypsin enabled a 72% increase in the number of detected phosphopeptides as compared to trypsin alone, whereas a trypsin replicate experiment only led to a 25% increase. Thus, when focusing solely on the trypsin and Lys-N data, we identified 4671 nonredundant phosphopeptides. Further analysis of the detected sites showed that the Lys-N and trypsin data sets were enriched in significantly different phosphorylation motifs, further evidencing that multiprotease approaches are very valuable in phosphoproteome analyses.'},\n",
       " {'19596240': 'Most inducible transcriptional programs consist of primary and secondary response genes (PRGs and SRGs) that differ in their kinetics of expression and in their requirements for new protein synthesis and chromatin remodeling. Here we show that many PRGs, in contrast to SRGs, have preassembled RNA polymerase II (Pol II) and positive histone modifications at their promoters in the basal state. Pol II at PRGs generates low levels of full-length unspliced transcripts but fails to make mature, protein-coding transcripts in the absence of stimulation. Induction of PRGs is controlled at the level of transcriptional elongation and mRNA processing, through the signal-dependent recruitment of P-TEFb. P-TEFb is in turn recruited by the bromodomain-containing protein Brd4, which detects H4K5/8/12Ac inducibly acquired at PRG promoters. Our findings suggest that the permissive structure of PRGs both stipulates their unique regulation in the basal state by corepressor complexes and enables their rapid induction in multiple cell types.'},\n",
       " {'19103749': 'Acetylation of the RelA subunit of NF-kappaB, especially at lysine-310, is critical for the transcriptional activation of NF-kappaB and the expression of inflammatory genes. In this study, we demonstrate that bromodomains of Brd4 bind to acetylated lysine-310. Brd4 enhances transcriptional activation of NF-kappaB and the expression of a subset of NF-kappaB-responsive inflammatory genes in an acetylated lysine-310-dependent manner. Bromodomains of Brd4 and acetylated lysine-310 of RelA are both required for the mutual interaction and coactivation function of Brd4. Finally, we demonstrate that Brd4 further recruits CDK9 to phosphorylate C-terminal domain of RNA polymerase II and facilitate the transcription of NF-kappaB-dependent inflammatory genes. Our results identify Brd4 as a novel coactivator of NF-kappaB through specifically binding to acetylated lysine-310 of RelA. In addition, these studies reveal a mechanism by which acetylated RelA stimulates the transcriptional activity of NF-kappaB and the NF-kappaB-dependent inflammatory response.'},\n",
       " {'19608861': \"Lysine acetylation is a reversible posttranslational modification of proteins and plays a key role in regulating gene expression. Technological limitations have so far prevented a global analysis of lysine acetylation's cellular roles. We used high-resolution mass spectrometry to identify 3600 lysine acetylation sites on 1750 proteins and quantified acetylation changes in response to the deacetylase inhibitors suberoylanilide hydroxamic acid and MS-275. Lysine acetylation preferentially targets large macromolecular complexes involved in diverse cellular processes, such as chromatin remodeling, cell cycle, splicing, nuclear transport, and actin nucleation. Acetylation impaired phosphorylation-dependent interactions of 14-3-3 and regulated the yeast cyclin-dependent kinase Cdc28. Our data demonstrate that the regulatory scope of lysine acetylation is broad and comparable with that of other major posttranslational modifications.\"},\n",
       " {'20068231': 'Eukaryotic cells replicate by a complex series of evolutionarily conserved events that are tightly regulated at defined stages of the cell division cycle. Progression through this cycle involves a large number of dedicated protein complexes and signaling pathways, and deregulation of this process is implicated in tumorigenesis. We applied high-resolution mass spectrometry-based proteomics to investigate the proteome and phosphoproteome of the human cell cycle on a global scale and quantified 6027 proteins and 20,443 unique phosphorylation sites and their dynamics. Co-regulated proteins and phosphorylation sites were grouped according to their cell cycle kinetics and compared to publicly available messenger RNA microarray data. Most detected phosphorylation sites and more than 20% of all quantified proteins showed substantial regulation, mainly in mitotic cells. Kinase-motif analysis revealed global activation during S phase of the DNA damage response network, which was mediated by phosphorylation by ATM or ATR or DNA-dependent protein kinases. We determined site-specific stoichiometry of more than 5000 sites and found that most of the up-regulated sites phosphorylated by cyclin-dependent kinase 1 (CDK1) or CDK2 were almost fully phosphorylated in mitotic cells. In particular, nuclear proteins and proteins involved in regulating metabolic processes have high phosphorylation site occupancy in mitosis. This suggests that these proteins may be inactivated by phosphorylation in mitotic cells.'},\n",
       " {'21269460': 'On the basis of large proteomics datasets measured from seven human cell lines we consider their intersection as an approximation of the human central proteome, which is the set of proteins ubiquitously expressed in all human cells. Composition and properties of the central proteome are investigated through bioinformatics analyses.'},\n",
       " {'21555454': 'Bromodomain protein 4 (Brd4) plays critical roles in development, cancer progression, and virus-host pathogenesis. To gain mechanistic insight into the various biological functions of Brd4, we performed a proteomic analysis to identify and characterize Brd4-associated cellular proteins. We found that the extraterminal (ET) domain, whose function has to date not been determined, interacts with NSD3, JMJD6, CHD4, GLTSCR1, and ATAD5. These ET-domain interactions were also conserved for Brd2 and Brd3, the other human BET proteins tested. We demonstrated that GLTSCR1, NSD3, and JMJD6 impart a pTEFb-independent transcriptional activation function on Brd4. NSD3 as well as JMJD6 is recruited to regulated genes in a Brd4-dependent manner. Moreover, we found that depletion of Brd4 or NSD3 reduces H3K36 methylation, demonstrating that the Brd4/NSD3 complex regulates this specific histone modification. Our results indicate that the Brd4 ET domain through the recruitment of the specific effectors regulates transcriptional activity. In particular, we show that one of these effectors, NSD3, regulates transcription by modifying the chromatin microenvironment at Brd4 target genes. Our study thus identifies the ET domain as a second important transcriptional regulatory domain for Brd4 in addition to the carboxyl-terminal domain (CTD) that interacts with pTEFb.'},\n",
       " {'21890894': 'Bromodomain-containing protein Brd4 is shown to persistently associate with chromosomes during mitosis for transmitting epigenetic memory across cell divisions. During interphase, Brd4 also plays a key role in regulating the transcription of signal-inducible genes by recruiting positive transcription elongation factor b (P-TEFb) to promoters. How the chromatin-bound Brd4 transits into a transcriptional regulation mode in response to stimulation, however, is largely unknown. Here, by analyzing the dynamics of Brd4 during ultraviolet or hexamethylene bisacetamide treatment, we show that the signal-induced release of chromatin-bound Brd4 is essential for its functional transition. In untreated cells, almost all Brd4 is observed in association with interphase chromatin. Upon treatment, Brd4 is released from chromatin, mostly due to signal-triggered deacetylation of nucleosomal histone H4 at acetylated-lysine 5/8 (H4K5ac/K8ac). Through selective association with the transcriptional active form of P-TEFb that has been liberated from the inactive multi-subunit complex in response to treatment, the released Brd4 mediates the recruitment of this active P-TEFb to promoter, which enhances transcription at the stage of elongation. Thus, through signal-induced release from chromatin and selective association with the active form of P-TEFb, the chromatin-bound Brd4 switches its role to mediate the recruitment of P-TEFb for regulating the transcriptional elongation of signal-inducible genes.'},\n",
       " {'21406692': 'To elucidate cellular events underlying the pluripotency of human embryonic stem cells (hESCs), we performed parallel quantitative proteomic and phosphoproteomic analyses of hESCs during differentiation initiated by a diacylglycerol analog or transfer to media that had not been conditioned by feeder cells. We profiled 6521 proteins and 23,522 phosphorylation sites, of which almost 50% displayed dynamic changes in phosphorylation status during 24 hours of differentiation. These data are a resource for studies of the events associated with the maintenance of hESC pluripotency and those accompanying their differentiation. From these data, we identified a core hESC phosphoproteome of sites with similar robust changes in response to the two distinct treatments. These sites exhibited distinct dynamic phosphorylation patterns, which were linked to known or predicted kinases on the basis of the matching sequence motif. In addition to identifying previously unknown phosphorylation sites on factors associated with differentiation, such as kinases and transcription factors, we observed dynamic phosphorylation of DNA methyltransferases (DNMTs). We found a specific interaction of DNMTs during early differentiation with the PAF1 (polymerase-associated factor 1) transcriptional elongation complex, which binds to promoters of the pluripotency and known DNMT target genes encoding OCT4 and NANOG, thereby providing a possible molecular link for the silencing of these genes during differentiation.'},\n",
       " {'22334664': 'Chromatin structure organization is crucial for regulating many fundamental cellular processes. However, the molecular mechanism that regulates the assembly of higher-order chromatin structure remains poorly understood. In this study, we demonstrate that Brd4 (bromodomain-containing protein 4) protein participates in the maintenance of the higher-order chromatin structure. Brd4, a member of the BET family of proteins, has been shown to play important roles in cellular growth control, cell cycle progression, and cancer development. We apply in situ single cell chromatin imaging and micrococcal nuclease (MNase) assay to show that Brd4 depletion leads to a large scale chromatin unfolding. A dominant-negative inhibitor encoding the double bromodomains (BDI/II) of Brd4 can competitively dissociate endogenous Brd4 from chromatin to trigger severely fragmented chromatin morphology. Mechanistic studies using Brd4 truncation mutants reveal that the Brd4 C-terminal domain is crucial for maintaining normal chromatin structure. Using bimolecular fluorescence complementation technology, we demonstrate that Brd4 molecules interact intermolecularly on chromatin and that replacing Brd4 molecules by BDI/II causes abnormal nucleosome aggregation and chromatin fragmentation. These studies establish a novel structural role of Brd4 in supporting the higher chromatin architecture.'},\n",
       " {'23086925': 'Transcriptional elongation by RNA polymerase II (Pol II) is regulated by positive transcription elongation factor b (P-TEFb) in association with bromodomain-containing protein 4 (BRD4). We used genome-wide chromatin immunoprecipitation sequencing in primary human CD4+ T cells to reveal that BRD4 co-localizes with Ser-2-phosphorylated Pol II (Pol II Ser-2) at both enhancers and promoters of active genes. Disruption of bromodomain-histone acetylation interactions by JQ1, a small-molecule bromodomain inhibitor, resulted in decreased BRD4 binding, reduced Pol II Ser-2, and reduced expression of lineage-specific genes in primary human CD4+ T cells. A large number of JQ1-disrupted BRD4 binding regions exhibited diacetylated H4 (lysine 5 and -8) and H3K27 acetylation (H3K27ac), which correlated with the presence of histone acetyltransferases and deacetylases. Genes associated with BRD4/H3K27ac co-occupancy exhibited significantly higher activity than those associated with H3K27ac or BRD4 binding alone. Comparison of BRD4 binding in T cells and in human embryonic stem cells revealed that enhancer BRD4 binding sites were predominantly lineage-specific. Our findings suggest that BRD4-driven Pol II phosphorylation at serine 2 plays an important role in regulating lineage-specific gene transcription in human CD4+ T cells.'},\n",
       " {'22509028': \"The bromodomain protein, BRD4, has been identified recently as a therapeutic target in acute myeloid leukemia, multiple myeloma, Burkitt's lymphoma, NUT midline carcinoma, colon cancer, and inflammatory disease; its loss is a prognostic signature for metastatic breast cancer. BRD4 also contributes to regulation of both cell cycle and transcription of oncogenes, HIV, and human papilloma virus (HPV). Despite its role in a broad range of biological processes, the precise molecular mechanism of BRD4 function remains unknown. We report that BRD4 is an atypical kinase that binds to the carboxyl-terminal domain (CTD) of RNA polymerase II and directly phosphorylates its serine 2 (Ser2) sites both in vitro and in vivo under conditions where other CTD kinases are inactive. Phosphorylation of the CTD Ser2 is inhibited in vivo by a BRD4 inhibitor that blocks its binding to chromatin. Our finding that BRD4 is an RNA polymerase II CTD Ser2 kinase implicates it as a regulator of eukaryotic transcription.\"},\n",
       " {'24360279': 'Distal enhancers characterized by the H3K4me(1) mark play critical roles in developmental and transcriptional programs. However, potential roles of specific distal regulatory elements in regulating RNA polymerase II (Pol II) promoter-proximal pause release remain poorly investigated. Here, we report that a unique cohort of jumonji C-domain-containing protein 6 (JMJD6) and bromodomain-containing protein 4 (Brd4) cobound distal enhancers, termed anti-pause enhancers (A-PEs), regulate promoter-proximal pause release of a large subset of transcription units via long-range interactions. Brd4-dependent JMJD6 recruitment on A-PEs mediates erasure of H4R3me(2(s)), which is directly read by 7SK snRNA, and decapping/demethylation of 7SK snRNA, ensuring the dismissal of the 7SK snRNA/HEXIM inhibitory complex. The interactions of both JMJD6 and Brd4 with the P-TEFb complex permit its activation and pause release of regulated coding genes. The functions of JMJD6/ Brd4-associated dual histone and RNA demethylase activity on anti-pause enhancers have intriguing implications for these proteins in development, homeostasis, and disease.'},\n",
       " {'23186163': 'Mass spectrometry (MS)-based phosphoproteomics has achieved extraordinary success in qualitative and quantitative analysis of cellular protein phosphorylation. Considering that an estimated level of phosphorylation in a cell is placed at well above 100,000 sites, there is still much room for improvement. Here, we attempt to extend the depth of phosphoproteome coverage while maintaining realistic aspirations in terms of available material, robustness, and instrument running time. We developed three strategies, where each provided a different balance between these three key parameters. The first strategy simply used enrichment by Ti(4+)-IMAC followed by reversed chromatography LC-MS (termed 1D). The second strategy incorporated an additional fractionation step through the use of HILIC (2D). Finally, a third strategy was designed employing first an SCX fractionation, followed by Ti(4+)-IMAC enrichment and additional fractionation by HILIC (3D). A preliminary evaluation was performed on the HeLa cell line. Detecting 3700 phosphopeptides in about 2 h, the 1D strategy was found to be the most sensitive but limited in comprehensivity, mainly due to issues with complexity and dynamic range. Overall, the best balance was achieved using the 2D based strategy, identifying close to 17,000 phosphopeptides with less than 1 mg of material in about 48 h. Subsequently, we confirmed the findings with the K562 cell sample. When sufficient material was available, the 3D strategy increased phosphoproteome allowing over 22,000 unique phosphopeptides to be identified. Unfortunately, the 3D strategy required more time and over 1 mg of material before it started to outperform 2D. Ultimately, combining all strategies, we were able to identify over 16,000 and nearly 24,000 unique phosphorylation sites from the cancer cell lines HeLa and K562, respectively. In summary, we demonstrate the need to carry out extensive fractionation for deep mining of the phosphoproteome and provide a guide for appropriate strategies depending on sample amount and/or analysis time.'},\n",
       " {'23317504': \"Bromodomain-containing protein 4 (Brd4) is an epigenetic reader and transcriptional regulator recently identified as a cancer therapeutic target for acute myeloid leukemia, multiple myeloma, and Burkitt's lymphoma. Although chromatin targeting is a crucial function of Brd4, there is little understanding of how bromodomains that bind acetylated histones are regulated, nor how the gene-specific activity of Brd4 is determined. Via interaction screen and domain mapping, we identified p53 as a functional partner of Brd4. Interestingly, Brd4 association with p53 is modulated by casein kinase II (CK2)-mediated phosphorylation of a conserved acidic region in Brd4 that selectively contacts either a juxtaposed bromodomain or an adjacent basic region to dictate the ability of Brd4 binding to chromatin and also the recruitment of p53 to regulated promoters. The unmasking of bromodomains and activator recruitment, concurrently triggered by the CK2 phospho switch, provide an intriguing mechanism for gene-specific targeting by a universal epigenetic reader.\"},\n",
       " {'23589332': 'RNA polymerase II (Pol II) and the pausing complex, NELF and DSIF, are detected near the transcription start site (TSS) of many active and silent genes. Active transcription starts when the pause release factor P-TEFb is recruited to initiate productive elongation. However, the mechanism of P-TEFb recruitment and regulation of NELF/DSIF during transcription is not fully understood. We investigated this question in interferon (IFN)-stimulated transcription, focusing on BRD4, a BET family protein that interacts with P-TEFb. Besides P-TEFb, BRD4 binds to acetylated histones through the bromodomain. We found that BRD4 and P-TEFb, although not present prior to IFN treatment, were robustly recruited to IFN-stimulated genes (ISGs) after stimulation. Likewise, NELF and DSIF prior to stimulation were hardly detectable on ISGs, which were strongly recruited after IFN treatment. A shRNA-based knockdown assay of NELF revealed that it negatively regulates the passage of Pol II and DSIF across the ISGs during elongation, reducing total ISG transcript output. Analyses with a BRD4 small-molecule inhibitor showed that IFN-induced recruitment of P-TEFb and NELF/DSIF was under the control of BRD4. We suggest a model where BRD4 coordinates both positive and negative regulation of ISG elongation.'},\n",
       " {'23728299': 'DNA damage activates a signalling network that blocks cell-cycle progression, recruits DNA repair factors and/or triggers senescence or programmed cell death. Alterations in chromatin structure are implicated in the initiation and propagation of the DNA damage response. Here we further investigate the role of chromatin structure in the DNA damage response by monitoring ionizing-radiation-induced signalling and response events with a high-content multiplex RNA-mediated interference screen of chromatin-modifying and -interacting genes. We discover that an isoform of Brd4, a bromodomain and extra-terminal (BET) family member, functions as an endogenous inhibitor of DNA damage response signalling by recruiting the condensin II chromatin remodelling complex to acetylated histones through bromodomain interactions. Loss of this isoform results in relaxed chromatin structure, rapid cell-cycle checkpoint recovery and enhanced survival after irradiation, whereas functional gain of this isoform compacted chromatin, attenuated DNA damage response signalling and enhanced radiation-induced lethality. These data implicate Brd4, previously known for its role in transcriptional control, as an insulator of chromatin that can modulate the signalling response to DNA damage.'},\n",
       " {'24275569': 'Protein phosphorylation is one of the most common post-translational modifications. It plays key roles in regulating diverse biological processes of liver tissues. To better understand the role of protein phosphorylation in liver functions, it is essential to perform in-depth phosphoproteome analysis of human liver. Here, an enzyme assisted reversed-phase-reversed-phase liquid chromatography (RP-RPLC) approach with both RPLC separations operated with optimized acidic mobile phase was developed. High orthogonal separation was achieved by trypsin digestion of the Glu-C generated peptides in the fractions collected from the first RPLC separation. The phosphoproteome coverage was further improved by using two types of instruments, i.e. TripleTOF 5600 and LTQ Orbitrap Velos. A total of 22,446 phosphorylation sites, corresponding to 6526 nonredundant phosphoproteins were finally identified from normal human liver tissues. Of these sites, 15,229 sites were confidently localized with Ascore≥13. This dataset was the largest phosphoproteome dataset of human liver. It can be a public resource for the liver research community and holds promise for further biology studies.'},\n",
       " {'25218447': 'SUMOylation is a reversible post-translational modification essential for genome stability. Using high-resolution MS, we have studied global SUMOylation in human cells in a site-specific manner, identifying a total of >4,300 SUMOylation sites in >1,600 proteins. To our knowledge, this is the first time that >1,000 SUMOylation sites have been identified under standard growth conditions. We quantitatively studied SUMOylation dynamics in response to SUMO protease inhibition, proteasome inhibition and heat shock. Many SUMOylated lysines have previously been reported to be ubiquitinated, acetylated or methylated, thus indicating cross-talk between SUMO and other post-translational modifications. We identified 70 phosphorylation and four acetylation events in proximity to SUMOylation sites, and we provide evidence for acetylation-dependent SUMOylation of endogenous histone H3. SUMOylation regulates target proteins involved in all nuclear processes including transcription, DNA repair, chromatin remodeling, precursor-mRNA splicing and ribosome assembly.'},\n",
       " {'25114211': 'SUMOylation is an essential ubiquitin-like modification involved in important biological processes in eukaryotic cells. Identification of small ubiquitin-related modifier (SUMO)-conjugated residues in proteins is critical for understanding the role of SUMOylation but remains experimentally challenging. We have set up a powerful and high-throughput method combining quantitative proteomics and peptide immunocapture to map SUMOylation sites and have analyzed changes in SUMOylation in response to stimuli. With this technique we identified 295 SUMO1 and 167 SUMO2 sites on endogenous substrates of human cells. We further used this strategy to characterize changes in SUMOylation induced by listeriolysin O, a bacterial toxin that impairs the host cell SUMOylation machinery, and identified several classes of host proteins specifically deSUMOylated in response to this toxin. Our approach constitutes an unprecedented tool, broadly applicable to various SUMO-regulated cellular processes in health and disease.'},\n",
       " {'25772364': 'Small ubiquitin-like modifiers play critical roles in the DNA damage response (DDR). To increase our understanding of SUMOylation in the mammalian DDR, we employed a quantitative proteomics approach in order to identify dynamically regulated SUMO-2 conjugates and modification sites upon treatment with the DNA damaging agent methyl methanesulfonate (MMS). We have uncovered a dynamic set of 20 upregulated and 33 downregulated SUMO-2 conjugates, and 755 SUMO-2 sites, of which 362 were dynamic in response to MMS. In contrast to yeast, where a response is centered on homologous recombination, we identified dynamically SUMOylated interaction networks of chromatin modifiers, transcription factors, DNA repair factors, and nuclear body components. SUMOylated chromatin modifiers include JARID1B/KDM5B, JARID1C/KDM5C, p300, CBP, PARP1, SetDB1, and MBD1. Whereas SUMOylated JARID1B was ubiquitylated by the SUMO-targeted ubiquitin ligase RNF4 and degraded by the proteasome in response to DNA damage, JARID1C was SUMOylated and recruited to the chromatin to demethylate histone H3K4.'},\n",
       " {'25593309': 'How chromatin shapes pathways that promote genome-epigenome integrity in response to DNA damage is an issue of crucial importance. We report that human bromodomain (BRD)-containing proteins, the primary \"readers\" of acetylated chromatin, are vital for the DNA damage response (DDR). We discovered that more than one-third of all human BRD proteins change localization in response to DNA damage. We identified ZMYND8 (zinc finger and MYND [myeloid, Nervy, and DEAF-1] domain containing 8) as a novel DDR factor that recruits the nucleosome remodeling and histone deacetylation (NuRD) complex to damaged chromatin. Our data define a transcription-associated DDR pathway mediated by ZMYND8 and the NuRD complex that targets DNA damage, including when it occurs within transcriptionally active chromatin, to repress transcription and promote repair by homologous recombination. Thus, our data identify human BRD proteins as key chromatin modulators of the DDR and provide novel insights into how DNA damage within actively transcribed regions requires chromatin-binding proteins to orchestrate the appropriate response in concordance with the damage-associated chromatin context.'},\n",
       " {'25755297': 'Genotoxic agents can cause replication fork stalling in dividing cells because of DNA lesions, eventually leading to replication fork collapse when the damage is not repaired. Small Ubiquitin-like Modifiers (SUMOs) are known to counteract replication stress, nevertheless, only a small number of relevant SUMO target proteins are known. To address this, we have purified and identified SUMO-2 target proteins regulated by replication stress in human cells. The developed methodology enabled single step purification of His10-SUMO-2 conjugates under denaturing conditions with high yield and high purity. Following statistical analysis on five biological replicates, a total of 566 SUMO-2 targets were identified. After 2 h of hydroxyurea treatment, 10 proteins were up-regulated for SUMOylation and two proteins were down-regulated for SUMOylation, whereas after 24 h, 35 proteins were up-regulated for SUMOylation, and 13 proteins were down-regulated for SUMOylation. A site-specific approach was used to map over 1000 SUMO-2 acceptor lysines in target proteins. The methodology is generic and is widely applicable in the ubiquitin field. A large subset of these identified proteins function in one network that consists of interacting replication factors, transcriptional regulators, DNA damage response factors including MDC1, ATR-interacting protein ATRIP, the Bloom syndrome protein and the BLM-binding partner RMI1, the crossover junction endonuclease EME1, BRCA1, and CHAF1A. Furthermore, centromeric proteins and signal transducers were dynamically regulated by SUMOylation upon replication stress. Our results uncover a comprehensive network of SUMO target proteins dealing with replication damage and provide a framework for detailed understanding of the role of SUMOylation to counteract replication stress. Ultimately, our study reveals how a post-translational modification is able to orchestrate a large variety of different proteins to integrate different nuclear processes with the aim of dealing with the induced DNA damage.'},\n",
       " {'28112733': 'Small ubiquitin-like modifiers (SUMOs) are post-translational modifications (PTMs) that regulate nuclear cellular processes. Here we used an augmented K0-SUMO proteomics strategy to identify 40,765 SUMO acceptor sites and quantify their fractional contribution for 6,747 human proteins. Structural-predictive analyses revealed that lysines residing in disordered regions are preferentially targeted by SUMO, in notable contrast to other widespread lysine modifications. In our data set, we identified 807 SUMOylated peptides that were co-modified by phosphorylation, along with dozens of SUMOylated peptides that were co-modified by ubiquitylation, acetylation and methylation. Notably, 9% of the identified SUMOylome occurred proximal to phosphorylation, and numerous SUMOylation sites were found to be fully dependent on prior phosphorylation events. SUMO-proximal phosphorylation occurred primarily in a proline-directed manner, and inhibition of cyclin-dependent kinases dynamically affected co-modification. Collectively, we present a comprehensive analysis of the SUMOylated proteome, uncovering the structural preferences for SUMO and providing system-wide evidence for a remarkable degree of cross-talk between SUMOylation and other major PTMs.'},\n",
       " {'29374058': 'The mammalian SWI/SNF chromatin remodeling complex is a heterogeneous collection of related protein complexes required for gene regulation and genome integrity. It contains a central ATPase (BRM or BRG1) and various combinations of 10-14 accessory subunits (BAFs for '},\n",
       " {'31969702': 'Proteins of the bromodomain and extra-terminal (BET) domain family are epigenetic readers that bind acetylated histones through their bromodomains to regulate gene transcription. Dual-bromodomain BET inhibitors (DbBi) that bind with similar affinities to the first (BD1) and second (BD2) bromodomains of BRD2, BRD3, BRD4 and BRDt have displayed modest clinical activity in monotherapy cancer trials. A reduced number of thrombocytes in the blood (thrombocytopenia) as well as symptoms of gastrointestinal toxicity are dose-limiting adverse events for some types of DbBi'},\n",
       " {'17344846': \"Cancers arise owing to mutations in a subset of genes that confer growth advantage. The availability of the human genome sequence led us to propose that systematic resequencing of cancer genomes for mutations would lead to the discovery of many additional cancer genes. Here we report more than 1,000 somatic mutations found in 274 megabases (Mb) of DNA corresponding to the coding exons of 518 protein kinase genes in 210 diverse human cancers. There was substantial variation in the number and pattern of mutations in individual cancers reflecting different exposures, DNA repair defects and cellular origins. Most somatic mutations are likely to be 'passengers' that do not contribute to oncogenesis. However, there was evidence for 'driver' mutations contributing to the development of the cancers studied in approximately 120 genes. Systematic sequencing of cancer genomes therefore reveals the evolutionary diversity of cancers and implicates a larger repertoire of cancer genes than previously anticipated.\"},\n",
       " {'18500820': \"Brd4 belongs to the BET family. It is a multifunctional protein involved in transcription, replication, the signal transduction pathway, and cell cycle progression. All of these functions are linked to its association with acetylated chromatin. With its tandem bromodomains, Brd4 avidly binds to diacetylated H4-AcK5/K12 and H3-AcK9/K14 peptides. Solution structure of the second bromodomain (BD) is reported in this research. In addition to the piD-helix, which is special for BET members, an incompact alphaZ' distinct from Brd2 BD2 is found, although they have identical sequences in this region. Both BD1 and BD2 bind to monoacetylated H4-AcK5 and H4-AcK12 peptides, but with subtle differences. An NMR perturbation study and mutational analysis identified the binding interface and revealed several residues important for binding specificity. By molecular dynamics simulations, a complex model composed of H4-AcK5/K12 and two molecules of BD2 is presented. Relaxation data and internal motions of BD2 are also discussed. Unlike Brd2 BD1, the two bromodomains of Brd4 are mainly monomeric in solution. They do not form heterodimers like TAFII250. It suggests that Brd4 should have its own mechanism to reinforce its chromatin association both in mitotic retention and related cellular regulation.\"},\n",
       " {'20871596': \"Epigenetic proteins are intently pursued targets in ligand discovery. So far, successful efforts have been limited to chromatin modifying enzymes, or so-called epigenetic 'writers' and 'erasers'. Potent inhibitors of histone binding modules have not yet been described. Here we report a cell-permeable small molecule (JQ1) that binds competitively to acetyl-lysine recognition motifs, or bromodomains. High potency and specificity towards a subset of human bromodomains is explained by co-crystal structures with bromodomain and extra-terminal (BET) family member BRD4, revealing excellent shape complementarity with the acetyl-lysine binding cavity. Recurrent translocation of BRD4 is observed in a genetically-defined, incurable subtype of human squamous carcinoma. Competitive binding by JQ1 displaces the BRD4 fusion oncoprotein from chromatin, prompting squamous differentiation and specific antiproliferative effects in BRD4-dependent cell lines and patient-derived xenograft models. These data establish proof-of-concept for targeting protein-protein interactions of epigenetic 'readers', and provide a versatile chemical scaffold for the development of chemical probes more broadly throughout the bromodomain family.\"},\n",
       " {'21068722': \"Interaction of pathogens with cells of the immune system results in activation of inflammatory gene expression. This response, although vital for immune defence, is frequently deleterious to the host due to the exaggerated production of inflammatory proteins. The scope of inflammatory responses reflects the activation state of signalling proteins upstream of inflammatory genes as well as signal-induced assembly of nuclear chromatin complexes that support mRNA expression. Recognition of post-translationally modified histones by nuclear proteins that initiate mRNA transcription and support mRNA elongation is a critical step in the regulation of gene expression. Here we present a novel pharmacological approach that targets inflammatory gene expression by interfering with the recognition of acetylated histones by the bromodomain and extra terminal domain (BET) family of proteins. We describe a synthetic compound (I-BET) that by 'mimicking' acetylated histones disrupts chromatin complexes responsible for the expression of key inflammatory genes in activated macrophages, and confers protection against lipopolysaccharide-induced endotoxic shock and bacteria-induced sepsis. Our findings suggest that synthetic compounds specifically targeting proteins that recognize post-translationally modified histones can serve as a new generation of immunomodulatory drugs.\"},\n",
       " {'21568322': 'Epigenetic mechanisms of gene regulation have a profound role in normal development and disease processes. An integral part of this mechanism occurs through lysine acetylation of histone tails which are recognized by bromodomains. While the biological and structural characterization of many bromodomain containing proteins has advanced considerably, the therapeutic tractability of this protein family is only now becoming understood. This paper describes the discovery and molecular characterization of potent (nM) small molecule inhibitors that disrupt the function of the BET family of bromodomains (Brd2, Brd3, and Brd4). By using a combination of phenotypic screening, chemoproteomics, and biophysical studies, we have discovered that the protein-protein interactions between bromodomains and acetylated histones can be antagonized by selective small molecules that bind at the acetylated lysine recognition pocket. X-ray crystal structures of compounds bound into bromodomains of Brd2 and Brd4 elucidate the molecular interactions of binding and explain the precisely defined stereochemistry required for activity.'},\n",
       " {'21964340': \"Recurrent chromosomal translocations involving the mixed lineage leukaemia (MLL) gene initiate aggressive forms of leukaemia, which are often refractory to conventional therapies. Many MLL-fusion partners are members of the super elongation complex (SEC), a critical regulator of transcriptional elongation, suggesting that aberrant control of this process has an important role in leukaemia induction. Here we use a global proteomic strategy to demonstrate that MLL fusions, as part of SEC and the polymerase-associated factor complex (PAFc), are associated with the BET family of acetyl-lysine recognizing, chromatin 'adaptor' proteins. These data provided the basis for therapeutic intervention in MLL-fusion leukaemia, via the displacement of the BET family of proteins from chromatin. We show that a novel small molecule inhibitor of the BET family, GSK1210151A (I-BET151), has profound efficacy against human and murine MLL-fusion leukaemic cell lines, through the induction of early cell cycle arrest and apoptosis. I-BET151 treatment in two human leukaemia cell lines with different MLL fusions alters the expression of a common set of genes whose function may account for these phenotypic changes. The mode of action of I-BET151 is, at least in part, due to the inhibition of transcription at key genes (BCL2, C-MYC and CDK6) through the displacement of BRD3/4, PAFc and SEC components from chromatin. In vivo studies indicate that I-BET151 has significant therapeutic value, providing survival benefit in two distinct mouse models of murine MLL-AF9 and human MLL-AF4 leukaemia. Finally, the efficacy of I-BET151 against human leukaemia stem cells is demonstrated, providing further evidence of its potent therapeutic potential. These findings establish the displacement of BET proteins from chromatin as a promising epigenetic therapy for these aggressive leukaemias.\"},\n",
       " {'22137933': 'Benzodiazepines are psychoactive drugs with anxiolytic, sedative, skeletal muscle relaxant and amnestic properties. Recently triazolo-benzodiazepines have been also described as potent and highly selective protein interaction inhibitors of bromodomain and extra-terminal (BET) proteins, a family of transcriptional co-regulators that play a key role in cancer cell survival and proliferation, but the requirements for high affinity interaction of this compound class with bromodomains has not been described. Here we provide insight into the structure-activity relationship (SAR) and selectivity of this versatile scaffold. In addition, using high resolution crystal structures we compared the binding mode of a series of benzodiazepine (BzD) and related triazolo-benzotriazepines (BzT) derivatives including clinically approved drugs such as alprazolam and midazolam. Our analysis revealed the importance of the 1-methyl triazolo ring system for BET binding and suggests modifications for the development of further high affinity bromodomain inhibitors.'},\n",
       " {'22645123': 'NF-κB-mediated inflammation is the major pathology in chronic kidney diseases, including HIV-associated nephropathy (HIVAN) that ultimately progresses to end stage renal disease. HIV infection in the kidney induces NF-κB activation, leading to the production of proinflammatory chemokines, cytokines, and adhesion molecules. In this study, we explored selective inhibition of NF-κB transcriptional activity by small molecule blocking NF-κB binding to the transcriptional cofactor BRD4, which is required for the assembly of the productive transcriptional complex comprising positive transcription elongation factor b and RNA polymerase II. We showed that our BET (Bromodomain and Extra-Terminal domain)-specific bromodomain inhibitor MS417, designed to block BRD4 binding to the acetylated NF-κB, effectively attenuates NF-κB transcriptional activation of proinflammatory genes in kidney cells treated with TNFα or infected by HIV. MS417 ameliorates inflammation and kidney injury in HIV-1 transgenic mice, an animal model for HIVAN. Our study suggests that BET bromodomain inhibition, targeting at the proinflammatory activity of NF-κB, represents a new therapeutic approach for treating NF-κB-mediated inflammation and kidney injury in HIVAN.'},\n",
       " {'22136404': 'Bromodomain-containing proteins are key epigenetic regulators of gene transcription and readers of the histone code. However, the therapeutic benefits of modulating this target class are largely unexplored due to the lack of suitable chemical probes. This article describes the generation of lead molecules for the BET bromodomains through screening a fragment set chosen using structural insights and computational approaches. Analysis of 40 BRD2/fragment X-ray complexes highlights both shared and disparate interaction features that may be exploited for affinity and selectivity. Six representative crystal structures are then exemplified in detail. Two of the fragments are completely new bromodomain chemotypes, and three have never before been crystallized in a bromodomain, so our results significantly extend the limited public knowledge-base of crystallographic small molecule/bromodomain interactions. Certain fragments (including paracetamol) bind in a consistent mode to different bromodomains such as CREBBP, suggesting their potential to act as generic bromodomain templates. An important implication is that the bromodomains are not only a phylogenetic family but also a system in which chemical and structural knowledge of one bromodomain gives insights transferrable to others.'},\n",
       " {'23095041': 'The posttranslational modification of chromatin through acetylation at selected histone lysine residues is governed by histone acetyltransferases (HATs) and histone deacetylases (HDACs). The significance of this subset of the epigenetic code is interrogated and interpreted by an acetyllysine-specific protein-protein interaction with bromodomain reader modules. Selective inhibition of the bromo and extra C-terminal domain (BET) family of bromodomains with a small molecule is feasible, and this may represent an opportunity for disease intervention through the recently disclosed antiproliferative and anti-inflammatory properties of such inhibitors. Herein, we describe the discovery and structure-activity relationship (SAR) of a novel, small-molecule chemical probe for BET family inhibition that was identified through the application of structure-based fragment assessment and optimization techniques. This has yielded a potent, selective compound with cell-based activity (PFI-1) that may further add to the understanding of BET family function within the bromodomains.'},\n",
       " {'22464331': 'Bromodomains (BRDs) are protein interaction modules that specifically recognize ε-N-lysine acetylation motifs, a key event in the reading process of epigenetic marks. The 61 BRDs in the human genome cluster into eight families based on structure/sequence similarity. Here, we present 29 high-resolution crystal structures, covering all BRD families. Comprehensive crossfamily structural analysis identifies conserved and family-specific structural features that are necessary for specific acetylation-dependent substrate recognition. Screening of\\xa0more than 30 representative BRDs against systematic histone-peptide arrays identifies new BRD substrates and reveals a strong influence of flanking posttranslational modifications, such as acetylation and phosphorylation, suggesting that BRDs recognize combinations of marks rather than singly acetylated sequences. We further uncovered a structural mechanism for the simultaneous binding and recognition of diverse diacetyl-containing peptides by BRD4. These data provide a foundation for structure-based drug design of specific inhibitors for this emerging target family.'},\n",
       " {'23517011': 'The bromodomain protein module, which binds to acetylated lysine, is emerging as an important epigenetic therapeutic target. We report the structure-guided optimization of 3,5-dimethylisoxazole derivatives to develop potent inhibitors of the BET (bromodomain and extra terminal domain) bromodomain family with good ligand efficiency. X-ray crystal structures of the most potent compounds reveal key interactions required for high affinity at BRD4(1). Cellular studies demonstrate that the phenol and acetate derivatives of the lead compounds showed strong antiproliferative effects on MV4;11 acute myeloid leukemia cells, as shown for other BET bromodomain inhibitors and genetic BRD4 knockdown, whereas the reported compounds showed no general cytotoxicity in other cancer cell lines tested.'},\n",
       " {'23530754': 'Recognizing acetyllysine of histone is a vital process of epigenetic regulation that is mediated by a protein module called bromodomain. To contribute novel scaffolds for developing into bromodomain inhibitors, we utilize a fragment-based drug discovery approach. By successively applying docking and X-ray crystallography, we were able to identify 9 fragment hits from diffracting more than 60 crystals. In the present work, we described four of them and carried out the integrated lead optimization for fragment 8, which bears a 2-thiazolidinone core. After several rounds of structure guided modifications, we assessed the druggability of 2-thiazolidinone by modulating in vitro pharmacokinetic studies and cellular activity assay. The results showed that two potent compounds of 2-thiazolidinones have good metabolic stability. Also, the cellular assay confirmed the activities of 2-thiazolidinones. Together, we hope the identified 2-thiazolidinone chemotype and other fragment hits described herein can stimulate researchers to develop more diversified bromodomain inhibitors.'},\n",
       " {'29176719': 'Jumonji domain-containing protein 6 (JMJD6) is a member of the Jumonji C family of Fe(II) and 2-oxoglutarate (2OG) dependent oxygenases. It possesses unique bi-functional oxygenase activities, acting as both an arginine demethylase and a lysyl-hydroxylase. JMJD6 has been reported to be over-expressed in oral, breast, lung, and colon cancers and plays important roles in regulation of transcription through interactions with transcription regulator BRD4, histones, U2AF65, Luc7L3, and SRSF11. Here, we report a structural mechanism revealed by NMR of JMJD6 recognition by the extraterminal (ET) domain of BRD4 in that a JMJD6 peptide (Lys84-Asn96) adapts an α-helix when bound to the ET domain. This intermolecular recognition is established through JMJD6 interactions with the conserved hydrophobic core of the ET domain, and reinforced by electrostatic interactions of JMJD6 with residues in the inter-helical α1-α2 loop of the ET domain. Notably, this mode of ligand recognition is different from that of ET domain recognition of NSD3, LANA of herpesvirus, and integrase of MLV, which involves formation of an intermolecular amphipathic two- or three- strand antiparallel β sheet. Furthermore, we demonstrate that the association between the BRD4 ET domain and JMJD6 likely requires a protein conformational change induced by single-stranded RNA binding.'},\n",
       " {'32193360': 'The two tandem bromodomains of the BET (bromodomain and extraterminal domain) proteins enable chromatin binding to facilitate transcription. Drugs that inhibit both bromodomains equally have shown efficacy in certain malignant and inflammatory conditions. To explore the individual functional contributions of the first (BD1) and second (BD2) bromodomains in biology and therapy, we developed selective BD1 and BD2 inhibitors. We found that steady-state gene expression primarily requires BD1, whereas the rapid increase of gene expression induced by inflammatory stimuli requires both BD1 and BD2 of all BET proteins. BD1 inhibitors phenocopied the effects of pan-BET inhibitors in cancer models, whereas BD2 inhibitors were predominantly effective in models of inflammatory and autoimmune disease. These insights into the differential requirement of BD1 and BD2 for the maintenance and induction of gene expression may guide future BET-targeted therapies.'},\n",
       " {'29379197': 'We found that the clinical phenotype associated with BRD4 haploinsufficiency overlapped with that of Cornelia de Lange syndrome (CdLS), which is most often caused by mutation of NIPBL. More typical CdLS was observed with a de novo BRD4 missense variant, which retained the ability to coimmunoprecipitate with NIPBL, but bound poorly to acetylated histones. BRD4 and NIPBL displayed correlated binding at super-enhancers and appeared to co-regulate developmental gene expression.'},\n",
       " {'29440723': 'In the version of this article initially published, Wendy Bickmore and Madapura Pradeepa were incorrectly not indicated as corresponding authors. The error has been corrected in the HTML and PDF versions of the paper.'},\n",
       " {'31168063': 'An amendment to this paper has been published and can be accessed via a link at the top of the paper.'},\n",
       " {'35470444': 'BRD4 is part of a multiprotein complex involved in loading the cohesin complex onto DNA, a fundamental process required for cohesin-mediated loop extrusion and formation of Topologically Associating Domains. Pathogenic variations in this complex have been associated with a growing number of syndromes, collectively known as cohesinopathies, the most classic being Cornelia de Lange syndrome. However, no cohort study has been conducted to delineate the clinical and molecular spectrum of BRD4-related disorder. We formed an international collaborative study, and collected 14 new patients, including two fetuses. We performed phenotype and genotype analysis, integrated prenatal findings from fetopathological examinations, phenotypes of pediatric patients and adults. We report the first cohort of patients with BRD4-related disorder and delineate the dysmorphic features at different ages. This work extends the phenotypic spectrum of cohesinopathies and characterize a new clinically relevant and recognizable pattern, distinguishable from the other cohesinopathies.'}]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "abstracts_to_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'11733348': []},\n",
       " {'15057824': []},\n",
       " {'15489334': []},\n",
       " {'12543779': []},\n",
       " {'16109376': []},\n",
       " {'16109377': ['complex, a potential']},\n",
       " {'17081983': []},\n",
       " {'16940503': []},\n",
       " {'17189189': []},\n",
       " {'18922874': []},\n",
       " {'18669648': []},\n",
       " {'19413330': []},\n",
       " {'19596240': []},\n",
       " {'19103749': []},\n",
       " {'19608861': []},\n",
       " {'20068231': []},\n",
       " {'21269460': ['Composition and properties']},\n",
       " {'21555454': []},\n",
       " {'21890894': []},\n",
       " {'21406692': []},\n",
       " {'22334664': []},\n",
       " {'23086925': []},\n",
       " {'22509028': []},\n",
       " {'24360279': []},\n",
       " {'23186163': []},\n",
       " {'23317504': []},\n",
       " {'23589332': []},\n",
       " {'23728299': []},\n",
       " {'24275569': []},\n",
       " {'25218447': []},\n",
       " {'25114211': []},\n",
       " {'25772364': []},\n",
       " {'25593309': []},\n",
       " {'25755297': []},\n",
       " {'28112733': []},\n",
       " {'29374058': []},\n",
       " {'31969702': []},\n",
       " {'17344846': []},\n",
       " {'18500820': []},\n",
       " {'20871596': []},\n",
       " {'21068722': []},\n",
       " {'21568322': []},\n",
       " {'21964340': []},\n",
       " {'22137933': []},\n",
       " {'22645123': []},\n",
       " {'22136404': []},\n",
       " {'23095041': []},\n",
       " {'22464331': []},\n",
       " {'23517011': []},\n",
       " {'23530754': []},\n",
       " {'29176719': ['cancers and plays']},\n",
       " {'32193360': []},\n",
       " {'29379197': []},\n",
       " {'29440723': []},\n",
       " {'31168063': []},\n",
       " {'35470444': []}]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "possible_matches = []\n",
    "gene = 'CAP'\n",
    "for abstract_dict in abstracts_to_check:\n",
    "    keys = list(abstract_dict.keys())\n",
    "    possible_match = {}\n",
    "    for key in keys:\n",
    "        result = find_matching_groups(abstract_dict[key],gene)\n",
    "        possible_match[key] = result\n",
    "    possible_matches.append(possible_match)\n",
    "\n",
    "possible_matches\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_matching_groups_og(abstract, gene_symbol):\n",
    "    abstract = abstract.replace('-', ' ')\n",
    "    # Split the abstract into individual words\n",
    "    words = abstract.split()\n",
    "    # Determine the length of the gene symbol to set the size of each word group\n",
    "    gene_length = len(gene_symbol)\n",
    "    # Initialize a list to store matching groups\n",
    "    matching_groups = []\n",
    "\n",
    "    # Loop through the words to form groups of size gene_length\n",
    "    for i in range(len(words) - gene_length + 1):\n",
    "        # Select the current group of words\n",
    "        group = words[i:i + gene_length]\n",
    "        # Check if the first letter of each word in the group matches the letters in the gene symbol\n",
    "        if all(group[j][0].lower() == gene_symbol[j].lower() for j in range(gene_length)):\n",
    "            # If the group matches, add it to the list as a string\n",
    "            matching_groups.append(' '.join(group))\n",
    "\n",
    "    return matching_groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chromosome associated protein']"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_abstract = 'We describe a novel nuclear factor called mitotic chromosome-associated protein (MCAP), which belongs to the poorly understood BET subgroup of the bromodomain superfamily. Expression of the 200-kDa MCAP was linked to cell division, as it was induced by growth stimulation and repressed by growth inhibition. The most notable feature of MCAP was its association with chromosomes during mitosis, observed at a time when the majority of nuclear regulatory factors were released into the cytoplasm, coinciding with global cessation of transcription. Indicative of its predominant interaction with euchromatin, MCAP localized on mitotic chromosomes with exquisite specificity: (i) MCAP-chromosome association became evident subsequent to the initiation of histone H3 phosphorylation and early chromosomal condensation; and (ii) MCAP was absent from centromeres, the sites of heterochromatin. Supporting a role for MCAP in G(2)/M transition, microinjection of anti-MCAP antibody into HeLa cell nuclei completely inhibited the entry into mitosis, without abrogating the ongoing DNA replication. These results suggest that MCAP plays a role in a process governing chromosomal dynamics during mitosis.'\n",
    "\n",
    "gene_symbol = 'CAP'\n",
    "\n",
    "find_matching_groups(test_abstract,gene_symbol)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Translocation t(15;19)(q13;p13.1) defines a lethal midline carcinoma arising adjacent to respiratory tract in young people. To characterize molecular alterations responsible for the distinctly aggressive biological behavior of this cancer, we mapped the chromosome 15 and 19 translocation breakpoints by fluorescence in situ hybridization (FISH) and Southern blotting. To evaluate preliminarily the frequency, anatomical distribution, and histological features of t(15;19) cancer, we developed a FISH assay for paraffin sections. Our findings reveal a novel oncogenic mechanism in which the chromosome 19 translocation breakpoint interrupts the coding sequence of a bromodomain gene, BRD4. These studies implicate BRD4 as a potential partner in a t(15;19)-associated fusion oncogene. In addition, we localized the chromosome 15 breakpoint to a 9-kb region in each of two cases, thereby identifying several candidate oncogenes which might represent the BRD4 fusion partner. FISH evaluation of 13 pediatric carcinomas revealed t(15;19) in one of four sinonasal carcinomas, whereas this translocation was not detected in thymic (n = 3), mucoepidermoid (n = 3), laryngeal (n = 2), or nasopharyngeal (n = 1) carcinomas. Our studies shed light on the oncogenic mechanism underlying t(15;19) and provide further evidence that this highly lethal cancer arises from respiratory mucosa.',\n",
       " 'Chromosome 19 has the highest gene density of all human chromosomes, more than double the genome-wide average. The large clustered gene families, corresponding high G + C content, CpG islands and density of repetitive DNA indicate a chromosome rich in biological and evolutionary significance. Here we describe 55.8 million base pairs of highly accurate finished sequence representing 99.9% of the euchromatin portion of the chromosome. Manual curation of gene loci reveals 1,461 protein-coding genes and 321 pseudogenes. Among these are genes directly implicated in mendelian disorders, including familial hypercholesterolaemia and insulin-resistant diabetes. Nearly one-quarter of these genes belong to tandemly arranged families, encompassing more than 25% of the chromosome. Comparative analyses show a fascinating picture of conservation and divergence, revealing large blocks of gene orthology with rodents, scattered regions with more recent gene family expansions and deletions, and segments of coding and non-coding conservation with the distant fish species Takifugu.',\n",
       " \"The National Institutes of Health's Mammalian Gene Collection (MGC) project was designed to generate and sequence a publicly accessible cDNA resource containing a complete open reading frame (ORF) for every human and mouse gene. The project initially used a random strategy to select clones from a large number of cDNA libraries from diverse tissues. Candidate clones were chosen based on 5'-EST sequences, and then fully sequenced to high accuracy and analyzed by algorithms developed for this project. Currently, more than 11,000 human and 10,000 mouse genes are represented in MGC by at least one clone with a full ORF. The random selection approach is now reaching a saturation point, and a transition to protocols targeted at the missing transcripts is now required to complete the mouse and human collections. Comparison of the sequence of the MGC clones to reference genome sequences reveals that most cDNA clones are of very high sequence quality, although it is likely that some cDNAs may carry missense variants as a consequence of experimental artifact, such as PCR, cloning, or reverse transcriptase errors. Recently, a rat cDNA component was added to the project, and ongoing frog (Xenopus) and zebrafish (Danio) cDNA projects were expanded to take advantage of the high-throughput MGC pipeline.\",\n",
       " 'The poorly differentiated carcinoma with t(15;19)(q13, p13.1) is characterized by its highly aggressive, invariably lethal clinical course. The chromosome 19 translocation breakpoint targets the BRD4 double bromodomain-containing gene, which functions in regulation of cell cycle progression. Herein we demonstrate that BRD4 is fused with nearly the entire transcript of the novel 15q13 gene, NUT (nuclear protein in testis), forming a 6.4-kb fusion oncogene, BRD4-NUT. NUT, like BRD4, is predicted to encode a nuclear protein but, unlike the ubiquitous BRD4 transcript, is expressed only in testis. These findings establish a model to elucidate the oncogenic consequences of unscheduled NUT expression and altered BRD4 function. Very few fusion oncogenes have been identified in epithelial tumors, and BRD4-NUT is the first fusion oncogene mechanism identified in a highly lethal form of carcinoma.',\n",
       " 'Brd4 is a mammalian bromodomain protein that binds to acetylated chromatin. Proteomic analysis revealed that Brd4 interacts with cyclinT1 and Cdk9 that constitutes core positive transcription elongation factor b (P-TEFb). Brd4 interacted with P-TEFb in the living nucleus through its bromodomain. About half of P-TEFb was bound to the inhibitory subunit and functionally inactive. Brd4 interacted with P-TEFb that was free of the inhibitory subunit. An increase in Brd4 expression led to increased P-TEFb-dependent phosphorylation of RNA polymerase II (RNAPII) CTD and stimulation of transcription from promoters in vivo. Conversely, a reduction in Brd4 expression by siRNA reduced CTD phosphorylation and transcription, revealing that Brd4 is a positive regulatory component of P-TEFb. In chromatin immunoprecipitation (ChIP) assays, the recruitment of P-TEFb to a promoter was dependent on Brd4 and was enhanced by an increase in chromatin acetylation. Together, P-TEFb alternately interacts with Brd4 and the inhibitory subunit to maintain functional equilibrium in the cell.',\n",
       " 'The cyclinT1/Cdk9 heterodimer that constitutes core P-TEFb is generally presumed to be the transcriptionally active form for stimulating RNA polymerase II elongation. About half of cellular P-TEFb also exists in an inactive complex with the 7SK snRNA and the HEXIM1 protein. Here, we show that the remaining half associates with the bromodomain protein Brd4. In stress-induced cells, the 7SK/HEXIM1-bound P-TEFb is quantitatively converted into the Brd4-associated form. The association with Brd4 is necessary to form the transcriptionally active P-TEFb, recruits P-TEFb to a promoter, and enables P-TEFb to contact the Mediator complex, a potential target for the Brd4-mediated recruitment. Although generally required for transcription, the P-TEFb-recruitment function of Brd4 can be substituted by that of HIV-1 Tat, which recruits P-TEFb directly for activated HIV-1 transcription. Brd4, HEXIM1, and 7SK are all implicated in regulating cell growth, which may result from their dynamic control of the general transcription factor P-TEFb.',\n",
       " 'Cell signaling mechanisms often transmit information via posttranslational protein modifications, most importantly reversible protein phosphorylation. Here we develop and apply a general mass spectrometric technology for identification and quantitation of phosphorylation sites as a function of stimulus, time, and subcellular location. We have detected 6,600 phosphorylation sites on 2,244 proteins and have determined their temporal dynamics after stimulating HeLa cells with epidermal growth factor (EGF) and recorded them in the Phosida database. Fourteen percent of phosphorylation sites are modulated at least 2-fold by EGF, and these were classified by their temporal profiles. Surprisingly, a majority of proteins contain multiple phosphorylation sites showing different kinetics, suggesting that they serve as platforms for integrating signals. In addition to protein kinase cascades, the targets of reversible phosphorylation include ubiquitin ligases, guanine nucleotide exchange factors, and at least 46 different transcriptional regulators. The dynamic phosphoproteome provides a missing link in a global, integrative view of cellular regulation.',\n",
       " \"The latency-associated nuclear antigen (LANA) of Kaposi's sarcoma-associated herpesvirus (KSHV) is required for viral episome maintenance in host cells during latent infection. Two regions of the protein have been implicated in tethering LANA/viral episomes to the host mitotic chromosomes, and LANA chromosome-binding sites are subjects of high interest. Because previous studies had identified bromodomain protein Brd4 as the mitotic chromosome anchor for the bovine papillomavirus E2 protein, which tethers the viral episomes to host mitotic chromosomes (J. You, J. L. Croyle, A. Nishimura, K. Ozato, and P. M. Howley, Cell 117:349-360, 2004, and J. You, M. R. Schweiger, and P. M. Howley, J. Virol. 79:14956-14961, 2005), we examined whether KSHV LANA interacts with Brd4. We found that LANA binds Brd4 in vivo and in vitro and that the binding is mediated by a direct protein-protein interaction between the ET (extraterminal) domain of Brd4 and a carboxyl-terminal region of LANA previously implicated in chromosome binding. Brd4 associates with mitotic chromosomes throughout mitosis and demonstrates a strong colocalization with LANA and the KSHV episomes on host mitotic chromosomes. Although another bromodomain protein, RING3/Brd2, binds to LANA in a similar fashion in vitro, it is largely excluded from the mitotic chromosomes in KSHV-uninfected cells and is partially recruited to the chromosomes in KSHV-infected cells. These data identify Brd4 as an interacting protein for the carboxyl terminus of LANA on mitotic chromosomes and suggest distinct functional roles for the two bromodomain proteins RING3/Brd2 and Brd4 in LANA binding. Additionally, because Brd4 has recently been shown to have a role in transcription, we examined whether Brd4 can regulate the CDK2 promoter, which can be transactivated by LANA.\",\n",
       " 'Autonomously replicating DNA viruses must evade mitotic checkpoints and actively partition their genomes to maintain persistent infection. The E2 protein serves these functions by tethering papillomavirus episomes to mitotic chromosomes; however, the mechanism remains unresolved. We show that E2 binds ChlR1, a DNA helicase that plays a role in sister chromatid cohesion. The E2 mutation W130R fails to bind ChlR1 and correspondingly does not associate with mitotic chromosomes. Viral genomes encoding this E2 mutation are not episomally maintained in cell culture. Notably, E2 W130R binds Brd4, which reportedly acts as a mitotic tether, indicating this interaction is insufficient for E2 association with mitotic chromosomes. RNAi-induced depletion of ChlR1 significantly reduced E2 localization to mitotic chromosomes. These studies provide compelling evidence that ChlR1 association is required for loading the papillomavirus E2 protein onto mitotic chromosomes and represents a kinetochore-independent mechanism for viral genome maintenance and segregation.',\n",
       " \"The EBNA1 protein of Epstein-Barr virus (EBV) is essential for EBV latent infection in ensuring the replication and stable segregation of the EBV genomes and in activating the transcription of other EBV latency genes. We have tested the ability of four host proteins (Brd2, Brd4, DEK, and MeCP2) implicated in the segregation of papillomavirus and Kaposi's sarcoma-associated herpesvirus to support EBNA1-mediated segregation of EBV-based plasmids in Saccharomyces cerevisiae. We found that Brd4 enabled EBNA1-mediated segregation while Brd2 and MeCP2 had a general stimulatory effect on plasmid maintenance. EBNA1 interacted with Brd4 in both yeast and human cells through N-terminal sequences previously shown to mediate transcriptional activation but not segregation. In keeping with this interaction site, silencing of Brd4 in human cells decreased transcriptional activation by EBNA1 but not the mitotic chromosome attachment of EBNA1 that is required for segregation. In addition, Brd4 was found to be preferentially localized to the FR enhancer element regulated by EBNA1, over other EBV sequences, in latently EBV-infected cells. The results indicate that EBNA1 can functionally interact with Brd4 in native and heterologous systems and that this interaction facilitates transcriptional activation by EBNA1 from the FR element.\",\n",
       " 'The eukaryotic cell division cycle is characterized by a sequence of orderly and highly regulated events resulting in the duplication and separation of all cellular material into two newly formed daughter cells. Protein phosphorylation by cyclin-dependent kinases (CDKs) drives this cycle. To gain further insight into how phosphorylation regulates the cell cycle, we sought to identify proteins whose phosphorylation is cell cycle regulated. Using stable isotope labeling along with a two-step strategy for phosphopeptide enrichment and high mass accuracy mass spectrometry, we examined protein phosphorylation in a human cell line arrested in the G(1) and mitotic phases of the cell cycle. We report the identification of >14,000 different phosphorylation events, more than half of which, to our knowledge, have not been described in the literature, along with relative quantitative data for the majority of these sites. We observed >1,000 proteins with increased phosphorylation in mitosis including many known cell cycle regulators. The majority of sites on regulated phosphopeptides lie in [S/T]P motifs, the minimum required sequence for CDKs, suggesting that many of the proteins may be CDK substrates. Analysis of non-proline site-containing phosphopeptides identified two unique motifs that suggest there are at least two undiscovered mitotic kinases.',\n",
       " 'The analysis of proteome-wide phosphorylation events is still a major analytical challenge because of the enormous complexity of protein phosphorylation networks. In this work, we evaluate the complementarity of Lys-N, Lys-C, and trypsin with regard to their ability to contribute to the global analysis of the phosphoproteome. A refined version of low-pH strong cation exchange was used to efficiently separate N-terminally acetylated, phosphorylated, and nonmodified peptides. A total of 5036 nonredundant phosphopeptides could be identified with a false discovery rate of <1% from 1 mg of protein using a combination of the three enzymes. Our data revealed that the overlap between the phosphopeptide data sets generated with different proteases was marginal, whereas the overlap between two similarly generated tryptic data sets was found to be at least 4 times higher. In this way, the parallel use of Lys-N and trypsin enabled a 72% increase in the number of detected phosphopeptides as compared to trypsin alone, whereas a trypsin replicate experiment only led to a 25% increase. Thus, when focusing solely on the trypsin and Lys-N data, we identified 4671 nonredundant phosphopeptides. Further analysis of the detected sites showed that the Lys-N and trypsin data sets were enriched in significantly different phosphorylation motifs, further evidencing that multiprotease approaches are very valuable in phosphoproteome analyses.',\n",
       " 'Most inducible transcriptional programs consist of primary and secondary response genes (PRGs and SRGs) that differ in their kinetics of expression and in their requirements for new protein synthesis and chromatin remodeling. Here we show that many PRGs, in contrast to SRGs, have preassembled RNA polymerase II (Pol II) and positive histone modifications at their promoters in the basal state. Pol II at PRGs generates low levels of full-length unspliced transcripts but fails to make mature, protein-coding transcripts in the absence of stimulation. Induction of PRGs is controlled at the level of transcriptional elongation and mRNA processing, through the signal-dependent recruitment of P-TEFb. P-TEFb is in turn recruited by the bromodomain-containing protein Brd4, which detects H4K5/8/12Ac inducibly acquired at PRG promoters. Our findings suggest that the permissive structure of PRGs both stipulates their unique regulation in the basal state by corepressor complexes and enables their rapid induction in multiple cell types.',\n",
       " 'Acetylation of the RelA subunit of NF-kappaB, especially at lysine-310, is critical for the transcriptional activation of NF-kappaB and the expression of inflammatory genes. In this study, we demonstrate that bromodomains of Brd4 bind to acetylated lysine-310. Brd4 enhances transcriptional activation of NF-kappaB and the expression of a subset of NF-kappaB-responsive inflammatory genes in an acetylated lysine-310-dependent manner. Bromodomains of Brd4 and acetylated lysine-310 of RelA are both required for the mutual interaction and coactivation function of Brd4. Finally, we demonstrate that Brd4 further recruits CDK9 to phosphorylate C-terminal domain of RNA polymerase II and facilitate the transcription of NF-kappaB-dependent inflammatory genes. Our results identify Brd4 as a novel coactivator of NF-kappaB through specifically binding to acetylated lysine-310 of RelA. In addition, these studies reveal a mechanism by which acetylated RelA stimulates the transcriptional activity of NF-kappaB and the NF-kappaB-dependent inflammatory response.',\n",
       " \"Lysine acetylation is a reversible posttranslational modification of proteins and plays a key role in regulating gene expression. Technological limitations have so far prevented a global analysis of lysine acetylation's cellular roles. We used high-resolution mass spectrometry to identify 3600 lysine acetylation sites on 1750 proteins and quantified acetylation changes in response to the deacetylase inhibitors suberoylanilide hydroxamic acid and MS-275. Lysine acetylation preferentially targets large macromolecular complexes involved in diverse cellular processes, such as chromatin remodeling, cell cycle, splicing, nuclear transport, and actin nucleation. Acetylation impaired phosphorylation-dependent interactions of 14-3-3 and regulated the yeast cyclin-dependent kinase Cdc28. Our data demonstrate that the regulatory scope of lysine acetylation is broad and comparable with that of other major posttranslational modifications.\",\n",
       " 'Eukaryotic cells replicate by a complex series of evolutionarily conserved events that are tightly regulated at defined stages of the cell division cycle. Progression through this cycle involves a large number of dedicated protein complexes and signaling pathways, and deregulation of this process is implicated in tumorigenesis. We applied high-resolution mass spectrometry-based proteomics to investigate the proteome and phosphoproteome of the human cell cycle on a global scale and quantified 6027 proteins and 20,443 unique phosphorylation sites and their dynamics. Co-regulated proteins and phosphorylation sites were grouped according to their cell cycle kinetics and compared to publicly available messenger RNA microarray data. Most detected phosphorylation sites and more than 20% of all quantified proteins showed substantial regulation, mainly in mitotic cells. Kinase-motif analysis revealed global activation during S phase of the DNA damage response network, which was mediated by phosphorylation by ATM or ATR or DNA-dependent protein kinases. We determined site-specific stoichiometry of more than 5000 sites and found that most of the up-regulated sites phosphorylated by cyclin-dependent kinase 1 (CDK1) or CDK2 were almost fully phosphorylated in mitotic cells. In particular, nuclear proteins and proteins involved in regulating metabolic processes have high phosphorylation site occupancy in mitosis. This suggests that these proteins may be inactivated by phosphorylation in mitotic cells.',\n",
       " 'On the basis of large proteomics datasets measured from seven human cell lines we consider their intersection as an approximation of the human central proteome, which is the set of proteins ubiquitously expressed in all human cells. Composition and properties of the central proteome are investigated through bioinformatics analyses.',\n",
       " 'Bromodomain protein 4 (Brd4) plays critical roles in development, cancer progression, and virus-host pathogenesis. To gain mechanistic insight into the various biological functions of Brd4, we performed a proteomic analysis to identify and characterize Brd4-associated cellular proteins. We found that the extraterminal (ET) domain, whose function has to date not been determined, interacts with NSD3, JMJD6, CHD4, GLTSCR1, and ATAD5. These ET-domain interactions were also conserved for Brd2 and Brd3, the other human BET proteins tested. We demonstrated that GLTSCR1, NSD3, and JMJD6 impart a pTEFb-independent transcriptional activation function on Brd4. NSD3 as well as JMJD6 is recruited to regulated genes in a Brd4-dependent manner. Moreover, we found that depletion of Brd4 or NSD3 reduces H3K36 methylation, demonstrating that the Brd4/NSD3 complex regulates this specific histone modification. Our results indicate that the Brd4 ET domain through the recruitment of the specific effectors regulates transcriptional activity. In particular, we show that one of these effectors, NSD3, regulates transcription by modifying the chromatin microenvironment at Brd4 target genes. Our study thus identifies the ET domain as a second important transcriptional regulatory domain for Brd4 in addition to the carboxyl-terminal domain (CTD) that interacts with pTEFb.',\n",
       " 'Bromodomain-containing protein Brd4 is shown to persistently associate with chromosomes during mitosis for transmitting epigenetic memory across cell divisions. During interphase, Brd4 also plays a key role in regulating the transcription of signal-inducible genes by recruiting positive transcription elongation factor b (P-TEFb) to promoters. How the chromatin-bound Brd4 transits into a transcriptional regulation mode in response to stimulation, however, is largely unknown. Here, by analyzing the dynamics of Brd4 during ultraviolet or hexamethylene bisacetamide treatment, we show that the signal-induced release of chromatin-bound Brd4 is essential for its functional transition. In untreated cells, almost all Brd4 is observed in association with interphase chromatin. Upon treatment, Brd4 is released from chromatin, mostly due to signal-triggered deacetylation of nucleosomal histone H4 at acetylated-lysine 5/8 (H4K5ac/K8ac). Through selective association with the transcriptional active form of P-TEFb that has been liberated from the inactive multi-subunit complex in response to treatment, the released Brd4 mediates the recruitment of this active P-TEFb to promoter, which enhances transcription at the stage of elongation. Thus, through signal-induced release from chromatin and selective association with the active form of P-TEFb, the chromatin-bound Brd4 switches its role to mediate the recruitment of P-TEFb for regulating the transcriptional elongation of signal-inducible genes.',\n",
       " 'To elucidate cellular events underlying the pluripotency of human embryonic stem cells (hESCs), we performed parallel quantitative proteomic and phosphoproteomic analyses of hESCs during differentiation initiated by a diacylglycerol analog or transfer to media that had not been conditioned by feeder cells. We profiled 6521 proteins and 23,522 phosphorylation sites, of which almost 50% displayed dynamic changes in phosphorylation status during 24 hours of differentiation. These data are a resource for studies of the events associated with the maintenance of hESC pluripotency and those accompanying their differentiation. From these data, we identified a core hESC phosphoproteome of sites with similar robust changes in response to the two distinct treatments. These sites exhibited distinct dynamic phosphorylation patterns, which were linked to known or predicted kinases on the basis of the matching sequence motif. In addition to identifying previously unknown phosphorylation sites on factors associated with differentiation, such as kinases and transcription factors, we observed dynamic phosphorylation of DNA methyltransferases (DNMTs). We found a specific interaction of DNMTs during early differentiation with the PAF1 (polymerase-associated factor 1) transcriptional elongation complex, which binds to promoters of the pluripotency and known DNMT target genes encoding OCT4 and NANOG, thereby providing a possible molecular link for the silencing of these genes during differentiation.',\n",
       " 'Chromatin structure organization is crucial for regulating many fundamental cellular processes. However, the molecular mechanism that regulates the assembly of higher-order chromatin structure remains poorly understood. In this study, we demonstrate that Brd4 (bromodomain-containing protein 4) protein participates in the maintenance of the higher-order chromatin structure. Brd4, a member of the BET family of proteins, has been shown to play important roles in cellular growth control, cell cycle progression, and cancer development. We apply in situ single cell chromatin imaging and micrococcal nuclease (MNase) assay to show that Brd4 depletion leads to a large scale chromatin unfolding. A dominant-negative inhibitor encoding the double bromodomains (BDI/II) of Brd4 can competitively dissociate endogenous Brd4 from chromatin to trigger severely fragmented chromatin morphology. Mechanistic studies using Brd4 truncation mutants reveal that the Brd4 C-terminal domain is crucial for maintaining normal chromatin structure. Using bimolecular fluorescence complementation technology, we demonstrate that Brd4 molecules interact intermolecularly on chromatin and that replacing Brd4 molecules by BDI/II causes abnormal nucleosome aggregation and chromatin fragmentation. These studies establish a novel structural role of Brd4 in supporting the higher chromatin architecture.',\n",
       " 'Transcriptional elongation by RNA polymerase II (Pol II) is regulated by positive transcription elongation factor b (P-TEFb) in association with bromodomain-containing protein 4 (BRD4). We used genome-wide chromatin immunoprecipitation sequencing in primary human CD4+ T cells to reveal that BRD4 co-localizes with Ser-2-phosphorylated Pol II (Pol II Ser-2) at both enhancers and promoters of active genes. Disruption of bromodomain-histone acetylation interactions by JQ1, a small-molecule bromodomain inhibitor, resulted in decreased BRD4 binding, reduced Pol II Ser-2, and reduced expression of lineage-specific genes in primary human CD4+ T cells. A large number of JQ1-disrupted BRD4 binding regions exhibited diacetylated H4 (lysine 5 and -8) and H3K27 acetylation (H3K27ac), which correlated with the presence of histone acetyltransferases and deacetylases. Genes associated with BRD4/H3K27ac co-occupancy exhibited significantly higher activity than those associated with H3K27ac or BRD4 binding alone. Comparison of BRD4 binding in T cells and in human embryonic stem cells revealed that enhancer BRD4 binding sites were predominantly lineage-specific. Our findings suggest that BRD4-driven Pol II phosphorylation at serine 2 plays an important role in regulating lineage-specific gene transcription in human CD4+ T cells.',\n",
       " \"The bromodomain protein, BRD4, has been identified recently as a therapeutic target in acute myeloid leukemia, multiple myeloma, Burkitt's lymphoma, NUT midline carcinoma, colon cancer, and inflammatory disease; its loss is a prognostic signature for metastatic breast cancer. BRD4 also contributes to regulation of both cell cycle and transcription of oncogenes, HIV, and human papilloma virus (HPV). Despite its role in a broad range of biological processes, the precise molecular mechanism of BRD4 function remains unknown. We report that BRD4 is an atypical kinase that binds to the carboxyl-terminal domain (CTD) of RNA polymerase II and directly phosphorylates its serine 2 (Ser2) sites both in vitro and in vivo under conditions where other CTD kinases are inactive. Phosphorylation of the CTD Ser2 is inhibited in vivo by a BRD4 inhibitor that blocks its binding to chromatin. Our finding that BRD4 is an RNA polymerase II CTD Ser2 kinase implicates it as a regulator of eukaryotic transcription.\",\n",
       " 'Distal enhancers characterized by the H3K4me(1) mark play critical roles in developmental and transcriptional programs. However, potential roles of specific distal regulatory elements in regulating RNA polymerase II (Pol II) promoter-proximal pause release remain poorly investigated. Here, we report that a unique cohort of jumonji C-domain-containing protein 6 (JMJD6) and bromodomain-containing protein 4 (Brd4) cobound distal enhancers, termed anti-pause enhancers (A-PEs), regulate promoter-proximal pause release of a large subset of transcription units via long-range interactions. Brd4-dependent JMJD6 recruitment on A-PEs mediates erasure of H4R3me(2(s)), which is directly read by 7SK snRNA, and decapping/demethylation of 7SK snRNA, ensuring the dismissal of the 7SK snRNA/HEXIM inhibitory complex. The interactions of both JMJD6 and Brd4 with the P-TEFb complex permit its activation and pause release of regulated coding genes. The functions of JMJD6/ Brd4-associated dual histone and RNA demethylase activity on anti-pause enhancers have intriguing implications for these proteins in development, homeostasis, and disease.',\n",
       " 'Mass spectrometry (MS)-based phosphoproteomics has achieved extraordinary success in qualitative and quantitative analysis of cellular protein phosphorylation. Considering that an estimated level of phosphorylation in a cell is placed at well above 100,000 sites, there is still much room for improvement. Here, we attempt to extend the depth of phosphoproteome coverage while maintaining realistic aspirations in terms of available material, robustness, and instrument running time. We developed three strategies, where each provided a different balance between these three key parameters. The first strategy simply used enrichment by Ti(4+)-IMAC followed by reversed chromatography LC-MS (termed 1D). The second strategy incorporated an additional fractionation step through the use of HILIC (2D). Finally, a third strategy was designed employing first an SCX fractionation, followed by Ti(4+)-IMAC enrichment and additional fractionation by HILIC (3D). A preliminary evaluation was performed on the HeLa cell line. Detecting 3700 phosphopeptides in about 2 h, the 1D strategy was found to be the most sensitive but limited in comprehensivity, mainly due to issues with complexity and dynamic range. Overall, the best balance was achieved using the 2D based strategy, identifying close to 17,000 phosphopeptides with less than 1 mg of material in about 48 h. Subsequently, we confirmed the findings with the K562 cell sample. When sufficient material was available, the 3D strategy increased phosphoproteome allowing over 22,000 unique phosphopeptides to be identified. Unfortunately, the 3D strategy required more time and over 1 mg of material before it started to outperform 2D. Ultimately, combining all strategies, we were able to identify over 16,000 and nearly 24,000 unique phosphorylation sites from the cancer cell lines HeLa and K562, respectively. In summary, we demonstrate the need to carry out extensive fractionation for deep mining of the phosphoproteome and provide a guide for appropriate strategies depending on sample amount and/or analysis time.',\n",
       " \"Bromodomain-containing protein 4 (Brd4) is an epigenetic reader and transcriptional regulator recently identified as a cancer therapeutic target for acute myeloid leukemia, multiple myeloma, and Burkitt's lymphoma. Although chromatin targeting is a crucial function of Brd4, there is little understanding of how bromodomains that bind acetylated histones are regulated, nor how the gene-specific activity of Brd4 is determined. Via interaction screen and domain mapping, we identified p53 as a functional partner of Brd4. Interestingly, Brd4 association with p53 is modulated by casein kinase II (CK2)-mediated phosphorylation of a conserved acidic region in Brd4 that selectively contacts either a juxtaposed bromodomain or an adjacent basic region to dictate the ability of Brd4 binding to chromatin and also the recruitment of p53 to regulated promoters. The unmasking of bromodomains and activator recruitment, concurrently triggered by the CK2 phospho switch, provide an intriguing mechanism for gene-specific targeting by a universal epigenetic reader.\",\n",
       " 'RNA polymerase II (Pol II) and the pausing complex, NELF and DSIF, are detected near the transcription start site (TSS) of many active and silent genes. Active transcription starts when the pause release factor P-TEFb is recruited to initiate productive elongation. However, the mechanism of P-TEFb recruitment and regulation of NELF/DSIF during transcription is not fully understood. We investigated this question in interferon (IFN)-stimulated transcription, focusing on BRD4, a BET family protein that interacts with P-TEFb. Besides P-TEFb, BRD4 binds to acetylated histones through the bromodomain. We found that BRD4 and P-TEFb, although not present prior to IFN treatment, were robustly recruited to IFN-stimulated genes (ISGs) after stimulation. Likewise, NELF and DSIF prior to stimulation were hardly detectable on ISGs, which were strongly recruited after IFN treatment. A shRNA-based knockdown assay of NELF revealed that it negatively regulates the passage of Pol II and DSIF across the ISGs during elongation, reducing total ISG transcript output. Analyses with a BRD4 small-molecule inhibitor showed that IFN-induced recruitment of P-TEFb and NELF/DSIF was under the control of BRD4. We suggest a model where BRD4 coordinates both positive and negative regulation of ISG elongation.',\n",
       " 'DNA damage activates a signalling network that blocks cell-cycle progression, recruits DNA repair factors and/or triggers senescence or programmed cell death. Alterations in chromatin structure are implicated in the initiation and propagation of the DNA damage response. Here we further investigate the role of chromatin structure in the DNA damage response by monitoring ionizing-radiation-induced signalling and response events with a high-content multiplex RNA-mediated interference screen of chromatin-modifying and -interacting genes. We discover that an isoform of Brd4, a bromodomain and extra-terminal (BET) family member, functions as an endogenous inhibitor of DNA damage response signalling by recruiting the condensin II chromatin remodelling complex to acetylated histones through bromodomain interactions. Loss of this isoform results in relaxed chromatin structure, rapid cell-cycle checkpoint recovery and enhanced survival after irradiation, whereas functional gain of this isoform compacted chromatin, attenuated DNA damage response signalling and enhanced radiation-induced lethality. These data implicate Brd4, previously known for its role in transcriptional control, as an insulator of chromatin that can modulate the signalling response to DNA damage.',\n",
       " 'Protein phosphorylation is one of the most common post-translational modifications. It plays key roles in regulating diverse biological processes of liver tissues. To better understand the role of protein phosphorylation in liver functions, it is essential to perform in-depth phosphoproteome analysis of human liver. Here, an enzyme assisted reversed-phase-reversed-phase liquid chromatography (RP-RPLC) approach with both RPLC separations operated with optimized acidic mobile phase was developed. High orthogonal separation was achieved by trypsin digestion of the Glu-C generated peptides in the fractions collected from the first RPLC separation. The phosphoproteome coverage was further improved by using two types of instruments, i.e. TripleTOF 5600 and LTQ Orbitrap Velos. A total of 22,446 phosphorylation sites, corresponding to 6526 nonredundant phosphoproteins were finally identified from normal human liver tissues. Of these sites, 15,229 sites were confidently localized with Ascore≥13. This dataset was the largest phosphoproteome dataset of human liver. It can be a public resource for the liver research community and holds promise for further biology studies.',\n",
       " 'SUMOylation is a reversible post-translational modification essential for genome stability. Using high-resolution MS, we have studied global SUMOylation in human cells in a site-specific manner, identifying a total of >4,300 SUMOylation sites in >1,600 proteins. To our knowledge, this is the first time that >1,000 SUMOylation sites have been identified under standard growth conditions. We quantitatively studied SUMOylation dynamics in response to SUMO protease inhibition, proteasome inhibition and heat shock. Many SUMOylated lysines have previously been reported to be ubiquitinated, acetylated or methylated, thus indicating cross-talk between SUMO and other post-translational modifications. We identified 70 phosphorylation and four acetylation events in proximity to SUMOylation sites, and we provide evidence for acetylation-dependent SUMOylation of endogenous histone H3. SUMOylation regulates target proteins involved in all nuclear processes including transcription, DNA repair, chromatin remodeling, precursor-mRNA splicing and ribosome assembly.',\n",
       " 'SUMOylation is an essential ubiquitin-like modification involved in important biological processes in eukaryotic cells. Identification of small ubiquitin-related modifier (SUMO)-conjugated residues in proteins is critical for understanding the role of SUMOylation but remains experimentally challenging. We have set up a powerful and high-throughput method combining quantitative proteomics and peptide immunocapture to map SUMOylation sites and have analyzed changes in SUMOylation in response to stimuli. With this technique we identified 295 SUMO1 and 167 SUMO2 sites on endogenous substrates of human cells. We further used this strategy to characterize changes in SUMOylation induced by listeriolysin O, a bacterial toxin that impairs the host cell SUMOylation machinery, and identified several classes of host proteins specifically deSUMOylated in response to this toxin. Our approach constitutes an unprecedented tool, broadly applicable to various SUMO-regulated cellular processes in health and disease.',\n",
       " 'Small ubiquitin-like modifiers play critical roles in the DNA damage response (DDR). To increase our understanding of SUMOylation in the mammalian DDR, we employed a quantitative proteomics approach in order to identify dynamically regulated SUMO-2 conjugates and modification sites upon treatment with the DNA damaging agent methyl methanesulfonate (MMS). We have uncovered a dynamic set of 20 upregulated and 33 downregulated SUMO-2 conjugates, and 755 SUMO-2 sites, of which 362 were dynamic in response to MMS. In contrast to yeast, where a response is centered on homologous recombination, we identified dynamically SUMOylated interaction networks of chromatin modifiers, transcription factors, DNA repair factors, and nuclear body components. SUMOylated chromatin modifiers include JARID1B/KDM5B, JARID1C/KDM5C, p300, CBP, PARP1, SetDB1, and MBD1. Whereas SUMOylated JARID1B was ubiquitylated by the SUMO-targeted ubiquitin ligase RNF4 and degraded by the proteasome in response to DNA damage, JARID1C was SUMOylated and recruited to the chromatin to demethylate histone H3K4.',\n",
       " 'How chromatin shapes pathways that promote genome-epigenome integrity in response to DNA damage is an issue of crucial importance. We report that human bromodomain (BRD)-containing proteins, the primary \"readers\" of acetylated chromatin, are vital for the DNA damage response (DDR). We discovered that more than one-third of all human BRD proteins change localization in response to DNA damage. We identified ZMYND8 (zinc finger and MYND [myeloid, Nervy, and DEAF-1] domain containing 8) as a novel DDR factor that recruits the nucleosome remodeling and histone deacetylation (NuRD) complex to damaged chromatin. Our data define a transcription-associated DDR pathway mediated by ZMYND8 and the NuRD complex that targets DNA damage, including when it occurs within transcriptionally active chromatin, to repress transcription and promote repair by homologous recombination. Thus, our data identify human BRD proteins as key chromatin modulators of the DDR and provide novel insights into how DNA damage within actively transcribed regions requires chromatin-binding proteins to orchestrate the appropriate response in concordance with the damage-associated chromatin context.',\n",
       " 'Genotoxic agents can cause replication fork stalling in dividing cells because of DNA lesions, eventually leading to replication fork collapse when the damage is not repaired. Small Ubiquitin-like Modifiers (SUMOs) are known to counteract replication stress, nevertheless, only a small number of relevant SUMO target proteins are known. To address this, we have purified and identified SUMO-2 target proteins regulated by replication stress in human cells. The developed methodology enabled single step purification of His10-SUMO-2 conjugates under denaturing conditions with high yield and high purity. Following statistical analysis on five biological replicates, a total of 566 SUMO-2 targets were identified. After 2 h of hydroxyurea treatment, 10 proteins were up-regulated for SUMOylation and two proteins were down-regulated for SUMOylation, whereas after 24 h, 35 proteins were up-regulated for SUMOylation, and 13 proteins were down-regulated for SUMOylation. A site-specific approach was used to map over 1000 SUMO-2 acceptor lysines in target proteins. The methodology is generic and is widely applicable in the ubiquitin field. A large subset of these identified proteins function in one network that consists of interacting replication factors, transcriptional regulators, DNA damage response factors including MDC1, ATR-interacting protein ATRIP, the Bloom syndrome protein and the BLM-binding partner RMI1, the crossover junction endonuclease EME1, BRCA1, and CHAF1A. Furthermore, centromeric proteins and signal transducers were dynamically regulated by SUMOylation upon replication stress. Our results uncover a comprehensive network of SUMO target proteins dealing with replication damage and provide a framework for detailed understanding of the role of SUMOylation to counteract replication stress. Ultimately, our study reveals how a post-translational modification is able to orchestrate a large variety of different proteins to integrate different nuclear processes with the aim of dealing with the induced DNA damage.',\n",
       " 'Small ubiquitin-like modifiers (SUMOs) are post-translational modifications (PTMs) that regulate nuclear cellular processes. Here we used an augmented K0-SUMO proteomics strategy to identify 40,765 SUMO acceptor sites and quantify their fractional contribution for 6,747 human proteins. Structural-predictive analyses revealed that lysines residing in disordered regions are preferentially targeted by SUMO, in notable contrast to other widespread lysine modifications. In our data set, we identified 807 SUMOylated peptides that were co-modified by phosphorylation, along with dozens of SUMOylated peptides that were co-modified by ubiquitylation, acetylation and methylation. Notably, 9% of the identified SUMOylome occurred proximal to phosphorylation, and numerous SUMOylation sites were found to be fully dependent on prior phosphorylation events. SUMO-proximal phosphorylation occurred primarily in a proline-directed manner, and inhibition of cyclin-dependent kinases dynamically affected co-modification. Collectively, we present a comprehensive analysis of the SUMOylated proteome, uncovering the structural preferences for SUMO and providing system-wide evidence for a remarkable degree of cross-talk between SUMOylation and other major PTMs.',\n",
       " 'The mammalian SWI/SNF chromatin remodeling complex is a heterogeneous collection of related protein complexes required for gene regulation and genome integrity. It contains a central ATPase (BRM or BRG1) and various combinations of 10-14 accessory subunits (BAFs for ',\n",
       " 'Proteins of the bromodomain and extra-terminal (BET) domain family are epigenetic readers that bind acetylated histones through their bromodomains to regulate gene transcription. Dual-bromodomain BET inhibitors (DbBi) that bind with similar affinities to the first (BD1) and second (BD2) bromodomains of BRD2, BRD3, BRD4 and BRDt have displayed modest clinical activity in monotherapy cancer trials. A reduced number of thrombocytes in the blood (thrombocytopenia) as well as symptoms of gastrointestinal toxicity are dose-limiting adverse events for some types of DbBi',\n",
       " \"Cancers arise owing to mutations in a subset of genes that confer growth advantage. The availability of the human genome sequence led us to propose that systematic resequencing of cancer genomes for mutations would lead to the discovery of many additional cancer genes. Here we report more than 1,000 somatic mutations found in 274 megabases (Mb) of DNA corresponding to the coding exons of 518 protein kinase genes in 210 diverse human cancers. There was substantial variation in the number and pattern of mutations in individual cancers reflecting different exposures, DNA repair defects and cellular origins. Most somatic mutations are likely to be 'passengers' that do not contribute to oncogenesis. However, there was evidence for 'driver' mutations contributing to the development of the cancers studied in approximately 120 genes. Systematic sequencing of cancer genomes therefore reveals the evolutionary diversity of cancers and implicates a larger repertoire of cancer genes than previously anticipated.\",\n",
       " \"Brd4 belongs to the BET family. It is a multifunctional protein involved in transcription, replication, the signal transduction pathway, and cell cycle progression. All of these functions are linked to its association with acetylated chromatin. With its tandem bromodomains, Brd4 avidly binds to diacetylated H4-AcK5/K12 and H3-AcK9/K14 peptides. Solution structure of the second bromodomain (BD) is reported in this research. In addition to the piD-helix, which is special for BET members, an incompact alphaZ' distinct from Brd2 BD2 is found, although they have identical sequences in this region. Both BD1 and BD2 bind to monoacetylated H4-AcK5 and H4-AcK12 peptides, but with subtle differences. An NMR perturbation study and mutational analysis identified the binding interface and revealed several residues important for binding specificity. By molecular dynamics simulations, a complex model composed of H4-AcK5/K12 and two molecules of BD2 is presented. Relaxation data and internal motions of BD2 are also discussed. Unlike Brd2 BD1, the two bromodomains of Brd4 are mainly monomeric in solution. They do not form heterodimers like TAFII250. It suggests that Brd4 should have its own mechanism to reinforce its chromatin association both in mitotic retention and related cellular regulation.\",\n",
       " \"Epigenetic proteins are intently pursued targets in ligand discovery. So far, successful efforts have been limited to chromatin modifying enzymes, or so-called epigenetic 'writers' and 'erasers'. Potent inhibitors of histone binding modules have not yet been described. Here we report a cell-permeable small molecule (JQ1) that binds competitively to acetyl-lysine recognition motifs, or bromodomains. High potency and specificity towards a subset of human bromodomains is explained by co-crystal structures with bromodomain and extra-terminal (BET) family member BRD4, revealing excellent shape complementarity with the acetyl-lysine binding cavity. Recurrent translocation of BRD4 is observed in a genetically-defined, incurable subtype of human squamous carcinoma. Competitive binding by JQ1 displaces the BRD4 fusion oncoprotein from chromatin, prompting squamous differentiation and specific antiproliferative effects in BRD4-dependent cell lines and patient-derived xenograft models. These data establish proof-of-concept for targeting protein-protein interactions of epigenetic 'readers', and provide a versatile chemical scaffold for the development of chemical probes more broadly throughout the bromodomain family.\",\n",
       " \"Interaction of pathogens with cells of the immune system results in activation of inflammatory gene expression. This response, although vital for immune defence, is frequently deleterious to the host due to the exaggerated production of inflammatory proteins. The scope of inflammatory responses reflects the activation state of signalling proteins upstream of inflammatory genes as well as signal-induced assembly of nuclear chromatin complexes that support mRNA expression. Recognition of post-translationally modified histones by nuclear proteins that initiate mRNA transcription and support mRNA elongation is a critical step in the regulation of gene expression. Here we present a novel pharmacological approach that targets inflammatory gene expression by interfering with the recognition of acetylated histones by the bromodomain and extra terminal domain (BET) family of proteins. We describe a synthetic compound (I-BET) that by 'mimicking' acetylated histones disrupts chromatin complexes responsible for the expression of key inflammatory genes in activated macrophages, and confers protection against lipopolysaccharide-induced endotoxic shock and bacteria-induced sepsis. Our findings suggest that synthetic compounds specifically targeting proteins that recognize post-translationally modified histones can serve as a new generation of immunomodulatory drugs.\",\n",
       " 'Epigenetic mechanisms of gene regulation have a profound role in normal development and disease processes. An integral part of this mechanism occurs through lysine acetylation of histone tails which are recognized by bromodomains. While the biological and structural characterization of many bromodomain containing proteins has advanced considerably, the therapeutic tractability of this protein family is only now becoming understood. This paper describes the discovery and molecular characterization of potent (nM) small molecule inhibitors that disrupt the function of the BET family of bromodomains (Brd2, Brd3, and Brd4). By using a combination of phenotypic screening, chemoproteomics, and biophysical studies, we have discovered that the protein-protein interactions between bromodomains and acetylated histones can be antagonized by selective small molecules that bind at the acetylated lysine recognition pocket. X-ray crystal structures of compounds bound into bromodomains of Brd2 and Brd4 elucidate the molecular interactions of binding and explain the precisely defined stereochemistry required for activity.',\n",
       " \"Recurrent chromosomal translocations involving the mixed lineage leukaemia (MLL) gene initiate aggressive forms of leukaemia, which are often refractory to conventional therapies. Many MLL-fusion partners are members of the super elongation complex (SEC), a critical regulator of transcriptional elongation, suggesting that aberrant control of this process has an important role in leukaemia induction. Here we use a global proteomic strategy to demonstrate that MLL fusions, as part of SEC and the polymerase-associated factor complex (PAFc), are associated with the BET family of acetyl-lysine recognizing, chromatin 'adaptor' proteins. These data provided the basis for therapeutic intervention in MLL-fusion leukaemia, via the displacement of the BET family of proteins from chromatin. We show that a novel small molecule inhibitor of the BET family, GSK1210151A (I-BET151), has profound efficacy against human and murine MLL-fusion leukaemic cell lines, through the induction of early cell cycle arrest and apoptosis. I-BET151 treatment in two human leukaemia cell lines with different MLL fusions alters the expression of a common set of genes whose function may account for these phenotypic changes. The mode of action of I-BET151 is, at least in part, due to the inhibition of transcription at key genes (BCL2, C-MYC and CDK6) through the displacement of BRD3/4, PAFc and SEC components from chromatin. In vivo studies indicate that I-BET151 has significant therapeutic value, providing survival benefit in two distinct mouse models of murine MLL-AF9 and human MLL-AF4 leukaemia. Finally, the efficacy of I-BET151 against human leukaemia stem cells is demonstrated, providing further evidence of its potent therapeutic potential. These findings establish the displacement of BET proteins from chromatin as a promising epigenetic therapy for these aggressive leukaemias.\",\n",
       " 'Benzodiazepines are psychoactive drugs with anxiolytic, sedative, skeletal muscle relaxant and amnestic properties. Recently triazolo-benzodiazepines have been also described as potent and highly selective protein interaction inhibitors of bromodomain and extra-terminal (BET) proteins, a family of transcriptional co-regulators that play a key role in cancer cell survival and proliferation, but the requirements for high affinity interaction of this compound class with bromodomains has not been described. Here we provide insight into the structure-activity relationship (SAR) and selectivity of this versatile scaffold. In addition, using high resolution crystal structures we compared the binding mode of a series of benzodiazepine (BzD) and related triazolo-benzotriazepines (BzT) derivatives including clinically approved drugs such as alprazolam and midazolam. Our analysis revealed the importance of the 1-methyl triazolo ring system for BET binding and suggests modifications for the development of further high affinity bromodomain inhibitors.',\n",
       " 'NF-κB-mediated inflammation is the major pathology in chronic kidney diseases, including HIV-associated nephropathy (HIVAN) that ultimately progresses to end stage renal disease. HIV infection in the kidney induces NF-κB activation, leading to the production of proinflammatory chemokines, cytokines, and adhesion molecules. In this study, we explored selective inhibition of NF-κB transcriptional activity by small molecule blocking NF-κB binding to the transcriptional cofactor BRD4, which is required for the assembly of the productive transcriptional complex comprising positive transcription elongation factor b and RNA polymerase II. We showed that our BET (Bromodomain and Extra-Terminal domain)-specific bromodomain inhibitor MS417, designed to block BRD4 binding to the acetylated NF-κB, effectively attenuates NF-κB transcriptional activation of proinflammatory genes in kidney cells treated with TNFα or infected by HIV. MS417 ameliorates inflammation and kidney injury in HIV-1 transgenic mice, an animal model for HIVAN. Our study suggests that BET bromodomain inhibition, targeting at the proinflammatory activity of NF-κB, represents a new therapeutic approach for treating NF-κB-mediated inflammation and kidney injury in HIVAN.',\n",
       " 'Bromodomain-containing proteins are key epigenetic regulators of gene transcription and readers of the histone code. However, the therapeutic benefits of modulating this target class are largely unexplored due to the lack of suitable chemical probes. This article describes the generation of lead molecules for the BET bromodomains through screening a fragment set chosen using structural insights and computational approaches. Analysis of 40 BRD2/fragment X-ray complexes highlights both shared and disparate interaction features that may be exploited for affinity and selectivity. Six representative crystal structures are then exemplified in detail. Two of the fragments are completely new bromodomain chemotypes, and three have never before been crystallized in a bromodomain, so our results significantly extend the limited public knowledge-base of crystallographic small molecule/bromodomain interactions. Certain fragments (including paracetamol) bind in a consistent mode to different bromodomains such as CREBBP, suggesting their potential to act as generic bromodomain templates. An important implication is that the bromodomains are not only a phylogenetic family but also a system in which chemical and structural knowledge of one bromodomain gives insights transferrable to others.',\n",
       " 'The posttranslational modification of chromatin through acetylation at selected histone lysine residues is governed by histone acetyltransferases (HATs) and histone deacetylases (HDACs). The significance of this subset of the epigenetic code is interrogated and interpreted by an acetyllysine-specific protein-protein interaction with bromodomain reader modules. Selective inhibition of the bromo and extra C-terminal domain (BET) family of bromodomains with a small molecule is feasible, and this may represent an opportunity for disease intervention through the recently disclosed antiproliferative and anti-inflammatory properties of such inhibitors. Herein, we describe the discovery and structure-activity relationship (SAR) of a novel, small-molecule chemical probe for BET family inhibition that was identified through the application of structure-based fragment assessment and optimization techniques. This has yielded a potent, selective compound with cell-based activity (PFI-1) that may further add to the understanding of BET family function within the bromodomains.',\n",
       " 'Bromodomains (BRDs) are protein interaction modules that specifically recognize ε-N-lysine acetylation motifs, a key event in the reading process of epigenetic marks. The 61 BRDs in the human genome cluster into eight families based on structure/sequence similarity. Here, we present 29 high-resolution crystal structures, covering all BRD families. Comprehensive crossfamily structural analysis identifies conserved and family-specific structural features that are necessary for specific acetylation-dependent substrate recognition. Screening of\\xa0more than 30 representative BRDs against systematic histone-peptide arrays identifies new BRD substrates and reveals a strong influence of flanking posttranslational modifications, such as acetylation and phosphorylation, suggesting that BRDs recognize combinations of marks rather than singly acetylated sequences. We further uncovered a structural mechanism for the simultaneous binding and recognition of diverse diacetyl-containing peptides by BRD4. These data provide a foundation for structure-based drug design of specific inhibitors for this emerging target family.',\n",
       " 'The bromodomain protein module, which binds to acetylated lysine, is emerging as an important epigenetic therapeutic target. We report the structure-guided optimization of 3,5-dimethylisoxazole derivatives to develop potent inhibitors of the BET (bromodomain and extra terminal domain) bromodomain family with good ligand efficiency. X-ray crystal structures of the most potent compounds reveal key interactions required for high affinity at BRD4(1). Cellular studies demonstrate that the phenol and acetate derivatives of the lead compounds showed strong antiproliferative effects on MV4;11 acute myeloid leukemia cells, as shown for other BET bromodomain inhibitors and genetic BRD4 knockdown, whereas the reported compounds showed no general cytotoxicity in other cancer cell lines tested.',\n",
       " 'Recognizing acetyllysine of histone is a vital process of epigenetic regulation that is mediated by a protein module called bromodomain. To contribute novel scaffolds for developing into bromodomain inhibitors, we utilize a fragment-based drug discovery approach. By successively applying docking and X-ray crystallography, we were able to identify 9 fragment hits from diffracting more than 60 crystals. In the present work, we described four of them and carried out the integrated lead optimization for fragment 8, which bears a 2-thiazolidinone core. After several rounds of structure guided modifications, we assessed the druggability of 2-thiazolidinone by modulating in vitro pharmacokinetic studies and cellular activity assay. The results showed that two potent compounds of 2-thiazolidinones have good metabolic stability. Also, the cellular assay confirmed the activities of 2-thiazolidinones. Together, we hope the identified 2-thiazolidinone chemotype and other fragment hits described herein can stimulate researchers to develop more diversified bromodomain inhibitors.',\n",
       " 'Jumonji domain-containing protein 6 (JMJD6) is a member of the Jumonji C family of Fe(II) and 2-oxoglutarate (2OG) dependent oxygenases. It possesses unique bi-functional oxygenase activities, acting as both an arginine demethylase and a lysyl-hydroxylase. JMJD6 has been reported to be over-expressed in oral, breast, lung, and colon cancers and plays important roles in regulation of transcription through interactions with transcription regulator BRD4, histones, U2AF65, Luc7L3, and SRSF11. Here, we report a structural mechanism revealed by NMR of JMJD6 recognition by the extraterminal (ET) domain of BRD4 in that a JMJD6 peptide (Lys84-Asn96) adapts an α-helix when bound to the ET domain. This intermolecular recognition is established through JMJD6 interactions with the conserved hydrophobic core of the ET domain, and reinforced by electrostatic interactions of JMJD6 with residues in the inter-helical α1-α2 loop of the ET domain. Notably, this mode of ligand recognition is different from that of ET domain recognition of NSD3, LANA of herpesvirus, and integrase of MLV, which involves formation of an intermolecular amphipathic two- or three- strand antiparallel β sheet. Furthermore, we demonstrate that the association between the BRD4 ET domain and JMJD6 likely requires a protein conformational change induced by single-stranded RNA binding.',\n",
       " 'The two tandem bromodomains of the BET (bromodomain and extraterminal domain) proteins enable chromatin binding to facilitate transcription. Drugs that inhibit both bromodomains equally have shown efficacy in certain malignant and inflammatory conditions. To explore the individual functional contributions of the first (BD1) and second (BD2) bromodomains in biology and therapy, we developed selective BD1 and BD2 inhibitors. We found that steady-state gene expression primarily requires BD1, whereas the rapid increase of gene expression induced by inflammatory stimuli requires both BD1 and BD2 of all BET proteins. BD1 inhibitors phenocopied the effects of pan-BET inhibitors in cancer models, whereas BD2 inhibitors were predominantly effective in models of inflammatory and autoimmune disease. These insights into the differential requirement of BD1 and BD2 for the maintenance and induction of gene expression may guide future BET-targeted therapies.',\n",
       " 'We found that the clinical phenotype associated with BRD4 haploinsufficiency overlapped with that of Cornelia de Lange syndrome (CdLS), which is most often caused by mutation of NIPBL. More typical CdLS was observed with a de novo BRD4 missense variant, which retained the ability to coimmunoprecipitate with NIPBL, but bound poorly to acetylated histones. BRD4 and NIPBL displayed correlated binding at super-enhancers and appeared to co-regulate developmental gene expression.',\n",
       " 'In the version of this article initially published, Wendy Bickmore and Madapura Pradeepa were incorrectly not indicated as corresponding authors. The error has been corrected in the HTML and PDF versions of the paper.',\n",
       " 'An amendment to this paper has been published and can be accessed via a link at the top of the paper.',\n",
       " 'BRD4 is part of a multiprotein complex involved in loading the cohesin complex onto DNA, a fundamental process required for cohesin-mediated loop extrusion and formation of Topologically Associating Domains. Pathogenic variations in this complex have been associated with a growing number of syndromes, collectively known as cohesinopathies, the most classic being Cornelia de Lange syndrome. However, no cohort study has been conducted to delineate the clinical and molecular spectrum of BRD4-related disorder. We formed an international collaborative study, and collected 14 new patients, including two fetuses. We performed phenotype and genotype analysis, integrated prenatal findings from fetopathological examinations, phenotypes of pediatric patients and adults. We report the first cohort of patients with BRD4-related disorder and delineate the dysmorphic features at different ages. This work extends the phenotypic spectrum of cohesinopathies and characterize a new clinically relevant and recognizable pattern, distinguishable from the other cohesinopathies.']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "possible_matches = []\n",
    "for abstract in abstracts_to_check:\n",
    "    possible_match = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import spacy\n",
    "\n",
    "def find_matching_groups_spacy(abstract, gene_symbol):\n",
    "    # Load the English tokenizer, tagger, parser, NER, and word vectors\n",
    "    nlp = spacy.load(\"en_core_web_sm\")\n",
    "    \n",
    "    # Process the abstract text with spaCy\n",
    "    doc = nlp(abstract)\n",
    "    \n",
    "    # Create a list of words (tokens) from the document\n",
    "    words = [token.text for token in doc if not token.is_punct and not token.is_space]\n",
    "    \n",
    "    # Determine the length of the gene symbol\n",
    "    gene_length = len(gene_symbol)\n",
    "    # Initialize a list to store matching groups\n",
    "    matching_groups = []\n",
    "    \n",
    "    # Loop through the words to form groups of size gene_length\n",
    "    for i in range(len(words) - gene_length + 1):\n",
    "        # Select the current group of words\n",
    "        group = words[i:i + gene_length]\n",
    "        # Check if the first letter of each word in the group matches the letters in the gene symbol\n",
    "        if all(group[j][0].lower() == gene_symbol[j].lower() for j in range(gene_length)):\n",
    "            # If the group matches, add it to the list as a string\n",
    "            matching_groups.append(' '.join(group))\n",
    "    \n",
    "    return matching_groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chromosome associated protein']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "find_matching_groups_spacy(test_abstract,gene_symbol)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test Run w/ Anastasia Gene Lists"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given some sample gene lists, grab a uniprot identifier and use it to query for alternative names and pmids related to the parent gene. Once the alternative names and pmids are grabbed, use the PMID to query pubmed for the publication abstract. Use the obtained abstract and spaCy to do a quick check for symbol first letter matching (i.e. do triplets of words from an abstract start with C A P to match CAP). More sophisticated methods can be employed to check for correct expansion of words."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "import spacy\n",
    "from tqdm import tqdm\n",
    "import pandas as pd\n",
    "\n",
    "def grab_uniprot_id(gene_symbol):\n",
    "    query = f'https://normalize.cancervariants.org/gene/normalize?q={gene_symbol}'\n",
    "    r = requests.get(query)\n",
    "    try:\n",
    "        uniprot_id = None\n",
    "        for extension in r.json()['gene_descriptor']['extensions']:\n",
    "            if extension['name'] == 'associated_with':\n",
    "                uniprot_id = [value.replace('uniprot:','') for value in extension['value'] if value.startswith('uniprot:')]\n",
    "                pass\n",
    "                \n",
    "    except:\n",
    "        uniprot_id = 'Yikes'\n",
    "    return uniprot_id\n",
    "\n",
    "def uniprot_request(uniprot_id):\n",
    "    uniprot_data = requests.get(f'https://rest.uniprot.org/uniprotkb/{uniprot_id}?format=json')\n",
    "    return uniprot_data    \n",
    "\n",
    "def grab_alternative_names(uniprot_data):\n",
    "    alt_names = []\n",
    "    try:\n",
    "        for name in uniprot_data.json()['proteinDescription']['alternativeNames']:\n",
    "            alt_names.append(name['fullName']['value'])\n",
    "    except:\n",
    "        alt_names = None\n",
    "    return(alt_names)\n",
    "\n",
    "def grab_pmids(uniprot_data):\n",
    "    pmids_to_check = []\n",
    "    for pmid in uniprot_data.json()['references']:\n",
    "        pmids_to_check.append(pmid['citation']['id'])\n",
    "    return(pmids_to_check)\n",
    "\n",
    "def get_pubmed_abstract(pmid, api_key='YOUR_API_KEY'):\n",
    "    \"\"\" Fetch the abstract for a given PMID from PubMed. \"\"\"\n",
    "    url = f\"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi\"\n",
    "    params = {\n",
    "        'db': 'pubmed',\n",
    "        'id': pmid,\n",
    "        'retmode': 'xml'\n",
    "        # 'api_key': api_key\n",
    "    }\n",
    "    \n",
    "    response = requests.get(url, params=params)\n",
    "    \n",
    "    if response.status_code == 200:\n",
    "        # Parse the XML response\n",
    "        from xml.etree import ElementTree\n",
    "        root = ElementTree.fromstring(response.text)\n",
    "        \n",
    "        # Find the AbstractText element and return its text\n",
    "        abstract_text = root.find('.//AbstractText')\n",
    "        if abstract_text is not None:\n",
    "            return abstract_text.text\n",
    "        else:\n",
    "            return \"No abstract found.\"\n",
    "    else:\n",
    "        return f\"Error: {response.status_code}\"\n",
    "\n",
    "def find_matching_groups(abstract, gene_symbol):\n",
    "    # Load the English tokenizer, tagger, parser, NER, and word vectors\n",
    "    nlp = spacy.load(\"en_core_web_sm\")\n",
    "    \n",
    "    # Process the abstract text with spaCy\n",
    "    doc = nlp(abstract)\n",
    "\n",
    "    abstract = abstract.replace('-',' ')\n",
    "    print(abstract)\n",
    "    # Create a list of words (tokens) from the document\n",
    "    words = [token.text for token in doc if not token.is_punct and not token.is_space]\n",
    "    \n",
    "    # Determine the length of the gene symbol\n",
    "    gene_length = len(gene_symbol)\n",
    "\n",
    "    # Initialize a list to store matching groups\n",
    "    matching_groups = []\n",
    "    \n",
    "    # Loop through the words to form groups of size gene_length\n",
    "    for i in range(len(words) - gene_length + 1):\n",
    "        # Select the current group of words\n",
    "        group = words[i:i + gene_length]\n",
    "        # Check if the first letter of each word in the group matches the letters in the gene symbol\n",
    "        if all(group[j][0].lower() == gene_symbol[j].lower() for j in range(gene_length)):\n",
    "            # If the group matches, add it to the list as a string\n",
    "            matching_groups.append(' '.join(group))\n",
    "    \n",
    "    return matching_groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collision_symbol</th>\n",
       "      <th>parent_symbol</th>\n",
       "      <th>gene_alias_pair</th>\n",
       "      <th>correct_alias_expansion</th>\n",
       "      <th>collision_association</th>\n",
       "      <th>curation_difficulty</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAP</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-CAP</td>\n",
       "      <td>Chromosome Associated Protein</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>easy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAP</td>\n",
       "      <td>LNPEP</td>\n",
       "      <td>LNPEP-CAP</td>\n",
       "      <td>Cystinyl Aminopeptidase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYM</td>\n",
       "      <td>ZMYM1</td>\n",
       "      <td>ZMYM1-MYM</td>\n",
       "      <td>Myeloproloferative syndrome and mental retarda...</td>\n",
       "      <td>Disease</td>\n",
       "      <td>hard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASP</td>\n",
       "      <td>ASIP</td>\n",
       "      <td>ASIP-ASP</td>\n",
       "      <td>Agouti Signaling Protein</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>easy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASP</td>\n",
       "      <td>TMPRSS11D</td>\n",
       "      <td>TMPRSS11D-ASP</td>\n",
       "      <td>Adrenal secretory serine protease</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASP</td>\n",
       "      <td>ASPM</td>\n",
       "      <td>ASPM-ASP</td>\n",
       "      <td>Drosophila abnormal spindle</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>hard</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALP</td>\n",
       "      <td>ATHS</td>\n",
       "      <td>ATHS-ALP</td>\n",
       "      <td>Atherogenic Lipoprotein Phenotype</td>\n",
       "      <td>Phenotype</td>\n",
       "      <td>easy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ALP</td>\n",
       "      <td>CCL27</td>\n",
       "      <td>CCL27-ALP</td>\n",
       "      <td>Alkaline Phosphatase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ALP</td>\n",
       "      <td>ALPI</td>\n",
       "      <td>ALPI-ALP</td>\n",
       "      <td>Antileukoproteinase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>hard</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  collision_symbol parent_symbol gene_alias_pair  \\\n",
       "0              CAP          BRD4        BRD4-CAP   \n",
       "1              CAP         LNPEP       LNPEP-CAP   \n",
       "2              MYM         ZMYM1       ZMYM1-MYM   \n",
       "3              ASP          ASIP        ASIP-ASP   \n",
       "4              ASP     TMPRSS11D   TMPRSS11D-ASP   \n",
       "5              ASP          ASPM        ASPM-ASP   \n",
       "6              ALP          ATHS        ATHS-ALP   \n",
       "7              ALP         CCL27       CCL27-ALP   \n",
       "8              ALP          ALPI        ALPI-ALP   \n",
       "\n",
       "                             correct_alias_expansion collision_association  \\\n",
       "0                      Chromosome Associated Protein       Protein Product   \n",
       "1                            Cystinyl Aminopeptidase       Protein Product   \n",
       "2  Myeloproloferative syndrome and mental retarda...               Disease   \n",
       "3                           Agouti Signaling Protein       Protein Product   \n",
       "4                  Adrenal secretory serine protease       Protein Product   \n",
       "5                        Drosophila abnormal spindle              Ortholog   \n",
       "6                  Atherogenic Lipoprotein Phenotype             Phenotype   \n",
       "7                               Alkaline Phosphatase       Protein Product   \n",
       "8                                Antileukoproteinase       Protein Product   \n",
       "\n",
       "  curation_difficulty  \n",
       "0                easy  \n",
       "1              medium  \n",
       "2                hard  \n",
       "3                easy  \n",
       "4              medium  \n",
       "5                hard  \n",
       "6                easy  \n",
       "7              medium  \n",
       "8                hard  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = {'collision_symbol': ['CAP','CAP','MYM','ASP','ASP','ASP','ALP','ALP','ALP'],\n",
    "        'parent_symbol': ['BRD4','LNPEP','ZMYM1','ASIP','TMPRSS11D','ASPM','ATHS','CCL27','ALPI'],\n",
    "        'gene_alias_pair': ['BRD4-CAP','LNPEP-CAP','ZMYM1-MYM','ASIP-ASP','TMPRSS11D-ASP','ASPM-ASP','ATHS-ALP','CCL27-ALP','ALPI-ALP'],\n",
    "        'correct_alias_expansion': [\"Chromosome Associated Protein\",\"Cystinyl Aminopeptidase\",\"Myeloproloferative syndrome and mental retardation\",\"Agouti Signaling Protein\",\"Adrenal secretory serine protease\",\"Drosophila abnormal spindle\",\"Atherogenic Lipoprotein Phenotype\",\"Alkaline Phosphatase\",\"Antileukoproteinase\"],\n",
    "        'collision_association': ['Protein Product','Protein Product','Disease', 'Protein Product', 'Protein Product', 'Ortholog', 'Phenotype', 'Protein Product', 'Protein Product'],\n",
    "        'curation_difficulty': ['easy','medium','hard','easy','medium','hard','easy','medium','hard']}\n",
    "df = pd.DataFrame(data)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for BRD4\n",
      "Checking abstracts for BRD4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 58/58 [01:11<00:00,  1.24s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for LNPEP\n",
      "Checking abstracts for LNPEP\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 15/15 [00:18<00:00,  1.21s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for ZMYM1\n",
      "Checking abstracts for ZMYM1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 5/5 [00:05<00:00,  1.05s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for ASIP\n",
      "Checking abstracts for ASIP\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 7/7 [00:08<00:00,  1.28s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for TMPRSS11D\n",
      "Checking abstracts for TMPRSS11D\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 5/5 [00:06<00:00,  1.28s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for ASPM\n",
      "Checking abstracts for ASPM\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 19/19 [00:22<00:00,  1.21s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for ATHS\n",
      "Retrieving uniprot ref data for CCL27\n",
      "Checking abstracts for CCL27\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 6/6 [00:06<00:00,  1.06s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieving uniprot ref data for ALPI\n",
      "Checking abstracts for ALPI\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 12/12 [00:14<00:00,  1.17s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>collision_symbol</th>\n",
       "      <th>parent_symbol</th>\n",
       "      <th>gene_alias_pair</th>\n",
       "      <th>correct_alias_expansion</th>\n",
       "      <th>collision_association</th>\n",
       "      <th>curation_difficulty</th>\n",
       "      <th>test_results</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>test_results_alt_names</th>\n",
       "      <th>test_results_pmids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAP</td>\n",
       "      <td>BRD4</td>\n",
       "      <td>BRD4-CAP</td>\n",
       "      <td>Chromosome Associated Protein</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>easy</td>\n",
       "      <td>None</td>\n",
       "      <td>[O60885]</td>\n",
       "      <td>[Protein HUNK1]</td>\n",
       "      <td>[{'11733348': []}, {'15057824': []}, {'1548933...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CAP</td>\n",
       "      <td>LNPEP</td>\n",
       "      <td>LNPEP-CAP</td>\n",
       "      <td>Cystinyl Aminopeptidase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "      <td>None</td>\n",
       "      <td>[Q9UIQ6]</td>\n",
       "      <td>[Insulin-regulated membrane aminopeptidase, In...</td>\n",
       "      <td>[{'8550619': []}, {'9177475': []}, {'10759854'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYM</td>\n",
       "      <td>ZMYM1</td>\n",
       "      <td>ZMYM1-MYM</td>\n",
       "      <td>Myeloproloferative syndrome and mental retarda...</td>\n",
       "      <td>Disease</td>\n",
       "      <td>hard</td>\n",
       "      <td>None</td>\n",
       "      <td>[Q5SVZ6]</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'17974005': []}, {'16710414': []}, {'2575529...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASP</td>\n",
       "      <td>ASIP</td>\n",
       "      <td>ASIP-ASP</td>\n",
       "      <td>Agouti Signaling Protein</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>easy</td>\n",
       "      <td>None</td>\n",
       "      <td>[P42127]</td>\n",
       "      <td>[Agouti switch protein]</td>\n",
       "      <td>[{'7937887': []}, {'7757071': ['Agouti Signali...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASP</td>\n",
       "      <td>TMPRSS11D</td>\n",
       "      <td>TMPRSS11D-ASP</td>\n",
       "      <td>Adrenal secretory serine protease</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "      <td>None</td>\n",
       "      <td>[O60235]</td>\n",
       "      <td>[Airway trypsin-like protease]</td>\n",
       "      <td>[{'9565616': []}, {'15489334': ['a saturation ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASP</td>\n",
       "      <td>ASPM</td>\n",
       "      <td>ASPM-ASP</td>\n",
       "      <td>Drosophila abnormal spindle</td>\n",
       "      <td>Ortholog</td>\n",
       "      <td>hard</td>\n",
       "      <td>None</td>\n",
       "      <td>[Q8IZT6]</td>\n",
       "      <td>[Abnormal spindle protein homolog]</td>\n",
       "      <td>[{'12355089': []}, {'14704186': []}, {'1597272...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ALP</td>\n",
       "      <td>ATHS</td>\n",
       "      <td>ATHS-ALP</td>\n",
       "      <td>Atherogenic Lipoprotein Phenotype</td>\n",
       "      <td>Phenotype</td>\n",
       "      <td>easy</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ALP</td>\n",
       "      <td>CCL27</td>\n",
       "      <td>CCL27-ALP</td>\n",
       "      <td>Alkaline Phosphatase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>medium</td>\n",
       "      <td>None</td>\n",
       "      <td>[Q9Y4X3]</td>\n",
       "      <td>[CC chemokine ILC, Cutaneous T-cell-attracting...</td>\n",
       "      <td>[{'10556532': []}, {'10588729': []}, {'1072569...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ALP</td>\n",
       "      <td>ALPI</td>\n",
       "      <td>ALPI-ALP</td>\n",
       "      <td>Antileukoproteinase</td>\n",
       "      <td>Protein Product</td>\n",
       "      <td>hard</td>\n",
       "      <td>None</td>\n",
       "      <td>[P09923]</td>\n",
       "      <td>None</td>\n",
       "      <td>[{'3468508': []}, {'3469665': []}, {'2841341':...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  collision_symbol parent_symbol gene_alias_pair  \\\n",
       "0              CAP          BRD4        BRD4-CAP   \n",
       "1              CAP         LNPEP       LNPEP-CAP   \n",
       "2              MYM         ZMYM1       ZMYM1-MYM   \n",
       "3              ASP          ASIP        ASIP-ASP   \n",
       "4              ASP     TMPRSS11D   TMPRSS11D-ASP   \n",
       "5              ASP          ASPM        ASPM-ASP   \n",
       "6              ALP          ATHS        ATHS-ALP   \n",
       "7              ALP         CCL27       CCL27-ALP   \n",
       "8              ALP          ALPI        ALPI-ALP   \n",
       "\n",
       "                             correct_alias_expansion collision_association  \\\n",
       "0                      Chromosome Associated Protein       Protein Product   \n",
       "1                            Cystinyl Aminopeptidase       Protein Product   \n",
       "2  Myeloproloferative syndrome and mental retarda...               Disease   \n",
       "3                           Agouti Signaling Protein       Protein Product   \n",
       "4                  Adrenal secretory serine protease       Protein Product   \n",
       "5                        Drosophila abnormal spindle              Ortholog   \n",
       "6                  Atherogenic Lipoprotein Phenotype             Phenotype   \n",
       "7                               Alkaline Phosphatase       Protein Product   \n",
       "8                                Antileukoproteinase       Protein Product   \n",
       "\n",
       "  curation_difficulty test_results uniprot_id  \\\n",
       "0                easy         None   [O60885]   \n",
       "1              medium         None   [Q9UIQ6]   \n",
       "2                hard         None   [Q5SVZ6]   \n",
       "3                easy         None   [P42127]   \n",
       "4              medium         None   [O60235]   \n",
       "5                hard         None   [Q8IZT6]   \n",
       "6                easy         None         []   \n",
       "7              medium         None   [Q9Y4X3]   \n",
       "8                hard         None   [P09923]   \n",
       "\n",
       "                              test_results_alt_names  \\\n",
       "0                                    [Protein HUNK1]   \n",
       "1  [Insulin-regulated membrane aminopeptidase, In...   \n",
       "2                                               None   \n",
       "3                            [Agouti switch protein]   \n",
       "4                     [Airway trypsin-like protease]   \n",
       "5                 [Abnormal spindle protein homolog]   \n",
       "6                                               None   \n",
       "7  [CC chemokine ILC, Cutaneous T-cell-attracting...   \n",
       "8                                               None   \n",
       "\n",
       "                                  test_results_pmids  \n",
       "0  [{'11733348': []}, {'15057824': []}, {'1548933...  \n",
       "1  [{'8550619': []}, {'9177475': []}, {'10759854'...  \n",
       "2  [{'17974005': []}, {'16710414': []}, {'2575529...  \n",
       "3  [{'7937887': []}, {'7757071': ['Agouti Signali...  \n",
       "4  [{'9565616': []}, {'15489334': ['a saturation ...  \n",
       "5  [{'12355089': []}, {'14704186': []}, {'1597272...  \n",
       "6                                               None  \n",
       "7  [{'10556532': []}, {'10588729': []}, {'1072569...  \n",
       "8  [{'3468508': []}, {'3469665': []}, {'2841341':...  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Main \n",
    "import time\n",
    "\n",
    "# Initialize\n",
    "df['test_results_alt_names'] = None\n",
    "df['test_results_pmids'] = None\n",
    "df['uniprot_id'] = df['parent_symbol'].apply(grab_uniprot_id)\n",
    "\n",
    "# Grab PMIDS and Alternative Names\n",
    "for index,row in df.iterrows():\n",
    "    print(f'Retrieving uniprot ref data for {row[\"parent_symbol\"]}')\n",
    "    if not row['uniprot_id']:\n",
    "        continue\n",
    "    data = uniprot_request(row['uniprot_id'][0])    \n",
    "    alt_names = grab_alternative_names(data)\n",
    "    df.at[index,'test_results_alt_names'] = alt_names\n",
    "    pmids = grab_pmids(data)\n",
    "\n",
    "    # Grab Abstracts from PubMed\n",
    "    print(f'Grabing abstracts for {row[\"parent_symbol\"]}')\n",
    "    abstracts_to_check = []\n",
    "    for pmid in tqdm(pmids):\n",
    "        entry = {}\n",
    "        if pmid.startswith('CI'):\n",
    "            continue\n",
    "        else:\n",
    "            time.sleep(1) # Pubmed will throw back \"too many requests\" error unless you space them out or find a way to batch them\n",
    "            abstract = get_pubmed_abstract(pmid)\n",
    "            entry[pmid] = abstract\n",
    "            abstracts_to_check.append(entry)\n",
    "\n",
    "    # Check Abstracts using spaCy\n",
    "    print(f'Checking abstracts for {row[\"parent_symbol\"]}')    \n",
    "    possible_matches = []\n",
    "    for abstract_dict in abstracts_to_check:\n",
    "        keys = list(abstract_dict.keys())\n",
    "        possible_match = {}\n",
    "        for key in keys:\n",
    "            result = find_matching_groups(abstract_dict[key],row['collision_symbol'])\n",
    "            possible_match[key] = result\n",
    "        possible_matches.append(possible_match)\n",
    "\n",
    "    df.at[index,'test_results_pmids'] = possible_matches\n",
    "    \n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Agouti switch protein']\n",
      "[{'7937887': []}, {'7757071': ['Agouti Signaling Protein']}, {'11780052': []}, {'15489334': ['a saturation point']}, {'15701517': ['agouti signaling protein']}, {'11833005': ['agouti signaling protein', 'agouti signaling protein']}, {'36536132': ['agouti signaling protein', 'a similar phenotype']}]\n"
     ]
    }
   ],
   "source": [
    "spot_check = 3\n",
    "\n",
    "print(df['test_results_alt_names'][spot_check])\n",
    "\n",
    "print(df['test_results_pmids'][spot_check])\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
