{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2613,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2614,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_ap_collision_df(mini_xxxx_df: pd.DataFrame, source: str) -> pd.DataFrame:\n",
    "    \"\"\"Create a df of alias-primary collision symbols \n",
    "\n",
    "    :param mini_xxxx_df: Processed df of gene records\n",
    "    :param source: Representation of the source of the gene records\n",
    "    :param split_on_character: Character that is used to seperate alias symbols in the mini_xxxx_df\n",
    "    :return: A df of genes that share an alias with another gene\n",
    "    \"\"\"\n",
    "\n",
    "    #Strip \"HGNC\" from HGNC ID\n",
    "    mini_xxxx_df[\"HGNC_ID\"] = mini_xxxx_df[\"HGNC_ID\"].str.lstrip(\"HGNC:\")\n",
    "\n",
    "    #Remove placeholder gene records (those with no primary gene symbol)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\" \", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"-\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.dropna(subset=[\"gene_symbol\"])\n",
    "\n",
    "    #Remove placeholder LOC genes\n",
    "    mini_xxxx_df = mini_xxxx_df[~mini_xxxx_df['gene_symbol'].str.startswith('LOC', na=False)]\n",
    "\n",
    "    mini_xxxx_df['HGNC_ID'] = mini_xxxx_df['HGNC_ID'].fillna(0).astype(int)\n",
    "    mini_xxxx_df['NCBI_ID'] = mini_xxxx_df['NCBI_ID'].fillna(0).astype(int)\n",
    "\n",
    "    #mini_xxx_df\n",
    "    ## no LOC genes (only present in NCBI)\n",
    "    ## no records with no primary symbols (only present in ENSG)\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv and save\n",
    "    mini_xxxx_df.to_csv(f'../output/mini_{source.lower()}_df.csv', index=True) \n",
    "\n",
    "    #Make a new df where the alias symbols are merged together for each record\n",
    "    merged_alias_xxxx_df = mini_xxxx_df.copy()\n",
    "    merged_alias_xxxx_df[\"alias_symbol\"] = merged_alias_xxxx_df[\"alias_symbol\"].fillna(\"\").astype(str)\n",
    "    merged_alias_xxxx_df = (\n",
    "    merged_alias_xxxx_df.groupby(\n",
    "        [f\"{source}_ID\"], dropna=False\n",
    "    )[\"alias_symbol\"]\n",
    "    .apply(lambda x: \",\".join(x.dropna())).reset_index())\n",
    "\n",
    "    #merged_alias_xxxx_df\n",
    "    ##merging aliases associated with the same record (same gene record identifier) into a list\n",
    "    #Convert the df into a csv and save\n",
    "    merged_alias_xxxx_df.to_csv(f'../output/merged_alias_{source}_df.csv', index=True) \n",
    "\n",
    "    #Make a set of the primary gene symbols\n",
    "    xxxx_gene_symbol_set = set(mini_xxxx_df[\"gene_symbol\"])\n",
    "    uppercased_xxxx_gene_symbol_set = {s.upper() for s in xxxx_gene_symbol_set}\n",
    "\n",
    "    \n",
    "    subset_genes_xxxx_df = mini_xxxx_df.copy()\n",
    "\n",
    "    #Remove alias symbols that are an exact match to their respective primary gene symbol\n",
    "    subset_genes_xxxx_df[\"alias_symbol_upper\"] = subset_genes_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df[\"gene_symbol_upper\"] = subset_genes_xxxx_df[\"gene_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df[subset_genes_xxxx_df['gene_symbol_upper'] != subset_genes_xxxx_df['alias_symbol_upper']]\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.drop([\"alias_symbol_upper\",\"gene_symbol_upper\"], axis=1)\n",
    "\n",
    "    #subset_genes_xxxx_df\n",
    "    ## no aliases that match the primary gene symbol\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv and save\n",
    "    subset_genes_xxxx_df.to_csv(f'../output/subset_genes_{source}_df.csv', index=True) \n",
    "\n",
    "    #Create df with genes that have an alias that can be found as another gene's primary gene symbol\n",
    "    ap_collision_xxxx_df = subset_genes_xxxx_df.copy()\n",
    "    ap_collision_xxxx_df[\"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "\n",
    "    #makes list of aliases into a set\n",
    "    ap_collision_xxxx_df.loc[:, \"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(lambda x: {x})\n",
    "\n",
    "    ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(\n",
    "        lambda x: x & uppercased_xxxx_gene_symbol_set\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.drop(\"alias_symbol_upper\", axis=1)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df[ap_collision_xxxx_df[\"collision\"].apply(lambda x: len(x) > 0)]\n",
    "    #changes back the set of aliases to a list, and make the set of collisions to a list\n",
    "    ap_collision_xxxx_df['collision'] = ap_collision_xxxx_df['collision'].apply(\n",
    "        lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.sort_values(\"collision\")\n",
    "\n",
    "    #Add a source tag for future merging efforts\n",
    "    ap_collision_xxxx_df[\"source\"] = str(source.upper())\n",
    "\n",
    "    #ap_collision_xxxx_df\n",
    "    ## only the gene records with an alias that matches another gene record's primary gene symbol\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv\n",
    "    ap_collision_xxxx_df.to_csv(f'../output/single_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "\n",
    "    #Create a secondary collision df that merges the alias symbols for each record\n",
    "    merged_alias_ap_collision_xxxx_df = ap_collision_xxxx_df.drop(columns=['alias_symbol'])\n",
    "    merged_alias_ap_collision_xxxx_df = pd.merge(merged_alias_ap_collision_xxxx_df, merged_alias_xxxx_df, on=[f\"{source}_ID\"], how=\"left\")\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df[[\"gene_symbol\",\"alias_symbol\",\"ENSG_ID\",\"HGNC_ID\",\"NCBI_ID\",\"collision\",\"source\"]]\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.drop_duplicates(subset=[f\"{source}_ID\"])\n",
    "\n",
    "    #Test to make sure all collisions are in the alias list for the record\n",
    "    test_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.copy()\n",
    "    test_ap_collision_xxxx_df[\"alias_symbol_upper\"] = test_ap_collision_xxxx_df[\"alias_symbol\"].str.upper()   \n",
    "    test_ap_collision_xxxx_df['alias_symbol_upper'] = test_ap_collision_xxxx_df['alias_symbol_upper'].apply(lambda x: x if isinstance(x, list) else [x])\n",
    "    test_ap_collision_xxxx_df['collision_in_alias'] = test_ap_collision_xxxx_df.apply(lambda row: row['collision'] in test_ap_collision_xxxx_df['alias_symbol_upper'].values, axis=1)\n",
    "    true_count = test_ap_collision_xxxx_df['collision_in_alias'].sum()\n",
    "    test_ap_collision_xxxx_df = test_ap_collision_xxxx_df.drop(\"alias_symbol_upper\", axis=1)\n",
    "\n",
    "    if true_count == 0:\n",
    "        #merged_alias_ap_collision_xxxx_df\n",
    "        ## only the gene records with an alias that matches another gene record's primary gene symbol\n",
    "        ## merging aliases associated with the same record (same gene record identifier) into a list\n",
    "        #Convert the df into a csv and save\n",
    "        merged_alias_ap_collision_xxxx_df.to_csv(f'../output/merged_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "        print(\"All collisions are present in gene alias lists.\")\n",
    "    else:\n",
    "        print(\"Some collisions are not present in gene alias lists.\")\n",
    "\n",
    "    return mini_xxxx_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2615,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133061</th>\n",
       "      <td>ENSG00000289291</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133064</th>\n",
       "      <td>ENSG00000296488</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133065 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049     <NA>   HGNC:7481         MTTF       MT-TF\n",
       "1       ENSG00000210049     <NA>   HGNC:7481         TRNF       MT-TF\n",
       "2       ENSG00000211459     <NA>   HGNC:7470          12S     MT-RNR1\n",
       "3       ENSG00000211459     <NA>   HGNC:7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459     <NA>   HGNC:7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...         ...          ...         ...\n",
       "133060  ENSG00000229388     <NA>  HGNC:52502    LINC01715    TAF12-DT\n",
       "133061  ENSG00000289291     <NA>         NaN          NaN         NaN\n",
       "133062  ENSG00000274978    26824  HGNC:10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824  HGNC:10108          U11       RNU11\n",
       "133064  ENSG00000296488     <NA>         NaN          NaN         NaN\n",
       "\n",
       "[133065 rows x 5 columns]"
      ]
     },
     "execution_count": 2615,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../input/ensg_biomart_gene20250625.txt\", sep=\"\\t\",dtype={\"NCBI gene (formerly Entrezgene) ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_ensg_df = mini_ensg_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Gene Synonym\": \"alias_symbol\",\n",
    "        \"Gene name\": \"gene_symbol\",\n",
    "        \"Gene stable ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene (formerly Entrezgene) ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2616,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df = mini_ensg_df.replace(\" \", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"-\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"<NA>\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the Ensembl data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2617,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86364"
      ]
     },
     "execution_count": 2617,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['ENSG_ID']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the Ensembl data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2618,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41165"
      ]
     },
     "execution_count": 2618,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['gene_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the Ensembl data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2619,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55413"
      ]
     },
     "execution_count": 2619,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the Ensembl data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2620,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "95942"
      ]
     },
     "execution_count": 2620,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['gene_symbol']) | set(mini_ensg_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2621,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38041"
      ]
     },
     "execution_count": 2621,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_ensg_df = mini_ensg_df[mini_ensg_df['ENSG_ID'].duplicated(keep=False) == False]\n",
    "\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_ensg_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2622,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "57009"
      ]
     },
     "execution_count": 2622,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now count rows in that subset where 'alias_symbol' is missing\n",
    "no_alias_symbol_count = unique_ensg_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2623,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "420"
      ]
     },
     "execution_count": 2623,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ensg_df[\n",
    "    mini_ensg_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ensg_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2624,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 2624,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_ensg_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2625,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "738"
      ]
     },
     "execution_count": 2625,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_ensg_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2626,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 2626,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_ensg_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2627,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92891</th>\n",
       "      <td>ENSG00000138641</td>\n",
       "      <td>8916</td>\n",
       "      <td>HGNC:4876</td>\n",
       "      <td>KIAA0032</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92878</th>\n",
       "      <td>ENSG00000287542</td>\n",
       "      <td>8916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52117</th>\n",
       "      <td>ENSG00000273529</td>\n",
       "      <td>388585</td>\n",
       "      <td>HGNC:19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14696</th>\n",
       "      <td>ENSG00000197921</td>\n",
       "      <td>388585</td>\n",
       "      <td>HGNC:19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4416</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4421</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4418</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4417</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43282</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43285</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4415</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4414</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4419</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43284</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43283</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43281</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43280</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43279</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43286</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4420</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72820</th>\n",
       "      <td>ENSG00000235657</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68794</th>\n",
       "      <td>ENSG00000224320</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66867</th>\n",
       "      <td>ENSG00000227715</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76339</th>\n",
       "      <td>ENSG00000231834</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101068</th>\n",
       "      <td>ENSG00000206503</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67552</th>\n",
       "      <td>ENSG00000206505</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79893</th>\n",
       "      <td>ENSG00000223980</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66597</th>\n",
       "      <td>ENSG00000229215</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79909</th>\n",
       "      <td>ENSG00000224608</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71168</th>\n",
       "      <td>ENSG00000234745</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66216</th>\n",
       "      <td>ENSG00000206450</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70494</th>\n",
       "      <td>ENSG00000232126</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79186</th>\n",
       "      <td>ENSG00000223532</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81469</th>\n",
       "      <td>ENSG00000228964</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62689</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78388</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78387</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78774</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76516</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78386</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78773</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70213</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80918</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76515</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80917</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81311</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81310</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70211</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76517</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80916</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62691</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70212</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62690</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81309</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78772</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76137</th>\n",
       "      <td>ENSG00000242685</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75324</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83537</th>\n",
       "      <td>ENSG00000243215</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74916</th>\n",
       "      <td>ENSG00000243719</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75325</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>RING6</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "92891   ENSG00000138641     8916   HGNC:4876     KIAA0032       HERC3\n",
       "92878   ENSG00000287542     8916         NaN          NaN       HERC3\n",
       "52117   ENSG00000273529   388585  HGNC:19764      BHLHB38        HES5\n",
       "14696   ENSG00000197921   388585  HGNC:19764      BHLHB38        HES5\n",
       "4416    ENSG00000280680    55733  HGNC:18270       MART-2        HHAT\n",
       "4421    ENSG00000280680    55733  HGNC:18270          SKN        HHAT\n",
       "4418    ENSG00000280680    55733  HGNC:18270         RASP        HHAT\n",
       "4417    ENSG00000280680    55733  HGNC:18270        MART2        HHAT\n",
       "43282   ENSG00000054392    55733  HGNC:18270        MART2        HHAT\n",
       "43285   ENSG00000054392    55733  HGNC:18270          SKI        HHAT\n",
       "4415    ENSG00000280680    55733  HGNC:18270         GUP2        HHAT\n",
       "4414    ENSG00000280680    55733  HGNC:18270     FLJ10724        HHAT\n",
       "4419    ENSG00000280680    55733  HGNC:18270          SIT        HHAT\n",
       "43284   ENSG00000054392    55733  HGNC:18270          SIT        HHAT\n",
       "43283   ENSG00000054392    55733  HGNC:18270         RASP        HHAT\n",
       "43281   ENSG00000054392    55733  HGNC:18270       MART-2        HHAT\n",
       "43280   ENSG00000054392    55733  HGNC:18270         GUP2        HHAT\n",
       "43279   ENSG00000054392    55733  HGNC:18270     FLJ10724        HHAT\n",
       "43286   ENSG00000054392    55733  HGNC:18270          SKN        HHAT\n",
       "4420    ENSG00000280680    55733  HGNC:18270          SKI        HHAT\n",
       "72820   ENSG00000235657     3105   HGNC:4931          NaN       HLA-A\n",
       "68794   ENSG00000224320     3105   HGNC:4931          NaN       HLA-A\n",
       "66867   ENSG00000227715     3105   HGNC:4931          NaN       HLA-A\n",
       "76339   ENSG00000231834     3105   HGNC:4931          NaN       HLA-A\n",
       "101068  ENSG00000206503     3105   HGNC:4931          NaN       HLA-A\n",
       "67552   ENSG00000206505     3105   HGNC:4931          NaN       HLA-A\n",
       "79893   ENSG00000223980     3105   HGNC:4931          NaN       HLA-A\n",
       "66597   ENSG00000229215     3105   HGNC:4931          NaN       HLA-A\n",
       "79909   ENSG00000224608     3106   HGNC:4932           AS       HLA-B\n",
       "71168   ENSG00000234745     3106   HGNC:4932           AS       HLA-B\n",
       "66216   ENSG00000206450     3106   HGNC:4932           AS       HLA-B\n",
       "70494   ENSG00000232126     3106   HGNC:4932           AS       HLA-B\n",
       "79186   ENSG00000223532     3106   HGNC:4932           AS       HLA-B\n",
       "81469   ENSG00000228964     3106   HGNC:4932           AS       HLA-B\n",
       "62689   ENSG00000204525     3107   HGNC:4933       D6S204       HLA-C\n",
       "78388   ENSG00000228299     3107   HGNC:4933       PSORS1       HLA-C\n",
       "78387   ENSG00000228299     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "78774   ENSG00000233841     3107   HGNC:4933       PSORS1       HLA-C\n",
       "76516   ENSG00000206452     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "78386   ENSG00000228299     3107   HGNC:4933       D6S204       HLA-C\n",
       "78773   ENSG00000233841     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "70213   ENSG00000225691     3107   HGNC:4933       PSORS1       HLA-C\n",
       "80918   ENSG00000206435     3107   HGNC:4933       PSORS1       HLA-C\n",
       "76515   ENSG00000206452     3107   HGNC:4933       D6S204       HLA-C\n",
       "80917   ENSG00000206435     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "81311   ENSG00000237022     3107   HGNC:4933       PSORS1       HLA-C\n",
       "81310   ENSG00000237022     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "70211   ENSG00000225691     3107   HGNC:4933       D6S204       HLA-C\n",
       "76517   ENSG00000206452     3107   HGNC:4933       PSORS1       HLA-C\n",
       "80916   ENSG00000206435     3107   HGNC:4933       D6S204       HLA-C\n",
       "62691   ENSG00000204525     3107   HGNC:4933       PSORS1       HLA-C\n",
       "70212   ENSG00000225691     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "62690   ENSG00000204525     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "81309   ENSG00000237022     3107   HGNC:4933       D6S204       HLA-C\n",
       "78772   ENSG00000233841     3107   HGNC:4933       D6S204       HLA-C\n",
       "76137   ENSG00000242685     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "75324   ENSG00000243189     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "83537   ENSG00000243215     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "74916   ENSG00000243719     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "75325   ENSG00000243189     3108   HGNC:4934        RING6     HLA-DMA"
      ]
     },
     "execution_count": 2627,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique ENSG ID\n",
    "multi_primary = mini_ensg_df.groupby([\"gene_symbol\"]).ENSG_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_ensg_df.loc[mini_ensg_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').iloc[6000:6060]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2628,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/3440414052.py\u001b[0m in \u001b[0;36m?\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcreate_ap_collision_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmini_ensg_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"ENSG\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/30448796.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(mini_xxxx_df, source)\u001b[0m\n\u001b[1;32m     72\u001b[0m     ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(\n\u001b[1;32m     73\u001b[0m         \u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m&\u001b[0m \u001b[0muppercased_xxxx_gene_symbol_set\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     74\u001b[0m     )\n\u001b[1;32m     75\u001b[0m     \u001b[0map_collision_xxxx_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0map_collision_xxxx_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdrop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"alias_symbol_upper\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m     \u001b[0map_collision_xxxx_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0map_collision_xxxx_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0map_collision_xxxx_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"collision\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m     \u001b[0;31m#changes back the set of aliases to a list, and make the set of collisions to a list\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m     ap_collision_xxxx_df['collision'] = ap_collision_xxxx_df['collision'].apply(\n\u001b[1;32m     79\u001b[0m         \u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m', '\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Desktop/gene-harmony-analysis/.venv/lib/python3.11/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, key)\u001b[0m\n\u001b[0;32m-> 4062\u001b[0;31m     \u001b[0;32mdef\u001b[0m \u001b[0m__getitem__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4063\u001b[0m         \u001b[0mcheck_dict_or_set_indexers\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4064\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitem_from_zerodim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4065\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ensg_df, \"ENSG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../output/mini_ensg_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028</td>\n",
       "      <td>30062</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028</td>\n",
       "      <td>30062</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>0</td>\n",
       "      <td>52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95005 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049        0     7481         MTTF       MT-TF\n",
       "1       ENSG00000210049        0     7481         TRNF       MT-TF\n",
       "2       ENSG00000211459        0     7470          12S     MT-RNR1\n",
       "3       ENSG00000211459        0     7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459        0     7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989    85028    30062    LINC00100      SNHG12\n",
       "133059  ENSG00000197989    85028    30062     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388        0    52502    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978    26824    10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824    10108          U11       RNU11\n",
       "\n",
       "[95005 rows x 5 columns]"
      ]
     },
     "execution_count": 2517,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ensg_df.csv\", index_col=[0])\n",
    "subset_genes_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>10037</td>\n",
       "      <td>125050</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>141</td>\n",
       "      <td>0</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>767 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol          alias_symbol          ENSG_ID  HGNC_ID  NCBI_ID  \\\n",
       "0         RN7SK                   7SK  ENSG00000283293    10037   125050   \n",
       "1         SOAT1  ACAT,ACAT1,SOAT,STAT  ENSG00000057252    11177     6646   \n",
       "2         SOAT2                 ACAT2  ENSG00000167780    11178     8435   \n",
       "3       NDUFAB1  ACP,ACP1,FASN2A,SDAP  ENSG00000004779     7694     4706   \n",
       "4        ACTBP8                ACTBP2  ENSG00000220267      141        0   \n",
       "..          ...                   ...              ...      ...      ...   \n",
       "806      ZNF121   D19S204,ZHC32,ZNF20  ENSG00000197961    12904     7675   \n",
       "807      RNF141          ZFP26,ZNF230  ENSG00000110315    21159    50862   \n",
       "808    ZNF322P1        ZNF322,ZNF322B  ENSG00000188801    14003        0   \n",
       "809      ZNF106  SH3BP3,ZFP106,ZNF474  ENSG00000103994    12886    64397   \n",
       "810        ZFP1       FLJ34243,ZNF475  ENSG00000184517    23328   162239   \n",
       "\n",
       "    collision source  \n",
       "0         7SK   ENSG  \n",
       "1       ACAT1   ENSG  \n",
       "2       ACAT2   ENSG  \n",
       "3        ACP1   ENSG  \n",
       "4      ACTBP2   ENSG  \n",
       "..        ...    ...  \n",
       "806     ZNF20   ENSG  \n",
       "807    ZNF230   ENSG  \n",
       "808    ZNF322   ENSG  \n",
       "809    ZNF474   ENSG  \n",
       "810    ZNF475   ENSG  \n",
       "\n",
       "[767 rows x 7 columns]"
      ]
     },
     "execution_count": 2518,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have an alias that is also a primary gene symbol of another gene?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "767"
      ]
     },
     "execution_count": 2519,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(merged_alias_ap_collision_ensg_df[\"ENSG_ID\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_gene_symbol_set = set(mini_ensg_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41164"
      ]
     },
     "execution_count": 2521,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ensg_gene_symbols = len(ensg_gene_symbol_set)\n",
    "total_number_ensg_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "606"
      ]
     },
     "execution_count": 2522,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_set = set(merged_alias_ap_collision_ensg_df[\"collision\"])\n",
    "len(ensg_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "676"
      ]
     },
     "execution_count": 2523,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_ensg_df[\"gene_symbol\"])\n",
    "len(ensg_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "767"
      ]
     },
     "execution_count": 2524,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_gene_record_set = set(merged_alias_ap_collision_ensg_df[\"ENSG_ID\"])\n",
    "len(ensg_alias_primary_collision_gene_record_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Why is the alias-gene collision set not the same length as the set of primary symbols with collisions ?\n",
    "2. Why is the length of the alias-gene collison set shorter?\n",
    " - A gene record with an alias-primary collision has an alias that matches a different gene's primary gene symbol.\n",
    " - Multiple gene records can share a single alias (alias-alias collision)\n",
    " - If that shared alias is an alias-primary collision, then there will be more unique gene symbols in the set of primary symbols with collisions than the set of alias-primary collisions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGNC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44232</th>\n",
       "      <td>HGNC:9997</td>\n",
       "      <td>RGS16</td>\n",
       "      <td>RGS-r</td>\n",
       "      <td>6004</td>\n",
       "      <td>ENSG00000143333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44233</th>\n",
       "      <td>HGNC:9998</td>\n",
       "      <td>RGS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5997</td>\n",
       "      <td>ENSG00000116741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>C2PA</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>FLJ20370</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>PDZ-RGS3</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66207 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0        HGNC:100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0        HGNC:100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1      HGNC:10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2      HGNC:10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3      HGNC:10002        RGS6          NaN     9628  ENSG00000182732\n",
       "...           ...         ...          ...      ...              ...\n",
       "44232   HGNC:9997       RGS16        RGS-r     6004  ENSG00000143333\n",
       "44233   HGNC:9998        RGS2          NaN     5997  ENSG00000116741\n",
       "44234   HGNC:9999        RGS3         C2PA     5998  ENSG00000138835\n",
       "44234   HGNC:9999        RGS3     FLJ20370     5998  ENSG00000138835\n",
       "44234   HGNC:9999        RGS3     PDZ-RGS3     5998  ENSG00000138835\n",
       "\n",
       "[66207 rows x 5 columns]"
      ]
     },
     "execution_count": 2525,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_file_path = \"../input/hgnc_biomart_gene20250625.txt\"\n",
    "\n",
    "mini_hgnc_df = pd.read_csv(\n",
    "    hgnc_file_path, sep=\"\\t\"\n",
    ")\n",
    "\n",
    "# Rename columns\n",
    "mini_hgnc_df = mini_hgnc_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Approved symbol\": \"gene_symbol\",\n",
    "        \"Ensembl gene ID\": \"ENSG_ID\",\n",
    "    }\n",
    ")\n",
    "\n",
    "#structure and labeling in HGNC download files changed between 2024 amd 2025\n",
    "if \"Alias symbol\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbol\": \"alias_symbol\"})\n",
    "elif \"Alias symbols\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbols\": \"alias_symbol\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"alias_symbol\"] = pd.NA\n",
    "\n",
    "if \"NCBI gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI gene ID\": \"NCBI_ID\"})\n",
    "elif \"NCBI Gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI Gene ID\": \"NCBI_ID\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"NCBI_ID\"] = pd.NA   \n",
    "\n",
    "mini_hgnc_df[\"NCBI_ID\"] = mini_hgnc_df[\"NCBI_ID\"].astype(pd.Int64Dtype())\n",
    "\n",
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', hgnc_file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_hgnc_df['alias_symbol'] = (\n",
    "        mini_hgnc_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split(',')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_hgnc_df = mini_hgnc_df.explode('alias_symbol')\n",
    "\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.replace(\" \", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"\", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"-\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the HGNC data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 2527,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['HGNC_ID']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the HGNC data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 2528,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['gene_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the HGNC data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64842"
      ]
     },
     "execution_count": 2529,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the HGNC data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "108609"
      ]
     },
     "execution_count": 2530,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['gene_symbol']) | set(mini_hgnc_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 2531,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_hcng_df = mini_hgnc_df[mini_hgnc_df['HGNC_ID'].duplicated(keep=False) == False]\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_hgnc_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21767"
      ]
     },
     "execution_count": 2532,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_count = unique_hcng_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "344"
      ]
     },
     "execution_count": 2533,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_hgnc_df[\n",
    "    mini_hgnc_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 2534,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_hgnc_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "477"
      ]
     },
     "execution_count": 2535,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_hgnc_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "47"
      ]
     },
     "execution_count": 2536,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_hgnc_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [HGNC_ID, gene_symbol, alias_symbol, NCBI_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 2537,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_hgnc_df.groupby([\"gene_symbol\"]).HGNC_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_hgnc_df.loc[mini_hgnc_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0      100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0      100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1    10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2    10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3    10002        RGS6          NaN     9628  ENSG00000182732"
      ]
     },
     "execution_count": 2538,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_hgnc_df, \"HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_hgnc_df = pd.read_csv(\n",
    "    \"../output/mini_hgnc_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44232</th>\n",
       "      <td>9997</td>\n",
       "      <td>RGS16</td>\n",
       "      <td>RGS-r</td>\n",
       "      <td>6004</td>\n",
       "      <td>ENSG00000143333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44233</th>\n",
       "      <td>9998</td>\n",
       "      <td>RGS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5997</td>\n",
       "      <td>ENSG00000116741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>C2PA</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>FLJ20370</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>PDZ-RGS3</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66192 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0          100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0          100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1        10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2        10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3        10002        RGS6          NaN     9628  ENSG00000182732\n",
       "...        ...         ...          ...      ...              ...\n",
       "44232     9997       RGS16        RGS-r     6004  ENSG00000143333\n",
       "44233     9998        RGS2          NaN     5997  ENSG00000116741\n",
       "44234     9999        RGS3         C2PA     5998  ENSG00000138835\n",
       "44234     9999        RGS3     FLJ20370     5998  ENSG00000138835\n",
       "44234     9999        RGS3     PDZ-RGS3     5998  ENSG00000138835\n",
       "\n",
       "[66192 rows x 5 columns]"
      ]
     },
     "execution_count": 2540,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df = pd.read_csv(\n",
    "    \"../output/subset_genes_hgnc_df.csv\", index_col=[0])\n",
    "subset_genes_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRL2</td>\n",
       "      <td>HCR,CRAM-B,CKRX,CRAM-A,ACKR5</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>1612</td>\n",
       "      <td>9034</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>ECEL1</td>\n",
       "      <td>XCE,DINE</td>\n",
       "      <td>ENSG00000171551</td>\n",
       "      <td>3147</td>\n",
       "      <td>9427</td>\n",
       "      <td>XCE</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474,SH3BP3</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>547 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol          ENSG_ID  HGNC_ID  \\\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1  ENSG00000125503    14947   \n",
       "1         SOAT1                         ACAT,ACAT1  ENSG00000057252    11177   \n",
       "2         SOAT2                              ACAT2  ENSG00000167780    11178   \n",
       "3         CCRL2       HCR,CRAM-B,CKRX,CRAM-A,ACKR5  ENSG00000121797     1612   \n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1  ENSG00000004779     7694   \n",
       "..          ...                                ...              ...      ...   \n",
       "556       ECEL1                           XCE,DINE  ENSG00000171551     3147   \n",
       "557      ZNF121                        ZHC32,ZNF20  ENSG00000197961    12904   \n",
       "558      RNF141                       ZFP26,ZNF230  ENSG00000110315    21159   \n",
       "559      ZNF106                      ZNF474,SH3BP3  ENSG00000103994    12886   \n",
       "560        ZFP1                    FLJ34243,ZNF475  ENSG00000184517    23328   \n",
       "\n",
       "     NCBI_ID collision source  \n",
       "0      54776     AAVS1   HGNC  \n",
       "1       6646     ACAT1   HGNC  \n",
       "2       8435     ACAT2   HGNC  \n",
       "3       9034     ACKR5   HGNC  \n",
       "4       4706      ACP1   HGNC  \n",
       "..       ...       ...    ...  \n",
       "556     9427       XCE   HGNC  \n",
       "557     7675     ZNF20   HGNC  \n",
       "558    50862    ZNF230   HGNC  \n",
       "559    64397    ZNF474   HGNC  \n",
       "560   162239    ZNF475   HGNC  \n",
       "\n",
       "[547 rows x 7 columns]"
      ]
     },
     "execution_count": 2541,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_hgnc_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_hgnc_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have an alias that is also a primary gene symbol of another gene?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "547"
      ]
     },
     "execution_count": 2542,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(merged_alias_ap_collision_hgnc_df[\"HGNC_ID\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_gene_symbol_set = set(mini_hgnc_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 2544,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_hgnc_gene_symbols = len(hgnc_gene_symbol_set)\n",
    "total_number_hgnc_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "492"
      ]
     },
     "execution_count": 2545,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_set = set(merged_alias_ap_collision_hgnc_df[\"collision\"])\n",
    "len(hgnc_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "547"
      ]
     },
     "execution_count": 2546,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_hgnc_df[\"gene_symbol\"])\n",
    "len(hgnc_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "547"
      ]
     },
     "execution_count": 2547,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_gene_record_set = set(merged_alias_ap_collision_hgnc_df[\"HGNC_ID\"])\n",
    "len(hgnc_alias_primary_collision_gene_record_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NCBI Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2738,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_file_path = \"../input/Homo_sapiens.gene_info20250625\"\n",
    "\n",
    "mini_ncbi_df = pd.read_csv(ncbi_file_path, sep=\"\\t\")\n",
    "# Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol\n",
    "mini_ncbi_df = mini_ncbi_df[\n",
    "[\"GeneID\", \"Symbol\", \"Synonyms\", \"dbXrefs\"]\n",
    "]\n",
    "mini_ncbi_df = mini_ncbi_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Synonyms\": \"alias_symbol\"})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split dbXrefs into individual columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2739,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.assign(\n",
    "    MIM=np.nan,\n",
    "    HGNC_ID=np.nan,\n",
    "    ENSG_ID=np.nan,\n",
    "    AllianceGenome=np.nan,\n",
    "    MIRbase=np.nan,\n",
    "    IMGTgene_db=np.nan,\n",
    "    dash=np.nan,\n",
    "    unknown=np.nan,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2740,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '138670' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ensg00000121410' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'hgnc:5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'cd1d' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '-' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'mi0000060' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    }
   ],
   "source": [
    "index_pos = 0\n",
    "\n",
    "print(len(mini_ncbi_df))\n",
    "while index_pos < len(mini_ncbi_df):\n",
    "    xrefs = mini_ncbi_df[\"dbXrefs\"][index_pos].split(\"|\")\n",
    "\n",
    "    for xref in xrefs:\n",
    "        xref = xref.lower()\n",
    "        if xref.startswith(\"mim:\"):\n",
    "            xref = xref.replace(\"mim:\", \"\")\n",
    "            mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
    "        elif xref.startswith(\"hgnc:hgnc:\"):\n",
    "            xref = xref.replace(\"hgnc:hgnc:\", \"\")\n",
    "            mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"ensembl:\"):\n",
    "            xref = xref.replace(\"ensembl:\", \"\")\n",
    "            mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"alliancegenome:\"):\n",
    "            xref = xref.replace(\"alliancegenome:\", \"\")\n",
    "            mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
    "        elif xref.startswith(\"mirbase\"):\n",
    "            xref = xref.replace(\"mirbase:\", \"\")\n",
    "            mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
    "        elif xref.startswith(\"imgt/gene-db:\"):\n",
    "            xref = xref.replace(\"imgt/gene-db:\", \"\")\n",
    "            mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
    "        elif xref.startswith(\"-\"):\n",
    "            mini_ncbi_df[\"dash\"][index_pos] = xref\n",
    "        else:\n",
    "            mini_ncbi_df[\"unknown\"][index_pos] = xref\n",
    "\n",
    "    index_pos += 1\n",
    "    pass\n",
    "\n",
    "print(index_pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2741,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df[\"ENSG_ID\"] = mini_ncbi_df[\"ENSG_ID\"].str.replace(\"ensg\", \"ENSG\", 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2742,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.drop(\n",
    "    [\n",
    "        \"AllianceGenome\",\n",
    "        \"MIRbase\",\n",
    "        \"IMGTgene_db\",\n",
    "        \"dash\",\n",
    "        \"unknown\",\n",
    "        \"dbXrefs\",\n",
    "        \"MIM\",\n",
    "    ],\n",
    "    axis=1,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "need to remove phenotype records. This file (ncbi_records_w_property_pheno_only_20250630.txt) was generated through the NCBI Gene web page by searching for these parameters \"phenotype only [Properties] AND \"Homo sapiens\"[porgn:__txid9606]\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2743,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_pheno_records_df = pd.read_csv(\"../input/ncbi_records_w_property_pheno_only_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_pheno_records_df = ncbi_pheno_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2744,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tax_id</th>\n",
       "      <th>Org_name</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>CurrentID</th>\n",
       "      <th>Status</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>description</th>\n",
       "      <th>other_designations</th>\n",
       "      <th>map_location</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>genomic_nucleotide_accession.version</th>\n",
       "      <th>start_position_on_the_genomic_accession</th>\n",
       "      <th>end_position_on_the_genomic_accession</th>\n",
       "      <th>orientation</th>\n",
       "      <th>exon_count</th>\n",
       "      <th>OMIM</th>\n",
       "      <th>Unnamed: 17</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>642</th>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>107648861</td>\n",
       "      <td>0</td>\n",
       "      <td>live</td>\n",
       "      <td>DEL11P13</td>\n",
       "      <td>C11DELp13</td>\n",
       "      <td>Chromosome 11p13 deletion syndrome, distal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11p13</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>616902.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     tax_id      Org_name    NCBI_ID  CurrentID Status gene_symbol  \\\n",
       "642    9606  Homo sapiens  107648861          0   live    DEL11P13   \n",
       "\n",
       "    alias_symbol                                 description  \\\n",
       "642    C11DELp13  Chromosome 11p13 deletion syndrome, distal   \n",
       "\n",
       "    other_designations map_location chromosome  \\\n",
       "642                NaN        11p13         11   \n",
       "\n",
       "     genomic_nucleotide_accession.version  \\\n",
       "642                                   NaN   \n",
       "\n",
       "     start_position_on_the_genomic_accession  \\\n",
       "642                                      NaN   \n",
       "\n",
       "     end_position_on_the_genomic_accession  orientation  exon_count      OMIM  \\\n",
       "642                                    NaN          NaN         NaN  616902.0   \n",
       "\n",
       "     Unnamed: 17  \n",
       "642          NaN  "
      ]
     },
     "execution_count": 2744,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_pheno_records_df.loc[ncbi_pheno_records_df[\"NCBI_ID\"]== 107648861]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2745,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_pheno_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2746,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [NCBI_ID, gene_symbol, alias_symbol, HGNC_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 2746,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df.loc[mini_ncbi_df[\"NCBI_ID\"]== 107648861]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2747,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239237 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B       5  ENSG00000121410\n",
       "0             1        A1BG          ABG       5  ENSG00000121410\n",
       "0             1        A1BG          GAB       5  ENSG00000121410\n",
       "0             1        A1BG     HYST2477       5  ENSG00000121410\n",
       "1             2         A2M         A2MD       7  ENSG00000175899\n",
       "...         ...         ...          ...     ...              ...\n",
       "193575  8923215        trnD            -     NaN              NaN\n",
       "193576  8923216        trnP            -     NaN              NaN\n",
       "193577  8923217        trnA            -     NaN              NaN\n",
       "193578  8923218        COX1            -     NaN              NaN\n",
       "193579  8923219    16S rRNA            -     NaN              NaN\n",
       "\n",
       "[239237 rows x 5 columns]"
      ]
     },
     "execution_count": 2747,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', ncbi_file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_ncbi_df['alias_symbol'] = (\n",
    "        mini_ncbi_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split('|')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_ncbi_df = mini_ncbi_df.explode('alias_symbol')\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2748,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.replace(\" \", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"\", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"-\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the NCBI data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2749,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "192410"
      ]
     },
     "execution_count": 2749,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_raw_gene_record_count_ncbi = mini_ncbi_df['NCBI_ID'].dropna().nunique()\n",
    "total_raw_gene_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the NCBI data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2750,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "192262"
      ]
     },
     "execution_count": 2750,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_primary_symbol_count_ncbi = mini_ncbi_df['gene_symbol'].dropna().nunique()\n",
    "raw_primary_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the NCBI data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2751,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "69365"
      ]
     },
     "execution_count": 2751,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_alias_symbol_count_ncbi = mini_ncbi_df['alias_symbol'].dropna().nunique()\n",
    "raw_alias_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the NCBI data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2752,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "260131"
      ]
     },
     "execution_count": 2752,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_total_symbol_count_ncbi = pd.concat([mini_ncbi_df[\"gene_symbol\"], mini_ncbi_df[\"alias_symbol\"]]).dropna().nunique()\n",
    "raw_total_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2753,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 2753,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_primary_symbol_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].isna()]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "no_primary_symbol_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2754,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "165021"
      ]
     },
     "execution_count": 2754,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"alias_symbol\"].isna()]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "no_alias_symbol_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2755,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "248"
      ]
     },
     "execution_count": 2755,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ncbi_df[\n",
    "    mini_ncbi_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "orf_record_count_ncbi = orf_rows[\"NCBI_ID\"].nunique()\n",
    "orf_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2756,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147698"
      ]
     },
     "execution_count": 2756,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('LOC', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "loc_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2757,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "396"
      ]
     },
     "execution_count": 2757,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('FAM', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "fam_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2758,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 2758,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kiaa_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('KIAA', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "kiaa_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop sub species records: \"homo sapiens ssp denisova\"[Organism] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2759,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_denisova_records_df = pd.read_csv(\"../input/ncbi_denisova_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_denisova_records_df = ncbi_denisova_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2760,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_denisova_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop sub species records: \"homo sapiens neanderthalensis\"[Organism] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2761,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_neanderthalensis_records_df = pd.read_csv(\"../input/ncbi_neanderthalensis_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_neanderthalensis_records_df = ncbi_neanderthalensis_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2762,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_neanderthalensis_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2763,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "110\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3550</th>\n",
       "      <td>4549</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4762</th>\n",
       "      <td>6052</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10082</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3551</th>\n",
       "      <td>4550</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>MTRNR2</td>\n",
       "      <td>7471</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>6053</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10083</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65892</th>\n",
       "      <td>124901565</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65891</th>\n",
       "      <td>124901564</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65890</th>\n",
       "      <td>124901563</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65889</th>\n",
       "      <td>124901562</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70782</th>\n",
       "      <td>124906586</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65888</th>\n",
       "      <td>124901561</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70114</th>\n",
       "      <td>124905878</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70102</th>\n",
       "      <td>124905866</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70099</th>\n",
       "      <td>124905863</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70108</th>\n",
       "      <td>124905872</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70096</th>\n",
       "      <td>124905860</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70093</th>\n",
       "      <td>124905857</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70111</th>\n",
       "      <td>124905875</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70090</th>\n",
       "      <td>124905854</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70117</th>\n",
       "      <td>124905881</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70120</th>\n",
       "      <td>124905884</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70123</th>\n",
       "      <td>124905887</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70105</th>\n",
       "      <td>124905869</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70126</th>\n",
       "      <td>124905890</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70138</th>\n",
       "      <td>124905902</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70132</th>\n",
       "      <td>124905896</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70135</th>\n",
       "      <td>124905899</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70129</th>\n",
       "      <td>124905893</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70103</th>\n",
       "      <td>124905867</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70106</th>\n",
       "      <td>124905870</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70100</th>\n",
       "      <td>124905864</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70139</th>\n",
       "      <td>124905903</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70121</th>\n",
       "      <td>124905885</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70097</th>\n",
       "      <td>124905861</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70136</th>\n",
       "      <td>124905900</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70109</th>\n",
       "      <td>124905873</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70124</th>\n",
       "      <td>124905888</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70133</th>\n",
       "      <td>124905897</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70091</th>\n",
       "      <td>124905855</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70127</th>\n",
       "      <td>124905891</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70112</th>\n",
       "      <td>124905876</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70115</th>\n",
       "      <td>124905879</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70130</th>\n",
       "      <td>124905894</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70094</th>\n",
       "      <td>124905858</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70118</th>\n",
       "      <td>124905882</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70144</th>\n",
       "      <td>124905908</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54286</th>\n",
       "      <td>107987368</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69857</th>\n",
       "      <td>124905580</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69860</th>\n",
       "      <td>124905583</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69861</th>\n",
       "      <td>124905584</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69863</th>\n",
       "      <td>124905586</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69858</th>\n",
       "      <td>124905581</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69855</th>\n",
       "      <td>124905578</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69865</th>\n",
       "      <td>124905588</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70143</th>\n",
       "      <td>124905907</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70146</th>\n",
       "      <td>124905910</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70083</th>\n",
       "      <td>124905847</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70148</th>\n",
       "      <td>124905912</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70150</th>\n",
       "      <td>124905914</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70085</th>\n",
       "      <td>124905849</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70152</th>\n",
       "      <td>124905916</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         NCBI_ID gene_symbol alias_symbol HGNC_ID ENSG_ID\n",
       "3550        4549        RNR1       MTRNR1    7470     NaN\n",
       "4762        6052        RNR1          NaN   10082     NaN\n",
       "3551        4550        RNR2       MTRNR2    7471     NaN\n",
       "4763        6053        RNR2          NaN   10083     NaN\n",
       "65892  124901565   TRNAA-AGC          NaN     NaN     NaN\n",
       "65891  124901564   TRNAA-AGC          NaN     NaN     NaN\n",
       "65890  124901563   TRNAA-AGC          NaN     NaN     NaN\n",
       "65889  124901562   TRNAA-AGC          NaN     NaN     NaN\n",
       "70782  124906586   TRNAA-AGC          NaN     NaN     NaN\n",
       "65888  124901561   TRNAA-AGC          NaN     NaN     NaN\n",
       "70114  124905878   TRNAD-GUC          NaN     NaN     NaN\n",
       "70102  124905866   TRNAD-GUC          NaN     NaN     NaN\n",
       "70099  124905863   TRNAD-GUC          NaN     NaN     NaN\n",
       "70108  124905872   TRNAD-GUC          NaN     NaN     NaN\n",
       "70096  124905860   TRNAD-GUC          NaN     NaN     NaN\n",
       "70093  124905857   TRNAD-GUC          NaN     NaN     NaN\n",
       "70111  124905875   TRNAD-GUC          NaN     NaN     NaN\n",
       "70090  124905854   TRNAD-GUC          NaN     NaN     NaN\n",
       "70117  124905881   TRNAD-GUC          NaN     NaN     NaN\n",
       "70120  124905884   TRNAD-GUC          NaN     NaN     NaN\n",
       "70123  124905887   TRNAD-GUC          NaN     NaN     NaN\n",
       "70105  124905869   TRNAD-GUC          NaN     NaN     NaN\n",
       "70126  124905890   TRNAD-GUC          NaN     NaN     NaN\n",
       "70138  124905902   TRNAD-GUC          NaN     NaN     NaN\n",
       "70132  124905896   TRNAD-GUC          NaN     NaN     NaN\n",
       "70135  124905899   TRNAD-GUC          NaN     NaN     NaN\n",
       "70129  124905893   TRNAD-GUC          NaN     NaN     NaN\n",
       "70103  124905867   TRNAE-CUC          NaN     NaN     NaN\n",
       "70106  124905870   TRNAE-CUC          NaN     NaN     NaN\n",
       "70100  124905864   TRNAE-CUC          NaN     NaN     NaN\n",
       "70139  124905903   TRNAE-CUC          NaN     NaN     NaN\n",
       "70121  124905885   TRNAE-CUC          NaN     NaN     NaN\n",
       "70097  124905861   TRNAE-CUC          NaN     NaN     NaN\n",
       "70136  124905900   TRNAE-CUC          NaN     NaN     NaN\n",
       "70109  124905873   TRNAE-CUC          NaN     NaN     NaN\n",
       "70124  124905888   TRNAE-CUC          NaN     NaN     NaN\n",
       "70133  124905897   TRNAE-CUC          NaN     NaN     NaN\n",
       "70091  124905855   TRNAE-CUC          NaN     NaN     NaN\n",
       "70127  124905891   TRNAE-CUC          NaN     NaN     NaN\n",
       "70112  124905876   TRNAE-CUC          NaN     NaN     NaN\n",
       "70115  124905879   TRNAE-CUC          NaN     NaN     NaN\n",
       "70130  124905894   TRNAE-CUC          NaN     NaN     NaN\n",
       "70094  124905858   TRNAE-CUC          NaN     NaN     NaN\n",
       "70118  124905882   TRNAE-CUC          NaN     NaN     NaN\n",
       "70144  124905908   TRNAE-UUC          NaN     NaN     NaN\n",
       "54286  107987368   TRNAE-UUC          NaN     NaN     NaN\n",
       "69857  124905580   TRNAE-UUC          NaN     NaN     NaN\n",
       "69860  124905583   TRNAE-UUC          NaN     NaN     NaN\n",
       "69861  124905584   TRNAE-UUC          NaN     NaN     NaN\n",
       "69863  124905586   TRNAE-UUC          NaN     NaN     NaN\n",
       "69858  124905581   TRNAG-CCC          NaN     NaN     NaN\n",
       "69855  124905578   TRNAG-CCC          NaN     NaN     NaN\n",
       "69865  124905588   TRNAG-CCC          NaN     NaN     NaN\n",
       "70143  124905907   TRNAG-GCC          NaN     NaN     NaN\n",
       "70146  124905910   TRNAG-GCC          NaN     NaN     NaN\n",
       "70083  124905847   TRNAG-GCC          NaN     NaN     NaN\n",
       "70148  124905912   TRNAG-GCC          NaN     NaN     NaN\n",
       "70150  124905914   TRNAG-GCC          NaN     NaN     NaN\n",
       "70085  124905849   TRNAG-GCC          NaN     NaN     NaN\n",
       "70152  124905916   TRNAG-GCC          NaN     NaN     NaN"
      ]
     },
     "execution_count": 2763,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_ncbi_df.groupby([\"gene_symbol\"]).NCBI_ID.nunique().gt(1)\n",
    "\n",
    "num_rows = mini_ncbi_df[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].shape[0]\n",
    "print(num_rows)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_country`:\n",
    "mini_ncbi_df.loc[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2764,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [NCBI_ID, gene_symbol, alias_symbol, HGNC_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 2764,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df.loc[mini_ncbi_df[\"gene_symbol\"] == \"ST2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2765,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0        1        A1BG          A1B        5  ENSG00000121410\n",
       "0        1        A1BG          ABG        5  ENSG00000121410\n",
       "0        1        A1BG          GAB        5  ENSG00000121410\n",
       "0        1        A1BG     HYST2477        5  ENSG00000121410\n",
       "1        2         A2M         A2MD        7  ENSG00000175899"
      ]
     },
     "execution_count": 2765,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ncbi_df, \"NCBI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2766,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = pd.read_csv(\n",
    "    \"../output/mini_ncbi_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2767,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193502</th>\n",
       "      <td>141732005</td>\n",
       "      <td>ADCY2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40064</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193503</th>\n",
       "      <td>141732006</td>\n",
       "      <td>NSG2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41074</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193504</th>\n",
       "      <td>141732007</td>\n",
       "      <td>ST18-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193505</th>\n",
       "      <td>141732008</td>\n",
       "      <td>MICAL2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58437</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193506</th>\n",
       "      <td>141732009</td>\n",
       "      <td>TRIM41-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40863</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91377 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0               1        A1BG          A1B        5  ENSG00000121410\n",
       "0               1        A1BG          ABG        5  ENSG00000121410\n",
       "0               1        A1BG          GAB        5  ENSG00000121410\n",
       "0               1        A1BG     HYST2477        5  ENSG00000121410\n",
       "1               2         A2M         A2MD        7  ENSG00000175899\n",
       "...           ...         ...          ...      ...              ...\n",
       "193502  141732005   ADCY2-AS1          NaN    40064              NaN\n",
       "193503  141732006    NSG2-AS1          NaN    41074              NaN\n",
       "193504  141732007    ST18-AS1          NaN    58430              NaN\n",
       "193505  141732008  MICAL2-AS1          NaN    58437              NaN\n",
       "193506  141732009  TRIM41-AS1          NaN    40863              NaN\n",
       "\n",
       "[91377 rows x 5 columns]"
      ]
     },
     "execution_count": 2767,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ncbi_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ncbi_df.csv\", index_col=[0])\n",
    "subset_genes_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2768,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IGHA2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000211890</td>\n",
       "      <td>5479</td>\n",
       "      <td>3494</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1,LENG3,MBS85,p84,p85</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACACT,ACAT,ACAT-1,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACACT2,ACAT2,ARGP2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCRL2</td>\n",
       "      <td>ACKR5,CKRX,CRAM,CRAM-A,CRAM-B,HCR</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>1612</td>\n",
       "      <td>9034</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2032</th>\n",
       "      <td>CCDC106</td>\n",
       "      <td>HSU79303,ZNF581</td>\n",
       "      <td>ENSG00000173581</td>\n",
       "      <td>30181</td>\n",
       "      <td>29903</td>\n",
       "      <td>ZNF581</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2033</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>ENSG00000197162</td>\n",
       "      <td>26496</td>\n",
       "      <td>146540</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2034</th>\n",
       "      <td>ZP4</td>\n",
       "      <td>ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4</td>\n",
       "      <td>ENSG00000116996</td>\n",
       "      <td>15770</td>\n",
       "      <td>57829</td>\n",
       "      <td>ZP1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2036</th>\n",
       "      <td>ZNF446</td>\n",
       "      <td>ZKSCAN20,ZSCAN30,ZSCAN52</td>\n",
       "      <td>ENSG00000083838</td>\n",
       "      <td>21036</td>\n",
       "      <td>55663</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2037</th>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>ZNF694,ZSCAN31,ZSCAN34</td>\n",
       "      <td>ENSG00000155592</td>\n",
       "      <td>25677</td>\n",
       "      <td>342357</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1614 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol                       alias_symbol          ENSG_ID  HGNC_ID  \\\n",
       "0          IGHA2                                A2M  ENSG00000211890     5479   \n",
       "1       PPP1R12C          AAVS1,LENG3,MBS85,p84,p85  ENSG00000125503    14947   \n",
       "2          SOAT1  ACACT,ACAT,ACAT-1,ACAT1,SOAT,STAT  ENSG00000057252    11177   \n",
       "3          SOAT2                 ACACT2,ACAT2,ARGP2  ENSG00000167780    11178   \n",
       "4          CCRL2  ACKR5,CKRX,CRAM,CRAM-A,CRAM-B,HCR  ENSG00000121797     1612   \n",
       "...          ...                                ...              ...      ...   \n",
       "2032     CCDC106                    HSU79303,ZNF581  ENSG00000173581    30181   \n",
       "2033      ZNF785                             ZNF688  ENSG00000197162    26496   \n",
       "2034         ZP4         ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4  ENSG00000116996    15770   \n",
       "2036      ZNF446           ZKSCAN20,ZSCAN30,ZSCAN52  ENSG00000083838    21036   \n",
       "2037     ZKSCAN2             ZNF694,ZSCAN31,ZSCAN34  ENSG00000155592    25677   \n",
       "\n",
       "      NCBI_ID collision source  \n",
       "0        3494       A2M   NCBI  \n",
       "1       54776     AAVS1   NCBI  \n",
       "2        6646     ACAT1   NCBI  \n",
       "3        8435     ACAT2   NCBI  \n",
       "4        9034     ACKR5   NCBI  \n",
       "...       ...       ...    ...  \n",
       "2032    29903    ZNF581   NCBI  \n",
       "2033   146540    ZNF688   NCBI  \n",
       "2034    57829       ZP1   NCBI  \n",
       "2036    55663   ZSCAN30   NCBI  \n",
       "2037   342357   ZSCAN31   NCBI  \n",
       "\n",
       "[1614 rows x 7 columns]"
      ]
     },
     "execution_count": 2768,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ncbi_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ncbi_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have an alias that is also a primary gene symbol of another gene?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2769,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1614"
      ]
     },
     "execution_count": 2769,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(merged_alias_ap_collision_ncbi_df[\"NCBI_ID\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2770,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_gene_symbol_set = set(mini_ncbi_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2771,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44541"
      ]
     },
     "execution_count": 2771,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ncbi_gene_symbols = len(ncbi_gene_symbol_set)\n",
    "total_number_ncbi_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2772,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1394"
      ]
     },
     "execution_count": 2772,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"collision\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2773,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1614"
      ]
     },
     "execution_count": 2773,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_ncbi_df[\"gene_symbol\"])\n",
    "len(ncbi_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2774,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1614"
      ]
     },
     "execution_count": 2774,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_gene_record_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"NCBI_ID\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_gene_record_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge 3 sets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2775,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRL2</td>\n",
       "      <td>HCR,CRAM-B,CKRX,CRAM-A,ACKR5</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>1612</td>\n",
       "      <td>9034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2928 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol collision source  \\\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1     AAVS1   HGNC   \n",
       "1         SOAT1                         ACAT,ACAT1     ACAT1   HGNC   \n",
       "2         SOAT2                              ACAT2     ACAT2   HGNC   \n",
       "3         CCRL2       HCR,CRAM-B,CKRX,CRAM-A,ACKR5     ACKR5   HGNC   \n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1      ACP1   HGNC   \n",
       "..          ...                                ...       ...    ...   \n",
       "806      ZNF121                D19S204,ZHC32,ZNF20     ZNF20   ENSG   \n",
       "807      RNF141                       ZFP26,ZNF230    ZNF230   ENSG   \n",
       "808    ZNF322P1                     ZNF322,ZNF322B    ZNF322   ENSG   \n",
       "809      ZNF106               SH3BP3,ZFP106,ZNF474    ZNF474   ENSG   \n",
       "810        ZFP1                    FLJ34243,ZNF475    ZNF475   ENSG   \n",
       "\n",
       "             ENSG_ID  HGNC_ID  NCBI_ID  \n",
       "0    ENSG00000125503    14947    54776  \n",
       "1    ENSG00000057252    11177     6646  \n",
       "2    ENSG00000167780    11178     8435  \n",
       "3    ENSG00000121797     1612     9034  \n",
       "4    ENSG00000004779     7694     4706  \n",
       "..               ...      ...      ...  \n",
       "806  ENSG00000197961    12904     7675  \n",
       "807  ENSG00000110315    21159    50862  \n",
       "808  ENSG00000188801    14003        0  \n",
       "809  ENSG00000103994    12886    64397  \n",
       "810  ENSG00000184517    23328   162239  \n",
       "\n",
       "[2928 rows x 7 columns]"
      ]
     },
     "execution_count": 2775,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df = pd.concat(\n",
    "    [\n",
    "        merged_alias_ap_collision_hgnc_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ncbi_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ensg_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "    ]\n",
    ")\n",
    "merged_alias_primary_collisions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2776,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene_symbol, alias_symbol, collision, source, ENSG_ID, HGNC_ID, NCBI_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 2776,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"CFM1\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2777,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>ALPS4,CMNS,KRAS,N-ras,NCMS,NRAS1,NS6</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>ENSG00000213281</td>\n",
       "      <td>7989</td>\n",
       "      <td>4893</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                          alias_symbol collision source  \\\n",
       "964        NRAS  ALPS4,CMNS,KRAS,N-ras,NCMS,NRAS1,NS6      KRAS   NCBI   \n",
       "\n",
       "             ENSG_ID  HGNC_ID  NCBI_ID  \n",
       "964  ENSG00000213281     7989     4893  "
      ]
     },
     "execution_count": 2777,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"KRAS\"\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2778,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_primary_collisions_df.to_csv(\n",
    "    \"../output/merged_alias_primary_collisions_df.csv\", index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2779,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ABCD1',\n",
       " 'ACD',\n",
       " 'ACKR2',\n",
       " 'ACOD1',\n",
       " 'ACTBP8',\n",
       " 'ADRA1D',\n",
       " 'AGXT',\n",
       " 'AIFM2',\n",
       " 'AKR1B1',\n",
       " 'AKR1B10',\n",
       " 'ALPK3',\n",
       " 'AMH',\n",
       " 'ANAPC2',\n",
       " 'ANKRD37',\n",
       " 'ANTXR1',\n",
       " 'AOC1',\n",
       " 'APEX1',\n",
       " 'AREG',\n",
       " 'ARHGAP21',\n",
       " 'ARHGEF7',\n",
       " 'ARID4A',\n",
       " 'ART4',\n",
       " 'ARTN',\n",
       " 'ASIC2',\n",
       " 'AURKAIP1',\n",
       " 'AZIN2',\n",
       " 'BANF1P1',\n",
       " 'BCAT2',\n",
       " 'BRIP1',\n",
       " 'BTF3P11',\n",
       " 'BTN3A3',\n",
       " 'C1D',\n",
       " 'C1QTNF1',\n",
       " 'C6orf89',\n",
       " 'CACNA1A',\n",
       " 'CADPS',\n",
       " 'CADPS2',\n",
       " 'CAPN5',\n",
       " 'CARD16',\n",
       " 'CCL13',\n",
       " 'CCL14',\n",
       " 'CCL15',\n",
       " 'CCM2',\n",
       " 'CD200R1',\n",
       " 'CDH19',\n",
       " 'CDH20',\n",
       " 'CDPF1',\n",
       " 'CELSR1',\n",
       " 'CES1',\n",
       " 'CFAP73',\n",
       " 'CFH',\n",
       " 'CHAF1B',\n",
       " 'CHAMP1',\n",
       " 'CHD6',\n",
       " 'CHEK2',\n",
       " 'CHORDC1',\n",
       " 'CLASP1',\n",
       " 'CLCF1',\n",
       " 'CNGB1',\n",
       " 'CNKSR2',\n",
       " 'CNOT6',\n",
       " 'CNRIP1',\n",
       " 'CNTN1',\n",
       " 'COASY',\n",
       " 'COPS2',\n",
       " 'COPS3',\n",
       " 'CORIN',\n",
       " 'COX7A2L',\n",
       " 'CPA4',\n",
       " 'CPAMD8',\n",
       " 'CPNE1',\n",
       " 'CPNE2',\n",
       " 'CPPED1',\n",
       " 'CREB3L4',\n",
       " 'CSNK2A2',\n",
       " 'CSTB',\n",
       " 'CTDSP2',\n",
       " 'CXCL10',\n",
       " 'CXXC1',\n",
       " 'CYCSP5',\n",
       " 'CYP11B1',\n",
       " 'CYP11B2',\n",
       " 'CYP21A2',\n",
       " 'CYP2A6',\n",
       " 'DCAF5',\n",
       " 'DCBLD2',\n",
       " 'DDOST',\n",
       " 'DDR2',\n",
       " 'DDX11',\n",
       " 'DDX18',\n",
       " 'DEAF1',\n",
       " 'DEFA6',\n",
       " 'DEPDC1B',\n",
       " 'DFFB',\n",
       " 'DHX8',\n",
       " 'DIDO1',\n",
       " 'DLC1',\n",
       " 'DLEC1',\n",
       " 'DLGAP3',\n",
       " 'DMAP1',\n",
       " 'DMTF1',\n",
       " 'DNPEP',\n",
       " 'DPAGT1',\n",
       " 'DPYSL2',\n",
       " 'DSC3',\n",
       " 'DUSP12',\n",
       " 'DUSP28',\n",
       " 'DYNLL1',\n",
       " 'E2F1',\n",
       " 'EIF2S1',\n",
       " 'EIF4G2',\n",
       " 'ELOF1',\n",
       " 'EPAS1',\n",
       " 'EPS8L1',\n",
       " 'EPX',\n",
       " 'ERC2',\n",
       " 'ERCC4',\n",
       " 'ETFA',\n",
       " 'F3',\n",
       " 'FA2H',\n",
       " 'FABP5P1',\n",
       " 'FADS1',\n",
       " 'FANCA',\n",
       " 'FASN',\n",
       " 'FBF1',\n",
       " 'FBXO5',\n",
       " 'FCRLA',\n",
       " 'FCRLB',\n",
       " 'FEM1C',\n",
       " 'FGF13',\n",
       " 'FGFR1',\n",
       " 'FLII',\n",
       " 'FOXM1',\n",
       " 'FYN',\n",
       " 'FZD9',\n",
       " 'FZR1',\n",
       " 'GAL3ST3',\n",
       " 'GALK2',\n",
       " 'GALNT15',\n",
       " 'GALNT18',\n",
       " 'GALNTL6',\n",
       " 'GALP',\n",
       " 'GARS1',\n",
       " 'GATD3',\n",
       " 'GC',\n",
       " 'GDF15',\n",
       " 'GDNF',\n",
       " 'GET1',\n",
       " 'GKN1',\n",
       " 'GLB1',\n",
       " 'GLS2',\n",
       " 'GMNN',\n",
       " 'GNAI2',\n",
       " 'GNPDA1',\n",
       " 'GOLT1A',\n",
       " 'GOLT1B',\n",
       " 'GPI',\n",
       " 'GPKOW',\n",
       " 'GPNMB',\n",
       " 'GPRIN1',\n",
       " 'GPX7',\n",
       " 'GRID1',\n",
       " 'GRID2',\n",
       " 'GTPBP4',\n",
       " 'GUCY2C',\n",
       " 'HCRT',\n",
       " 'HCRTR1',\n",
       " 'HECA',\n",
       " 'HFE',\n",
       " 'HID1',\n",
       " 'HNF1A-AS1',\n",
       " 'HOMER2',\n",
       " 'HOOK1',\n",
       " 'HOOK2',\n",
       " 'HOOK3',\n",
       " 'HRNR',\n",
       " 'HSD17B4',\n",
       " 'HSD17B7P2',\n",
       " 'IAPP',\n",
       " 'IFNA1',\n",
       " 'IGF2BP3',\n",
       " 'IL22',\n",
       " 'IL6',\n",
       " 'ILF3',\n",
       " 'INSL3',\n",
       " 'INSL6',\n",
       " 'IPO4',\n",
       " 'ITM2C',\n",
       " 'KANSL1',\n",
       " 'KCNA4',\n",
       " 'KCNA5',\n",
       " 'KCNH4',\n",
       " 'KCNH8',\n",
       " 'KCNN4',\n",
       " 'KCTD11',\n",
       " 'KHSRP',\n",
       " 'KIF20B',\n",
       " 'KIF2A',\n",
       " 'KLF5',\n",
       " 'KLHL7',\n",
       " 'KLK1',\n",
       " 'KLK6',\n",
       " 'KLRG1',\n",
       " 'KRTAP5-11',\n",
       " 'KRTAP5-2',\n",
       " 'KRTAP5-3',\n",
       " 'KRTAP5-5',\n",
       " 'KRTAP5-7',\n",
       " 'KRTAP5-8',\n",
       " 'KRTAP5-9',\n",
       " 'LAMA4',\n",
       " 'LDB2',\n",
       " 'LGALS12',\n",
       " 'LGALS13',\n",
       " 'LGALSL',\n",
       " 'LNPK',\n",
       " 'LNX1',\n",
       " 'LONP1',\n",
       " 'LRIF1',\n",
       " 'LRIG1',\n",
       " 'LRP12',\n",
       " 'LRP5',\n",
       " 'LTB4R2',\n",
       " 'LZTS1',\n",
       " 'MAFIP',\n",
       " 'MAGI1',\n",
       " 'MAP4K2',\n",
       " 'MAP4K5',\n",
       " 'MAPRE2',\n",
       " 'MB',\n",
       " 'MBD1',\n",
       " 'MBD4',\n",
       " 'MBL2',\n",
       " 'MED12',\n",
       " 'MED22',\n",
       " 'MED9',\n",
       " 'MELTF',\n",
       " 'METAP1',\n",
       " 'METAP2',\n",
       " 'MGAM',\n",
       " 'MIPEP',\n",
       " 'MLF2',\n",
       " 'MMAB',\n",
       " 'MMACHC',\n",
       " 'MOCOS',\n",
       " 'MOGAT1',\n",
       " 'MOGAT2',\n",
       " 'MOGAT3',\n",
       " 'MRLN',\n",
       " 'MRPL13',\n",
       " 'MRPL41',\n",
       " 'MRPS12',\n",
       " 'MRPS28',\n",
       " 'MRPS35',\n",
       " 'MRTFA',\n",
       " 'MS4A12',\n",
       " 'MS4A4A',\n",
       " 'MS4A7',\n",
       " 'MTCH1',\n",
       " 'MYCBP2',\n",
       " 'MYDGF',\n",
       " 'MYH7B',\n",
       " 'MYL1',\n",
       " 'MYO1C',\n",
       " 'MYO1E',\n",
       " 'MYT1',\n",
       " 'MYZAP',\n",
       " 'NANOGP2',\n",
       " 'NANOGP4',\n",
       " 'NANOGP7',\n",
       " 'NANOS1',\n",
       " 'NANOS2',\n",
       " 'NANOS3',\n",
       " 'NAPSA',\n",
       " 'NAPSB',\n",
       " 'NAT8',\n",
       " 'NCOA2',\n",
       " 'NCOA3',\n",
       " 'NDRG1',\n",
       " 'NDUFAB1',\n",
       " 'NECTIN4',\n",
       " 'NEDD4',\n",
       " 'NELL2',\n",
       " 'NFE2L1',\n",
       " 'NKX2-1',\n",
       " 'NLRP2B',\n",
       " 'NLRP3',\n",
       " 'NLRP4',\n",
       " 'NLRP6',\n",
       " 'NOL3',\n",
       " 'NOL6',\n",
       " 'NPPC',\n",
       " 'NPRL2',\n",
       " 'NPRL3',\n",
       " 'NR5A1',\n",
       " 'NRG1',\n",
       " 'NT5C3A',\n",
       " 'NTN1',\n",
       " 'NUDT19',\n",
       " 'OSER1',\n",
       " 'PAEP',\n",
       " 'PAK6',\n",
       " 'PAPPA-AS1',\n",
       " 'PARD3',\n",
       " 'PCDHA4',\n",
       " 'PCDHA6',\n",
       " 'PCDHB11',\n",
       " 'PCDHB16',\n",
       " 'PCDHGA5',\n",
       " 'PCDHGB4',\n",
       " 'PDHX',\n",
       " 'PDPK1',\n",
       " 'PGAM4',\n",
       " 'PGAP1',\n",
       " 'PGF',\n",
       " 'PGPEP1',\n",
       " 'PIGU',\n",
       " 'PITRM1',\n",
       " 'PKMYT1',\n",
       " 'PKN1',\n",
       " 'PLPP6',\n",
       " 'PLPPR3',\n",
       " 'PM20D1',\n",
       " 'PMEL',\n",
       " 'PMEPA1',\n",
       " 'PMP22',\n",
       " 'PMPCB',\n",
       " 'PMS2CL',\n",
       " 'PNKD',\n",
       " 'PODXL',\n",
       " 'POLK',\n",
       " 'POLR1A',\n",
       " 'POLR1B',\n",
       " 'POLR1G',\n",
       " 'PPIAP10',\n",
       " 'PPP1R9B',\n",
       " 'PPY',\n",
       " 'PRDX5',\n",
       " 'PRDX6',\n",
       " 'PRG2',\n",
       " 'PRICKLE1',\n",
       " 'PRKD2',\n",
       " 'PRKRA',\n",
       " 'PRPF38B',\n",
       " 'PRSS8',\n",
       " 'PSG2',\n",
       " 'PSMA7',\n",
       " 'PSMD1',\n",
       " 'PSMD2',\n",
       " 'PTEN',\n",
       " 'PTENP1',\n",
       " 'PTPMT1',\n",
       " 'PTPN18',\n",
       " 'PTPRU',\n",
       " 'PTRH2',\n",
       " 'PXDN',\n",
       " 'PXMP2',\n",
       " 'PYDC1',\n",
       " 'RAB43',\n",
       " 'RAI1',\n",
       " 'RBL1',\n",
       " 'RDH14',\n",
       " 'REXO2',\n",
       " 'RGL4',\n",
       " 'RGPD1',\n",
       " 'RGS18',\n",
       " 'RHEBP1',\n",
       " 'RHOD',\n",
       " 'RIMBP2',\n",
       " 'RIN2',\n",
       " 'RMRP',\n",
       " 'RNASE1',\n",
       " 'RNASE10',\n",
       " 'RNASE12',\n",
       " 'RNASE3',\n",
       " 'RNASE6',\n",
       " 'RNASE7',\n",
       " 'RNASEH1P1',\n",
       " 'RNF138',\n",
       " 'RNF141',\n",
       " 'RNF2',\n",
       " 'RNF4',\n",
       " 'RNF7',\n",
       " 'RORA',\n",
       " 'RPL15',\n",
       " 'RPL17',\n",
       " 'RPL23',\n",
       " 'RRP1B',\n",
       " 'RSC1A1',\n",
       " 'RSL24D1',\n",
       " 'S100A9',\n",
       " 'S100G',\n",
       " 'SBDS',\n",
       " 'SCGB1D1',\n",
       " 'SEC14L2',\n",
       " 'SEC14L3',\n",
       " 'SELENOP',\n",
       " 'SEM1',\n",
       " 'SEMA3F-AS1',\n",
       " 'SEPSECS',\n",
       " 'SERPINA2',\n",
       " 'SERPINB8',\n",
       " 'SF3B2',\n",
       " 'SF3B3',\n",
       " 'SFI1',\n",
       " 'SGPP2',\n",
       " 'SH2D1A',\n",
       " 'SH3D19',\n",
       " 'SLC19A1',\n",
       " 'SLC25A1',\n",
       " 'SLC25A15',\n",
       " 'SLC25A16',\n",
       " 'SLC25A21',\n",
       " 'SLC25A23',\n",
       " 'SLC35A1',\n",
       " 'SLC38A1',\n",
       " 'SLC38A2',\n",
       " 'SLC6A20',\n",
       " 'SLC7A8',\n",
       " 'SLIT1',\n",
       " 'SLIT3',\n",
       " 'SLTM',\n",
       " 'SLURP1',\n",
       " 'SMIM29',\n",
       " 'SMOC2',\n",
       " 'SMOX',\n",
       " 'SND1',\n",
       " 'SNORA58',\n",
       " 'SNORD115-45',\n",
       " 'SNRPB2',\n",
       " 'SNX33',\n",
       " 'SOAT1',\n",
       " 'SOAT2',\n",
       " 'SOS1',\n",
       " 'SOX9',\n",
       " 'SP6',\n",
       " 'SPATA17',\n",
       " 'SPATS1',\n",
       " 'SPNS1',\n",
       " 'SPPL2A',\n",
       " 'SPPL2B',\n",
       " 'SRFBP1',\n",
       " 'SRMS',\n",
       " 'STK24',\n",
       " 'STK4',\n",
       " 'SYCP2',\n",
       " 'TACR1',\n",
       " 'TADA2A',\n",
       " 'TDO2',\n",
       " 'TENM1',\n",
       " 'TENT4A',\n",
       " 'TGFBRAP1',\n",
       " 'THPO',\n",
       " 'TIMP1',\n",
       " 'TINAGL1',\n",
       " 'TIRAP',\n",
       " 'TMEM184B-AS1',\n",
       " 'TMPRSS4',\n",
       " 'TNIP1',\n",
       " 'TNNI3',\n",
       " 'TNPO1',\n",
       " 'TOX4',\n",
       " 'TP53RK',\n",
       " 'TPGS1',\n",
       " 'TRAPPC8',\n",
       " 'TRAV1-1',\n",
       " 'TSPO',\n",
       " 'TTC5',\n",
       " 'UBASH3A',\n",
       " 'UBE2Z',\n",
       " 'UCMA',\n",
       " 'UGT2B4',\n",
       " 'USP2-AS1',\n",
       " 'USP21',\n",
       " 'USP25',\n",
       " 'USP32',\n",
       " 'UTS2',\n",
       " 'UXT',\n",
       " 'VSIG2',\n",
       " 'XPO1',\n",
       " 'ZBTB32',\n",
       " 'ZFP1',\n",
       " 'ZNF106',\n",
       " 'ZNF121',\n",
       " 'ZNF160',\n",
       " 'ZNF83'}"
      ]
     },
     "execution_count": 2779,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_ap_collisions = (\n",
    "    ncbi_alias_primary_collision_primary_symbol_set\n",
    "    & hgnc_alias_primary_collision_primary_symbol_set\n",
    "    & ensg_alias_primary_collision_primary_symbol_set\n",
    ")\n",
    "common_ap_collisions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2780,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "486"
      ]
     },
     "execution_count": 2780,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_ap_collisions)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
