{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_ap_collision_df(mini_xxxx_df: pd.DataFrame, source: str) -> pd.DataFrame:\n",
    "    \"\"\"Create a df of alias-primary collision symbols \n",
    "\n",
    "    :param mini_xxxx_df: Processed df of gene records\n",
    "    :param source: Representation of the source of the gene records\n",
    "    :param split_on_character: Character that is used to seperate alias symbols in the mini_xxxx_df\n",
    "    :return: A df of genes that share an alias with another gene\n",
    "    \"\"\"\n",
    "\n",
    "    #Strip \"HGNC\" from HGNC ID\n",
    "    mini_xxxx_df[\"HGNC_ID\"] = mini_xxxx_df[\"HGNC_ID\"].str.lstrip(\"HGNC:\")\n",
    "\n",
    "    #Remove placeholder LOC genes\n",
    "    mini_xxxx_df = mini_xxxx_df[~mini_xxxx_df['gene_symbol'].str.startswith('LOC', na=False)]\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    mini_xxxx_df.to_csv(f'../output/mini_{source.lower()}_df.csv', index=True) \n",
    "\n",
    "    #Make a new df where the alias symbols are merged together for each record\n",
    "    merged_alias_xxxx_df = mini_xxxx_df.copy()\n",
    "    merged_alias_xxxx_df[\"alias_symbol\"] = merged_alias_xxxx_df[\"alias_symbol\"].fillna(\"\").astype(str)\n",
    "    merged_alias_xxxx_df = (\n",
    "    merged_alias_xxxx_df.groupby(\n",
    "        [\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], dropna=False\n",
    "    )[\"alias_symbol\"]\n",
    "    .apply(lambda x: \",\".join(x.dropna())).reset_index())\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    merged_alias_xxxx_df.to_csv(f'../output/merged_alias_{source}_df.csv', index=True) \n",
    "\n",
    "    #Make a set of the primary gene symbols\n",
    "    xxxx_gene_symbol_set = set(mini_xxxx_df[\"gene_symbol\"])\n",
    "\n",
    "    #Drop genes without any aliases\n",
    "    subset_genes_xxxx_df = mini_xxxx_df.copy()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.replace(\" \", np.nan)\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.replace(\"\", np.nan)\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.replace(\"-\", np.nan)\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.dropna(subset=[\"alias_symbol\"])\n",
    "\n",
    "    #Remove duplicate instances of primary gene symbol- alias pairs\n",
    "    #(occur because the same primary gene symbol may have multiple different ENSG IDs, \n",
    "    #see gene RFLNB for example)\n",
    "    subset_genes_xxxx_df[\"gene_symbol\"] = subset_genes_xxxx_df[\"gene_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df[\"alias_symbol\"] = subset_genes_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.drop_duplicates(subset=['gene_symbol', 'alias_symbol'], keep='first')\n",
    "\n",
    "    #Remove instances where the primary gene symbol is an alias for that gene record\n",
    "    #the pair would be the same symbol twice\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df[subset_genes_xxxx_df['gene_symbol'] != subset_genes_xxxx_df['alias_symbol']]\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    subset_genes_xxxx_df.to_csv(f'../output/subset_genes_{source}_df.csv', index=True) \n",
    "\n",
    "    #Create df with genes that have an alias that can be found as another gene's primary gene symbol\n",
    "    ap_collision_xxxx_df = subset_genes_xxxx_df.copy()\n",
    "    ap_collision_xxxx_df.loc[:, \"alias_symbol\"] = ap_collision_xxxx_df[\"alias_symbol\"].apply(lambda x: {x})\n",
    "    ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol\"].apply(\n",
    "        lambda x: x & xxxx_gene_symbol_set\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df[ap_collision_xxxx_df[\"collision\"].apply(lambda x: len(x) > 0)]\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n",
    "    ap_collision_xxxx_df['NCBI_ID'] = ap_collision_xxxx_df['NCBI_ID'].fillna(0).astype(int)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.sort_values(\"collision\")\n",
    "\n",
    "    #Add a source tag for future merging efforts\n",
    "    ap_collision_xxxx_df[\"source\"] = str(source.upper())\n",
    "\n",
    "    #Convert the df into a csv\n",
    "    ap_collision_xxxx_df.to_csv(f'../output/single_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "\n",
    "    #Create a secondary collision df that merges the alias symbols for each record\n",
    "    merged_alias_ap_collision_xxxx_df = ap_collision_xxxx_df.drop(columns=['alias_symbol'])\n",
    "    merged_alias_ap_collision_xxxx_df = pd.merge(merged_alias_ap_collision_xxxx_df, merged_alias_xxxx_df, on=[\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], how=\"left\")\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df[[\"gene_symbol\",\"alias_symbol\",\"ENSG_ID\",\"HGNC_ID\",\"NCBI_ID\",\"collision\",\"source\"]]\n",
    "\n",
    "    #Test to make sure all collisions are in the alias list for the record\n",
    "    test_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.copy()\n",
    "    test_ap_collision_xxxx_df['alias_symbol'] = test_ap_collision_xxxx_df['alias_symbol'].apply(lambda x: x if isinstance(x, list) else [x])\n",
    "    test_ap_collision_xxxx_df['collision_in_alias'] = test_ap_collision_xxxx_df.apply(lambda row: row['collision'] in test_ap_collision_xxxx_df['alias_symbol'].values, axis=1)\n",
    "    true_count = test_ap_collision_xxxx_df['collision_in_alias'].sum()\n",
    "\n",
    "    if true_count == 0:\n",
    "        #Convert the df into a csv and save\n",
    "        merged_alias_ap_collision_xxxx_df.to_csv(f'../output/merged_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "        print(\"All collisions are present in gene alias lists.\")\n",
    "    else:\n",
    "        print(\"Some collisions are not present in gene alias lists.\")\n",
    "\n",
    "    return mini_xxxx_df.head(), subset_genes_xxxx_df.head(), ap_collision_xxxx_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>12S</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117135</th>\n",
       "      <td>ENSG00000200033</td>\n",
       "      <td>RNU6-403P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:47366</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117136</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>LNCSLCC1</td>\n",
       "      <td>HGNC:53417</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117137</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>RP11-400N13.2</td>\n",
       "      <td>HGNC:53417</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117138</th>\n",
       "      <td>ENSG00000229463</td>\n",
       "      <td>LYST-AS1</td>\n",
       "      <td>LYST-IT2</td>\n",
       "      <td>HGNC:41320</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117139</th>\n",
       "      <td>ENSG00000229291</td>\n",
       "      <td>LINC02768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HGNC:54288</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117140 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID gene_symbol   alias_symbol     HGNC_ID  NCBI_ID\n",
       "0       ENSG00000210049       MT-TF           MTTF   HGNC:7481     <NA>\n",
       "1       ENSG00000210049       MT-TF           TRNF   HGNC:7481     <NA>\n",
       "2       ENSG00000211459     MT-RNR1            12S   HGNC:7470     <NA>\n",
       "3       ENSG00000211459     MT-RNR1         MOTS-C   HGNC:7470     <NA>\n",
       "4       ENSG00000211459     MT-RNR1         MTRNR1   HGNC:7470     <NA>\n",
       "...                 ...         ...            ...         ...      ...\n",
       "117135  ENSG00000200033   RNU6-403P            NaN  HGNC:47366     <NA>\n",
       "117136  ENSG00000228437   LINC02474       LNCSLCC1  HGNC:53417     <NA>\n",
       "117137  ENSG00000228437   LINC02474  RP11-400N13.2  HGNC:53417     <NA>\n",
       "117138  ENSG00000229463    LYST-AS1       LYST-IT2  HGNC:41320     <NA>\n",
       "117139  ENSG00000229291   LINC02768            NaN  HGNC:54288     <NA>\n",
       "\n",
       "[117140 rows x 5 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../input/ensg_biomart_gene20240626.txt\", sep=\"\\t\",dtype={\"NCBI gene (formerly Entrezgene) ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_ensg_df = mini_ensg_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Gene Synonym\": \"alias_symbol\",\n",
    "        \"Gene name\": \"gene_symbol\",\n",
    "        \"Gene stable ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene (formerly Entrezgene) ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/932853458.py:62: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(           ENSG_ID gene_symbol alias_symbol HGNC_ID  NCBI_ID\n",
       " 0  ENSG00000210049       MT-TF         MTTF    7481     <NA>\n",
       " 1  ENSG00000210049       MT-TF         TRNF    7481     <NA>\n",
       " 2  ENSG00000211459     MT-RNR1          12S    7470     <NA>\n",
       " 3  ENSG00000211459     MT-RNR1       MOTS-C    7470     <NA>\n",
       " 4  ENSG00000211459     MT-RNR1       MTRNR1    7470     <NA>,\n",
       "            ENSG_ID gene_symbol alias_symbol HGNC_ID  NCBI_ID\n",
       " 0  ENSG00000210049       MT-TF         MTTF    7481     <NA>\n",
       " 1  ENSG00000210049       MT-TF         TRNF    7481     <NA>\n",
       " 2  ENSG00000211459     MT-RNR1          12S    7470     <NA>\n",
       " 3  ENSG00000211459     MT-RNR1       MOTS-C    7470     <NA>\n",
       " 4  ENSG00000211459     MT-RNR1       MTRNR1    7470     <NA>,\n",
       "                ENSG_ID gene_symbol alias_symbol HGNC_ID  NCBI_ID collision  \\\n",
       " 33522  ENSG00000283293       RN7SK          7SK   10037   125050       7SK   \n",
       " 51305  ENSG00000057252       SOAT1        ACAT1   11177     6646     ACAT1   \n",
       " 29843  ENSG00000167780       SOAT2        ACAT2   11178     8435     ACAT2   \n",
       " 48092  ENSG00000004779     NDUFAB1         ACP1    7694     4706      ACP1   \n",
       " 56636  ENSG00000220267      ACTBP8       ACTBP2     141        0    ACTBP2   \n",
       " \n",
       "       source  \n",
       " 33522   ENSG  \n",
       " 51305   ENSG  \n",
       " 29843   ENSG  \n",
       " 48092   ENSG  \n",
       " 56636   ENSG  )"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ensg_df, \"ENSG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>12S</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117135</th>\n",
       "      <td>ENSG00000200033</td>\n",
       "      <td>RNU6-403P</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47366.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117136</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>LNCSLCC1</td>\n",
       "      <td>53417.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117137</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>RP11-400N13.2</td>\n",
       "      <td>53417.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117138</th>\n",
       "      <td>ENSG00000229463</td>\n",
       "      <td>LYST-AS1</td>\n",
       "      <td>LYST-IT2</td>\n",
       "      <td>41320.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117139</th>\n",
       "      <td>ENSG00000229291</td>\n",
       "      <td>LINC02768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>54288.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117140 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID gene_symbol   alias_symbol  HGNC_ID  NCBI_ID\n",
       "0       ENSG00000210049       MT-TF           MTTF   7481.0      NaN\n",
       "1       ENSG00000210049       MT-TF           TRNF   7481.0      NaN\n",
       "2       ENSG00000211459     MT-RNR1            12S   7470.0      NaN\n",
       "3       ENSG00000211459     MT-RNR1         MOTS-C   7470.0      NaN\n",
       "4       ENSG00000211459     MT-RNR1         MTRNR1   7470.0      NaN\n",
       "...                 ...         ...            ...      ...      ...\n",
       "117135  ENSG00000200033   RNU6-403P            NaN  47366.0      NaN\n",
       "117136  ENSG00000228437   LINC02474       LNCSLCC1  53417.0      NaN\n",
       "117137  ENSG00000228437   LINC02474  RP11-400N13.2  53417.0      NaN\n",
       "117138  ENSG00000229463    LYST-AS1       LYST-IT2  41320.0      NaN\n",
       "117139  ENSG00000229291   LINC02768            NaN  54288.0      NaN\n",
       "\n",
       "[117140 rows x 5 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df= pd.read_csv(\n",
    "    \"../output/mini_ensg_df.csv\", index_col=[0])\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>12S</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117133</th>\n",
       "      <td>ENSG00000232679</td>\n",
       "      <td>LINC01705</td>\n",
       "      <td>ERLR</td>\n",
       "      <td>52493.0</td>\n",
       "      <td>105372950.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117134</th>\n",
       "      <td>ENSG00000232679</td>\n",
       "      <td>LINC01705</td>\n",
       "      <td>RP11-400N13.3</td>\n",
       "      <td>52493.0</td>\n",
       "      <td>105372950.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117136</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>LNCSLCC1</td>\n",
       "      <td>53417.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117137</th>\n",
       "      <td>ENSG00000228437</td>\n",
       "      <td>LINC02474</td>\n",
       "      <td>RP11-400N13.2</td>\n",
       "      <td>53417.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117138</th>\n",
       "      <td>ENSG00000229463</td>\n",
       "      <td>LYST-AS1</td>\n",
       "      <td>LYST-IT2</td>\n",
       "      <td>41320.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57275 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID gene_symbol   alias_symbol  HGNC_ID      NCBI_ID\n",
       "0       ENSG00000210049       MT-TF           MTTF   7481.0          NaN\n",
       "1       ENSG00000210049       MT-TF           TRNF   7481.0          NaN\n",
       "2       ENSG00000211459     MT-RNR1            12S   7470.0          NaN\n",
       "3       ENSG00000211459     MT-RNR1         MOTS-C   7470.0          NaN\n",
       "4       ENSG00000211459     MT-RNR1         MTRNR1   7470.0          NaN\n",
       "...                 ...         ...            ...      ...          ...\n",
       "117133  ENSG00000232679   LINC01705           ERLR  52493.0  105372950.0\n",
       "117134  ENSG00000232679   LINC01705  RP11-400N13.3  52493.0  105372950.0\n",
       "117136  ENSG00000228437   LINC02474       LNCSLCC1  53417.0          NaN\n",
       "117137  ENSG00000228437   LINC02474  RP11-400N13.2  53417.0          NaN\n",
       "117138  ENSG00000229463    LYST-AS1       LYST-IT2  41320.0          NaN\n",
       "\n",
       "[57275 rows x 5 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ensg_df.csv\", index_col=[0])\n",
    "subset_genes_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4932</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>494470.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4933</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "      <td>LNCAMPC</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>494470.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4934</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>494470.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4935</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "      <td>RNF165</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>494470.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ENSG_ID gene_symbol alias_symbol  HGNC_ID   NCBI_ID\n",
       "4932  ENSG00000141622       ARK2C        ARKL2  31696.0  494470.0\n",
       "4933  ENSG00000141622       ARK2C      LNCAMPC  31696.0  494470.0\n",
       "4934  ENSG00000141622       ARK2C     RNF111L2  31696.0  494470.0\n",
       "4935  ENSG00000141622       ARK2C       RNF165  31696.0  494470.0"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df.loc[\n",
    "    subset_genes_ensg_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>10037.0</td>\n",
       "      <td>125050</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177.0</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178.0</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694.0</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>141.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904.0</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159.0</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886.0</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328.0</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>694 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol          alias_symbol          ENSG_ID  HGNC_ID  NCBI_ID  \\\n",
       "0         RN7SK                   7SK  ENSG00000283293  10037.0   125050   \n",
       "1         SOAT1  ACAT,ACAT1,SOAT,STAT  ENSG00000057252  11177.0     6646   \n",
       "2         SOAT2                 ACAT2  ENSG00000167780  11178.0     8435   \n",
       "3       NDUFAB1  ACP,ACP1,FASN2A,SDAP  ENSG00000004779   7694.0     4706   \n",
       "4        ACTBP8                ACTBP2  ENSG00000220267    141.0        0   \n",
       "..          ...                   ...              ...      ...      ...   \n",
       "689      ZNF121   D19S204,ZHC32,ZNF20  ENSG00000197961  12904.0     7675   \n",
       "690      RNF141          ZFP26,ZNF230  ENSG00000110315  21159.0    50862   \n",
       "691    ZNF322P1        ZNF322,ZNF322B  ENSG00000188801  14003.0        0   \n",
       "692      ZNF106  SH3BP3,ZFP106,ZNF474  ENSG00000103994  12886.0    64397   \n",
       "693        ZFP1       FLJ34243,ZNF475  ENSG00000184517  23328.0   162239   \n",
       "\n",
       "    collision source  \n",
       "0         7SK   ENSG  \n",
       "1       ACAT1   ENSG  \n",
       "2       ACAT2   ENSG  \n",
       "3        ACP1   ENSG  \n",
       "4      ACTBP2   ENSG  \n",
       "..        ...    ...  \n",
       "689     ZNF20   ENSG  \n",
       "690    ZNF230   ENSG  \n",
       "691    ZNF322   ENSG  \n",
       "692    ZNF474   ENSG  \n",
       "693    ZNF475   ENSG  \n",
       "\n",
       "[694 rows x 7 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_gene_symbol_set = set(mini_ensg_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41068"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ensembl_gene_symbols = len(ensg_gene_symbol_set)\n",
    "total_number_ensembl_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "621"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_set = set(merged_alias_ap_collision_ensg_df[\"collision\"])\n",
    "len(ensg_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "678"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_ensg_df[\"gene_symbol\"])\n",
    "len(ensg_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Why is the alias-gene collision set not the same length as the set of primary symbols with collisions ?\n",
    "2. Why is the length of the alias-gene collison set shorter?\n",
    " - A priamry gene symbol with an alias-gene collision has an alias that matches a different gene's primary gene symbol.\n",
    " - Multiple genes can share a single alias (alias-alias collision)\n",
    " - If that shared alias is an alias-gene collision, then there will be more unique gene symbols in the set of primary symbols with collisions than the set of alias-gene collisions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGNC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HGNC_ID  alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       "0          HGNC:5           NaN        1  ENSG00000121410        A1BG\n",
       "1      HGNC:37133      FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       "2      HGNC:24086           ACF    29974  ENSG00000148584        A1CF\n",
       "3      HGNC:24086           ASP    29974  ENSG00000148584        A1CF\n",
       "4      HGNC:24086         ACF64    29974  ENSG00000148584        A1CF\n",
       "...           ...           ...      ...              ...         ...\n",
       "67578  HGNC:29027      KIAA0399    23140  ENSG00000074755       ZZEF1\n",
       "67579  HGNC:29027          ZZZ4    23140  ENSG00000074755       ZZEF1\n",
       "67580  HGNC:29027      FLJ10821    23140  ENSG00000074755       ZZEF1\n",
       "67581  HGNC:24523  DKFZP564I052    26009  ENSG00000036549        ZZZ3\n",
       "67582  HGNC:24523         ATAC1    26009  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = pd.read_csv(\n",
    "    \"../input/hgnc_biomart_gene20240626.txt\", sep=\"\\t\",dtype={\"NCBI gene ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_hgnc_df = mini_hgnc_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Approved symbol\": \"gene_symbol\",\n",
    "        \"Alias symbol\": \"alias_symbol\",\n",
    "        \"Ensembl gene ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/932853458.py:62: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(  HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 0       5          NaN        1  ENSG00000121410        A1BG\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF,\n",
       "   HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF\n",
       " 5   24086        ACF65    29974  ENSG00000148584        A1CF,\n",
       "       HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol collision  \\\n",
       " 42104   14947        AAVS1    54776  ENSG00000125503    PPP1R12C     AAVS1   \n",
       " 56627   11177        ACAT1     6646  ENSG00000057252       SOAT1     ACAT1   \n",
       " 56628   11178        ACAT2     8435  ENSG00000167780       SOAT2     ACAT2   \n",
       " 17689    4319         ACLS     2737  ENSG00000106571        GLI3      ACLS   \n",
       " 35323    7694         ACP1     4706  ENSG00000004779     NDUFAB1      ACP1   \n",
       " \n",
       "       source  \n",
       " 42104   HGNC  \n",
       " 56627   HGNC  \n",
       " 56628   HGNC  \n",
       " 17689   HGNC  \n",
       " 35323   HGNC  )"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_hgnc_df, \"HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "0            5           NaN       1.0  ENSG00000121410        A1BG\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df= pd.read_csv(\n",
    "    \"../output/mini_hgnc_df.csv\", index_col=[0])\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF65</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44542 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "5        24086         ACF65   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[44542 rows x 5 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df = pd.read_csv(\n",
    "    \"../output/subset_genes_hgnc_df.csv\", index_col=[0])\n",
    "subset_genes_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>31696</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709</th>\n",
       "      <td>31696</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711</th>\n",
       "      <td>31696</td>\n",
       "      <td>LNCAMPC</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC_ID alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "2708    31696        ARKL2  494470.0  ENSG00000141622       ARK2C\n",
       "2709    31696     RNF111L2  494470.0  ENSG00000141622       ARK2C\n",
       "2711    31696      LNCAMPC  494470.0  ENSG00000141622       ARK2C"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df.loc[\n",
    "    subset_genes_hgnc_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ENSG00000106571</td>\n",
       "      <td>4319</td>\n",
       "      <td>2737</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474,SH3BP3</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>ZNF208</td>\n",
       "      <td>PMIDP,ZNF95</td>\n",
       "      <td>ENSG00000160321</td>\n",
       "      <td>12999</td>\n",
       "      <td>7757</td>\n",
       "      <td>ZNF95</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>653 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol          ENSG_ID  HGNC_ID  \\\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1  ENSG00000125503    14947   \n",
       "1         SOAT1                         ACAT,ACAT1  ENSG00000057252    11177   \n",
       "2         SOAT2                              ACAT2  ENSG00000167780    11178   \n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV  ENSG00000106571     4319   \n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1  ENSG00000004779     7694   \n",
       "..          ...                                ...              ...      ...   \n",
       "648      ZNF121                        ZHC32,ZNF20  ENSG00000197961    12904   \n",
       "649      RNF141                       ZFP26,ZNF230  ENSG00000110315    21159   \n",
       "650      ZNF106                      ZNF474,SH3BP3  ENSG00000103994    12886   \n",
       "651        ZFP1                    FLJ34243,ZNF475  ENSG00000184517    23328   \n",
       "652      ZNF208                        PMIDP,ZNF95  ENSG00000160321    12999   \n",
       "\n",
       "     NCBI_ID collision source  \n",
       "0      54776     AAVS1   HGNC  \n",
       "1       6646     ACAT1   HGNC  \n",
       "2       8435     ACAT2   HGNC  \n",
       "3       2737      ACLS   HGNC  \n",
       "4       4706      ACP1   HGNC  \n",
       "..       ...       ...    ...  \n",
       "648     7675     ZNF20   HGNC  \n",
       "649    50862    ZNF230   HGNC  \n",
       "650    64397    ZNF474   HGNC  \n",
       "651   162239    ZNF475   HGNC  \n",
       "652     7757     ZNF95   HGNC  \n",
       "\n",
       "[653 rows x 7 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_hgnc_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_hgnc_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_gene_symbol_set = set(mini_hgnc_df[\"gene_symbol\"])\n",
    "# all_gene_symbols_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_hgnc_gene_symbols = len(hgnc_gene_symbol_set)\n",
    "total_number_hgnc_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "576"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_set = set(merged_alias_ap_collision_hgnc_df[\"collision\"])\n",
    "len(hgnc_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "637"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_hgnc_df[\"gene_symbol\"])\n",
    "len(hgnc_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NCBI Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = pd.read_csv(\"../input/Homo_sapiens.gene_info20240627\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>dbXrefs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>MIM:138670|HGNC:HGNC:5|Ensembl:ENSG00000121410...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>MIM:103950|HGNC:HGNC:7|Ensembl:ENSG00000175899...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>HGNC:HGNC:8|Ensembl:ENSG00000291190|AllianceGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>MIM:108345|HGNC:HGNC:7645|Ensembl:ENSG00000171...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>MIM:612182|HGNC:HGNC:7646|Ensembl:ENSG00000156...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193451</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193452</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193453</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193454</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193455</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193456 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol               alias_symbol  \\\n",
       "0             1        A1BG       A1B|ABG|GAB|HYST2477   \n",
       "1             2         A2M  A2MD|CPAMD5|FWP007|S863-7   \n",
       "2             3       A2MP1                       A2MP   \n",
       "3             9        NAT1       AAC1|MNAT|NAT-1|NATI   \n",
       "4            10        NAT2            AAC2|NAT-2|PNAT   \n",
       "...         ...         ...                        ...   \n",
       "193451  8923215        trnD                          -   \n",
       "193452  8923216        trnP                          -   \n",
       "193453  8923217        trnA                          -   \n",
       "193454  8923218        COX1                          -   \n",
       "193455  8923219    16S rRNA                          -   \n",
       "\n",
       "                                                  dbXrefs  \n",
       "0       MIM:138670|HGNC:HGNC:5|Ensembl:ENSG00000121410...  \n",
       "1       MIM:103950|HGNC:HGNC:7|Ensembl:ENSG00000175899...  \n",
       "2       HGNC:HGNC:8|Ensembl:ENSG00000291190|AllianceGe...  \n",
       "3       MIM:108345|HGNC:HGNC:7645|Ensembl:ENSG00000171...  \n",
       "4       MIM:612182|HGNC:HGNC:7646|Ensembl:ENSG00000156...  \n",
       "...                                                   ...  \n",
       "193451                                                  -  \n",
       "193452                                                  -  \n",
       "193453                                                  -  \n",
       "193454                                                  -  \n",
       "193455                                                  -  \n",
       "\n",
       "[193456 rows x 4 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[\n",
    "[\"GeneID\", \"Symbol\", \"Synonyms\", \"dbXrefs\"]\n",
    "]\n",
    "mini_ncbi_df = mini_ncbi_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Synonyms\": \"alias_symbol\"}\n",
    ")\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split dbXrefs into individual columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.assign(\n",
    "    MIM=np.nan,\n",
    "    HGNC_ID=np.nan,\n",
    "    ENSG_ID=np.nan,\n",
    "    AllianceGenome=np.nan,\n",
    "    MIRbase=np.nan,\n",
    "    IMGTgene_db=np.nan,\n",
    "    dash=np.nan,\n",
    "    unknown=np.nan,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193456\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '138670' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ensg00000121410' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'hgnc:5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'cd1d' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '-' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'mi0000060' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193456\n"
     ]
    }
   ],
   "source": [
    "index_pos = 0\n",
    "\n",
    "print(len(mini_ncbi_df))\n",
    "while index_pos < len(mini_ncbi_df):\n",
    "    xrefs = mini_ncbi_df[\"dbXrefs\"][index_pos].split(\"|\")\n",
    "\n",
    "    for xref in xrefs:\n",
    "        xref = xref.lower()\n",
    "        if xref.startswith(\"mim:\"):\n",
    "            xref = xref.replace(\"mim:\", \"\")\n",
    "            mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
    "        elif xref.startswith(\"hgnc:hgnc:\"):\n",
    "            xref = xref.replace(\"hgnc:hgnc:\", \"\")\n",
    "            mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"ensembl:\"):\n",
    "            xref = xref.replace(\"ensembl:\", \"\")\n",
    "            mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"alliancegenome:\"):\n",
    "            xref = xref.replace(\"alliancegenome:\", \"\")\n",
    "            mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
    "        elif xref.startswith(\"mirbase\"):\n",
    "            xref = xref.replace(\"mirbase:\", \"\")\n",
    "            mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
    "        elif xref.startswith(\"imgt/gene-db:\"):\n",
    "            xref = xref.replace(\"imgt/gene-db:\", \"\")\n",
    "            mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
    "        elif xref.startswith(\"-\"):\n",
    "            mini_ncbi_df[\"dash\"][index_pos] = xref\n",
    "        else:\n",
    "            mini_ncbi_df[\"unknown\"][index_pos] = xref\n",
    "\n",
    "    index_pos += 1\n",
    "    pass\n",
    "\n",
    "print(index_pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df[\"ENSG_ID\"] = mini_ncbi_df[\"ENSG_ID\"].str.replace(\"ensg\", \"ENSG\", 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>A2MP1</td>\n",
       "      <td>A2MP</td>\n",
       "      <td>8</td>\n",
       "      <td>ENSG00000291190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>7645</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>7646</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193451</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193452</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193453</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193454</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193455</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193456 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol               alias_symbol HGNC_ID  \\\n",
       "0             1        A1BG       A1B|ABG|GAB|HYST2477       5   \n",
       "1             2         A2M  A2MD|CPAMD5|FWP007|S863-7       7   \n",
       "2             3       A2MP1                       A2MP       8   \n",
       "3             9        NAT1       AAC1|MNAT|NAT-1|NATI    7645   \n",
       "4            10        NAT2            AAC2|NAT-2|PNAT    7646   \n",
       "...         ...         ...                        ...     ...   \n",
       "193451  8923215        trnD                          -     NaN   \n",
       "193452  8923216        trnP                          -     NaN   \n",
       "193453  8923217        trnA                          -     NaN   \n",
       "193454  8923218        COX1                          -     NaN   \n",
       "193455  8923219    16S rRNA                          -     NaN   \n",
       "\n",
       "                ENSG_ID  \n",
       "0       ENSG00000121410  \n",
       "1       ENSG00000175899  \n",
       "2       ENSG00000291190  \n",
       "3       ENSG00000171428  \n",
       "4       ENSG00000156006  \n",
       "...                 ...  \n",
       "193451              NaN  \n",
       "193452              NaN  \n",
       "193453              NaN  \n",
       "193454              NaN  \n",
       "193455              NaN  \n",
       "\n",
       "[193456 rows x 5 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.drop(\n",
    "    [\n",
    "        \"AllianceGenome\",\n",
    "        \"MIRbase\",\n",
    "        \"IMGTgene_db\",\n",
    "        \"dash\",\n",
    "        \"unknown\",\n",
    "        \"dbXrefs\",\n",
    "        \"MIM\",\n",
    "    ],\n",
    "    axis=1,\n",
    ")\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193451</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193452</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193453</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193454</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193455</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239929 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B       5  ENSG00000121410\n",
       "0             1        A1BG          ABG       5  ENSG00000121410\n",
       "0             1        A1BG          GAB       5  ENSG00000121410\n",
       "0             1        A1BG     HYST2477       5  ENSG00000121410\n",
       "1             2         A2M         A2MD       7  ENSG00000175899\n",
       "...         ...         ...          ...     ...              ...\n",
       "193451  8923215        trnD            -     NaN              NaN\n",
       "193452  8923216        trnP            -     NaN              NaN\n",
       "193453  8923217        trnA            -     NaN              NaN\n",
       "193454  8923218        COX1            -     NaN              NaN\n",
       "193455  8923219    16S rRNA            -     NaN              NaN\n",
       "\n",
       "[239929 rows x 5 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['alias_symbol'] = mini_ncbi_df['alias_symbol'].str.split('|')\n",
    "mini_ncbi_df = mini_ncbi_df.explode('alias_symbol')\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/932853458.py:62: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(   NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "    NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID collision  \\\n",
       " 2826      3494       IGHA2          A2M    5479  ENSG00000211890       A2M   \n",
       " 24278   404744   NPSR1-AS1         AAA1   22128  ENSG00000197085      AAA1   \n",
       " 16430    89876      CFAP91         AAT1   24010  ENSG00000183833      AAT1   \n",
       " 2323      2875         GPT         AAT1    4552  ENSG00000167701      AAT1   \n",
       " 12272    54776    PPP1R12C        AAVS1   14947  ENSG00000125503     AAVS1   \n",
       " \n",
       "       source  \n",
       " 2826    NCBI  \n",
       " 24278   NCBI  \n",
       " 16430   NCBI  \n",
       " 2323    NCBI  \n",
       " 12272   NCBI  )"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ncbi_df, \"NCBI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193451</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193452</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193453</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193454</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193455</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91925 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0             1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0             1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0             1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1             2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...         ...         ...          ...      ...              ...\n",
       "193451  8923215        trnD            -      NaN              NaN\n",
       "193452  8923216        trnP            -      NaN              NaN\n",
       "193453  8923217        trnA            -      NaN              NaN\n",
       "193454  8923218        COX1            -      NaN              NaN\n",
       "193455  8923219    16S rRNA            -      NaN              NaN\n",
       "\n",
       "[91925 rows x 5 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df= pd.read_csv(\n",
    "    \"../output/mini_ncbi_df.csv\", index_col=[0])\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190961</th>\n",
       "      <td>131840634</td>\n",
       "      <td>GLTC1</td>\n",
       "      <td>GLTC</td>\n",
       "      <td>56861.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193342</th>\n",
       "      <td>132532400</td>\n",
       "      <td>GABRA6-AS1</td>\n",
       "      <td>ARBAG</td>\n",
       "      <td>40248.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193377</th>\n",
       "      <td>133395150</td>\n",
       "      <td>LNCARGI</td>\n",
       "      <td>ARGI</td>\n",
       "      <td>56890.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193378</th>\n",
       "      <td>133834869</td>\n",
       "      <td>MLDHR</td>\n",
       "      <td>MP31</td>\n",
       "      <td>55481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193378</th>\n",
       "      <td>133834869</td>\n",
       "      <td>MLDHR</td>\n",
       "      <td>PTENURF</td>\n",
       "      <td>55481.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>73829 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0               1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0               1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0               1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0               1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1               2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...           ...         ...          ...      ...              ...\n",
       "190961  131840634       GLTC1         GLTC  56861.0              NaN\n",
       "193342  132532400  GABRA6-AS1        ARBAG  40248.0              NaN\n",
       "193377  133395150     LNCARGI         ARGI  56890.0              NaN\n",
       "193378  133834869       MLDHR         MP31  55481.0              NaN\n",
       "193378  133834869       MLDHR      PTENURF  55481.0              NaN\n",
       "\n",
       "[73829 rows x 5 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ncbi_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ncbi_df.csv\", index_col=[0])\n",
    "subset_genes_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IGHA2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000211890</td>\n",
       "      <td>5479.0</td>\n",
       "      <td>3494</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPSR1-AS1</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>ENSG00000197085</td>\n",
       "      <td>22128.0</td>\n",
       "      <td>404744</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFAP91</td>\n",
       "      <td>AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...</td>\n",
       "      <td>ENSG00000183833</td>\n",
       "      <td>24010.0</td>\n",
       "      <td>89876</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPT</td>\n",
       "      <td>AAT1,ALT,ALT1,GPT1,SGPT</td>\n",
       "      <td>ENSG00000167701</td>\n",
       "      <td>4552.0</td>\n",
       "      <td>2875</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1,LENG3,MBS85,p84,p85</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947.0</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2119</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>ENSG00000197162</td>\n",
       "      <td>26496.0</td>\n",
       "      <td>146540</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2120</th>\n",
       "      <td>ZP4</td>\n",
       "      <td>ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4</td>\n",
       "      <td>ENSG00000116996</td>\n",
       "      <td>15770.0</td>\n",
       "      <td>57829</td>\n",
       "      <td>ZP1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2121</th>\n",
       "      <td>LMBR1</td>\n",
       "      <td>ACHP,C7orf2,DIF14,LSS,PPD2,THYP,TPT,TPTPS,ZRS</td>\n",
       "      <td>ENSG00000105983</td>\n",
       "      <td>13243.0</td>\n",
       "      <td>64327</td>\n",
       "      <td>ZRS</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2122</th>\n",
       "      <td>ZNF446</td>\n",
       "      <td>ZKSCAN20,ZSCAN30,ZSCAN52</td>\n",
       "      <td>ENSG00000083838</td>\n",
       "      <td>21036.0</td>\n",
       "      <td>55663</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2123</th>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>ZNF694,ZSCAN31,ZSCAN34</td>\n",
       "      <td>ENSG00000155592</td>\n",
       "      <td>25677.0</td>\n",
       "      <td>342357</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2124 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol                                       alias_symbol  \\\n",
       "0          IGHA2                                                A2M   \n",
       "1      NPSR1-AS1                                               AAA1   \n",
       "2         CFAP91  AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...   \n",
       "3            GPT                            AAT1,ALT,ALT1,GPT1,SGPT   \n",
       "4       PPP1R12C                          AAVS1,LENG3,MBS85,p84,p85   \n",
       "...          ...                                                ...   \n",
       "2119      ZNF785                                             ZNF688   \n",
       "2120         ZP4                         ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4   \n",
       "2121       LMBR1      ACHP,C7orf2,DIF14,LSS,PPD2,THYP,TPT,TPTPS,ZRS   \n",
       "2122      ZNF446                           ZKSCAN20,ZSCAN30,ZSCAN52   \n",
       "2123     ZKSCAN2                             ZNF694,ZSCAN31,ZSCAN34   \n",
       "\n",
       "              ENSG_ID  HGNC_ID  NCBI_ID collision source  \n",
       "0     ENSG00000211890   5479.0     3494       A2M   NCBI  \n",
       "1     ENSG00000197085  22128.0   404744      AAA1   NCBI  \n",
       "2     ENSG00000183833  24010.0    89876      AAT1   NCBI  \n",
       "3     ENSG00000167701   4552.0     2875      AAT1   NCBI  \n",
       "4     ENSG00000125503  14947.0    54776     AAVS1   NCBI  \n",
       "...               ...      ...      ...       ...    ...  \n",
       "2119  ENSG00000197162  26496.0   146540    ZNF688   NCBI  \n",
       "2120  ENSG00000116996  15770.0    57829       ZP1   NCBI  \n",
       "2121  ENSG00000105983  13243.0    64327       ZRS   NCBI  \n",
       "2122  ENSG00000083838  21036.0    55663   ZSCAN30   NCBI  \n",
       "2123  ENSG00000155592  25677.0   342357   ZSCAN31   NCBI  \n",
       "\n",
       "[2124 rows x 7 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ncbi_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ncbi_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_gene_symbol_set = set(mini_ncbi_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45390"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ncbi_gene_symbols = len(ncbi_gene_symbol_set)\n",
    "total_number_ncbi_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1566"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"collision\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1689"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_primary_symbol_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"gene_symbol\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge 3 sets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3471 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol collision source\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1     AAVS1   HGNC\n",
       "1         SOAT1                         ACAT,ACAT1     ACAT1   HGNC\n",
       "2         SOAT2                              ACAT2     ACAT2   HGNC\n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV      ACLS   HGNC\n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1      ACP1   HGNC\n",
       "..          ...                                ...       ...    ...\n",
       "689      ZNF121                D19S204,ZHC32,ZNF20     ZNF20   ENSG\n",
       "690      RNF141                       ZFP26,ZNF230    ZNF230   ENSG\n",
       "691    ZNF322P1                     ZNF322,ZNF322B    ZNF322   ENSG\n",
       "692      ZNF106               SH3BP3,ZFP106,ZNF474    ZNF474   ENSG\n",
       "693        ZFP1                    FLJ34243,ZNF475    ZNF475   ENSG\n",
       "\n",
       "[3471 rows x 4 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df = pd.concat(\n",
    "    [\n",
    "        merged_alias_ap_collision_hgnc_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ncbi_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ensg_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "    ]\n",
    ")\n",
    "merged_alias_primary_collisions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>MGC45871,RefilinB,Cfm1</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>CFM1,FAM101B</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol            alias_symbol collision source\n",
       "84        RFLNB  MGC45871,RefilinB,Cfm1      CFM1   HGNC\n",
       "215       RFLNB            CFM1,FAM101B      CFM1   NCBI"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"CFM1\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DANCR</td>\n",
       "      <td>ANCR,AGU2,lncRNA-ANCR</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>UBE3A</td>\n",
       "      <td>AS,ANCR,E6-AP,FLJ26981</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SLC25A23</td>\n",
       "      <td>FLJ30339,MGC2615,APC2</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANAPC2</td>\n",
       "      <td>APC2,KIAA1406</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AREG</td>\n",
       "      <td>AR,CRDGF</td>\n",
       "      <td>AR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>665</th>\n",
       "      <td>RNU4-5P</td>\n",
       "      <td>RNU4P5,U4,U4/6</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>666</th>\n",
       "      <td>RNU4-4P</td>\n",
       "      <td>RNU4P4,U4,U4/5</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>667</th>\n",
       "      <td>RNU4-1</td>\n",
       "      <td>RNU4A,RNU4B2,U4,U4BL</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>668</th>\n",
       "      <td>RNU6-50P</td>\n",
       "      <td>RNU6-50,RNU6P1,U6</td>\n",
       "      <td>U6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>669</th>\n",
       "      <td>RNU6-1</td>\n",
       "      <td>RNU6A,U6,U6-1</td>\n",
       "      <td>U6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1071 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol            alias_symbol collision source\n",
       "13        DANCR   ANCR,AGU2,lncRNA-ANCR      ANCR   HGNC\n",
       "14        UBE3A  AS,ANCR,E6-AP,FLJ26981      ANCR   HGNC\n",
       "15     SLC25A23   FLJ30339,MGC2615,APC2      APC2   HGNC\n",
       "16       ANAPC2           APC2,KIAA1406      APC2   HGNC\n",
       "19         AREG                AR,CRDGF        AR   HGNC\n",
       "..          ...                     ...       ...    ...\n",
       "665     RNU4-5P          RNU4P5,U4,U4/6        U4   ENSG\n",
       "666     RNU4-4P          RNU4P4,U4,U4/5        U4   ENSG\n",
       "667      RNU4-1    RNU4A,RNU4B2,U4,U4BL        U4   ENSG\n",
       "668    RNU6-50P       RNU6-50,RNU6P1,U6        U6   ENSG\n",
       "669      RNU6-1           RNU6A,U6,U6-1        U6   ENSG\n",
       "\n",
       "[1071 rows x 4 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['collision','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>ATF1,ATF2,HFB1-GDNF</td>\n",
       "      <td>ATF1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>ATF1,ATF2,HFB1-GDNF</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>TMPRSS4</td>\n",
       "      <td>TMPRSS3,MT-SP2,CAP2</td>\n",
       "      <td>CAP2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>ACKR2</td>\n",
       "      <td>CCR10,D6,CCR9</td>\n",
       "      <td>CCR10</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>ACKR2</td>\n",
       "      <td>CCR10,D6,CCR9</td>\n",
       "      <td>CCR9</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>583</th>\n",
       "      <td>SLIT3</td>\n",
       "      <td>MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2</td>\n",
       "      <td>SLIT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>584</th>\n",
       "      <td>SLIT3</td>\n",
       "      <td>MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2</td>\n",
       "      <td>SLIT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>SPNS1</td>\n",
       "      <td>HSPIN1,LAT,NRS,PP2030,SLC63A1,SPIN1,SPINL</td>\n",
       "      <td>SPIN1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>615</th>\n",
       "      <td>UXT</td>\n",
       "      <td>ART-27,SKP2,STAP1</td>\n",
       "      <td>STAP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>TMPRSS4</td>\n",
       "      <td>CAP2,MT-SP2,TMPRSS3</td>\n",
       "      <td>TMPRSS3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>644 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                               alias_symbol collision source\n",
       "28         GDNF                        ATF1,ATF2,HFB1-GDNF      ATF1   HGNC\n",
       "29         GDNF                        ATF1,ATF2,HFB1-GDNF      ATF2   HGNC\n",
       "62      TMPRSS4                        TMPRSS3,MT-SP2,CAP2      CAP2   HGNC\n",
       "72        ACKR2                              CCR10,D6,CCR9     CCR10   HGNC\n",
       "74        ACKR2                              CCR10,D6,CCR9      CCR9   HGNC\n",
       "..          ...                                        ...       ...    ...\n",
       "583       SLIT3             MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2     SLIT1   ENSG\n",
       "584       SLIT3             MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2     SLIT2   ENSG\n",
       "602       SPNS1  HSPIN1,LAT,NRS,PP2030,SLC63A1,SPIN1,SPINL     SPIN1   ENSG\n",
       "615         UXT                          ART-27,SKP2,STAP1     STAP1   ENSG\n",
       "635     TMPRSS4                        CAP2,MT-SP2,TMPRSS3   TMPRSS3   ENSG\n",
       "\n",
       "[644 rows x 4 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['gene_symbol','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_primary_collisions_df.to_csv(\n",
    "    \"../output/merged_alias_primary_collisions_df.csv\", index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ABCD1',\n",
       " 'ACD',\n",
       " 'ACKR2',\n",
       " 'ACOD1',\n",
       " 'ACTBP8',\n",
       " 'ADRA1D',\n",
       " 'AGXT',\n",
       " 'AIFM2',\n",
       " 'AKR1B1',\n",
       " 'AKR1B10',\n",
       " 'ALPK3',\n",
       " 'AMH',\n",
       " 'ANAPC2',\n",
       " 'ANKRD37',\n",
       " 'ANTXR1',\n",
       " 'AOC1',\n",
       " 'APEX1',\n",
       " 'AREG',\n",
       " 'ARHGAP21',\n",
       " 'ARHGEF7',\n",
       " 'ARID4A',\n",
       " 'ART4',\n",
       " 'ARTN',\n",
       " 'ASIC2',\n",
       " 'AURKAIP1',\n",
       " 'AZIN2',\n",
       " 'B3GNTL1',\n",
       " 'BANF1P1',\n",
       " 'BCAT2',\n",
       " 'BRIP1',\n",
       " 'BTF3P11',\n",
       " 'BTN3A3',\n",
       " 'BVES',\n",
       " 'C1D',\n",
       " 'C1QTNF1',\n",
       " 'C6ORF89',\n",
       " 'CACNA1A',\n",
       " 'CADPS',\n",
       " 'CADPS2',\n",
       " 'CAPN5',\n",
       " 'CARD16',\n",
       " 'CCL13',\n",
       " 'CCL14',\n",
       " 'CCL15',\n",
       " 'CCM2',\n",
       " 'CD200R1',\n",
       " 'CDH19',\n",
       " 'CDH20',\n",
       " 'CDKN2AIP',\n",
       " 'CDPF1',\n",
       " 'CELSR1',\n",
       " 'CES1',\n",
       " 'CFAP73',\n",
       " 'CFH',\n",
       " 'CHAF1B',\n",
       " 'CHAMP1',\n",
       " 'CHD6',\n",
       " 'CHEK2',\n",
       " 'CHORDC1',\n",
       " 'CLASP1',\n",
       " 'CLCF1',\n",
       " 'CNGB1',\n",
       " 'CNKSR2',\n",
       " 'CNOT6',\n",
       " 'CNRIP1',\n",
       " 'CNTN1',\n",
       " 'COASY',\n",
       " 'COPS2',\n",
       " 'COPS3',\n",
       " 'CORIN',\n",
       " 'COX7A2L',\n",
       " 'CPA4',\n",
       " 'CPAMD8',\n",
       " 'CPNE1',\n",
       " 'CPNE2',\n",
       " 'CPPED1',\n",
       " 'CREB3L4',\n",
       " 'CSNK2A2',\n",
       " 'CSTB',\n",
       " 'CTDSP2',\n",
       " 'CXCL10',\n",
       " 'CXXC1',\n",
       " 'CYCSP5',\n",
       " 'CYP11B1',\n",
       " 'CYP11B2',\n",
       " 'CYP21A2',\n",
       " 'CYP2A6',\n",
       " 'DCAF5',\n",
       " 'DCBLD2',\n",
       " 'DDOST',\n",
       " 'DDR2',\n",
       " 'DDX11',\n",
       " 'DDX18',\n",
       " 'DEAF1',\n",
       " 'DEFA6',\n",
       " 'DEPDC1B',\n",
       " 'DFFB',\n",
       " 'DHX8',\n",
       " 'DIDO1',\n",
       " 'DLC1',\n",
       " 'DLEC1',\n",
       " 'DLGAP3',\n",
       " 'DMAP1',\n",
       " 'DMTF1',\n",
       " 'DNPEP',\n",
       " 'DPAGT1',\n",
       " 'DPYSL2',\n",
       " 'DSC3',\n",
       " 'DUSP12',\n",
       " 'DUSP28',\n",
       " 'DYNLL1',\n",
       " 'E2F1',\n",
       " 'EIF2S1',\n",
       " 'EIF4G2',\n",
       " 'ELOF1',\n",
       " 'EPAS1',\n",
       " 'EPS8L1',\n",
       " 'EPX',\n",
       " 'ERC2',\n",
       " 'ERCC4',\n",
       " 'ETFA',\n",
       " 'F3',\n",
       " 'FA2H',\n",
       " 'FABP5P1',\n",
       " 'FADS1',\n",
       " 'FANCA',\n",
       " 'FASN',\n",
       " 'FBF1',\n",
       " 'FBXO5',\n",
       " 'FCRLA',\n",
       " 'FCRLB',\n",
       " 'FEM1C',\n",
       " 'FGF13',\n",
       " 'FGFR1',\n",
       " 'FLII',\n",
       " 'FOXM1',\n",
       " 'FYN',\n",
       " 'FZD9',\n",
       " 'FZR1',\n",
       " 'GAL3ST3',\n",
       " 'GALK2',\n",
       " 'GALNT15',\n",
       " 'GALNT18',\n",
       " 'GALNTL6',\n",
       " 'GALP',\n",
       " 'GARS1',\n",
       " 'GATD3',\n",
       " 'GC',\n",
       " 'GDF15',\n",
       " 'GDNF',\n",
       " 'GET1',\n",
       " 'GJD3',\n",
       " 'GKN1',\n",
       " 'GLB1',\n",
       " 'GLS2',\n",
       " 'GMNN',\n",
       " 'GNAI2',\n",
       " 'GNPDA1',\n",
       " 'GOLT1A',\n",
       " 'GOLT1B',\n",
       " 'GPI',\n",
       " 'GPKOW',\n",
       " 'GPNMB',\n",
       " 'GPRIN1',\n",
       " 'GPX7',\n",
       " 'GRID1',\n",
       " 'GRID2',\n",
       " 'GTPBP4',\n",
       " 'GUCY2C',\n",
       " 'HCRT',\n",
       " 'HCRTR1',\n",
       " 'HECA',\n",
       " 'HFE',\n",
       " 'HID1',\n",
       " 'HNF1A-AS1',\n",
       " 'HOMER2',\n",
       " 'HOOK1',\n",
       " 'HOOK2',\n",
       " 'HOOK3',\n",
       " 'HRNR',\n",
       " 'HSD17B4',\n",
       " 'HSD17B7P2',\n",
       " 'IAPP',\n",
       " 'IFNA1',\n",
       " 'IGF2BP3',\n",
       " 'IL22',\n",
       " 'IL6',\n",
       " 'ILF3',\n",
       " 'INSL3',\n",
       " 'INSL6',\n",
       " 'IPO4',\n",
       " 'ITM2C',\n",
       " 'KANSL1',\n",
       " 'KCNA4',\n",
       " 'KCNA5',\n",
       " 'KCNH4',\n",
       " 'KCNH8',\n",
       " 'KCNN4',\n",
       " 'KCTD11',\n",
       " 'KHSRP',\n",
       " 'KIF20B',\n",
       " 'KIF2A',\n",
       " 'KLF5',\n",
       " 'KLHL7',\n",
       " 'KLK1',\n",
       " 'KLK6',\n",
       " 'KLRG1',\n",
       " 'KRTAP5-11',\n",
       " 'KRTAP5-2',\n",
       " 'KRTAP5-3',\n",
       " 'KRTAP5-5',\n",
       " 'KRTAP5-7',\n",
       " 'KRTAP5-8',\n",
       " 'KRTAP5-9',\n",
       " 'LAMA4',\n",
       " 'LDB2',\n",
       " 'LGALS12',\n",
       " 'LGALS13',\n",
       " 'LGALSL',\n",
       " 'LNPK',\n",
       " 'LNX1',\n",
       " 'LONP1',\n",
       " 'LRIF1',\n",
       " 'LRIG1',\n",
       " 'LRP12',\n",
       " 'LRP5',\n",
       " 'LTB4R2',\n",
       " 'LZTS1',\n",
       " 'MAFIP',\n",
       " 'MAGI1',\n",
       " 'MAP4K2',\n",
       " 'MAP4K5',\n",
       " 'MAPRE2',\n",
       " 'MB',\n",
       " 'MBD1',\n",
       " 'MBD4',\n",
       " 'MBL2',\n",
       " 'MED12',\n",
       " 'MED22',\n",
       " 'MED9',\n",
       " 'MELTF',\n",
       " 'METAP1',\n",
       " 'METAP2',\n",
       " 'MGAM',\n",
       " 'MIPEP',\n",
       " 'MLF2',\n",
       " 'MMAB',\n",
       " 'MMACHC',\n",
       " 'MOCOS',\n",
       " 'MOGAT1',\n",
       " 'MOGAT2',\n",
       " 'MOGAT3',\n",
       " 'MRLN',\n",
       " 'MRPL13',\n",
       " 'MRPL41',\n",
       " 'MRPS12',\n",
       " 'MRPS28',\n",
       " 'MRPS35',\n",
       " 'MRTFA',\n",
       " 'MS4A12',\n",
       " 'MS4A4A',\n",
       " 'MS4A7',\n",
       " 'MTCH1',\n",
       " 'MYCBP2',\n",
       " 'MYDGF',\n",
       " 'MYH7B',\n",
       " 'MYL1',\n",
       " 'MYO1C',\n",
       " 'MYO1E',\n",
       " 'MYT1',\n",
       " 'MYZAP',\n",
       " 'NANOGP2',\n",
       " 'NANOGP4',\n",
       " 'NANOGP7',\n",
       " 'NANOS1',\n",
       " 'NANOS2',\n",
       " 'NANOS3',\n",
       " 'NAPSA',\n",
       " 'NAPSB',\n",
       " 'NAT8',\n",
       " 'NCOA2',\n",
       " 'NCOA3',\n",
       " 'NDRG1',\n",
       " 'NDUFAB1',\n",
       " 'NECTIN4',\n",
       " 'NEDD4',\n",
       " 'NELL2',\n",
       " 'NFE2L1',\n",
       " 'NKX2-1',\n",
       " 'NLRP2B',\n",
       " 'NLRP3',\n",
       " 'NLRP4',\n",
       " 'NLRP6',\n",
       " 'NOL3',\n",
       " 'NOL6',\n",
       " 'NPPC',\n",
       " 'NPRL2',\n",
       " 'NPRL3',\n",
       " 'NR5A1',\n",
       " 'NRG1',\n",
       " 'NT5C3A',\n",
       " 'NTN1',\n",
       " 'NUDT19',\n",
       " 'OSER1',\n",
       " 'PAEP',\n",
       " 'PAK6',\n",
       " 'PAPPA-AS1',\n",
       " 'PARD3',\n",
       " 'PCDHA4',\n",
       " 'PCDHA6',\n",
       " 'PCDHB11',\n",
       " 'PCDHB16',\n",
       " 'PCDHGA5',\n",
       " 'PCDHGB4',\n",
       " 'PDHX',\n",
       " 'PDPK1',\n",
       " 'PEDS1',\n",
       " 'PGAM4',\n",
       " 'PGAP1',\n",
       " 'PGF',\n",
       " 'PGPEP1',\n",
       " 'PIGU',\n",
       " 'PITRM1',\n",
       " 'PKMYT1',\n",
       " 'PKN1',\n",
       " 'PLPP6',\n",
       " 'PLPPR3',\n",
       " 'PM20D1',\n",
       " 'PMEL',\n",
       " 'PMEPA1',\n",
       " 'PMP22',\n",
       " 'PMPCB',\n",
       " 'PMS2CL',\n",
       " 'PNKD',\n",
       " 'PODXL',\n",
       " 'POLK',\n",
       " 'POLR1A',\n",
       " 'POLR1B',\n",
       " 'POLR1G',\n",
       " 'PPIAP10',\n",
       " 'PPP1R9B',\n",
       " 'PPY',\n",
       " 'PRDX5',\n",
       " 'PRDX6',\n",
       " 'PRG2',\n",
       " 'PRICKLE1',\n",
       " 'PRKD2',\n",
       " 'PRKRA',\n",
       " 'PRODH',\n",
       " 'PRPF38B',\n",
       " 'PRSS8',\n",
       " 'PSG2',\n",
       " 'PSMA7',\n",
       " 'PSMD1',\n",
       " 'PSMD2',\n",
       " 'PTEN',\n",
       " 'PTENP1',\n",
       " 'PTPMT1',\n",
       " 'PTPN18',\n",
       " 'PTPRU',\n",
       " 'PTRH2',\n",
       " 'PXDN',\n",
       " 'PXMP2',\n",
       " 'PYDC1',\n",
       " 'RAB43',\n",
       " 'RAI1',\n",
       " 'RBL1',\n",
       " 'RDH14',\n",
       " 'REXO2',\n",
       " 'RGL4',\n",
       " 'RGPD1',\n",
       " 'RGS18',\n",
       " 'RHEBP1',\n",
       " 'RHOD',\n",
       " 'RIMBP2',\n",
       " 'RIN2',\n",
       " 'RMRP',\n",
       " 'RNASE1',\n",
       " 'RNASE10',\n",
       " 'RNASE12',\n",
       " 'RNASE3',\n",
       " 'RNASE6',\n",
       " 'RNASE7',\n",
       " 'RNASEH1P1',\n",
       " 'RNF138',\n",
       " 'RNF141',\n",
       " 'RNF2',\n",
       " 'RNF4',\n",
       " 'RNF7',\n",
       " 'RORA',\n",
       " 'RPL15',\n",
       " 'RPL17',\n",
       " 'RPL23',\n",
       " 'RRP1B',\n",
       " 'RSC1A1',\n",
       " 'RSL24D1',\n",
       " 'S100A9',\n",
       " 'S100G',\n",
       " 'SBDS',\n",
       " 'SCGB1D1',\n",
       " 'SEC14L2',\n",
       " 'SEC14L3',\n",
       " 'SELENOP',\n",
       " 'SEM1',\n",
       " 'SEMA3F-AS1',\n",
       " 'SEPSECS',\n",
       " 'SERPINA2',\n",
       " 'SERPINB8',\n",
       " 'SF3B2',\n",
       " 'SF3B3',\n",
       " 'SFI1',\n",
       " 'SGCG',\n",
       " 'SGPP2',\n",
       " 'SH2D1A',\n",
       " 'SH3D19',\n",
       " 'SLC19A1',\n",
       " 'SLC25A1',\n",
       " 'SLC25A15',\n",
       " 'SLC25A16',\n",
       " 'SLC25A21',\n",
       " 'SLC25A23',\n",
       " 'SLC35A1',\n",
       " 'SLC38A1',\n",
       " 'SLC38A2',\n",
       " 'SLC6A20',\n",
       " 'SLC7A8',\n",
       " 'SLIT1',\n",
       " 'SLIT3',\n",
       " 'SLTM',\n",
       " 'SLURP1',\n",
       " 'SMIM29',\n",
       " 'SMIM6',\n",
       " 'SMOC2',\n",
       " 'SMOX',\n",
       " 'SND1',\n",
       " 'SNORA58',\n",
       " 'SNORD115-45',\n",
       " 'SNRPB2',\n",
       " 'SNX33',\n",
       " 'SOAT1',\n",
       " 'SOAT2',\n",
       " 'SOS1',\n",
       " 'SOX9',\n",
       " 'SP6',\n",
       " 'SPATA17',\n",
       " 'SPATS1',\n",
       " 'SPNS1',\n",
       " 'SPPL2A',\n",
       " 'SPPL2B',\n",
       " 'SRFBP1',\n",
       " 'SRMS',\n",
       " 'STK19',\n",
       " 'STK19B',\n",
       " 'STK24',\n",
       " 'STK4',\n",
       " 'SYCP2',\n",
       " 'TACR1',\n",
       " 'TADA2A',\n",
       " 'TDO2',\n",
       " 'TENM1',\n",
       " 'TENT4A',\n",
       " 'TGFBRAP1',\n",
       " 'THPO',\n",
       " 'TIMP1',\n",
       " 'TINAGL1',\n",
       " 'TIRAP',\n",
       " 'TMEM184B-AS1',\n",
       " 'TMPRSS4',\n",
       " 'TNIP1',\n",
       " 'TNNI3',\n",
       " 'TNPO1',\n",
       " 'TOX4',\n",
       " 'TP53RK',\n",
       " 'TPGS1',\n",
       " 'TRAPPC8',\n",
       " 'TRAV1-1',\n",
       " 'TSPO',\n",
       " 'TTC5',\n",
       " 'UBASH3A',\n",
       " 'UBE2Z',\n",
       " 'UCMA',\n",
       " 'UGT2B4',\n",
       " 'USP2-AS1',\n",
       " 'USP21',\n",
       " 'USP25',\n",
       " 'USP32',\n",
       " 'UTS2',\n",
       " 'UXT',\n",
       " 'VSIG2',\n",
       " 'XPO1',\n",
       " 'ZBTB32',\n",
       " 'ZFP1',\n",
       " 'ZNF106',\n",
       " 'ZNF121',\n",
       " 'ZNF160',\n",
       " 'ZNF83'}"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_ap_collisions = (\n",
    "    ncbi_alias_primary_collision_primary_symbol_set\n",
    "    & hgnc_alias_primary_collision_primary_symbol_set\n",
    "    & ensg_alias_primary_collision_primary_symbol_set\n",
    ")\n",
    "common_ap_collisions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "496"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_ap_collisions)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
