{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1026,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1027,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_ap_collision_df(mini_xxxx_df: pd.DataFrame, source: str) -> pd.DataFrame:\n",
    "    \"\"\"Create a df of alias-primary collision symbols \n",
    "\n",
    "    :param mini_xxxx_df: Processed df of gene records\n",
    "    :param source: Representation of the source of the gene records\n",
    "    :param split_on_character: Character that is used to seperate alias symbols in the mini_xxxx_df\n",
    "    :return: A df of genes that share an alias with another gene\n",
    "    \"\"\"\n",
    "\n",
    "    #Strip \"HGNC\" from HGNC ID\n",
    "    mini_xxxx_df[\"HGNC_ID\"] = mini_xxxx_df[\"HGNC_ID\"].str.lstrip(\"HGNC:\")\n",
    "\n",
    "    #Remove placeholder gene records (those with no primary gene symbol)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\" \", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"-\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.dropna(subset=[\"gene_symbol\"])\n",
    "\n",
    "    #Remove placeholder LOC genes\n",
    "    mini_xxxx_df = mini_xxxx_df[~mini_xxxx_df['gene_symbol'].str.startswith('LOC', na=False)]\n",
    "\n",
    "    #mini_xxx_df\n",
    "    ## no LOC genes (only present in NCBI)\n",
    "    ## no records with no primary symbols (only present in ENSG)\n",
    "    #Convert the df into a csv and save\n",
    "    mini_xxxx_df.to_csv(f'../output/mini_{source.lower()}_df.csv', index=True) \n",
    "\n",
    "    #Make a new df where the alias symbols are merged together for each record\n",
    "    merged_alias_xxxx_df = mini_xxxx_df.copy()\n",
    "    merged_alias_xxxx_df[\"alias_symbol\"] = merged_alias_xxxx_df[\"alias_symbol\"].fillna(\"\").astype(str)\n",
    "    merged_alias_xxxx_df = (\n",
    "    merged_alias_xxxx_df.groupby(\n",
    "        [\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], dropna=False\n",
    "    )[\"alias_symbol\"]\n",
    "    .apply(lambda x: \",\".join(x.dropna())).reset_index())\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    merged_alias_xxxx_df.to_csv(f'../output/merged_alias_{source}_df.csv', index=True) \n",
    "\n",
    "    #Make a set of the primary gene symbols\n",
    "    xxxx_gene_symbol_set = set(mini_xxxx_df[\"gene_symbol\"])\n",
    "    uppercased_xxxx_gene_symbol_set = {s.upper() for s in xxxx_gene_symbol_set}\n",
    "\n",
    "    \n",
    "    subset_genes_xxxx_df = mini_xxxx_df.copy()\n",
    "\n",
    "    # #Drop genes without any aliases\n",
    "    # subset_genes_xxxx_df = subset_genes_xxxx_df.dropna(subset=[\"alias_symbol\"])\n",
    "\n",
    "    # # Remove duplicate instances of primary gene symbol- alias pairs\n",
    "    # # (occur because the same primary gene symbol may have multiple different ENSG IDs, \n",
    "    # # see gene RFLNB for example)\n",
    "    # subset_genes_xxxx_df[\"gene_symbol\"] = subset_genes_xxxx_df[\"gene_symbol\"].str.upper()\n",
    "    # subset_genes_xxxx_df[\"alias_symbol\"] = subset_genes_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    # subset_genes_xxxx_df = subset_genes_xxxx_df.drop_duplicates(subset=['gene_symbol', 'alias_symbol'], keep='first')\n",
    "\n",
    "    #Remove alias symbols that are an exact match to their respective primary gene symbol\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df[subset_genes_xxxx_df['gene_symbol'] != subset_genes_xxxx_df['alias_symbol']]\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    subset_genes_xxxx_df.to_csv(f'../output/subset_genes_{source}_df.csv', index=True) \n",
    "\n",
    "    #Create df with genes that have an alias that can be found as another gene's primary gene symbol\n",
    "    ap_collision_xxxx_df = subset_genes_xxxx_df.copy()\n",
    "    ap_collision_xxxx_df[\"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    ap_collision_xxxx_df.loc[:, \"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(lambda x: {x})\n",
    "    ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(\n",
    "        lambda x: x & uppercased_xxxx_gene_symbol_set\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.drop(\"alias_symbol_upper\", axis=1)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df[ap_collision_xxxx_df[\"collision\"].apply(lambda x: len(x) > 0)]\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n",
    "    ap_collision_xxxx_df['NCBI_ID'] = ap_collision_xxxx_df['NCBI_ID'].fillna(0).astype(int)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.sort_values(\"collision\")\n",
    "\n",
    "    #Add a source tag for future merging efforts\n",
    "    ap_collision_xxxx_df[\"source\"] = str(source.upper())\n",
    "\n",
    "    #Convert the df into a csv\n",
    "    ap_collision_xxxx_df.to_csv(f'../output/single_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "\n",
    "    #Create a secondary collision df that merges the alias symbols for each record\n",
    "    merged_alias_ap_collision_xxxx_df = ap_collision_xxxx_df.drop(columns=['alias_symbol'])\n",
    "    merged_alias_ap_collision_xxxx_df = pd.merge(merged_alias_ap_collision_xxxx_df, merged_alias_xxxx_df, on=[\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], how=\"left\")\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df[[\"gene_symbol\",\"alias_symbol\",\"ENSG_ID\",\"HGNC_ID\",\"NCBI_ID\",\"collision\",\"source\"]]\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.drop_duplicates(subset=[f\"{source}_ID\"])\n",
    "\n",
    "    #Test to make sure all collisions are in the alias list for the record\n",
    "    test_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.copy()\n",
    "    test_ap_collision_xxxx_df['alias_symbol'] = test_ap_collision_xxxx_df['alias_symbol'].apply(lambda x: x if isinstance(x, list) else [x])\n",
    "    test_ap_collision_xxxx_df['collision_in_alias'] = test_ap_collision_xxxx_df.apply(lambda row: row['collision'] in test_ap_collision_xxxx_df['alias_symbol'].values, axis=1)\n",
    "    true_count = test_ap_collision_xxxx_df['collision_in_alias'].sum()\n",
    "\n",
    "    if true_count == 0:\n",
    "        #Convert the df into a csv and save\n",
    "        merged_alias_ap_collision_xxxx_df.to_csv(f'../output/merged_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "        print(\"All collisions are present in gene alias lists.\")\n",
    "    else:\n",
    "        print(\"Some collisions are not present in gene alias lists.\")\n",
    "\n",
    "    return mini_xxxx_df.head(), subset_genes_xxxx_df.head(), ap_collision_xxxx_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1028,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133061</th>\n",
       "      <td>ENSG00000289291</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133064</th>\n",
       "      <td>ENSG00000296488</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133065 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049     <NA>   HGNC:7481         MTTF       MT-TF\n",
       "1       ENSG00000210049     <NA>   HGNC:7481         TRNF       MT-TF\n",
       "2       ENSG00000211459     <NA>   HGNC:7470          12S     MT-RNR1\n",
       "3       ENSG00000211459     <NA>   HGNC:7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459     <NA>   HGNC:7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...         ...          ...         ...\n",
       "133060  ENSG00000229388     <NA>  HGNC:52502    LINC01715    TAF12-DT\n",
       "133061  ENSG00000289291     <NA>         NaN          NaN         NaN\n",
       "133062  ENSG00000274978    26824  HGNC:10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824  HGNC:10108          U11       RNU11\n",
       "133064  ENSG00000296488     <NA>         NaN          NaN         NaN\n",
       "\n",
       "[133065 rows x 5 columns]"
      ]
     },
     "execution_count": 1028,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../input/ensg_biomart_gene20250625.txt\", sep=\"\\t\",dtype={\"NCBI gene (formerly Entrezgene) ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_ensg_df = mini_ensg_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Gene Synonym\": \"alias_symbol\",\n",
    "        \"Gene name\": \"gene_symbol\",\n",
    "        \"Gene stable ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene (formerly Entrezgene) ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1029,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86364"
      ]
     },
     "execution_count": 1029,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['ENSG_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1030,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41165"
      ]
     },
     "execution_count": 1030,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['gene_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1031,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55413"
      ]
     },
     "execution_count": 1031,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1032,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "95942"
      ]
     },
     "execution_count": 1032,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['gene_symbol']) | set(mini_ensg_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1033,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38041"
      ]
     },
     "execution_count": 1033,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df.replace(\" \", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"-\", np.nan)\n",
    "\n",
    "unique_ensg_df = mini_ensg_df[mini_ensg_df['ENSG_ID'].duplicated(keep=False) == False]\n",
    "\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_ensg_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1034,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "57009"
      ]
     },
     "execution_count": 1034,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now count rows in that subset where 'alias_symbol' is missing\n",
    "no_alias_symbol_count = unique_ensg_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1035,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "420"
      ]
     },
     "execution_count": 1035,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ensg_df[\n",
    "    mini_ensg_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ensg_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1036,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 1036,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_ensg_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1037,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "738"
      ]
     },
     "execution_count": 1037,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_ensg_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1038,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 1038,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_ensg_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1039,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92891</th>\n",
       "      <td>ENSG00000138641</td>\n",
       "      <td>8916</td>\n",
       "      <td>HGNC:4876</td>\n",
       "      <td>KIAA0032</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92878</th>\n",
       "      <td>ENSG00000287542</td>\n",
       "      <td>8916</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52117</th>\n",
       "      <td>ENSG00000273529</td>\n",
       "      <td>388585</td>\n",
       "      <td>HGNC:19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14696</th>\n",
       "      <td>ENSG00000197921</td>\n",
       "      <td>388585</td>\n",
       "      <td>HGNC:19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4416</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4421</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4418</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4417</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43282</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43285</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4415</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4414</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4419</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43284</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43283</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43281</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43280</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43279</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43286</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4420</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>HGNC:18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72820</th>\n",
       "      <td>ENSG00000235657</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68794</th>\n",
       "      <td>ENSG00000224320</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66867</th>\n",
       "      <td>ENSG00000227715</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76339</th>\n",
       "      <td>ENSG00000231834</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101068</th>\n",
       "      <td>ENSG00000206503</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67552</th>\n",
       "      <td>ENSG00000206505</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79893</th>\n",
       "      <td>ENSG00000223980</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66597</th>\n",
       "      <td>ENSG00000229215</td>\n",
       "      <td>3105</td>\n",
       "      <td>HGNC:4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79909</th>\n",
       "      <td>ENSG00000224608</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71168</th>\n",
       "      <td>ENSG00000234745</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66216</th>\n",
       "      <td>ENSG00000206450</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70494</th>\n",
       "      <td>ENSG00000232126</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79186</th>\n",
       "      <td>ENSG00000223532</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81469</th>\n",
       "      <td>ENSG00000228964</td>\n",
       "      <td>3106</td>\n",
       "      <td>HGNC:4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62689</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78388</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78387</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78774</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76516</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78386</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78773</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70213</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80918</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76515</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80917</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81311</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81310</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70211</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76517</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80916</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62691</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70212</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62690</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81309</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78772</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>HGNC:4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76137</th>\n",
       "      <td>ENSG00000242685</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75324</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83537</th>\n",
       "      <td>ENSG00000243215</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74916</th>\n",
       "      <td>ENSG00000243719</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75325</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>HGNC:4934</td>\n",
       "      <td>RING6</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "92891   ENSG00000138641     8916   HGNC:4876     KIAA0032       HERC3\n",
       "92878   ENSG00000287542     8916         NaN          NaN       HERC3\n",
       "52117   ENSG00000273529   388585  HGNC:19764      BHLHB38        HES5\n",
       "14696   ENSG00000197921   388585  HGNC:19764      BHLHB38        HES5\n",
       "4416    ENSG00000280680    55733  HGNC:18270       MART-2        HHAT\n",
       "4421    ENSG00000280680    55733  HGNC:18270          SKN        HHAT\n",
       "4418    ENSG00000280680    55733  HGNC:18270         RASP        HHAT\n",
       "4417    ENSG00000280680    55733  HGNC:18270        MART2        HHAT\n",
       "43282   ENSG00000054392    55733  HGNC:18270        MART2        HHAT\n",
       "43285   ENSG00000054392    55733  HGNC:18270          SKI        HHAT\n",
       "4415    ENSG00000280680    55733  HGNC:18270         GUP2        HHAT\n",
       "4414    ENSG00000280680    55733  HGNC:18270     FLJ10724        HHAT\n",
       "4419    ENSG00000280680    55733  HGNC:18270          SIT        HHAT\n",
       "43284   ENSG00000054392    55733  HGNC:18270          SIT        HHAT\n",
       "43283   ENSG00000054392    55733  HGNC:18270         RASP        HHAT\n",
       "43281   ENSG00000054392    55733  HGNC:18270       MART-2        HHAT\n",
       "43280   ENSG00000054392    55733  HGNC:18270         GUP2        HHAT\n",
       "43279   ENSG00000054392    55733  HGNC:18270     FLJ10724        HHAT\n",
       "43286   ENSG00000054392    55733  HGNC:18270          SKN        HHAT\n",
       "4420    ENSG00000280680    55733  HGNC:18270          SKI        HHAT\n",
       "72820   ENSG00000235657     3105   HGNC:4931          NaN       HLA-A\n",
       "68794   ENSG00000224320     3105   HGNC:4931          NaN       HLA-A\n",
       "66867   ENSG00000227715     3105   HGNC:4931          NaN       HLA-A\n",
       "76339   ENSG00000231834     3105   HGNC:4931          NaN       HLA-A\n",
       "101068  ENSG00000206503     3105   HGNC:4931          NaN       HLA-A\n",
       "67552   ENSG00000206505     3105   HGNC:4931          NaN       HLA-A\n",
       "79893   ENSG00000223980     3105   HGNC:4931          NaN       HLA-A\n",
       "66597   ENSG00000229215     3105   HGNC:4931          NaN       HLA-A\n",
       "79909   ENSG00000224608     3106   HGNC:4932           AS       HLA-B\n",
       "71168   ENSG00000234745     3106   HGNC:4932           AS       HLA-B\n",
       "66216   ENSG00000206450     3106   HGNC:4932           AS       HLA-B\n",
       "70494   ENSG00000232126     3106   HGNC:4932           AS       HLA-B\n",
       "79186   ENSG00000223532     3106   HGNC:4932           AS       HLA-B\n",
       "81469   ENSG00000228964     3106   HGNC:4932           AS       HLA-B\n",
       "62689   ENSG00000204525     3107   HGNC:4933       D6S204       HLA-C\n",
       "78388   ENSG00000228299     3107   HGNC:4933       PSORS1       HLA-C\n",
       "78387   ENSG00000228299     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "78774   ENSG00000233841     3107   HGNC:4933       PSORS1       HLA-C\n",
       "76516   ENSG00000206452     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "78386   ENSG00000228299     3107   HGNC:4933       D6S204       HLA-C\n",
       "78773   ENSG00000233841     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "70213   ENSG00000225691     3107   HGNC:4933       PSORS1       HLA-C\n",
       "80918   ENSG00000206435     3107   HGNC:4933       PSORS1       HLA-C\n",
       "76515   ENSG00000206452     3107   HGNC:4933       D6S204       HLA-C\n",
       "80917   ENSG00000206435     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "81311   ENSG00000237022     3107   HGNC:4933       PSORS1       HLA-C\n",
       "81310   ENSG00000237022     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "70211   ENSG00000225691     3107   HGNC:4933       D6S204       HLA-C\n",
       "76517   ENSG00000206452     3107   HGNC:4933       PSORS1       HLA-C\n",
       "80916   ENSG00000206435     3107   HGNC:4933       D6S204       HLA-C\n",
       "62691   ENSG00000204525     3107   HGNC:4933       PSORS1       HLA-C\n",
       "70212   ENSG00000225691     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "62690   ENSG00000204525     3107   HGNC:4933      HLA-JY3       HLA-C\n",
       "81309   ENSG00000237022     3107   HGNC:4933       D6S204       HLA-C\n",
       "78772   ENSG00000233841     3107   HGNC:4933       D6S204       HLA-C\n",
       "76137   ENSG00000242685     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "75324   ENSG00000243189     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "83537   ENSG00000243215     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "74916   ENSG00000243719     3108   HGNC:4934      D6S222E     HLA-DMA\n",
       "75325   ENSG00000243189     3108   HGNC:4934        RING6     HLA-DMA"
      ]
     },
     "execution_count": 1039,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique ENSG ID\n",
    "multi_primary = mini_ensg_df.groupby([\"gene_symbol\"]).ENSG_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_ensg_df.loc[mini_ensg_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').iloc[6000:6060]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1040,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/1363854315.py:69: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(           ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol\n",
       " 0  ENSG00000210049     <NA>    7481         MTTF       MT-TF\n",
       " 1  ENSG00000210049     <NA>    7481         TRNF       MT-TF\n",
       " 2  ENSG00000211459     <NA>    7470          12S     MT-RNR1\n",
       " 3  ENSG00000211459     <NA>    7470       MOTS-C     MT-RNR1\n",
       " 4  ENSG00000211459     <NA>    7470       MTRNR1     MT-RNR1,\n",
       "            ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol\n",
       " 0  ENSG00000210049     <NA>    7481         MTTF       MT-TF\n",
       " 1  ENSG00000210049     <NA>    7481         TRNF       MT-TF\n",
       " 2  ENSG00000211459     <NA>    7470          12S     MT-RNR1\n",
       " 3  ENSG00000211459     <NA>    7470       MOTS-C     MT-RNR1\n",
       " 4  ENSG00000211459     <NA>    7470       MTRNR1     MT-RNR1,\n",
       "                ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol collision  \\\n",
       " 24761  ENSG00000283293   125050   10037          7SK       RN7SK       7SK   \n",
       " 37365  ENSG00000057252     6646   11177        ACAT1       SOAT1     ACAT1   \n",
       " 17666  ENSG00000167780     8435   11178        ACAT2       SOAT2     ACAT2   \n",
       " 35063  ENSG00000004779     4706    7694         ACP1     NDUFAB1      ACP1   \n",
       " 47647  ENSG00000220267        0     141       ACTBP2      ACTBP8    ACTBP2   \n",
       " \n",
       "       source  \n",
       " 24761   ENSG  \n",
       " 37365   ENSG  \n",
       " 17666   ENSG  \n",
       " 35063   ENSG  \n",
       " 47647   ENSG  )"
      ]
     },
     "execution_count": 1040,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ensg_df, \"ENSG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1041,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52502.0</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95024 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049      NaN   7481.0         MTTF       MT-TF\n",
       "1       ENSG00000210049      NaN   7481.0         TRNF       MT-TF\n",
       "2       ENSG00000211459      NaN   7470.0          12S     MT-RNR1\n",
       "3       ENSG00000211459      NaN   7470.0       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459      NaN   7470.0       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989  85028.0  30062.0    LINC00100      SNHG12\n",
       "133059  ENSG00000197989  85028.0  30062.0     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388      NaN  52502.0    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978  26824.0  10108.0      RNU11-1       RNU11\n",
       "133063  ENSG00000274978  26824.0  10108.0          U11       RNU11\n",
       "\n",
       "[95024 rows x 5 columns]"
      ]
     },
     "execution_count": 1041,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df= pd.read_csv(\n",
    "    \"../output/mini_ensg_df.csv\", index_col=[0])\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1042,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52502.0</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95006 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049      NaN   7481.0         MTTF       MT-TF\n",
       "1       ENSG00000210049      NaN   7481.0         TRNF       MT-TF\n",
       "2       ENSG00000211459      NaN   7470.0          12S     MT-RNR1\n",
       "3       ENSG00000211459      NaN   7470.0       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459      NaN   7470.0       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989  85028.0  30062.0    LINC00100      SNHG12\n",
       "133059  ENSG00000197989  85028.0  30062.0     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388      NaN  52502.0    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978  26824.0  10108.0      RNU11-1       RNU11\n",
       "133063  ENSG00000274978  26824.0  10108.0          U11       RNU11\n",
       "\n",
       "[95006 rows x 5 columns]"
      ]
     },
     "execution_count": 1042,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ensg_df.csv\", index_col=[0])\n",
    "subset_genes_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1043,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45334</th>\n",
       "      <td>ENSG00000232629</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68189</th>\n",
       "      <td>ENSG00000230675</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73816</th>\n",
       "      <td>ENSG00000228813</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74541</th>\n",
       "      <td>ENSG00000224305</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76050</th>\n",
       "      <td>ENSG00000228254</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76102</th>\n",
       "      <td>ENSG00000229493</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77670</th>\n",
       "      <td>ENSG00000196610</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78251</th>\n",
       "      <td>ENSG00000226165</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "45334  ENSG00000232629   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "68189  ENSG00000230675   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "73816  ENSG00000228813   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "74541  ENSG00000224305   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "76050  ENSG00000228254   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "76102  ENSG00000229493   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "77670  ENSG00000196610   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "78251  ENSG00000226165   3120.0   4945.0      HLA-DXB    HLA-DQB2"
      ]
     },
     "execution_count": 1043,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df.loc[mini_ensg_df[\"gene_symbol\"] == \"HLA-DQB2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1044,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4202</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4203</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>LNCAMPC</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4204</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4205</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>RNF165</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ENSG_ID   NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "4202  ENSG00000141622  494470.0  31696.0        ARKL2       ARK2C\n",
       "4203  ENSG00000141622  494470.0  31696.0      LNCAMPC       ARK2C\n",
       "4204  ENSG00000141622  494470.0  31696.0     RNF111L2       ARK2C\n",
       "4205  ENSG00000141622  494470.0  31696.0       RNF165       ARK2C"
      ]
     },
     "execution_count": 1044,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df.loc[\n",
    "    subset_genes_ensg_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1045,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>10037.0</td>\n",
       "      <td>125050</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177.0</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178.0</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694.0</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>141.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904.0</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159.0</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886.0</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328.0</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>768 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol          alias_symbol          ENSG_ID  HGNC_ID  NCBI_ID  \\\n",
       "0         RN7SK                   7SK  ENSG00000283293  10037.0   125050   \n",
       "1         SOAT1  ACAT,ACAT1,SOAT,STAT  ENSG00000057252  11177.0     6646   \n",
       "2         SOAT2                 ACAT2  ENSG00000167780  11178.0     8435   \n",
       "3       NDUFAB1  ACP,ACP1,FASN2A,SDAP  ENSG00000004779   7694.0     4706   \n",
       "4        ACTBP8                ACTBP2  ENSG00000220267    141.0        0   \n",
       "..          ...                   ...              ...      ...      ...   \n",
       "807      ZNF121   D19S204,ZHC32,ZNF20  ENSG00000197961  12904.0     7675   \n",
       "808      RNF141          ZFP26,ZNF230  ENSG00000110315  21159.0    50862   \n",
       "809    ZNF322P1        ZNF322,ZNF322B  ENSG00000188801  14003.0        0   \n",
       "810      ZNF106  SH3BP3,ZFP106,ZNF474  ENSG00000103994  12886.0    64397   \n",
       "811        ZFP1       FLJ34243,ZNF475  ENSG00000184517  23328.0   162239   \n",
       "\n",
       "    collision source  \n",
       "0         7SK   ENSG  \n",
       "1       ACAT1   ENSG  \n",
       "2       ACAT2   ENSG  \n",
       "3        ACP1   ENSG  \n",
       "4      ACTBP2   ENSG  \n",
       "..        ...    ...  \n",
       "807     ZNF20   ENSG  \n",
       "808    ZNF230   ENSG  \n",
       "809    ZNF322   ENSG  \n",
       "810    ZNF474   ENSG  \n",
       "811    ZNF475   ENSG  \n",
       "\n",
       "[768 rows x 7 columns]"
      ]
     },
     "execution_count": 1045,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1046,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>10037.0</td>\n",
       "      <td>125050</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177.0</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178.0</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694.0</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>141.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TADA2A</td>\n",
       "      <td>ADA2,ADA2A,HADA2,TADA2L</td>\n",
       "      <td>ENSG00000277104</td>\n",
       "      <td>11531.0</td>\n",
       "      <td>6871</td>\n",
       "      <td>ADA2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TADA2A</td>\n",
       "      <td>ADA2,ADA2A,HADA2,TADA2L</td>\n",
       "      <td>ENSG00000276234</td>\n",
       "      <td>11531.0</td>\n",
       "      <td>6871</td>\n",
       "      <td>ADA2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ADAM28</td>\n",
       "      <td>ADAM23,EMDCII,MDC-LM,MDC-LS</td>\n",
       "      <td>ENSG00000042980</td>\n",
       "      <td>206.0</td>\n",
       "      <td>10863</td>\n",
       "      <td>ADAM23</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ADCY8</td>\n",
       "      <td>AC8,ADCY3,HBAC1</td>\n",
       "      <td>ENSG00000155897</td>\n",
       "      <td>239.0</td>\n",
       "      <td>114</td>\n",
       "      <td>ADCY3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ADRA1D</td>\n",
       "      <td>ADRA1,ADRA1A,ADRA1R</td>\n",
       "      <td>ENSG00000171873</td>\n",
       "      <td>280.0</td>\n",
       "      <td>146</td>\n",
       "      <td>ADRA1A</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AGXT</td>\n",
       "      <td>AGT,AGT1,AGXT1,PH1,SER-PYRAT,SPAT,SPT,TLH6</td>\n",
       "      <td>ENSG00000172482</td>\n",
       "      <td>341.0</td>\n",
       "      <td>189</td>\n",
       "      <td>AGT</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AURKAIP1</td>\n",
       "      <td>AIP,AKIP,FLJ20608,MS38</td>\n",
       "      <td>ENSG00000175756</td>\n",
       "      <td>24114.0</td>\n",
       "      <td>54998</td>\n",
       "      <td>AIP</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AK4</td>\n",
       "      <td>AK3,AK3L1</td>\n",
       "      <td>ENSG00000162433</td>\n",
       "      <td>363.0</td>\n",
       "      <td>205</td>\n",
       "      <td>AK3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AK3</td>\n",
       "      <td>AK3L1,AK6,AKL3L1</td>\n",
       "      <td>ENSG00000147853</td>\n",
       "      <td>17376.0</td>\n",
       "      <td>50808</td>\n",
       "      <td>AK6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>FBF1</td>\n",
       "      <td>ALB,FBF-1,FLJ00103,KIAA1863</td>\n",
       "      <td>ENSG00000188878</td>\n",
       "      <td>24674.0</td>\n",
       "      <td>85302</td>\n",
       "      <td>ALB</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ABCD1</td>\n",
       "      <td>ADRENOLEUKODYSTROPHY,ALD,ALDP,AMN</td>\n",
       "      <td>ENSG00000101986</td>\n",
       "      <td>61.0</td>\n",
       "      <td>215</td>\n",
       "      <td>AMN</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANKRD30BL</td>\n",
       "      <td>ANKRD30BP3,NCRNA00164</td>\n",
       "      <td>ENSG00000163046</td>\n",
       "      <td>35167.0</td>\n",
       "      <td>554226</td>\n",
       "      <td>ANKRD30BP3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ANAPC2</td>\n",
       "      <td>APC2,KIAA1406</td>\n",
       "      <td>ENSG00000176248</td>\n",
       "      <td>19989.0</td>\n",
       "      <td>29882</td>\n",
       "      <td>APC2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SLC25A23</td>\n",
       "      <td>APC2,FLJ30339,MGC2615</td>\n",
       "      <td>ENSG00000125648</td>\n",
       "      <td>19375.0</td>\n",
       "      <td>79085</td>\n",
       "      <td>APC2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>APLP2</td>\n",
       "      <td>APPH,APPL2</td>\n",
       "      <td>ENSG00000084234</td>\n",
       "      <td>598.0</td>\n",
       "      <td>334</td>\n",
       "      <td>APPL2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AQP7</td>\n",
       "      <td>AQP7L,AQP9,AQPAP</td>\n",
       "      <td>ENSG00000165269</td>\n",
       "      <td>640.0</td>\n",
       "      <td>364</td>\n",
       "      <td>AQP9</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>AREG</td>\n",
       "      <td>AR,AREGB,CRDGF,SDGF</td>\n",
       "      <td>ENSG00000109321</td>\n",
       "      <td>651.0</td>\n",
       "      <td>374</td>\n",
       "      <td>AR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AKR1B1</td>\n",
       "      <td>ALDR1,AR</td>\n",
       "      <td>ENSG00000085662</td>\n",
       "      <td>381.0</td>\n",
       "      <td>231</td>\n",
       "      <td>AR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>FDXR</td>\n",
       "      <td>ADR,ADXR,AR</td>\n",
       "      <td>ENSG00000161513</td>\n",
       "      <td>3642.0</td>\n",
       "      <td>2232</td>\n",
       "      <td>AR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NOL3</td>\n",
       "      <td>ARC,CARD2,MYP,NOP30</td>\n",
       "      <td>ENSG00000140939</td>\n",
       "      <td>7869.0</td>\n",
       "      <td>8996</td>\n",
       "      <td>ARC</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ARL17A</td>\n",
       "      <td>ARF1P2,ARL17P1,ARF1P2,ARL17P1</td>\n",
       "      <td>ENSG00000185829</td>\n",
       "      <td>24096.0</td>\n",
       "      <td>51326</td>\n",
       "      <td>ARF1P2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ARFGAP3</td>\n",
       "      <td>ARFGAP1</td>\n",
       "      <td>ENSG00000242247</td>\n",
       "      <td>661.0</td>\n",
       "      <td>26286</td>\n",
       "      <td>ARFGAP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TINAGL1</td>\n",
       "      <td>ARG1,LCN7,LIECG3,P3ECSL,TINAGRP</td>\n",
       "      <td>ENSG00000142910</td>\n",
       "      <td>19168.0</td>\n",
       "      <td>64129</td>\n",
       "      <td>ARG1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ARHGAP21</td>\n",
       "      <td>ARHGAP10,KIAA1424</td>\n",
       "      <td>ENSG00000107863</td>\n",
       "      <td>23725.0</td>\n",
       "      <td>57584</td>\n",
       "      <td>ARHGAP10</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>AKR1B10</td>\n",
       "      <td>AKR1B11,AKR1B12,ALDRLN,ARL-1,ARL1,HIS,HSI</td>\n",
       "      <td>ENSG00000198074</td>\n",
       "      <td>382.0</td>\n",
       "      <td>57016</td>\n",
       "      <td>ARL1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SLURP1</td>\n",
       "      <td>ANUP,ARS,ARSB,LY6-MT,LY6LS,MDM</td>\n",
       "      <td>ENSG00000126233</td>\n",
       "      <td>18746.0</td>\n",
       "      <td>57152</td>\n",
       "      <td>ARSB</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PARD3</td>\n",
       "      <td>ASIP,BAZ,BAZOOKA,PAR3,PARD3A,PPP1R118</td>\n",
       "      <td>ENSG00000148498</td>\n",
       "      <td>16051.0</td>\n",
       "      <td>56288</td>\n",
       "      <td>ASIP</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>ATF1,ATF2,HFB1-GDNF</td>\n",
       "      <td>ENSG00000168621</td>\n",
       "      <td>4232.0</td>\n",
       "      <td>2668</td>\n",
       "      <td>ATF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ATP2B4</td>\n",
       "      <td>ATP2B2,MXRA1,PMCA4</td>\n",
       "      <td>ENSG00000058668</td>\n",
       "      <td>817.0</td>\n",
       "      <td>493</td>\n",
       "      <td>ATP2B2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>SERPINA2</td>\n",
       "      <td>ARGS,ATR,PIL,SERPINA2P</td>\n",
       "      <td>ENSG00000258597</td>\n",
       "      <td>8985.0</td>\n",
       "      <td>390502</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SERPINA2</td>\n",
       "      <td>ARGS,ATR,PIL,SERPINA2P</td>\n",
       "      <td>ENSG00000274821</td>\n",
       "      <td>8985.0</td>\n",
       "      <td>390502</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ANTXR1</td>\n",
       "      <td>ATR,FLJ10601,FLJ21776,TEM8</td>\n",
       "      <td>ENSG00000169604</td>\n",
       "      <td>21014.0</td>\n",
       "      <td>84168</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>NLRP3</td>\n",
       "      <td>AGTAVPRL,AII,AVP,C1ORF7,CIAS1,CLR1.1,DFNA34,FC...</td>\n",
       "      <td>ENSG00000162711</td>\n",
       "      <td>16400.0</td>\n",
       "      <td>114548</td>\n",
       "      <td>AVP</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>B4GAT1</td>\n",
       "      <td>B3GN-T1,B3GNT1,B3GNT6,BETA3GNTI,IGAT,IGNT</td>\n",
       "      <td>ENSG00000174684</td>\n",
       "      <td>15685.0</td>\n",
       "      <td>11041</td>\n",
       "      <td>B3GNT6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>B3GNTL1</td>\n",
       "      <td>B3GNT8,B3GNT8</td>\n",
       "      <td>ENSG00000175711</td>\n",
       "      <td>21727.0</td>\n",
       "      <td>128966610</td>\n",
       "      <td>B3GNT8</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>B3GNTL1</td>\n",
       "      <td>B3GNT8,B3GNT8</td>\n",
       "      <td>ENSG00000277033</td>\n",
       "      <td>21727.0</td>\n",
       "      <td>128966610</td>\n",
       "      <td>B3GNT8</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>BRIP1</td>\n",
       "      <td>BACH1,FANCJ,OF</td>\n",
       "      <td>ENSG00000136492</td>\n",
       "      <td>20473.0</td>\n",
       "      <td>83990</td>\n",
       "      <td>BACH1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>RNF2</td>\n",
       "      <td>BAP-1,BAP1,DING,HIPI3,RING1B,RING2</td>\n",
       "      <td>ENSG00000121481</td>\n",
       "      <td>10061.0</td>\n",
       "      <td>6045</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>MAGI1</td>\n",
       "      <td>AIP3,BAIAP1,BAP1,MAGI-1,TNRC19,WWP3</td>\n",
       "      <td>ENSG00000151276</td>\n",
       "      <td>946.0</td>\n",
       "      <td>9223</td>\n",
       "      <td>BAP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>BCAT2</td>\n",
       "      <td>BCAM,BCATM,BCT2</td>\n",
       "      <td>ENSG00000105552</td>\n",
       "      <td>977.0</td>\n",
       "      <td>587</td>\n",
       "      <td>BCAM</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>BANF1P1</td>\n",
       "      <td>BCRG1,BCRP1,D14S1460,D14S1460E</td>\n",
       "      <td>ENSG00000258531</td>\n",
       "      <td>20252.0</td>\n",
       "      <td>0</td>\n",
       "      <td>BCRP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>DCAF5</td>\n",
       "      <td>BCRG2,BCRP2,D14S1461E,KIAA1824,WDR22</td>\n",
       "      <td>ENSG00000139990</td>\n",
       "      <td>20224.0</td>\n",
       "      <td>8816</td>\n",
       "      <td>BCRP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PTPN18</td>\n",
       "      <td>BDP1</td>\n",
       "      <td>ENSG00000072135</td>\n",
       "      <td>9649.0</td>\n",
       "      <td>26469</td>\n",
       "      <td>BDP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>ASIC2</td>\n",
       "      <td>ACCN,ACCN1,ASIC2A,BNAC1,BNC1,HBNAC1,MDEG</td>\n",
       "      <td>ENSG00000108684</td>\n",
       "      <td>99.0</td>\n",
       "      <td>40</td>\n",
       "      <td>BNC1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>C6orf89</td>\n",
       "      <td>BRAP,FLJ25357</td>\n",
       "      <td>ENSG00000198663</td>\n",
       "      <td>21114.0</td>\n",
       "      <td>221477</td>\n",
       "      <td>BRAP</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>DEPDC1B</td>\n",
       "      <td>BRCC3,XTP1</td>\n",
       "      <td>ENSG00000035499</td>\n",
       "      <td>24902.0</td>\n",
       "      <td>55789</td>\n",
       "      <td>BRCC3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ZFP36L1</td>\n",
       "      <td>BERG36,BRF1,CMG1,ERF1,RNF162B,TIS11B</td>\n",
       "      <td>ENSG00000185650</td>\n",
       "      <td>1107.0</td>\n",
       "      <td>677</td>\n",
       "      <td>BRF1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>ZFP36L2</td>\n",
       "      <td>BRF2,ERF2,RNF162C,TIS11D</td>\n",
       "      <td>ENSG00000152518</td>\n",
       "      <td>1108.0</td>\n",
       "      <td>678</td>\n",
       "      <td>BRF2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>ITM2C</td>\n",
       "      <td>BRI3,BRICD2C,E25,HRPC.1050_D_4,ITM3</td>\n",
       "      <td>ENSG00000135916</td>\n",
       "      <td>6175.0</td>\n",
       "      <td>81618</td>\n",
       "      <td>BRI3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>BRMS1L</td>\n",
       "      <td>BRMS1,FLJ39177,MGC11296,P40</td>\n",
       "      <td>ENSG00000100916</td>\n",
       "      <td>20512.0</td>\n",
       "      <td>84312</td>\n",
       "      <td>BRMS1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>PGAP1</td>\n",
       "      <td>BST1,FLJ12377,SPG67</td>\n",
       "      <td>ENSG00000197121</td>\n",
       "      <td>25712.0</td>\n",
       "      <td>80055</td>\n",
       "      <td>BST1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>BTN3A3</td>\n",
       "      <td>BTF3,BTN3.3</td>\n",
       "      <td>ENSG00000111801</td>\n",
       "      <td>1140.0</td>\n",
       "      <td>10384</td>\n",
       "      <td>BTF3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>MFRP</td>\n",
       "      <td>C1QTNF5,FLJ30570,NNO2,RD6</td>\n",
       "      <td>ENSG00000235718</td>\n",
       "      <td>18121.0</td>\n",
       "      <td>83552</td>\n",
       "      <td>C1QTNF5</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>C2orf72</td>\n",
       "      <td>C2ORF72,LOC257407</td>\n",
       "      <td>ENSG00000204128</td>\n",
       "      <td>27418.0</td>\n",
       "      <td>257407</td>\n",
       "      <td>C2ORF72</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>C4A</td>\n",
       "      <td>C4,C4A2,C4A3,C4A4,C4A6,C4B,C4S,CO4,CPAMD2,RG,C...</td>\n",
       "      <td>ENSG00000244207</td>\n",
       "      <td>1323.0</td>\n",
       "      <td>720</td>\n",
       "      <td>C4B</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_symbol                                       alias_symbol  \\\n",
       "0        RN7SK                                                7SK   \n",
       "1        SOAT1                               ACAT,ACAT1,SOAT,STAT   \n",
       "2        SOAT2                                              ACAT2   \n",
       "3      NDUFAB1                               ACP,ACP1,FASN2A,SDAP   \n",
       "4       ACTBP8                                             ACTBP2   \n",
       "5       TADA2A                            ADA2,ADA2A,HADA2,TADA2L   \n",
       "6       TADA2A                            ADA2,ADA2A,HADA2,TADA2L   \n",
       "7       ADAM28                        ADAM23,EMDCII,MDC-LM,MDC-LS   \n",
       "8        ADCY8                                    AC8,ADCY3,HBAC1   \n",
       "9       ADRA1D                                ADRA1,ADRA1A,ADRA1R   \n",
       "10        AGXT         AGT,AGT1,AGXT1,PH1,SER-PYRAT,SPAT,SPT,TLH6   \n",
       "11    AURKAIP1                             AIP,AKIP,FLJ20608,MS38   \n",
       "12         AK4                                          AK3,AK3L1   \n",
       "13         AK3                                   AK3L1,AK6,AKL3L1   \n",
       "14        FBF1                        ALB,FBF-1,FLJ00103,KIAA1863   \n",
       "15       ABCD1                  ADRENOLEUKODYSTROPHY,ALD,ALDP,AMN   \n",
       "16   ANKRD30BL                              ANKRD30BP3,NCRNA00164   \n",
       "17      ANAPC2                                      APC2,KIAA1406   \n",
       "18    SLC25A23                              APC2,FLJ30339,MGC2615   \n",
       "19       APLP2                                         APPH,APPL2   \n",
       "20        AQP7                                   AQP7L,AQP9,AQPAP   \n",
       "21        AREG                                AR,AREGB,CRDGF,SDGF   \n",
       "22      AKR1B1                                           ALDR1,AR   \n",
       "23        FDXR                                        ADR,ADXR,AR   \n",
       "24        NOL3                                ARC,CARD2,MYP,NOP30   \n",
       "25      ARL17A                      ARF1P2,ARL17P1,ARF1P2,ARL17P1   \n",
       "27     ARFGAP3                                            ARFGAP1   \n",
       "28     TINAGL1                    ARG1,LCN7,LIECG3,P3ECSL,TINAGRP   \n",
       "29    ARHGAP21                                  ARHGAP10,KIAA1424   \n",
       "30     AKR1B10          AKR1B11,AKR1B12,ALDRLN,ARL-1,ARL1,HIS,HSI   \n",
       "31      SLURP1                     ANUP,ARS,ARSB,LY6-MT,LY6LS,MDM   \n",
       "32       PARD3              ASIP,BAZ,BAZOOKA,PAR3,PARD3A,PPP1R118   \n",
       "33        GDNF                                ATF1,ATF2,HFB1-GDNF   \n",
       "35      ATP2B4                                 ATP2B2,MXRA1,PMCA4   \n",
       "36    SERPINA2                             ARGS,ATR,PIL,SERPINA2P   \n",
       "37    SERPINA2                             ARGS,ATR,PIL,SERPINA2P   \n",
       "38      ANTXR1                         ATR,FLJ10601,FLJ21776,TEM8   \n",
       "39       NLRP3  AGTAVPRL,AII,AVP,C1ORF7,CIAS1,CLR1.1,DFNA34,FC...   \n",
       "40      B4GAT1          B3GN-T1,B3GNT1,B3GNT6,BETA3GNTI,IGAT,IGNT   \n",
       "41     B3GNTL1                                      B3GNT8,B3GNT8   \n",
       "42     B3GNTL1                                      B3GNT8,B3GNT8   \n",
       "45       BRIP1                                     BACH1,FANCJ,OF   \n",
       "46        RNF2                 BAP-1,BAP1,DING,HIPI3,RING1B,RING2   \n",
       "47       MAGI1                AIP3,BAIAP1,BAP1,MAGI-1,TNRC19,WWP3   \n",
       "48       BCAT2                                    BCAM,BCATM,BCT2   \n",
       "49     BANF1P1                     BCRG1,BCRP1,D14S1460,D14S1460E   \n",
       "50       DCAF5               BCRG2,BCRP2,D14S1461E,KIAA1824,WDR22   \n",
       "51      PTPN18                                               BDP1   \n",
       "52       ASIC2           ACCN,ACCN1,ASIC2A,BNAC1,BNC1,HBNAC1,MDEG   \n",
       "53     C6orf89                                      BRAP,FLJ25357   \n",
       "54     DEPDC1B                                         BRCC3,XTP1   \n",
       "55     ZFP36L1               BERG36,BRF1,CMG1,ERF1,RNF162B,TIS11B   \n",
       "56     ZFP36L2                           BRF2,ERF2,RNF162C,TIS11D   \n",
       "57       ITM2C                BRI3,BRICD2C,E25,HRPC.1050_D_4,ITM3   \n",
       "58      BRMS1L                        BRMS1,FLJ39177,MGC11296,P40   \n",
       "59       PGAP1                                BST1,FLJ12377,SPG67   \n",
       "60      BTN3A3                                        BTF3,BTN3.3   \n",
       "61        MFRP                          C1QTNF5,FLJ30570,NNO2,RD6   \n",
       "62     C2orf72                                  C2ORF72,LOC257407   \n",
       "63         C4A  C4,C4A2,C4A3,C4A4,C4A6,C4B,C4S,CO4,CPAMD2,RG,C...   \n",
       "\n",
       "            ENSG_ID  HGNC_ID    NCBI_ID   collision source  \n",
       "0   ENSG00000283293  10037.0     125050         7SK   ENSG  \n",
       "1   ENSG00000057252  11177.0       6646       ACAT1   ENSG  \n",
       "2   ENSG00000167780  11178.0       8435       ACAT2   ENSG  \n",
       "3   ENSG00000004779   7694.0       4706        ACP1   ENSG  \n",
       "4   ENSG00000220267    141.0          0      ACTBP2   ENSG  \n",
       "5   ENSG00000277104  11531.0       6871        ADA2   ENSG  \n",
       "6   ENSG00000276234  11531.0       6871        ADA2   ENSG  \n",
       "7   ENSG00000042980    206.0      10863      ADAM23   ENSG  \n",
       "8   ENSG00000155897    239.0        114       ADCY3   ENSG  \n",
       "9   ENSG00000171873    280.0        146      ADRA1A   ENSG  \n",
       "10  ENSG00000172482    341.0        189         AGT   ENSG  \n",
       "11  ENSG00000175756  24114.0      54998         AIP   ENSG  \n",
       "12  ENSG00000162433    363.0        205         AK3   ENSG  \n",
       "13  ENSG00000147853  17376.0      50808         AK6   ENSG  \n",
       "14  ENSG00000188878  24674.0      85302         ALB   ENSG  \n",
       "15  ENSG00000101986     61.0        215         AMN   ENSG  \n",
       "16  ENSG00000163046  35167.0     554226  ANKRD30BP3   ENSG  \n",
       "17  ENSG00000176248  19989.0      29882        APC2   ENSG  \n",
       "18  ENSG00000125648  19375.0      79085        APC2   ENSG  \n",
       "19  ENSG00000084234    598.0        334       APPL2   ENSG  \n",
       "20  ENSG00000165269    640.0        364        AQP9   ENSG  \n",
       "21  ENSG00000109321    651.0        374          AR   ENSG  \n",
       "22  ENSG00000085662    381.0        231          AR   ENSG  \n",
       "23  ENSG00000161513   3642.0       2232          AR   ENSG  \n",
       "24  ENSG00000140939   7869.0       8996         ARC   ENSG  \n",
       "25  ENSG00000185829  24096.0      51326      ARF1P2   ENSG  \n",
       "27  ENSG00000242247    661.0      26286     ARFGAP1   ENSG  \n",
       "28  ENSG00000142910  19168.0      64129        ARG1   ENSG  \n",
       "29  ENSG00000107863  23725.0      57584    ARHGAP10   ENSG  \n",
       "30  ENSG00000198074    382.0      57016        ARL1   ENSG  \n",
       "31  ENSG00000126233  18746.0      57152        ARSB   ENSG  \n",
       "32  ENSG00000148498  16051.0      56288        ASIP   ENSG  \n",
       "33  ENSG00000168621   4232.0       2668        ATF1   ENSG  \n",
       "35  ENSG00000058668    817.0        493      ATP2B2   ENSG  \n",
       "36  ENSG00000258597   8985.0     390502         ATR   ENSG  \n",
       "37  ENSG00000274821   8985.0     390502         ATR   ENSG  \n",
       "38  ENSG00000169604  21014.0      84168         ATR   ENSG  \n",
       "39  ENSG00000162711  16400.0     114548         AVP   ENSG  \n",
       "40  ENSG00000174684  15685.0      11041      B3GNT6   ENSG  \n",
       "41  ENSG00000175711  21727.0  128966610      B3GNT8   ENSG  \n",
       "42  ENSG00000277033  21727.0  128966610      B3GNT8   ENSG  \n",
       "45  ENSG00000136492  20473.0      83990       BACH1   ENSG  \n",
       "46  ENSG00000121481  10061.0       6045        BAP1   ENSG  \n",
       "47  ENSG00000151276    946.0       9223        BAP1   ENSG  \n",
       "48  ENSG00000105552    977.0        587        BCAM   ENSG  \n",
       "49  ENSG00000258531  20252.0          0       BCRP1   ENSG  \n",
       "50  ENSG00000139990  20224.0       8816       BCRP2   ENSG  \n",
       "51  ENSG00000072135   9649.0      26469        BDP1   ENSG  \n",
       "52  ENSG00000108684     99.0         40        BNC1   ENSG  \n",
       "53  ENSG00000198663  21114.0     221477        BRAP   ENSG  \n",
       "54  ENSG00000035499  24902.0      55789       BRCC3   ENSG  \n",
       "55  ENSG00000185650   1107.0        677        BRF1   ENSG  \n",
       "56  ENSG00000152518   1108.0        678        BRF2   ENSG  \n",
       "57  ENSG00000135916   6175.0      81618        BRI3   ENSG  \n",
       "58  ENSG00000100916  20512.0      84312       BRMS1   ENSG  \n",
       "59  ENSG00000197121  25712.0      80055        BST1   ENSG  \n",
       "60  ENSG00000111801   1140.0      10384        BTF3   ENSG  \n",
       "61  ENSG00000235718  18121.0      83552     C1QTNF5   ENSG  \n",
       "62  ENSG00000204128  27418.0     257407     C2ORF72   ENSG  \n",
       "63  ENSG00000244207   1323.0        720         C4B   ENSG  "
      ]
     },
     "execution_count": 1046,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df.head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1047,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "677"
      ]
     },
     "execution_count": 1047,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(merged_alias_ap_collision_ensg_df[\"gene_symbol\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1048,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "768"
      ]
     },
     "execution_count": 1048,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(merged_alias_ap_collision_ensg_df[\"ENSG_ID\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1049,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_gene_symbol_set = set(mini_ensg_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1050,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41164"
      ]
     },
     "execution_count": 1050,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ensg_gene_symbols = len(ensg_gene_symbol_set)\n",
    "total_number_ensg_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1051,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "607"
      ]
     },
     "execution_count": 1051,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_set = set(merged_alias_ap_collision_ensg_df[\"collision\"])\n",
    "len(ensg_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1052,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "677"
      ]
     },
     "execution_count": 1052,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_ensg_df[\"gene_symbol\"])\n",
    "len(ensg_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Why is the alias-gene collision set not the same length as the set of primary symbols with collisions ?\n",
    "2. Why is the length of the alias-gene collison set shorter?\n",
    " - A priamry gene symbol with an alias-gene collision has an alias that matches a different gene's primary gene symbol.\n",
    " - Multiple genes can share a single alias (alias-alias collision)\n",
    " - If that shared alias is an alias-gene collision, then there will be more unique gene symbols in the set of primary symbols with collisions than the set of alias-gene collisions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGNC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1053,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HGNC_ID  alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       "0          HGNC:5           NaN        1  ENSG00000121410        A1BG\n",
       "1      HGNC:37133      FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       "2      HGNC:24086           ACF    29974  ENSG00000148584        A1CF\n",
       "3      HGNC:24086           ASP    29974  ENSG00000148584        A1CF\n",
       "4      HGNC:24086         ACF64    29974  ENSG00000148584        A1CF\n",
       "...           ...           ...      ...              ...         ...\n",
       "67578  HGNC:29027      KIAA0399    23140  ENSG00000074755       ZZEF1\n",
       "67579  HGNC:29027          ZZZ4    23140  ENSG00000074755       ZZEF1\n",
       "67580  HGNC:29027      FLJ10821    23140  ENSG00000074755       ZZEF1\n",
       "67581  HGNC:24523  DKFZP564I052    26009  ENSG00000036549        ZZZ3\n",
       "67582  HGNC:24523         ATAC1    26009  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 1053,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_path = \"../input/hgnc_biomart_gene20240626.txt\"\n",
    "\n",
    "mini_hgnc_df = pd.read_csv(\n",
    "    file_path, sep=\"\\t\"\n",
    ")\n",
    "\n",
    "# Rename columns\n",
    "mini_hgnc_df = mini_hgnc_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Approved symbol\": \"gene_symbol\",\n",
    "        \"Alias symbols\": \"alias_symbol\",\n",
    "        \"Ensembl gene ID\": \"ENSG_ID\",\n",
    "    }\n",
    ")\n",
    "\n",
    "#structure and labeling in HGNC download files changed between 2024 amd 2025\n",
    "if \"Alias symbol\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbol\": \"alias_symbol\"})\n",
    "elif \"Alias symbols\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbols\": \"alias_symbol\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"alias_symbol\"] = pd.NA\n",
    "\n",
    "if \"NCBI gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI gene ID\": \"NCBI_ID\"})\n",
    "elif \"NCBI Gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI Gene ID\": \"NCBI_ID\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"NCBI_ID\"] = pd.NA   \n",
    "\n",
    "mini_hgnc_df[\"NCBI_ID\"] = mini_hgnc_df[\"NCBI_ID\"].astype(pd.Int64Dtype())\n",
    "\n",
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_hgnc_df['alias_symbol'] = (\n",
    "        mini_hgnc_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split(',')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_hgnc_df = mini_hgnc_df.explode('alias_symbol')\n",
    "\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1054,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 1054,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['HGNC_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1055,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 1055,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['gene_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1056,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43187"
      ]
     },
     "execution_count": 1056,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1057,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "88307"
      ]
     },
     "execution_count": 1057,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['gene_symbol']) | set(mini_hgnc_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1058,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 1058,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.replace(\" \", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"\", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"-\", np.nan)\n",
    "unique_hcng_df = mini_hgnc_df[mini_hgnc_df['HGNC_ID'].duplicated(keep=False) == False]\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_hgnc_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1059,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22999"
      ]
     },
     "execution_count": 1059,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_count = unique_hcng_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1060,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "470"
      ]
     },
     "execution_count": 1060,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_hgnc_df[\n",
    "    mini_hgnc_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1061,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 1061,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_hgnc_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1062,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "494"
      ]
     },
     "execution_count": 1062,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_hgnc_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1063,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46"
      ]
     },
     "execution_count": 1063,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_hgnc_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1064,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6727</th>\n",
       "      <td>HGNC:1624</td>\n",
       "      <td>CTRCT8</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        HGNC_ID alias_symbol  NCBI_ID ENSG_ID gene_symbol\n",
       "6727  HGNC:1624       CTRCT8     <NA>     NaN         CCV"
      ]
     },
     "execution_count": 1064,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mask = mini_hgnc_df['gene_symbol'].str.contains('CCV', na=False) \n",
    "\n",
    "# Use the boolean mask to filter the DataFrame\n",
    "filtered_df = mini_hgnc_df[mask]\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1065,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/1363854315.py:69: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(  HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 0       5          NaN        1  ENSG00000121410        A1BG\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF,\n",
       "   HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 0       5          NaN        1  ENSG00000121410        A1BG\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF,\n",
       "       HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol collision  \\\n",
       " 42104   14947        AAVS1    54776  ENSG00000125503    PPP1R12C     AAVS1   \n",
       " 56627   11177        ACAT1     6646  ENSG00000057252       SOAT1     ACAT1   \n",
       " 56628   11178        ACAT2     8435  ENSG00000167780       SOAT2     ACAT2   \n",
       " 17689    4319         ACLS     2737  ENSG00000106571        GLI3      ACLS   \n",
       " 35323    7694         ACP1     4706  ENSG00000004779     NDUFAB1      ACP1   \n",
       " \n",
       "       source  \n",
       " 42104   HGNC  \n",
       " 56627   HGNC  \n",
       " 56628   HGNC  \n",
       " 17689   HGNC  \n",
       " 35323   HGNC  )"
      ]
     },
     "execution_count": 1065,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_hgnc_df, \"HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1066,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "0            5           NaN       1.0  ENSG00000121410        A1BG\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 1066,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df= pd.read_csv(\n",
    "    \"../output/mini_hgnc_df.csv\", index_col=[0])\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1067,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [HGNC_ID, alias_symbol, NCBI_ID, ENSG_ID, gene_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 1067,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_hgnc_df.groupby([\"gene_symbol\"]).HGNC_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_hgnc_df.loc[mini_hgnc_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1068,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "0            5           NaN       1.0  ENSG00000121410        A1BG\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 1068,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df = pd.read_csv(\n",
    "    \"../output/subset_genes_hgnc_df.csv\", index_col=[0])\n",
    "subset_genes_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1069,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>31696</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709</th>\n",
       "      <td>31696</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2710</th>\n",
       "      <td>31696</td>\n",
       "      <td>Ark2C</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711</th>\n",
       "      <td>31696</td>\n",
       "      <td>lncAMPC</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC_ID alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "2708    31696        ARKL2  494470.0  ENSG00000141622       ARK2C\n",
       "2709    31696     RNF111L2  494470.0  ENSG00000141622       ARK2C\n",
       "2710    31696        Ark2C  494470.0  ENSG00000141622       ARK2C\n",
       "2711    31696      lncAMPC  494470.0  ENSG00000141622       ARK2C"
      ]
     },
     "execution_count": 1069,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df.loc[\n",
    "    subset_genes_hgnc_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1070,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ENSG00000106571</td>\n",
       "      <td>4319</td>\n",
       "      <td>2737</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>655</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>656</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>657</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474,SH3BP3</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>658</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>659</th>\n",
       "      <td>ZNF208</td>\n",
       "      <td>PMIDP,ZNF95</td>\n",
       "      <td>ENSG00000160321</td>\n",
       "      <td>12999</td>\n",
       "      <td>7757</td>\n",
       "      <td>ZNF95</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>644 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol          ENSG_ID  HGNC_ID  \\\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1  ENSG00000125503    14947   \n",
       "1         SOAT1                         ACAT,ACAT1  ENSG00000057252    11177   \n",
       "2         SOAT2                              ACAT2  ENSG00000167780    11178   \n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV  ENSG00000106571     4319   \n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1  ENSG00000004779     7694   \n",
       "..          ...                                ...              ...      ...   \n",
       "655      ZNF121                        ZHC32,ZNF20  ENSG00000197961    12904   \n",
       "656      RNF141                       ZFP26,ZNF230  ENSG00000110315    21159   \n",
       "657      ZNF106                      ZNF474,SH3BP3  ENSG00000103994    12886   \n",
       "658        ZFP1                    FLJ34243,ZNF475  ENSG00000184517    23328   \n",
       "659      ZNF208                        PMIDP,ZNF95  ENSG00000160321    12999   \n",
       "\n",
       "     NCBI_ID collision source  \n",
       "0      54776     AAVS1   HGNC  \n",
       "1       6646     ACAT1   HGNC  \n",
       "2       8435     ACAT2   HGNC  \n",
       "3       2737      ACLS   HGNC  \n",
       "4       4706      ACP1   HGNC  \n",
       "..       ...       ...    ...  \n",
       "655     7675     ZNF20   HGNC  \n",
       "656    50862    ZNF230   HGNC  \n",
       "657    64397    ZNF474   HGNC  \n",
       "658   162239    ZNF475   HGNC  \n",
       "659     7757     ZNF95   HGNC  \n",
       "\n",
       "[644 rows x 7 columns]"
      ]
     },
     "execution_count": 1070,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_hgnc_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_hgnc_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1071,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_gene_symbol_set = set(mini_hgnc_df[\"gene_symbol\"])\n",
    "# all_gene_symbols_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1072,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 1072,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_hgnc_gene_symbols = len(hgnc_gene_symbol_set)\n",
    "total_number_hgnc_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1073,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "568"
      ]
     },
     "execution_count": 1073,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_set = set(merged_alias_ap_collision_hgnc_df[\"collision\"])\n",
    "len(hgnc_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1074,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "644"
      ]
     },
     "execution_count": 1074,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_hgnc_df[\"gene_symbol\"])\n",
    "len(hgnc_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NCBI Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1075,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_path = \"../input/Homo_sapiens.gene_info20250625\"\n",
    "\n",
    "mini_ncbi_df = pd.read_csv(file_path, sep=\"\\t\")\n",
    "# Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol\n",
    "mini_ncbi_df = mini_ncbi_df[\n",
    "[\"GeneID\", \"Symbol\", \"Synonyms\", \"dbXrefs\"]\n",
    "]\n",
    "mini_ncbi_df = mini_ncbi_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Synonyms\": \"alias_symbol\"})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split dbXrefs into individual columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1076,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.assign(\n",
    "    MIM=np.nan,\n",
    "    HGNC_ID=np.nan,\n",
    "    ENSG_ID=np.nan,\n",
    "    AllianceGenome=np.nan,\n",
    "    MIRbase=np.nan,\n",
    "    IMGTgene_db=np.nan,\n",
    "    dash=np.nan,\n",
    "    unknown=np.nan,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1077,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '138670' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ensg00000121410' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'hgnc:5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'cd1d' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '-' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'mi0000060' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    }
   ],
   "source": [
    "index_pos = 0\n",
    "\n",
    "print(len(mini_ncbi_df))\n",
    "while index_pos < len(mini_ncbi_df):\n",
    "    xrefs = mini_ncbi_df[\"dbXrefs\"][index_pos].split(\"|\")\n",
    "\n",
    "    for xref in xrefs:\n",
    "        xref = xref.lower()\n",
    "        if xref.startswith(\"mim:\"):\n",
    "            xref = xref.replace(\"mim:\", \"\")\n",
    "            mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
    "        elif xref.startswith(\"hgnc:hgnc:\"):\n",
    "            xref = xref.replace(\"hgnc:hgnc:\", \"\")\n",
    "            mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"ensembl:\"):\n",
    "            xref = xref.replace(\"ensembl:\", \"\")\n",
    "            mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"alliancegenome:\"):\n",
    "            xref = xref.replace(\"alliancegenome:\", \"\")\n",
    "            mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
    "        elif xref.startswith(\"mirbase\"):\n",
    "            xref = xref.replace(\"mirbase:\", \"\")\n",
    "            mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
    "        elif xref.startswith(\"imgt/gene-db:\"):\n",
    "            xref = xref.replace(\"imgt/gene-db:\", \"\")\n",
    "            mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
    "        elif xref.startswith(\"-\"):\n",
    "            mini_ncbi_df[\"dash\"][index_pos] = xref\n",
    "        else:\n",
    "            mini_ncbi_df[\"unknown\"][index_pos] = xref\n",
    "\n",
    "    index_pos += 1\n",
    "    pass\n",
    "\n",
    "print(index_pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1078,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df[\"ENSG_ID\"] = mini_ncbi_df[\"ENSG_ID\"].str.replace(\"ensg\", \"ENSG\", 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1079,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>7645</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>7646</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>NATP</td>\n",
       "      <td>AACP|NATP1</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193580 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol               alias_symbol HGNC_ID  \\\n",
       "0             1        A1BG       A1B|ABG|GAB|HYST2477       5   \n",
       "1             2         A2M  A2MD|CPAMD5|FWP007|S863-7       7   \n",
       "2             9        NAT1       AAC1|MNAT|NAT-1|NATI    7645   \n",
       "3            10        NAT2            AAC2|NAT-2|PNAT    7646   \n",
       "4            11        NATP                 AACP|NATP1      15   \n",
       "...         ...         ...                        ...     ...   \n",
       "193575  8923215        trnD                          -     NaN   \n",
       "193576  8923216        trnP                          -     NaN   \n",
       "193577  8923217        trnA                          -     NaN   \n",
       "193578  8923218        COX1                          -     NaN   \n",
       "193579  8923219    16S rRNA                          -     NaN   \n",
       "\n",
       "                ENSG_ID  \n",
       "0       ENSG00000121410  \n",
       "1       ENSG00000175899  \n",
       "2       ENSG00000171428  \n",
       "3       ENSG00000156006  \n",
       "4                   NaN  \n",
       "...                 ...  \n",
       "193575              NaN  \n",
       "193576              NaN  \n",
       "193577              NaN  \n",
       "193578              NaN  \n",
       "193579              NaN  \n",
       "\n",
       "[193580 rows x 5 columns]"
      ]
     },
     "execution_count": 1079,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.drop(\n",
    "    [\n",
    "        \"AllianceGenome\",\n",
    "        \"MIRbase\",\n",
    "        \"IMGTgene_db\",\n",
    "        \"dash\",\n",
    "        \"unknown\",\n",
    "        \"dbXrefs\",\n",
    "        \"MIM\",\n",
    "    ],\n",
    "    axis=1,\n",
    ")\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1080,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "193580"
      ]
     },
     "execution_count": 1080,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ncbi_df['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1081,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "193427"
      ]
     },
     "execution_count": 1081,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ncbi_df['gene_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1082,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "27600"
      ]
     },
     "execution_count": 1082,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ncbi_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1083,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "220799"
      ]
     },
     "execution_count": 1083,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ncbi_df['gene_symbol']) | set(mini_ncbi_df['alias_symbol']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1084,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 1084,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.replace(\" \", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"\", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"-\", np.nan)\n",
    "\n",
    "unique_ncbi_df = mini_ncbi_df[mini_ncbi_df['NCBI_ID'].duplicated(keep=False) == False]\n",
    "\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_ncbi_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1085,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "165885"
      ]
     },
     "execution_count": 1085,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_count = unique_ncbi_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1086,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "248"
      ]
     },
     "execution_count": 1086,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ncbi_df[\n",
    "    mini_ncbi_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1087,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147700"
      ]
     },
     "execution_count": 1087,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_ncbi_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1088,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "396"
      ]
     },
     "execution_count": 1088,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_ncbi_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1089,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 1089,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_ncbi_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1090,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>193532</th>\n",
       "      <td>6775087</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193573</th>\n",
       "      <td>8923213</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193548</th>\n",
       "      <td>8923188</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3522</th>\n",
       "      <td>4508</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>ATPase6|MTATP6</td>\n",
       "      <td>7414</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193519</th>\n",
       "      <td>6775074</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523</th>\n",
       "      <td>4509</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>ATPase8|MTATP8</td>\n",
       "      <td>7415</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193522</th>\n",
       "      <td>6775077</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193571</th>\n",
       "      <td>8923211</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193528</th>\n",
       "      <td>6775083</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3525</th>\n",
       "      <td>4512</td>\n",
       "      <td>COX1</td>\n",
       "      <td>COI|MTCO1</td>\n",
       "      <td>7419</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193562</th>\n",
       "      <td>8923202</td>\n",
       "      <td>COX2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193524</th>\n",
       "      <td>6775079</td>\n",
       "      <td>COX2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3526</th>\n",
       "      <td>4513</td>\n",
       "      <td>COX2</td>\n",
       "      <td>COII|MTCO2</td>\n",
       "      <td>7421</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193547</th>\n",
       "      <td>8923187</td>\n",
       "      <td>COX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3527</th>\n",
       "      <td>4514</td>\n",
       "      <td>COX3</td>\n",
       "      <td>COIII|MTCO3</td>\n",
       "      <td>7422</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193518</th>\n",
       "      <td>6775073</td>\n",
       "      <td>COX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3529</th>\n",
       "      <td>4519</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>MTCYB</td>\n",
       "      <td>7427</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193510</th>\n",
       "      <td>6775065</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193565</th>\n",
       "      <td>8923205</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52286</th>\n",
       "      <td>107648861</td>\n",
       "      <td>DEL11P13</td>\n",
       "      <td>C11DELp13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36284</th>\n",
       "      <td>100528024</td>\n",
       "      <td>DEL11P13</td>\n",
       "      <td>C11DELp13|WAGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31435</th>\n",
       "      <td>100240737</td>\n",
       "      <td>DEL1P36</td>\n",
       "      <td>C1DELp36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63076</th>\n",
       "      <td>123670537</td>\n",
       "      <td>DEL1P36</td>\n",
       "      <td>C1DELp36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2459</th>\n",
       "      <td>3045</td>\n",
       "      <td>HBD</td>\n",
       "      <td>HBK</td>\n",
       "      <td>4829</td>\n",
       "      <td>ENSG00000223609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30652</th>\n",
       "      <td>100187828</td>\n",
       "      <td>HBD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35568</th>\n",
       "      <td>100505381</td>\n",
       "      <td>MMD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19942</th>\n",
       "      <td>221938</td>\n",
       "      <td>MMD2</td>\n",
       "      <td>PAQR10</td>\n",
       "      <td>30133</td>\n",
       "      <td>ENSG00000136297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3538</th>\n",
       "      <td>4535</td>\n",
       "      <td>ND1</td>\n",
       "      <td>MTND1</td>\n",
       "      <td>7455</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193569</th>\n",
       "      <td>8923209</td>\n",
       "      <td>ND1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193543</th>\n",
       "      <td>6775098</td>\n",
       "      <td>ND1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193539</th>\n",
       "      <td>6775094</td>\n",
       "      <td>ND2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>4536</td>\n",
       "      <td>ND2</td>\n",
       "      <td>MTND2</td>\n",
       "      <td>7456</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193572</th>\n",
       "      <td>8923212</td>\n",
       "      <td>ND2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193566</th>\n",
       "      <td>8923206</td>\n",
       "      <td>ND3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3540</th>\n",
       "      <td>4537</td>\n",
       "      <td>ND3</td>\n",
       "      <td>MTND3</td>\n",
       "      <td>7458</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193521</th>\n",
       "      <td>6775076</td>\n",
       "      <td>ND3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193517</th>\n",
       "      <td>6775072</td>\n",
       "      <td>ND4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3541</th>\n",
       "      <td>4538</td>\n",
       "      <td>ND4</td>\n",
       "      <td>MTND4</td>\n",
       "      <td>7459</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193557</th>\n",
       "      <td>8923197</td>\n",
       "      <td>ND4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193516</th>\n",
       "      <td>6775071</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193561</th>\n",
       "      <td>8923201</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3542</th>\n",
       "      <td>4539</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>MTND4L</td>\n",
       "      <td>7460</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193511</th>\n",
       "      <td>6775066</td>\n",
       "      <td>ND5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3543</th>\n",
       "      <td>4540</td>\n",
       "      <td>ND5</td>\n",
       "      <td>MTND5</td>\n",
       "      <td>7461</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193570</th>\n",
       "      <td>8923210</td>\n",
       "      <td>ND5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193508</th>\n",
       "      <td>6775063</td>\n",
       "      <td>ND6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3544</th>\n",
       "      <td>4541</td>\n",
       "      <td>ND6</td>\n",
       "      <td>MTND6</td>\n",
       "      <td>7462</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193558</th>\n",
       "      <td>8923198</td>\n",
       "      <td>ND6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3550</th>\n",
       "      <td>4549</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4762</th>\n",
       "      <td>6052</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10082</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3551</th>\n",
       "      <td>4550</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>MTRNR2</td>\n",
       "      <td>7471</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>6053</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10083</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5560</th>\n",
       "      <td>7006</td>\n",
       "      <td>TEC</td>\n",
       "      <td>PSCTK4</td>\n",
       "      <td>11719</td>\n",
       "      <td>ENSG00000135605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28617</th>\n",
       "      <td>100124696</td>\n",
       "      <td>TEC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65888</th>\n",
       "      <td>124901561</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65889</th>\n",
       "      <td>124901562</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70782</th>\n",
       "      <td>124906586</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65891</th>\n",
       "      <td>124901564</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65892</th>\n",
       "      <td>124901565</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol    alias_symbol HGNC_ID          ENSG_ID\n",
       "193532    6775087    12S rRNA             NaN     NaN              NaN\n",
       "193573    8923213    12S rRNA             NaN     NaN              NaN\n",
       "193548    8923188        ATP6             NaN     NaN              NaN\n",
       "3522         4508        ATP6  ATPase6|MTATP6    7414              NaN\n",
       "193519    6775074        ATP6             NaN     NaN              NaN\n",
       "3523         4509        ATP8  ATPase8|MTATP8    7415              NaN\n",
       "193522    6775077        ATP8             NaN     NaN              NaN\n",
       "193571    8923211        ATP8             NaN     NaN              NaN\n",
       "193528    6775083        COX1             NaN     NaN              NaN\n",
       "193578    8923218        COX1             NaN     NaN              NaN\n",
       "3525         4512        COX1       COI|MTCO1    7419              NaN\n",
       "193562    8923202        COX2             NaN     NaN              NaN\n",
       "193524    6775079        COX2             NaN     NaN              NaN\n",
       "3526         4513        COX2      COII|MTCO2    7421              NaN\n",
       "193547    8923187        COX3             NaN     NaN              NaN\n",
       "3527         4514        COX3     COIII|MTCO3    7422              NaN\n",
       "193518    6775073        COX3             NaN     NaN              NaN\n",
       "3529         4519        CYTB           MTCYB    7427              NaN\n",
       "193510    6775065        CYTB             NaN     NaN              NaN\n",
       "193565    8923205        CYTB             NaN     NaN              NaN\n",
       "52286   107648861    DEL11P13       C11DELp13     NaN              NaN\n",
       "36284   100528024    DEL11P13  C11DELp13|WAGR     NaN              NaN\n",
       "31435   100240737     DEL1P36        C1DELp36     NaN              NaN\n",
       "63076   123670537     DEL1P36        C1DELp36     NaN              NaN\n",
       "2459         3045         HBD             HBK    4829  ENSG00000223609\n",
       "30652   100187828         HBD             NaN     NaN              NaN\n",
       "35568   100505381        MMD2             NaN     NaN              NaN\n",
       "19942      221938        MMD2          PAQR10   30133  ENSG00000136297\n",
       "3538         4535         ND1           MTND1    7455              NaN\n",
       "193569    8923209         ND1             NaN     NaN              NaN\n",
       "193543    6775098         ND1             NaN     NaN              NaN\n",
       "193539    6775094         ND2             NaN     NaN              NaN\n",
       "3539         4536         ND2           MTND2    7456              NaN\n",
       "193572    8923212         ND2             NaN     NaN              NaN\n",
       "193566    8923206         ND3             NaN     NaN              NaN\n",
       "3540         4537         ND3           MTND3    7458              NaN\n",
       "193521    6775076         ND3             NaN     NaN              NaN\n",
       "193517    6775072         ND4             NaN     NaN              NaN\n",
       "3541         4538         ND4           MTND4    7459              NaN\n",
       "193557    8923197         ND4             NaN     NaN              NaN\n",
       "193516    6775071        ND4L             NaN     NaN              NaN\n",
       "193561    8923201        ND4L             NaN     NaN              NaN\n",
       "3542         4539        ND4L          MTND4L    7460              NaN\n",
       "193511    6775066         ND5             NaN     NaN              NaN\n",
       "3543         4540         ND5           MTND5    7461              NaN\n",
       "193570    8923210         ND5             NaN     NaN              NaN\n",
       "193508    6775063         ND6             NaN     NaN              NaN\n",
       "3544         4541         ND6           MTND6    7462              NaN\n",
       "193558    8923198         ND6             NaN     NaN              NaN\n",
       "3550         4549        RNR1          MTRNR1    7470              NaN\n",
       "4762         6052        RNR1             NaN   10082              NaN\n",
       "3551         4550        RNR2          MTRNR2    7471              NaN\n",
       "4763         6053        RNR2             NaN   10083              NaN\n",
       "5560         7006         TEC          PSCTK4   11719  ENSG00000135605\n",
       "28617   100124696         TEC             NaN     NaN              NaN\n",
       "65888   124901561   TRNAA-AGC             NaN     NaN              NaN\n",
       "65889   124901562   TRNAA-AGC             NaN     NaN              NaN\n",
       "70782   124906586   TRNAA-AGC             NaN     NaN              NaN\n",
       "65891   124901564   TRNAA-AGC             NaN     NaN              NaN\n",
       "65892   124901565   TRNAA-AGC             NaN     NaN              NaN"
      ]
     },
     "execution_count": 1090,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_ncbi_df.groupby([\"gene_symbol\"]).NCBI_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_country`:\n",
    "mini_ncbi_df.loc[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1091,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>240508 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B       5  ENSG00000121410\n",
       "0             1        A1BG          ABG       5  ENSG00000121410\n",
       "0             1        A1BG          GAB       5  ENSG00000121410\n",
       "0             1        A1BG     HYST2477       5  ENSG00000121410\n",
       "1             2         A2M         A2MD       7  ENSG00000175899\n",
       "...         ...         ...          ...     ...              ...\n",
       "193575  8923215        trnD          NaN     NaN              NaN\n",
       "193576  8923216        trnP          NaN     NaN              NaN\n",
       "193577  8923217        trnA          NaN     NaN              NaN\n",
       "193578  8923218        COX1          NaN     NaN              NaN\n",
       "193579  8923219    16S rRNA          NaN     NaN              NaN\n",
       "\n",
       "[240508 rows x 5 columns]"
      ]
     },
     "execution_count": 1091,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['alias_symbol'] = mini_ncbi_df['alias_symbol'].str.split('|')\n",
    "mini_ncbi_df = mini_ncbi_df.explode('alias_symbol')\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1092,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/1363854315.py:69: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(   NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "    NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID collision  \\\n",
       " 2809      3494       IGHA2          A2M    5479  ENSG00000211890       A2M   \n",
       " 24213   404744   NPSR1-AS1         AAA1   22128  ENSG00000197085      AAA1   \n",
       " 16372    89876      CFAP91         AAT1   24010  ENSG00000183833      AAT1   \n",
       " 2307      2875         GPT         AAT1    4552  ENSG00000167701      AAT1   \n",
       " 12222    54776    PPP1R12C        AAVS1   14947  ENSG00000125503     AAVS1   \n",
       " \n",
       "       source  \n",
       " 2809    NCBI  \n",
       " 24213   NCBI  \n",
       " 16372   NCBI  \n",
       " 2307    NCBI  \n",
       " 12222   NCBI  )"
      ]
     },
     "execution_count": 1092,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ncbi_df, \"NCBI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1093,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92719 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0             1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0             1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0             1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1             2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...         ...         ...          ...      ...              ...\n",
       "193575  8923215        trnD          NaN      NaN              NaN\n",
       "193576  8923216        trnP          NaN      NaN              NaN\n",
       "193577  8923217        trnA          NaN      NaN              NaN\n",
       "193578  8923218        COX1          NaN      NaN              NaN\n",
       "193579  8923219    16S rRNA          NaN      NaN              NaN\n",
       "\n",
       "[92719 rows x 5 columns]"
      ]
     },
     "execution_count": 1093,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df= pd.read_csv(\n",
    "    \"../output/mini_ncbi_df.csv\", index_col=[0])\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1094,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [NCBI_ID, gene_symbol, alias_symbol, HGNC_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 1094,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df[mini_ncbi_df['gene_symbol'].str.startswith('LOC')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1095,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92719 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0             1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0             1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0             1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1             2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...         ...         ...          ...      ...              ...\n",
       "193575  8923215        trnD          NaN      NaN              NaN\n",
       "193576  8923216        trnP          NaN      NaN              NaN\n",
       "193577  8923217        trnA          NaN      NaN              NaN\n",
       "193578  8923218        COX1          NaN      NaN              NaN\n",
       "193579  8923219    16S rRNA          NaN      NaN              NaN\n",
       "\n",
       "[92719 rows x 5 columns]"
      ]
     },
     "execution_count": 1095,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ncbi_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ncbi_df.csv\", index_col=[0])\n",
    "subset_genes_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1096,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IGHA2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000211890</td>\n",
       "      <td>5479.0</td>\n",
       "      <td>3494</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPSR1-AS1</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>ENSG00000197085</td>\n",
       "      <td>22128.0</td>\n",
       "      <td>404744</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFAP91</td>\n",
       "      <td>AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...</td>\n",
       "      <td>ENSG00000183833</td>\n",
       "      <td>24010.0</td>\n",
       "      <td>89876</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GPT</td>\n",
       "      <td>AAT1,ALT,ALT1,GPT1,SGPT</td>\n",
       "      <td>ENSG00000167701</td>\n",
       "      <td>4552.0</td>\n",
       "      <td>2875</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1,LENG3,MBS85,p84,p85</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947.0</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2167</th>\n",
       "      <td>CCDC106</td>\n",
       "      <td>HSU79303,ZNF581</td>\n",
       "      <td>ENSG00000173581</td>\n",
       "      <td>30181.0</td>\n",
       "      <td>29903</td>\n",
       "      <td>ZNF581</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2168</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>ENSG00000197162</td>\n",
       "      <td>26496.0</td>\n",
       "      <td>146540</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2169</th>\n",
       "      <td>ZP4</td>\n",
       "      <td>ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4</td>\n",
       "      <td>ENSG00000116996</td>\n",
       "      <td>15770.0</td>\n",
       "      <td>57829</td>\n",
       "      <td>ZP1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2171</th>\n",
       "      <td>ZNF446</td>\n",
       "      <td>ZKSCAN20,ZSCAN30,ZSCAN52</td>\n",
       "      <td>ENSG00000083838</td>\n",
       "      <td>21036.0</td>\n",
       "      <td>55663</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2172</th>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>ZNF694,ZSCAN31,ZSCAN34</td>\n",
       "      <td>ENSG00000155592</td>\n",
       "      <td>25677.0</td>\n",
       "      <td>342357</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1735 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol                                       alias_symbol  \\\n",
       "0          IGHA2                                                A2M   \n",
       "1      NPSR1-AS1                                               AAA1   \n",
       "2         CFAP91  AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...   \n",
       "3            GPT                            AAT1,ALT,ALT1,GPT1,SGPT   \n",
       "4       PPP1R12C                          AAVS1,LENG3,MBS85,p84,p85   \n",
       "...          ...                                                ...   \n",
       "2167     CCDC106                                    HSU79303,ZNF581   \n",
       "2168      ZNF785                                             ZNF688   \n",
       "2169         ZP4                         ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4   \n",
       "2171      ZNF446                           ZKSCAN20,ZSCAN30,ZSCAN52   \n",
       "2172     ZKSCAN2                             ZNF694,ZSCAN31,ZSCAN34   \n",
       "\n",
       "              ENSG_ID  HGNC_ID  NCBI_ID collision source  \n",
       "0     ENSG00000211890   5479.0     3494       A2M   NCBI  \n",
       "1     ENSG00000197085  22128.0   404744      AAA1   NCBI  \n",
       "2     ENSG00000183833  24010.0    89876      AAT1   NCBI  \n",
       "3     ENSG00000167701   4552.0     2875      AAT1   NCBI  \n",
       "4     ENSG00000125503  14947.0    54776     AAVS1   NCBI  \n",
       "...               ...      ...      ...       ...    ...  \n",
       "2167  ENSG00000173581  30181.0    29903    ZNF581   NCBI  \n",
       "2168  ENSG00000197162  26496.0   146540    ZNF688   NCBI  \n",
       "2169  ENSG00000116996  15770.0    57829       ZP1   NCBI  \n",
       "2171  ENSG00000083838  21036.0    55663   ZSCAN30   NCBI  \n",
       "2172  ENSG00000155592  25677.0   342357   ZSCAN31   NCBI  \n",
       "\n",
       "[1735 rows x 7 columns]"
      ]
     },
     "execution_count": 1096,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ncbi_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ncbi_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1097,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_gene_symbol_set = set(mini_ncbi_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1098,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45727"
      ]
     },
     "execution_count": 1098,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ncbi_gene_symbols = len(ncbi_gene_symbol_set)\n",
    "total_number_ncbi_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1099,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1490"
      ]
     },
     "execution_count": 1099,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"collision\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: make the records set based on ID not symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1735"
      ]
     },
     "execution_count": 1100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_primary_symbol_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"gene_symbol\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge 3 sets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3147 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol collision source\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1     AAVS1   HGNC\n",
       "1         SOAT1                         ACAT,ACAT1     ACAT1   HGNC\n",
       "2         SOAT2                              ACAT2     ACAT2   HGNC\n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV      ACLS   HGNC\n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1      ACP1   HGNC\n",
       "..          ...                                ...       ...    ...\n",
       "807      ZNF121                D19S204,ZHC32,ZNF20     ZNF20   ENSG\n",
       "808      RNF141                       ZFP26,ZNF230    ZNF230   ENSG\n",
       "809    ZNF322P1                     ZNF322,ZNF322B    ZNF322   ENSG\n",
       "810      ZNF106               SH3BP3,ZFP106,ZNF474    ZNF474   ENSG\n",
       "811        ZFP1                    FLJ34243,ZNF475    ZNF475   ENSG\n",
       "\n",
       "[3147 rows x 4 columns]"
      ]
     },
     "execution_count": 1101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df = pd.concat(\n",
    "    [\n",
    "        merged_alias_ap_collision_hgnc_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ncbi_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ensg_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "    ]\n",
    ")\n",
    "merged_alias_primary_collisions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>MGC45871,RefilinB,Cfm1</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>CFM1,FAM101B</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol            alias_symbol collision source\n",
       "87        RFLNB  MGC45871,RefilinB,Cfm1      CFM1   HGNC\n",
       "215       RFLNB            CFM1,FAM101B      CFM1   NCBI"
      ]
     },
     "execution_count": 1102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"CFM1\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>UBE3A</td>\n",
       "      <td>AS,ANCR,E6-AP,FLJ26981</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DANCR</td>\n",
       "      <td>ANCR,AGU2,lncRNA-ANCR</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SLC25A23</td>\n",
       "      <td>FLJ30339,MGC2615,APC2</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANAPC2</td>\n",
       "      <td>APC2,KIAA1406</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AKR1B1</td>\n",
       "      <td>AR</td>\n",
       "      <td>AR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>WDR82P1</td>\n",
       "      <td>HCG26824,SW2,WDR82,WDR82B</td>\n",
       "      <td>WDR82</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>WDR82P1</td>\n",
       "      <td>HCG26824,SW2,WDR82,WDR82B</td>\n",
       "      <td>WDR82</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>820 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol               alias_symbol collision source\n",
       "13        UBE3A     AS,ANCR,E6-AP,FLJ26981      ANCR   HGNC\n",
       "14        DANCR      ANCR,AGU2,lncRNA-ANCR      ANCR   HGNC\n",
       "15     SLC25A23      FLJ30339,MGC2615,APC2      APC2   HGNC\n",
       "16       ANAPC2              APC2,KIAA1406      APC2   HGNC\n",
       "19       AKR1B1                         AR        AR   HGNC\n",
       "..          ...                        ...       ...    ...\n",
       "796       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "797       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "798       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "804     WDR82P1  HCG26824,SW2,WDR82,WDR82B     WDR82   ENSG\n",
       "805     WDR82P1  HCG26824,SW2,WDR82,WDR82B     WDR82   ENSG\n",
       "\n",
       "[820 rows x 4 columns]"
      ]
     },
     "execution_count": 1103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['collision','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TADA2A</td>\n",
       "      <td>ADA2,ADA2A,HADA2,TADA2L</td>\n",
       "      <td>ADA2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TADA2A</td>\n",
       "      <td>ADA2,ADA2A,HADA2,TADA2L</td>\n",
       "      <td>ADA2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>SERPINA2</td>\n",
       "      <td>ARGS,ATR,PIL,SERPINA2P</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SERPINA2</td>\n",
       "      <td>ARGS,ATR,PIL,SERPINA2P</td>\n",
       "      <td>ATR</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>B3GNTL1</td>\n",
       "      <td>B3GNT8,B3GNT8</td>\n",
       "      <td>B3GNT8</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>797</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>798</th>\n",
       "      <td>VARS1</td>\n",
       "      <td>VARS,VARS2</td>\n",
       "      <td>VARS2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>804</th>\n",
       "      <td>WDR82P1</td>\n",
       "      <td>HCG26824,SW2,WDR82,WDR82B</td>\n",
       "      <td>WDR82</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>WDR82P1</td>\n",
       "      <td>HCG26824,SW2,WDR82,WDR82B</td>\n",
       "      <td>WDR82</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>151 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol               alias_symbol collision source\n",
       "5        TADA2A    ADA2,ADA2A,HADA2,TADA2L      ADA2   ENSG\n",
       "6        TADA2A    ADA2,ADA2A,HADA2,TADA2L      ADA2   ENSG\n",
       "36     SERPINA2     ARGS,ATR,PIL,SERPINA2P       ATR   ENSG\n",
       "37     SERPINA2     ARGS,ATR,PIL,SERPINA2P       ATR   ENSG\n",
       "41      B3GNTL1              B3GNT8,B3GNT8    B3GNT8   ENSG\n",
       "..          ...                        ...       ...    ...\n",
       "796       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "797       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "798       VARS1                 VARS,VARS2     VARS2   ENSG\n",
       "804     WDR82P1  HCG26824,SW2,WDR82,WDR82B     WDR82   ENSG\n",
       "805     WDR82P1  HCG26824,SW2,WDR82,WDR82B     WDR82   ENSG\n",
       "\n",
       "[151 rows x 4 columns]"
      ]
     },
     "execution_count": 1104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['gene_symbol','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1105,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_primary_collisions_df.to_csv(\n",
    "    \"../output/merged_alias_primary_collisions_df.csv\", index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ABCD1',\n",
       " 'ACD',\n",
       " 'ACKR2',\n",
       " 'ACOD1',\n",
       " 'ACTBP8',\n",
       " 'ADRA1D',\n",
       " 'AGXT',\n",
       " 'AIFM2',\n",
       " 'AKR1B1',\n",
       " 'AKR1B10',\n",
       " 'ALPK3',\n",
       " 'AMH',\n",
       " 'ANAPC2',\n",
       " 'ANKRD37',\n",
       " 'ANTXR1',\n",
       " 'AOC1',\n",
       " 'APEX1',\n",
       " 'AREG',\n",
       " 'ARHGAP21',\n",
       " 'ARHGEF7',\n",
       " 'ARID4A',\n",
       " 'ART4',\n",
       " 'ARTN',\n",
       " 'ASIC2',\n",
       " 'AURKAIP1',\n",
       " 'AZIN2',\n",
       " 'BANF1P1',\n",
       " 'BCAT2',\n",
       " 'BRIP1',\n",
       " 'BTF3P11',\n",
       " 'BTN3A3',\n",
       " 'C1D',\n",
       " 'C1QTNF1',\n",
       " 'C6orf89',\n",
       " 'CACNA1A',\n",
       " 'CADPS',\n",
       " 'CADPS2',\n",
       " 'CAPN5',\n",
       " 'CARD16',\n",
       " 'CCL13',\n",
       " 'CCL14',\n",
       " 'CCL15',\n",
       " 'CCM2',\n",
       " 'CD200R1',\n",
       " 'CDH19',\n",
       " 'CDH20',\n",
       " 'CDPF1',\n",
       " 'CELSR1',\n",
       " 'CES1',\n",
       " 'CFAP73',\n",
       " 'CFH',\n",
       " 'CHAF1B',\n",
       " 'CHAMP1',\n",
       " 'CHD6',\n",
       " 'CHEK2',\n",
       " 'CHORDC1',\n",
       " 'CLASP1',\n",
       " 'CLCF1',\n",
       " 'CNGB1',\n",
       " 'CNKSR2',\n",
       " 'CNOT6',\n",
       " 'CNRIP1',\n",
       " 'CNTN1',\n",
       " 'COASY',\n",
       " 'COPS2',\n",
       " 'COPS3',\n",
       " 'CORIN',\n",
       " 'COX7A2L',\n",
       " 'CPA4',\n",
       " 'CPAMD8',\n",
       " 'CPNE1',\n",
       " 'CPNE2',\n",
       " 'CPPED1',\n",
       " 'CREB3L4',\n",
       " 'CSNK2A2',\n",
       " 'CSTB',\n",
       " 'CTDSP2',\n",
       " 'CXCL10',\n",
       " 'CXXC1',\n",
       " 'CYCSP5',\n",
       " 'CYP11B1',\n",
       " 'CYP11B2',\n",
       " 'CYP21A2',\n",
       " 'CYP2A6',\n",
       " 'DCAF5',\n",
       " 'DCBLD2',\n",
       " 'DDOST',\n",
       " 'DDR2',\n",
       " 'DDX11',\n",
       " 'DDX18',\n",
       " 'DEAF1',\n",
       " 'DEFA6',\n",
       " 'DEPDC1B',\n",
       " 'DFFB',\n",
       " 'DHX8',\n",
       " 'DIDO1',\n",
       " 'DLC1',\n",
       " 'DLEC1',\n",
       " 'DLGAP3',\n",
       " 'DMAP1',\n",
       " 'DMTF1',\n",
       " 'DNPEP',\n",
       " 'DPAGT1',\n",
       " 'DPYSL2',\n",
       " 'DSC3',\n",
       " 'DUSP12',\n",
       " 'DUSP28',\n",
       " 'DYNLL1',\n",
       " 'E2F1',\n",
       " 'EIF2S1',\n",
       " 'EIF4G2',\n",
       " 'ELOF1',\n",
       " 'EPAS1',\n",
       " 'EPS8L1',\n",
       " 'EPX',\n",
       " 'ERC2',\n",
       " 'ERCC4',\n",
       " 'ETFA',\n",
       " 'F3',\n",
       " 'FA2H',\n",
       " 'FABP5P1',\n",
       " 'FADS1',\n",
       " 'FANCA',\n",
       " 'FASN',\n",
       " 'FBF1',\n",
       " 'FBXO5',\n",
       " 'FCRLA',\n",
       " 'FCRLB',\n",
       " 'FEM1C',\n",
       " 'FGF13',\n",
       " 'FGFR1',\n",
       " 'FLII',\n",
       " 'FOXM1',\n",
       " 'FYN',\n",
       " 'FZD9',\n",
       " 'FZR1',\n",
       " 'GAL3ST3',\n",
       " 'GALK2',\n",
       " 'GALNT15',\n",
       " 'GALNT18',\n",
       " 'GALNTL6',\n",
       " 'GALP',\n",
       " 'GARS1',\n",
       " 'GATD3',\n",
       " 'GC',\n",
       " 'GDF15',\n",
       " 'GDNF',\n",
       " 'GET1',\n",
       " 'GJD3',\n",
       " 'GKN1',\n",
       " 'GLB1',\n",
       " 'GLS2',\n",
       " 'GMNN',\n",
       " 'GNAI2',\n",
       " 'GNPDA1',\n",
       " 'GOLT1A',\n",
       " 'GOLT1B',\n",
       " 'GPI',\n",
       " 'GPKOW',\n",
       " 'GPNMB',\n",
       " 'GPRIN1',\n",
       " 'GPX7',\n",
       " 'GRID1',\n",
       " 'GRID2',\n",
       " 'GTPBP4',\n",
       " 'GUCY2C',\n",
       " 'HCRT',\n",
       " 'HCRTR1',\n",
       " 'HECA',\n",
       " 'HFE',\n",
       " 'HID1',\n",
       " 'HNF1A-AS1',\n",
       " 'HOMER2',\n",
       " 'HOOK1',\n",
       " 'HOOK2',\n",
       " 'HOOK3',\n",
       " 'HRNR',\n",
       " 'HSD17B4',\n",
       " 'HSD17B7P2',\n",
       " 'IAPP',\n",
       " 'IFNA1',\n",
       " 'IGF2BP3',\n",
       " 'IL22',\n",
       " 'IL6',\n",
       " 'ILF3',\n",
       " 'INSL3',\n",
       " 'INSL6',\n",
       " 'IPO4',\n",
       " 'ITM2C',\n",
       " 'KANSL1',\n",
       " 'KCNA4',\n",
       " 'KCNA5',\n",
       " 'KCNH4',\n",
       " 'KCNH8',\n",
       " 'KCNN4',\n",
       " 'KCTD11',\n",
       " 'KHSRP',\n",
       " 'KIF20B',\n",
       " 'KIF2A',\n",
       " 'KLF5',\n",
       " 'KLHL7',\n",
       " 'KLK1',\n",
       " 'KLK6',\n",
       " 'KLRG1',\n",
       " 'KRTAP5-11',\n",
       " 'KRTAP5-2',\n",
       " 'KRTAP5-3',\n",
       " 'KRTAP5-5',\n",
       " 'KRTAP5-7',\n",
       " 'KRTAP5-8',\n",
       " 'KRTAP5-9',\n",
       " 'LAMA4',\n",
       " 'LDB2',\n",
       " 'LGALS12',\n",
       " 'LGALS13',\n",
       " 'LGALSL',\n",
       " 'LNPK',\n",
       " 'LNX1',\n",
       " 'LONP1',\n",
       " 'LRIF1',\n",
       " 'LRIG1',\n",
       " 'LRP12',\n",
       " 'LRP5',\n",
       " 'LTB4R2',\n",
       " 'LZTS1',\n",
       " 'MAFIP',\n",
       " 'MAGI1',\n",
       " 'MAP4K2',\n",
       " 'MAP4K5',\n",
       " 'MAPRE2',\n",
       " 'MB',\n",
       " 'MBD1',\n",
       " 'MBD4',\n",
       " 'MBL2',\n",
       " 'MED12',\n",
       " 'MED22',\n",
       " 'MED9',\n",
       " 'MELTF',\n",
       " 'METAP1',\n",
       " 'METAP2',\n",
       " 'MGAM',\n",
       " 'MIPEP',\n",
       " 'MLF2',\n",
       " 'MMAB',\n",
       " 'MMACHC',\n",
       " 'MOCOS',\n",
       " 'MOGAT1',\n",
       " 'MOGAT2',\n",
       " 'MOGAT3',\n",
       " 'MRLN',\n",
       " 'MRPL13',\n",
       " 'MRPL41',\n",
       " 'MRPS12',\n",
       " 'MRPS28',\n",
       " 'MRPS35',\n",
       " 'MRTFA',\n",
       " 'MS4A12',\n",
       " 'MS4A4A',\n",
       " 'MS4A7',\n",
       " 'MTCH1',\n",
       " 'MYCBP2',\n",
       " 'MYDGF',\n",
       " 'MYH7B',\n",
       " 'MYL1',\n",
       " 'MYO1C',\n",
       " 'MYO1E',\n",
       " 'MYT1',\n",
       " 'MYZAP',\n",
       " 'NANOGP2',\n",
       " 'NANOGP4',\n",
       " 'NANOGP7',\n",
       " 'NANOS1',\n",
       " 'NANOS2',\n",
       " 'NANOS3',\n",
       " 'NAPSA',\n",
       " 'NAPSB',\n",
       " 'NAT8',\n",
       " 'NCOA2',\n",
       " 'NCOA3',\n",
       " 'NDRG1',\n",
       " 'NDUFAB1',\n",
       " 'NECTIN4',\n",
       " 'NEDD4',\n",
       " 'NELL2',\n",
       " 'NFE2L1',\n",
       " 'NKX2-1',\n",
       " 'NLRP2B',\n",
       " 'NLRP3',\n",
       " 'NLRP4',\n",
       " 'NLRP6',\n",
       " 'NOL3',\n",
       " 'NOL6',\n",
       " 'NPPC',\n",
       " 'NPRL2',\n",
       " 'NPRL3',\n",
       " 'NR5A1',\n",
       " 'NRG1',\n",
       " 'NT5C3A',\n",
       " 'NTN1',\n",
       " 'NUDT19',\n",
       " 'OSER1',\n",
       " 'PAEP',\n",
       " 'PAK6',\n",
       " 'PAPPA-AS1',\n",
       " 'PARD3',\n",
       " 'PCDHA4',\n",
       " 'PCDHA6',\n",
       " 'PCDHB11',\n",
       " 'PCDHB16',\n",
       " 'PCDHGA5',\n",
       " 'PCDHGB4',\n",
       " 'PDHX',\n",
       " 'PDPK1',\n",
       " 'PEDS1',\n",
       " 'PGAM4',\n",
       " 'PGAP1',\n",
       " 'PGF',\n",
       " 'PGPEP1',\n",
       " 'PIGU',\n",
       " 'PITRM1',\n",
       " 'PKMYT1',\n",
       " 'PKN1',\n",
       " 'PLPP6',\n",
       " 'PLPPR3',\n",
       " 'PM20D1',\n",
       " 'PMEL',\n",
       " 'PMEPA1',\n",
       " 'PMP22',\n",
       " 'PMPCB',\n",
       " 'PMS2CL',\n",
       " 'PNKD',\n",
       " 'PODXL',\n",
       " 'POLK',\n",
       " 'POLR1A',\n",
       " 'POLR1B',\n",
       " 'POLR1G',\n",
       " 'PPIAP10',\n",
       " 'PPP1R9B',\n",
       " 'PPY',\n",
       " 'PRDX5',\n",
       " 'PRDX6',\n",
       " 'PRG2',\n",
       " 'PRICKLE1',\n",
       " 'PRKD2',\n",
       " 'PRKRA',\n",
       " 'PRPF38B',\n",
       " 'PRSS8',\n",
       " 'PSG2',\n",
       " 'PSMA7',\n",
       " 'PSMD1',\n",
       " 'PSMD2',\n",
       " 'PTEN',\n",
       " 'PTENP1',\n",
       " 'PTPMT1',\n",
       " 'PTPN18',\n",
       " 'PTPRU',\n",
       " 'PTRH2',\n",
       " 'PXDN',\n",
       " 'PXMP2',\n",
       " 'PYDC1',\n",
       " 'RAB43',\n",
       " 'RAI1',\n",
       " 'RBL1',\n",
       " 'RDH14',\n",
       " 'REXO2',\n",
       " 'RGL4',\n",
       " 'RGPD1',\n",
       " 'RGS18',\n",
       " 'RHEBP1',\n",
       " 'RHOD',\n",
       " 'RIMBP2',\n",
       " 'RIN2',\n",
       " 'RMRP',\n",
       " 'RNASE1',\n",
       " 'RNASE10',\n",
       " 'RNASE12',\n",
       " 'RNASE3',\n",
       " 'RNASE6',\n",
       " 'RNASE7',\n",
       " 'RNASEH1P1',\n",
       " 'RNF138',\n",
       " 'RNF141',\n",
       " 'RNF2',\n",
       " 'RNF4',\n",
       " 'RNF7',\n",
       " 'RORA',\n",
       " 'RPL15',\n",
       " 'RPL17',\n",
       " 'RPL23',\n",
       " 'RRP1B',\n",
       " 'RSC1A1',\n",
       " 'RSL24D1',\n",
       " 'S100A9',\n",
       " 'S100G',\n",
       " 'SBDS',\n",
       " 'SCGB1D1',\n",
       " 'SEC14L2',\n",
       " 'SEC14L3',\n",
       " 'SELENOP',\n",
       " 'SEM1',\n",
       " 'SEMA3F-AS1',\n",
       " 'SEPSECS',\n",
       " 'SERPINA2',\n",
       " 'SERPINB8',\n",
       " 'SF3B2',\n",
       " 'SF3B3',\n",
       " 'SFI1',\n",
       " 'SGPP2',\n",
       " 'SH2D1A',\n",
       " 'SH3D19',\n",
       " 'SLC19A1',\n",
       " 'SLC25A1',\n",
       " 'SLC25A15',\n",
       " 'SLC25A16',\n",
       " 'SLC25A21',\n",
       " 'SLC25A23',\n",
       " 'SLC35A1',\n",
       " 'SLC38A1',\n",
       " 'SLC38A2',\n",
       " 'SLC6A20',\n",
       " 'SLC7A8',\n",
       " 'SLIT1',\n",
       " 'SLIT3',\n",
       " 'SLTM',\n",
       " 'SLURP1',\n",
       " 'SMIM29',\n",
       " 'SMOC2',\n",
       " 'SMOX',\n",
       " 'SND1',\n",
       " 'SNORA58',\n",
       " 'SNORD115-45',\n",
       " 'SNRPB2',\n",
       " 'SNX33',\n",
       " 'SOAT1',\n",
       " 'SOAT2',\n",
       " 'SOS1',\n",
       " 'SOX9',\n",
       " 'SP6',\n",
       " 'SPATA17',\n",
       " 'SPATS1',\n",
       " 'SPNS1',\n",
       " 'SPPL2A',\n",
       " 'SPPL2B',\n",
       " 'SRFBP1',\n",
       " 'SRMS',\n",
       " 'STK24',\n",
       " 'STK4',\n",
       " 'SYCP2',\n",
       " 'TACR1',\n",
       " 'TADA2A',\n",
       " 'TDO2',\n",
       " 'TENM1',\n",
       " 'TENT4A',\n",
       " 'TGFBRAP1',\n",
       " 'THPO',\n",
       " 'TIMP1',\n",
       " 'TINAGL1',\n",
       " 'TIRAP',\n",
       " 'TMEM184B-AS1',\n",
       " 'TMPRSS4',\n",
       " 'TNIP1',\n",
       " 'TNNI3',\n",
       " 'TNPO1',\n",
       " 'TOX4',\n",
       " 'TP53RK',\n",
       " 'TPGS1',\n",
       " 'TRAPPC8',\n",
       " 'TRAV1-1',\n",
       " 'TSPO',\n",
       " 'TTC5',\n",
       " 'UBASH3A',\n",
       " 'UBE2Z',\n",
       " 'UCMA',\n",
       " 'UGT2B4',\n",
       " 'USP2-AS1',\n",
       " 'USP21',\n",
       " 'USP25',\n",
       " 'USP32',\n",
       " 'UTS2',\n",
       " 'UXT',\n",
       " 'VSIG2',\n",
       " 'XPO1',\n",
       " 'ZBTB32',\n",
       " 'ZFP1',\n",
       " 'ZNF106',\n",
       " 'ZNF121',\n",
       " 'ZNF160',\n",
       " 'ZNF83'}"
      ]
     },
     "execution_count": 1106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_ap_collisions = (\n",
    "    ncbi_alias_primary_collision_primary_symbol_set\n",
    "    & hgnc_alias_primary_collision_primary_symbol_set\n",
    "    & ensg_alias_primary_collision_primary_symbol_set\n",
    ")\n",
    "common_ap_collisions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "488"
      ]
     },
     "execution_count": 1107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_ap_collisions)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
