{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 631,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 632,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_ap_collision_df(mini_xxxx_df: pd.DataFrame, source: str) -> pd.DataFrame:\n",
    "    \"\"\"Create a df of alias-primary collision symbols \n",
    "\n",
    "    :param mini_xxxx_df: Processed df of gene records\n",
    "    :param source: Representation of the source of the gene records\n",
    "    :param split_on_character: Character that is used to seperate alias symbols in the mini_xxxx_df\n",
    "    :return: A df of genes that share an alias with another gene\n",
    "    \"\"\"\n",
    "\n",
    "    #Strip \"HGNC\" from HGNC ID\n",
    "    mini_xxxx_df[\"HGNC_ID\"] = mini_xxxx_df[\"HGNC_ID\"].str.lstrip(\"HGNC:\")\n",
    "\n",
    "    #Remove placeholder gene records (those with no primary gene symbol)\n",
    "    mini_xxxx_df = mini_xxxx_df.dropna(subset=[\"gene_symbol\"])\n",
    "\n",
    "    #Remove placeholder LOC genes\n",
    "    mini_xxxx_df = mini_xxxx_df[~mini_xxxx_df['gene_symbol'].str.startswith('LOC', na=False)]\n",
    "\n",
    "    mini_xxxx_df['HGNC_ID'] = mini_xxxx_df['HGNC_ID'].fillna(0).astype(int)\n",
    "    mini_xxxx_df['NCBI_ID'] = mini_xxxx_df['NCBI_ID'].fillna(0).astype(int)\n",
    "\n",
    "    #mini_xxx_df\n",
    "    ## no LOC genes (only present in NCBI)\n",
    "    ## no records with no primary symbols (only present in ENSG)\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv and save\n",
    "    mini_xxxx_df.to_csv(f'../output/mini_{source.lower()}_df.csv', index=True) \n",
    "\n",
    "    #Make a new df where the alias symbols are merged together for each record\n",
    "    merged_alias_xxxx_df = (\n",
    "        mini_xxxx_df\n",
    "        .copy()\n",
    "        .fillna(\"\")\n",
    "        .groupby(f\"{source}_ID\", dropna=False)\n",
    "        .agg(lambda col: \",\".join(sorted(set(map(str, col)))))\n",
    "        .reset_index()\n",
    "    )\n",
    "\n",
    "    #merged_alias_xxxx_df\n",
    "    ##merging aliases associated with the same record (same gene record identifier) into a list\n",
    "    #Convert the df into a csv and save\n",
    "    merged_alias_xxxx_df.to_csv(f'../output/merged_alias_{source}_df.csv', index=True) \n",
    "\n",
    "    #Make a set of the primary gene symbols\n",
    "    xxxx_gene_symbol_set = set(mini_xxxx_df[\"gene_symbol\"])\n",
    "    uppercased_xxxx_gene_symbol_set = {s.upper() for s in xxxx_gene_symbol_set}\n",
    "\n",
    "    \n",
    "    subset_genes_xxxx_df = mini_xxxx_df.copy()\n",
    "\n",
    "    #Remove alias symbols that are an exact match to their respective primary gene symbol\n",
    "    subset_genes_xxxx_df[\"alias_symbol_upper\"] = subset_genes_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df[\"gene_symbol_upper\"] = subset_genes_xxxx_df[\"gene_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df[subset_genes_xxxx_df['gene_symbol_upper'] != subset_genes_xxxx_df['alias_symbol_upper']]\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.drop([\"alias_symbol_upper\",\"gene_symbol_upper\"], axis=1)\n",
    "\n",
    "    #subset_genes_xxxx_df\n",
    "    ## no aliases that match the primary gene symbol\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv and save\n",
    "    subset_genes_xxxx_df.to_csv(f'../output/subset_genes_{source}_df.csv', index=True) \n",
    "\n",
    "    #Create df with genes that have an alias that can be found as another gene's primary gene symbol\n",
    "    ap_collision_xxxx_df = subset_genes_xxxx_df.copy()\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.dropna(subset=[\"alias_symbol\"])\n",
    "    ap_collision_xxxx_df[\"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "\n",
    "    #makes list of aliases into a set\n",
    "    ap_collision_xxxx_df.loc[:, \"alias_symbol_upper\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(lambda x: {x})\n",
    "\n",
    "    ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol_upper\"].apply(\n",
    "        lambda x: x & uppercased_xxxx_gene_symbol_set\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.drop(\"alias_symbol_upper\", axis=1)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df[ap_collision_xxxx_df[\"collision\"].apply(lambda x: len(x) > 0)]\n",
    "    #changes back the set of aliases to a list, and make the set of collisions to a list\n",
    "    ap_collision_xxxx_df['collision'] = ap_collision_xxxx_df['collision'].apply(\n",
    "        lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.sort_values(\"collision\")\n",
    "\n",
    "    #Add a source tag for future merging efforts\n",
    "    ap_collision_xxxx_df[\"source\"] = str(source.upper())\n",
    "\n",
    "    #ap_collision_xxxx_df\n",
    "    ## only the gene records with an alias that matches another gene record's primary gene symbol\n",
    "    ## each alias on a separate row\n",
    "    #Convert the df into a csv\n",
    "    ap_collision_xxxx_df.to_csv(f'../output/single_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "\n",
    "    #Create a secondary collision df that merges the alias symbols for each record\n",
    "    columns_map = {\n",
    "        'ENSG': ['NCBI_ID', 'HGNC_ID'],\n",
    "        'HGNC': ['NCBI_ID', 'ENSG_ID'],\n",
    "        'NCBI': ['HGNC_ID', 'ENSG_ID']\n",
    "    }\n",
    "\n",
    "    cols_of_interest = columns_map.get(source, [])\n",
    "\n",
    "    merged_alias_ap_collision_xxxx_df = ap_collision_xxxx_df.drop(columns=cols_of_interest + ['alias_symbol'])\n",
    "\n",
    "    merged_alias_ap_collision_xxxx_df = pd.merge(\n",
    "        merged_alias_ap_collision_xxxx_df,\n",
    "        merged_alias_xxxx_df[[f\"{source}_ID\"] + cols_of_interest + ['alias_symbol']],\n",
    "        on=[f\"{source}_ID\"],\n",
    "        how=\"left\"\n",
    "    )\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.drop_duplicates(subset=[f\"{source}_ID\"])\n",
    "\n",
    "    #Test to make sure all collisions are in the alias list for the record\n",
    "    test_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.copy()\n",
    "    test_ap_collision_xxxx_df[\"alias_symbol_upper\"] = test_ap_collision_xxxx_df[\"alias_symbol\"].str.upper()   \n",
    "    test_ap_collision_xxxx_df['alias_symbol_upper'] = test_ap_collision_xxxx_df['alias_symbol_upper'].apply(lambda x: x if isinstance(x, list) else [x])\n",
    "    test_ap_collision_xxxx_df['collision_in_alias'] = test_ap_collision_xxxx_df.apply(lambda row: row['collision'] in test_ap_collision_xxxx_df['alias_symbol_upper'].values, axis=1)\n",
    "    true_count = test_ap_collision_xxxx_df['collision_in_alias'].sum()\n",
    "    test_ap_collision_xxxx_df = test_ap_collision_xxxx_df.drop(\"alias_symbol_upper\", axis=1)\n",
    "\n",
    "    if true_count == 0:\n",
    "        #merged_alias_ap_collision_xxxx_df\n",
    "        ## only the gene records with an alias that matches another gene record's primary gene symbol\n",
    "        ## merging aliases associated with the same record (same gene record identifier) into a list\n",
    "        #Convert the df into a csv and save\n",
    "        merged_alias_ap_collision_xxxx_df.to_csv(f'../output/merged_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "        print(\"All collisions are present in gene alias lists.\")\n",
    "    else:\n",
    "        print(\"Some collisions are not present in gene alias lists.\")\n",
    "\n",
    "    return mini_xxxx_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 633,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133061</th>\n",
       "      <td>ENSG00000289291</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133064</th>\n",
       "      <td>ENSG00000296488</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133065 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049     <NA>   HGNC:7481         MTTF       MT-TF\n",
       "1       ENSG00000210049     <NA>   HGNC:7481         TRNF       MT-TF\n",
       "2       ENSG00000211459     <NA>   HGNC:7470          12S     MT-RNR1\n",
       "3       ENSG00000211459     <NA>   HGNC:7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459     <NA>   HGNC:7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...         ...          ...         ...\n",
       "133060  ENSG00000229388     <NA>  HGNC:52502    LINC01715    TAF12-DT\n",
       "133061  ENSG00000289291     <NA>         NaN          NaN         NaN\n",
       "133062  ENSG00000274978    26824  HGNC:10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824  HGNC:10108          U11       RNU11\n",
       "133064  ENSG00000296488     <NA>         NaN          NaN         NaN\n",
       "\n",
       "[133065 rows x 5 columns]"
      ]
     },
     "execution_count": 633,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../input/ensg_biomart_gene20250625.txt\", sep=\"\\t\",dtype={\"NCBI gene (formerly Entrezgene) ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_ensg_df = mini_ensg_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Gene Synonym\": \"alias_symbol\",\n",
    "        \"Gene name\": \"gene_symbol\",\n",
    "        \"Gene stable ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene (formerly Entrezgene) ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 634,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df = mini_ensg_df.replace(\" \", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"-\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"<NA>\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the Ensembl data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 635,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86364"
      ]
     },
     "execution_count": 635,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_raw_gene_record_count_ensg = len(set(mini_ensg_df['ENSG_ID']))\n",
    "total_raw_gene_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 636,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37963"
      ]
     },
     "execution_count": 636,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_no_primary_symbol_record_count_ensg = (\n",
    "    mini_ensg_df[mini_ensg_df[\"gene_symbol\"].isna()]\n",
    "    .dropna(subset=[\"ENSG_ID\"])\n",
    "    [\"ENSG_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_no_primary_symbol_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 637,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 637,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_loc_record_count_ensg = (\n",
    "    mini_ensg_df[mini_ensg_df[\"gene_symbol\"].str.startswith('LOC', na=False)&\n",
    "    ~mini_ensg_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"ENSG_ID\"])\n",
    "    [\"ENSG_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_loc_record_count_ensg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 638,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0  ENSG00000210049        0     7481         MTTF       MT-TF\n",
       "1  ENSG00000210049        0     7481         TRNF       MT-TF\n",
       "2  ENSG00000211459        0     7470          12S     MT-RNR1\n",
       "3  ENSG00000211459        0     7470       MOTS-C     MT-RNR1\n",
       "4  ENSG00000211459        0     7470       MTRNR1     MT-RNR1"
      ]
     },
     "execution_count": 638,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ensg_df, \"ENSG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 639,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../output/mini_ensg_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 640,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>0</td>\n",
       "      <td>7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028</td>\n",
       "      <td>30062</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028</td>\n",
       "      <td>30062</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>0</td>\n",
       "      <td>52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95005 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049        0     7481         MTTF       MT-TF\n",
       "1       ENSG00000210049        0     7481         TRNF       MT-TF\n",
       "2       ENSG00000211459        0     7470          12S     MT-RNR1\n",
       "3       ENSG00000211459        0     7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459        0     7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989    85028    30062    LINC00100      SNHG12\n",
       "133059  ENSG00000197989    85028    30062     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388        0    52502    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978    26824    10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824    10108          U11       RNU11\n",
       "\n",
       "[95005 rows x 5 columns]"
      ]
     },
     "execution_count": 640,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ensg_df.csv\", index_col=[0])\n",
    "subset_genes_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 641,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24761</th>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>125050</td>\n",
       "      <td>10037</td>\n",
       "      <td>7SK</td>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37365</th>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>6646</td>\n",
       "      <td>11177</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17666</th>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>8435</td>\n",
       "      <td>11178</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35063</th>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>4706</td>\n",
       "      <td>7694</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47647</th>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>0</td>\n",
       "      <td>141</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53226</th>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>7675</td>\n",
       "      <td>12904</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35374</th>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>50862</td>\n",
       "      <td>21159</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96229</th>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>0</td>\n",
       "      <td>14003</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81813</th>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>64397</td>\n",
       "      <td>12886</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14437</th>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>162239</td>\n",
       "      <td>23328</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ZFP1</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>811 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol collision  \\\n",
       "24761  ENSG00000283293   125050    10037          7SK       RN7SK       7SK   \n",
       "37365  ENSG00000057252     6646    11177        ACAT1       SOAT1     ACAT1   \n",
       "17666  ENSG00000167780     8435    11178        ACAT2       SOAT2     ACAT2   \n",
       "35063  ENSG00000004779     4706     7694         ACP1     NDUFAB1      ACP1   \n",
       "47647  ENSG00000220267        0      141       ACTBP2      ACTBP8    ACTBP2   \n",
       "...                ...      ...      ...          ...         ...       ...   \n",
       "53226  ENSG00000197961     7675    12904        ZNF20      ZNF121     ZNF20   \n",
       "35374  ENSG00000110315    50862    21159       ZNF230      RNF141    ZNF230   \n",
       "96229  ENSG00000188801        0    14003       ZNF322    ZNF322P1    ZNF322   \n",
       "81813  ENSG00000103994    64397    12886       ZNF474      ZNF106    ZNF474   \n",
       "14437  ENSG00000184517   162239    23328       ZNF475        ZFP1    ZNF475   \n",
       "\n",
       "      source  \n",
       "24761   ENSG  \n",
       "37365   ENSG  \n",
       "17666   ENSG  \n",
       "35063   ENSG  \n",
       "47647   ENSG  \n",
       "...      ...  \n",
       "53226   ENSG  \n",
       "35374   ENSG  \n",
       "96229   ENSG  \n",
       "81813   ENSG  \n",
       "14437   ENSG  \n",
       "\n",
       "[811 rows x 7 columns]"
      ]
     },
     "execution_count": 641,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/single_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 642,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>125050</td>\n",
       "      <td>10037</td>\n",
       "      <td>7SK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>6646</td>\n",
       "      <td>11177</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>8435</td>\n",
       "      <td>11178</td>\n",
       "      <td>ACAT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>4706</td>\n",
       "      <td>7694</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>0</td>\n",
       "      <td>141</td>\n",
       "      <td>ACTBP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>7675</td>\n",
       "      <td>12904</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>50862</td>\n",
       "      <td>21159</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>0</td>\n",
       "      <td>14003</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>64397</td>\n",
       "      <td>12886</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>ZFP1</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>162239</td>\n",
       "      <td>23328</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>767 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             ENSG_ID gene_symbol collision source NCBI_ID  HGNC_ID  \\\n",
       "0    ENSG00000283293       RN7SK       7SK   ENSG  125050    10037   \n",
       "1    ENSG00000057252       SOAT1     ACAT1   ENSG    6646    11177   \n",
       "2    ENSG00000167780       SOAT2     ACAT2   ENSG    8435    11178   \n",
       "3    ENSG00000004779     NDUFAB1      ACP1   ENSG    4706     7694   \n",
       "4    ENSG00000220267      ACTBP8    ACTBP2   ENSG       0      141   \n",
       "..               ...         ...       ...    ...     ...      ...   \n",
       "806  ENSG00000197961      ZNF121     ZNF20   ENSG    7675    12904   \n",
       "807  ENSG00000110315      RNF141    ZNF230   ENSG   50862    21159   \n",
       "808  ENSG00000188801    ZNF322P1    ZNF322   ENSG       0    14003   \n",
       "809  ENSG00000103994      ZNF106    ZNF474   ENSG   64397    12886   \n",
       "810  ENSG00000184517        ZFP1    ZNF475   ENSG  162239    23328   \n",
       "\n",
       "             alias_symbol  \n",
       "0                     7SK  \n",
       "1    ACAT,ACAT1,SOAT,STAT  \n",
       "2                   ACAT2  \n",
       "3    ACP,ACP1,FASN2A,SDAP  \n",
       "4                  ACTBP2  \n",
       "..                    ...  \n",
       "806   D19S204,ZHC32,ZNF20  \n",
       "807          ZFP26,ZNF230  \n",
       "808        ZNF322,ZNF322B  \n",
       "809  SH3BP3,ZFP106,ZNF474  \n",
       "810       FLJ34243,ZNF475  \n",
       "\n",
       "[767 rows x 7 columns]"
      ]
     },
     "execution_count": 642,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records are in the Ensembl database?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 643,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "48401"
      ]
     },
     "execution_count": 643,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_record_set_ensg = set(mini_ensg_df['ENSG_ID'])\n",
    "gene_record_count_ensg = len(gene_record_set_ensg)\n",
    "gene_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the Ensembl data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 644,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41164"
      ]
     },
     "execution_count": 644,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primary_symbol_set_ensg = set(mini_ensg_df['gene_symbol'])\n",
    "primary_symbol_count_ensg = len(primary_symbol_set_ensg)\n",
    "primary_symbol_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the Ensembl data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 645,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "55413"
      ]
     },
     "execution_count": 645,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alias_symbol_set_ensg = set(mini_ensg_df['alias_symbol'])\n",
    "alias_symbol_count_ensg = len(alias_symbol_set_ensg)\n",
    "alias_symbol_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the Ensembl data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 646,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "95941"
      ]
     },
     "execution_count": 646,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_symbol_count_ensg = pd.concat([mini_ensg_df[\"gene_symbol\"], mini_ensg_df[\"alias_symbol\"]]).dropna().nunique()\n",
    "total_symbol_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 647,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19363"
      ]
     },
     "execution_count": 647,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_record_count_ensg = (\n",
    "    mini_ensg_df[mini_ensg_df[\"alias_symbol\"].isna()]\n",
    "    .dropna(subset=[\"ENSG_ID\"])\n",
    "    [\"ENSG_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "no_alias_symbol_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 648,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "288"
      ]
     },
     "execution_count": 648,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_record_set_ensg = set(\n",
    "    mini_ensg_df[\n",
    "        mini_ensg_df[\"gene_symbol\"].str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "        ~mini_ensg_df[\"gene_symbol\"].str.contains(\"-\", na=False)\n",
    "    ]\n",
    "    .dropna(subset=[\"ENSG_ID\"])[\"ENSG_ID\"]\n",
    ")\n",
    "\n",
    "orf_record_count_ensg = len(orf_record_set_ensg)\n",
    "orf_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 649,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "430"
      ]
     },
     "execution_count": 649,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam_record_set_ensg = set(\n",
    "    mini_ensg_df[\n",
    "        mini_ensg_df[\"gene_symbol\"].str.startswith(\"FAM\", na=False) &\n",
    "        ~mini_ensg_df[\"gene_symbol\"].str.contains(\"-\", na=False)\n",
    "    ]\n",
    "    .dropna(subset=[\"ENSG_ID\"])[\"ENSG_ID\"]\n",
    ")\n",
    "\n",
    "fam_record_count_ensg = len(fam_record_set_ensg)\n",
    "fam_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 650,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33"
      ]
     },
     "execution_count": 650,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kiaa_record_set_ensg = set(\n",
    "    mini_ensg_df[\n",
    "        mini_ensg_df[\"gene_symbol\"].str.startswith(\"KIAA\", na=False) &\n",
    "        ~mini_ensg_df[\"gene_symbol\"].str.contains(\"-\", na=False)\n",
    "    ]\n",
    "    .dropna(subset=[\"ENSG_ID\"])[\"ENSG_ID\"]\n",
    ")\n",
    "\n",
    "kiaa_record_count_ensg = len(kiaa_record_set_ensg)\n",
    "kiaa_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 651,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25007\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92891</th>\n",
       "      <td>ENSG00000138641</td>\n",
       "      <td>8916</td>\n",
       "      <td>4876</td>\n",
       "      <td>KIAA0032</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92878</th>\n",
       "      <td>ENSG00000287542</td>\n",
       "      <td>8916</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HERC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52117</th>\n",
       "      <td>ENSG00000273529</td>\n",
       "      <td>388585</td>\n",
       "      <td>19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14696</th>\n",
       "      <td>ENSG00000197921</td>\n",
       "      <td>388585</td>\n",
       "      <td>19764</td>\n",
       "      <td>BHLHB38</td>\n",
       "      <td>HES5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4416</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4421</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4418</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4417</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43282</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>MART2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43285</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4415</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4414</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4419</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43284</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SIT</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43283</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>RASP</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43281</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>MART-2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43280</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>GUP2</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43279</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>FLJ10724</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43286</th>\n",
       "      <td>ENSG00000054392</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SKN</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4420</th>\n",
       "      <td>ENSG00000280680</td>\n",
       "      <td>55733</td>\n",
       "      <td>18270</td>\n",
       "      <td>SKI</td>\n",
       "      <td>HHAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72820</th>\n",
       "      <td>ENSG00000235657</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68794</th>\n",
       "      <td>ENSG00000224320</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66867</th>\n",
       "      <td>ENSG00000227715</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76339</th>\n",
       "      <td>ENSG00000231834</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101068</th>\n",
       "      <td>ENSG00000206503</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67552</th>\n",
       "      <td>ENSG00000206505</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79893</th>\n",
       "      <td>ENSG00000223980</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66597</th>\n",
       "      <td>ENSG00000229215</td>\n",
       "      <td>3105</td>\n",
       "      <td>4931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79909</th>\n",
       "      <td>ENSG00000224608</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71168</th>\n",
       "      <td>ENSG00000234745</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66216</th>\n",
       "      <td>ENSG00000206450</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70494</th>\n",
       "      <td>ENSG00000232126</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79186</th>\n",
       "      <td>ENSG00000223532</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81469</th>\n",
       "      <td>ENSG00000228964</td>\n",
       "      <td>3106</td>\n",
       "      <td>4932</td>\n",
       "      <td>AS</td>\n",
       "      <td>HLA-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62689</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78388</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78387</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78774</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76516</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78386</th>\n",
       "      <td>ENSG00000228299</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78773</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70213</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80918</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76515</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80917</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81311</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81310</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70211</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76517</th>\n",
       "      <td>ENSG00000206452</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80916</th>\n",
       "      <td>ENSG00000206435</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62691</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>PSORS1</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70212</th>\n",
       "      <td>ENSG00000225691</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62690</th>\n",
       "      <td>ENSG00000204525</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>HLA-JY3</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81309</th>\n",
       "      <td>ENSG00000237022</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78772</th>\n",
       "      <td>ENSG00000233841</td>\n",
       "      <td>3107</td>\n",
       "      <td>4933</td>\n",
       "      <td>D6S204</td>\n",
       "      <td>HLA-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76137</th>\n",
       "      <td>ENSG00000242685</td>\n",
       "      <td>3108</td>\n",
       "      <td>4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75324</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83537</th>\n",
       "      <td>ENSG00000243215</td>\n",
       "      <td>3108</td>\n",
       "      <td>4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74916</th>\n",
       "      <td>ENSG00000243719</td>\n",
       "      <td>3108</td>\n",
       "      <td>4934</td>\n",
       "      <td>D6S222E</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75325</th>\n",
       "      <td>ENSG00000243189</td>\n",
       "      <td>3108</td>\n",
       "      <td>4934</td>\n",
       "      <td>RING6</td>\n",
       "      <td>HLA-DMA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "92891   ENSG00000138641     8916     4876     KIAA0032       HERC3\n",
       "92878   ENSG00000287542     8916        0          NaN       HERC3\n",
       "52117   ENSG00000273529   388585    19764      BHLHB38        HES5\n",
       "14696   ENSG00000197921   388585    19764      BHLHB38        HES5\n",
       "4416    ENSG00000280680    55733    18270       MART-2        HHAT\n",
       "4421    ENSG00000280680    55733    18270          SKN        HHAT\n",
       "4418    ENSG00000280680    55733    18270         RASP        HHAT\n",
       "4417    ENSG00000280680    55733    18270        MART2        HHAT\n",
       "43282   ENSG00000054392    55733    18270        MART2        HHAT\n",
       "43285   ENSG00000054392    55733    18270          SKI        HHAT\n",
       "4415    ENSG00000280680    55733    18270         GUP2        HHAT\n",
       "4414    ENSG00000280680    55733    18270     FLJ10724        HHAT\n",
       "4419    ENSG00000280680    55733    18270          SIT        HHAT\n",
       "43284   ENSG00000054392    55733    18270          SIT        HHAT\n",
       "43283   ENSG00000054392    55733    18270         RASP        HHAT\n",
       "43281   ENSG00000054392    55733    18270       MART-2        HHAT\n",
       "43280   ENSG00000054392    55733    18270         GUP2        HHAT\n",
       "43279   ENSG00000054392    55733    18270     FLJ10724        HHAT\n",
       "43286   ENSG00000054392    55733    18270          SKN        HHAT\n",
       "4420    ENSG00000280680    55733    18270          SKI        HHAT\n",
       "72820   ENSG00000235657     3105     4931          NaN       HLA-A\n",
       "68794   ENSG00000224320     3105     4931          NaN       HLA-A\n",
       "66867   ENSG00000227715     3105     4931          NaN       HLA-A\n",
       "76339   ENSG00000231834     3105     4931          NaN       HLA-A\n",
       "101068  ENSG00000206503     3105     4931          NaN       HLA-A\n",
       "67552   ENSG00000206505     3105     4931          NaN       HLA-A\n",
       "79893   ENSG00000223980     3105     4931          NaN       HLA-A\n",
       "66597   ENSG00000229215     3105     4931          NaN       HLA-A\n",
       "79909   ENSG00000224608     3106     4932           AS       HLA-B\n",
       "71168   ENSG00000234745     3106     4932           AS       HLA-B\n",
       "66216   ENSG00000206450     3106     4932           AS       HLA-B\n",
       "70494   ENSG00000232126     3106     4932           AS       HLA-B\n",
       "79186   ENSG00000223532     3106     4932           AS       HLA-B\n",
       "81469   ENSG00000228964     3106     4932           AS       HLA-B\n",
       "62689   ENSG00000204525     3107     4933       D6S204       HLA-C\n",
       "78388   ENSG00000228299     3107     4933       PSORS1       HLA-C\n",
       "78387   ENSG00000228299     3107     4933      HLA-JY3       HLA-C\n",
       "78774   ENSG00000233841     3107     4933       PSORS1       HLA-C\n",
       "76516   ENSG00000206452     3107     4933      HLA-JY3       HLA-C\n",
       "78386   ENSG00000228299     3107     4933       D6S204       HLA-C\n",
       "78773   ENSG00000233841     3107     4933      HLA-JY3       HLA-C\n",
       "70213   ENSG00000225691     3107     4933       PSORS1       HLA-C\n",
       "80918   ENSG00000206435     3107     4933       PSORS1       HLA-C\n",
       "76515   ENSG00000206452     3107     4933       D6S204       HLA-C\n",
       "80917   ENSG00000206435     3107     4933      HLA-JY3       HLA-C\n",
       "81311   ENSG00000237022     3107     4933       PSORS1       HLA-C\n",
       "81310   ENSG00000237022     3107     4933      HLA-JY3       HLA-C\n",
       "70211   ENSG00000225691     3107     4933       D6S204       HLA-C\n",
       "76517   ENSG00000206452     3107     4933       PSORS1       HLA-C\n",
       "80916   ENSG00000206435     3107     4933       D6S204       HLA-C\n",
       "62691   ENSG00000204525     3107     4933       PSORS1       HLA-C\n",
       "70212   ENSG00000225691     3107     4933      HLA-JY3       HLA-C\n",
       "62690   ENSG00000204525     3107     4933      HLA-JY3       HLA-C\n",
       "81309   ENSG00000237022     3107     4933       D6S204       HLA-C\n",
       "78772   ENSG00000233841     3107     4933       D6S204       HLA-C\n",
       "76137   ENSG00000242685     3108     4934      D6S222E     HLA-DMA\n",
       "75324   ENSG00000243189     3108     4934      D6S222E     HLA-DMA\n",
       "83537   ENSG00000243215     3108     4934      D6S222E     HLA-DMA\n",
       "74916   ENSG00000243719     3108     4934      D6S222E     HLA-DMA\n",
       "75325   ENSG00000243189     3108     4934        RING6     HLA-DMA"
      ]
     },
     "execution_count": 651,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique ENSG ID\n",
    "multi_primary = mini_ensg_df.groupby([\"gene_symbol\"]).ENSG_ID.nunique().gt(1)\n",
    "\n",
    "num_rows = mini_ensg_df[mini_ensg_df.gene_symbol.isin(multi_primary[multi_primary].index)].shape[0]\n",
    "print(num_rows)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_ensg_df.loc[mini_ensg_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').iloc[6000:6060]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many ambiguous symbols result from alias-primary collisions?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 652,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "606"
      ]
     },
     "execution_count": 652,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_collision_ambiguous_symbol_set_ensg = set(merged_alias_ap_collision_ensg_df[\"collision\"])\n",
    "ap_collision_ambiguous_symbol_count_ensg = len(ap_collision_ambiguous_symbol_set_ensg)\n",
    "ap_collision_ambiguous_symbol_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many records have at least one alias-primary collision (alias that matches another record's primary gene symbol)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 653,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "767"
      ]
     },
     "execution_count": 653,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_record_set_ensg = set(merged_alias_ap_collision_ensg_df[\"ENSG_ID\"])\n",
    "ap_record_count_ensg = len(ap_record_set_ensg)\n",
    "ap_record_count_ensg"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Why is the alias-gene collision set not the same length as the set of primary symbols with collisions ?\n",
    "2. Why is the length of the alias-gene collison set shorter?\n",
    " - A gene record with an alias-primary collision has an alias that matches a different gene's primary gene symbol.\n",
    " - Multiple gene records can share a single alias (alias-alias collision)\n",
    " - If that shared alias is an alias-primary collision, then there will be more unique gene symbols in the set of primary symbols with collisions than the set of alias-primary collisions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGNC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 654,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44232</th>\n",
       "      <td>HGNC:9997</td>\n",
       "      <td>RGS16</td>\n",
       "      <td>RGS-r</td>\n",
       "      <td>6004</td>\n",
       "      <td>ENSG00000143333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44233</th>\n",
       "      <td>HGNC:9998</td>\n",
       "      <td>RGS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5997</td>\n",
       "      <td>ENSG00000116741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>C2PA</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>FLJ20370</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>HGNC:9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>PDZ-RGS3</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66207 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0        HGNC:100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0        HGNC:100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1      HGNC:10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2      HGNC:10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3      HGNC:10002        RGS6          NaN     9628  ENSG00000182732\n",
       "...           ...         ...          ...      ...              ...\n",
       "44232   HGNC:9997       RGS16        RGS-r     6004  ENSG00000143333\n",
       "44233   HGNC:9998        RGS2          NaN     5997  ENSG00000116741\n",
       "44234   HGNC:9999        RGS3         C2PA     5998  ENSG00000138835\n",
       "44234   HGNC:9999        RGS3     FLJ20370     5998  ENSG00000138835\n",
       "44234   HGNC:9999        RGS3     PDZ-RGS3     5998  ENSG00000138835\n",
       "\n",
       "[66207 rows x 5 columns]"
      ]
     },
     "execution_count": 654,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_file_path = \"../input/hgnc_biomart_gene20250625.txt\"\n",
    "\n",
    "mini_hgnc_df = pd.read_csv(\n",
    "    hgnc_file_path, sep=\"\\t\"\n",
    ")\n",
    "\n",
    "# Rename columns\n",
    "mini_hgnc_df = mini_hgnc_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Approved symbol\": \"gene_symbol\",\n",
    "        \"Ensembl gene ID\": \"ENSG_ID\",\n",
    "    }\n",
    ")\n",
    "\n",
    "#structure and labeling in HGNC download files changed between 2024 amd 2025\n",
    "if \"Alias symbol\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbol\": \"alias_symbol\"})\n",
    "elif \"Alias symbols\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbols\": \"alias_symbol\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"alias_symbol\"] = pd.NA\n",
    "\n",
    "if \"NCBI gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI gene ID\": \"NCBI_ID\"})\n",
    "elif \"NCBI Gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI Gene ID\": \"NCBI_ID\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"NCBI_ID\"] = pd.NA   \n",
    "\n",
    "mini_hgnc_df[\"NCBI_ID\"] = mini_hgnc_df[\"NCBI_ID\"].astype(pd.Int64Dtype())\n",
    "\n",
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', hgnc_file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_hgnc_df['alias_symbol'] = (\n",
    "        mini_hgnc_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split(',')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_hgnc_df = mini_hgnc_df.explode('alias_symbol')\n",
    "\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 655,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.replace(\" \", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"\", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"-\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the HGNC data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 656,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 656,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_raw_gene_record_set_hgnc = set(mini_hgnc_df['HGNC_ID'])\n",
    "total_raw_gene_record_count_hgnc = len(total_raw_gene_record_set_hgnc)\n",
    "total_raw_gene_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 657,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 657,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_no_primary_symbol_record_count_hgnc = (\n",
    "    mini_hgnc_df[mini_hgnc_df[\"gene_symbol\"].isna()]\n",
    "    .dropna(subset=[\"HGNC_ID\"])\n",
    "    [\"HGNC_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_no_primary_symbol_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 658,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 658,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_loc_record_count_hgnc = (\n",
    "    mini_hgnc_df[mini_hgnc_df[\"gene_symbol\"].str.startswith('LOC', na=False)&\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"HGNC_ID\"])\n",
    "    [\"HGNC_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_loc_record_count_hgnc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 659,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0      100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0      100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1    10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2    10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3    10002        RGS6          NaN     9628  ENSG00000182732"
      ]
     },
     "execution_count": 659,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_hgnc_df, \"HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 660,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_hgnc_df = pd.read_csv(\n",
    "    \"../output/mini_hgnc_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 661,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>BNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>ASIC1</td>\n",
       "      <td>hBNaC2</td>\n",
       "      <td>41</td>\n",
       "      <td>ENSG00000110881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10000</td>\n",
       "      <td>RGS4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5999</td>\n",
       "      <td>ENSG00000117152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10001</td>\n",
       "      <td>RGS5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8490</td>\n",
       "      <td>ENSG00000143248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10002</td>\n",
       "      <td>RGS6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9628</td>\n",
       "      <td>ENSG00000182732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44232</th>\n",
       "      <td>9997</td>\n",
       "      <td>RGS16</td>\n",
       "      <td>RGS-r</td>\n",
       "      <td>6004</td>\n",
       "      <td>ENSG00000143333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44233</th>\n",
       "      <td>9998</td>\n",
       "      <td>RGS2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5997</td>\n",
       "      <td>ENSG00000116741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>C2PA</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>FLJ20370</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44234</th>\n",
       "      <td>9999</td>\n",
       "      <td>RGS3</td>\n",
       "      <td>PDZ-RGS3</td>\n",
       "      <td>5998</td>\n",
       "      <td>ENSG00000138835</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66192 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID gene_symbol alias_symbol  NCBI_ID          ENSG_ID\n",
       "0          100       ASIC1        BNaC2       41  ENSG00000110881\n",
       "0          100       ASIC1       hBNaC2       41  ENSG00000110881\n",
       "1        10000        RGS4          NaN     5999  ENSG00000117152\n",
       "2        10001        RGS5          NaN     8490  ENSG00000143248\n",
       "3        10002        RGS6          NaN     9628  ENSG00000182732\n",
       "...        ...         ...          ...      ...              ...\n",
       "44232     9997       RGS16        RGS-r     6004  ENSG00000143333\n",
       "44233     9998        RGS2          NaN     5997  ENSG00000116741\n",
       "44234     9999        RGS3         C2PA     5998  ENSG00000138835\n",
       "44234     9999        RGS3     FLJ20370     5998  ENSG00000138835\n",
       "44234     9999        RGS3     PDZ-RGS3     5998  ENSG00000138835\n",
       "\n",
       "[66192 rows x 5 columns]"
      ]
     },
     "execution_count": 661,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df = pd.read_csv(\n",
    "    \"../output/subset_genes_hgnc_df.csv\", index_col=[0])\n",
    "subset_genes_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 662,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14947</td>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>54776</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>AAVS1,DKFZP434D0412,MBS85,p84,p85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11177</td>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>6646</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11178</td>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>8435</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>ACAT2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1612</td>\n",
       "      <td>CCRL2</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>9034</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>ACKR5,CKRX,CRAM-A,CRAM-B,HCR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7694</td>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>4706</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>556</th>\n",
       "      <td>3147</td>\n",
       "      <td>ECEL1</td>\n",
       "      <td>XCE</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>9427</td>\n",
       "      <td>ENSG00000171551</td>\n",
       "      <td>DINE,XCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>12904</td>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>7675</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>ZHC32,ZNF20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>21159</td>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>50862</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>12886</td>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>64397</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>SH3BP3,ZNF474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>560</th>\n",
       "      <td>23328</td>\n",
       "      <td>ZFP1</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>162239</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>547 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     HGNC_ID gene_symbol collision source  NCBI_ID          ENSG_ID  \\\n",
       "0      14947    PPP1R12C     AAVS1   HGNC    54776  ENSG00000125503   \n",
       "1      11177       SOAT1     ACAT1   HGNC     6646  ENSG00000057252   \n",
       "2      11178       SOAT2     ACAT2   HGNC     8435  ENSG00000167780   \n",
       "3       1612       CCRL2     ACKR5   HGNC     9034  ENSG00000121797   \n",
       "4       7694     NDUFAB1      ACP1   HGNC     4706  ENSG00000004779   \n",
       "..       ...         ...       ...    ...      ...              ...   \n",
       "556     3147       ECEL1       XCE   HGNC     9427  ENSG00000171551   \n",
       "557    12904      ZNF121     ZNF20   HGNC     7675  ENSG00000197961   \n",
       "558    21159      RNF141    ZNF230   HGNC    50862  ENSG00000110315   \n",
       "559    12886      ZNF106    ZNF474   HGNC    64397  ENSG00000103994   \n",
       "560    23328        ZFP1    ZNF475   HGNC   162239  ENSG00000184517   \n",
       "\n",
       "                          alias_symbol  \n",
       "0    AAVS1,DKFZP434D0412,MBS85,p84,p85  \n",
       "1                           ACAT,ACAT1  \n",
       "2                                ACAT2  \n",
       "3         ACKR5,CKRX,CRAM-A,CRAM-B,HCR  \n",
       "4                 ACP,ACP1,FASN2A,SDAP  \n",
       "..                                 ...  \n",
       "556                           DINE,XCE  \n",
       "557                        ZHC32,ZNF20  \n",
       "558                       ZFP26,ZNF230  \n",
       "559                      SH3BP3,ZNF474  \n",
       "560                    FLJ34243,ZNF475  \n",
       "\n",
       "[547 rows x 7 columns]"
      ]
     },
     "execution_count": 662,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_hgnc_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_hgnc_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records are in the HGNC database?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 663,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 663,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_record_set_hgnc = set(mini_hgnc_df['HGNC_ID'])\n",
    "gene_record_count_hgnc = len(gene_record_set_hgnc)\n",
    "gene_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the HGNC data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 664,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44235"
      ]
     },
     "execution_count": 664,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primary_symbol_set_hgnc = set(mini_hgnc_df['gene_symbol'])\n",
    "primary_symbol_count_hgnc = len(primary_symbol_set_hgnc)\n",
    "primary_symbol_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the HGNC data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 665,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43075"
      ]
     },
     "execution_count": 665,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alias_symbol_set_hgnc = set(mini_hgnc_df['alias_symbol'])\n",
    "alias_symbol_count_hgnc = len(alias_symbol_set_hgnc)\n",
    "alias_symbol_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the HGNC data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 666,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86842"
      ]
     },
     "execution_count": 666,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_symbol_count_hgnc = len(set(mini_hgnc_df['gene_symbol']) | set(mini_hgnc_df['alias_symbol']))\n",
    "total_symbol_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 667,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21768"
      ]
     },
     "execution_count": 667,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_record_count_hgnc = (\n",
    "    mini_hgnc_df[mini_hgnc_df[\"alias_symbol\"].isna()]\n",
    "    .dropna(subset=[\"HGNC_ID\"])\n",
    "    [\"HGNC_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "no_alias_symbol_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 668,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "246"
      ]
     },
     "execution_count": 668,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_record_set_hgnc = set(\n",
    "    mini_hgnc_df[\n",
    "        mini_hgnc_df[\"gene_symbol\"].str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "        ~mini_hgnc_df[\"gene_symbol\"].str.contains(\"-\", na=False)\n",
    "    ]\n",
    "    .dropna(subset=[\"HGNC_ID\"])[\"HGNC_ID\"]\n",
    ")\n",
    "\n",
    "orf_record_count_hgnc = len(orf_record_set_hgnc)\n",
    "orf_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 669,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "371"
      ]
     },
     "execution_count": 669,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam_record_set_hgnc = set(\n",
    "    mini_hgnc_df[mini_hgnc_df[\"gene_symbol\"].str.startswith('FAM', na=False)&\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"HGNC_ID\"])\n",
    "    [\"HGNC_ID\"]\n",
    ")\n",
    "fam_record_count_hgnc = len(fam_record_set_hgnc)\n",
    "fam_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 670,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32"
      ]
     },
     "execution_count": 670,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kiaa_record_set_hgnc = set(\n",
    "    mini_hgnc_df[mini_hgnc_df[\"gene_symbol\"].str.startswith('KIAA', na=False)&\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"HGNC_ID\"])\n",
    "    [\"HGNC_ID\"]\n",
    ")\n",
    "kiaa_record_count_hgnc = len(kiaa_record_set_hgnc)\n",
    "kiaa_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 671,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [HGNC_ID, gene_symbol, alias_symbol, NCBI_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 671,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_hgnc_df.groupby([\"gene_symbol\"]).HGNC_ID.nunique().gt(1)\n",
    "\n",
    "num_rows = mini_hgnc_df[mini_hgnc_df.gene_symbol.isin(multi_primary[multi_primary].index)].shape[0]\n",
    "print(num_rows)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_country`:\n",
    "multi_record_same_symbol_df = mini_hgnc_df.loc[mini_hgnc_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol')\n",
    "multi_record_same_symbol_df.head(60)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many ambiguous symbols result from alias-primary collisions?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "492"
      ]
     },
     "execution_count": 672,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_collision_ambiguous_symbol_set_hgnc = set(\n",
    "    merged_alias_ap_collision_hgnc_df[\"collision\"]\n",
    ")\n",
    "ap_collision_ambiguous_symbol_count_hgnc = len(ap_collision_ambiguous_symbol_set_hgnc)\n",
    "ap_collision_ambiguous_symbol_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many records have at least one alias-primary collision (alias that matches another record's primary gene symbol)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 673,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "547"
      ]
     },
     "execution_count": 673,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_record_set_hgnc = set(merged_alias_ap_collision_hgnc_df[\"HGNC_ID\"])\n",
    "ap_record_count_hgnc = len(ap_record_set_hgnc)\n",
    "ap_record_count_hgnc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NCBI Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 674,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_file_path = \"../input/Homo_sapiens.gene_info20250625\"\n",
    "\n",
    "mini_ncbi_df = pd.read_csv(ncbi_file_path, sep=\"\\t\")\n",
    "# Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol\n",
    "mini_ncbi_df = mini_ncbi_df[\n",
    "[\"GeneID\", \"Symbol\", \"Synonyms\", \"dbXrefs\"]\n",
    "]\n",
    "mini_ncbi_df = mini_ncbi_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Synonyms\": \"alias_symbol\"})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split dbXrefs into individual columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 675,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.assign(\n",
    "    MIM=np.nan,\n",
    "    HGNC_ID=np.nan,\n",
    "    ENSG_ID=np.nan,\n",
    "    AllianceGenome=np.nan,\n",
    "    MIRbase=np.nan,\n",
    "    IMGTgene_db=np.nan,\n",
    "    dash=np.nan,\n",
    "    unknown=np.nan,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 676,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:11: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '138670' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:14: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:14: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:17: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:17: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ensg00000121410' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:20: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:20: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'hgnc:5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:26: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:26: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'cd1d' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:28: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:28: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '-' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:23: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_27750/2507567125.py:23: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'mi0000060' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    }
   ],
   "source": [
    "index_pos = 0\n",
    "\n",
    "print(len(mini_ncbi_df))\n",
    "while index_pos < len(mini_ncbi_df):\n",
    "    xrefs = mini_ncbi_df[\"dbXrefs\"][index_pos].split(\"|\")\n",
    "\n",
    "    for xref in xrefs:\n",
    "        xref = xref.lower()\n",
    "        if xref.startswith(\"mim:\"):\n",
    "            xref = xref.replace(\"mim:\", \"\")\n",
    "            mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
    "        elif xref.startswith(\"hgnc:hgnc:\"):\n",
    "            xref = xref.replace(\"hgnc:hgnc:\", \"\")\n",
    "            mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"ensembl:\"):\n",
    "            xref = xref.replace(\"ensembl:\", \"\")\n",
    "            mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"alliancegenome:\"):\n",
    "            xref = xref.replace(\"alliancegenome:\", \"\")\n",
    "            mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
    "        elif xref.startswith(\"mirbase\"):\n",
    "            xref = xref.replace(\"mirbase:\", \"\")\n",
    "            mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
    "        elif xref.startswith(\"imgt/gene-db:\"):\n",
    "            xref = xref.replace(\"imgt/gene-db:\", \"\")\n",
    "            mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
    "        elif xref.startswith(\"-\"):\n",
    "            mini_ncbi_df[\"dash\"][index_pos] = xref\n",
    "        else:\n",
    "            mini_ncbi_df[\"unknown\"][index_pos] = xref\n",
    "\n",
    "    index_pos += 1\n",
    "    pass\n",
    "\n",
    "print(index_pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 677,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df[\"ENSG_ID\"] = mini_ncbi_df[\"ENSG_ID\"].str.replace(\"ensg\", \"ENSG\", 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 678,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.drop(\n",
    "    [\n",
    "        \"AllianceGenome\",\n",
    "        \"MIRbase\",\n",
    "        \"IMGTgene_db\",\n",
    "        \"dash\",\n",
    "        \"unknown\",\n",
    "        \"dbXrefs\",\n",
    "        \"MIM\",\n",
    "    ],\n",
    "    axis=1,\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "need to remove phenotype records. This file (ncbi_records_w_property_pheno_only_20250630.txt) was generated through the NCBI Gene web page by searching for these parameters \"phenotype only [Properties] AND \"Homo sapiens\"[porgn:__txid9606]\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 679,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_pheno_records_df = pd.read_csv(\"../input/ncbi_records_w_property_pheno_only_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_pheno_records_df = ncbi_pheno_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 680,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_pheno_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 681,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>239237 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B       5  ENSG00000121410\n",
       "0             1        A1BG          ABG       5  ENSG00000121410\n",
       "0             1        A1BG          GAB       5  ENSG00000121410\n",
       "0             1        A1BG     HYST2477       5  ENSG00000121410\n",
       "1             2         A2M         A2MD       7  ENSG00000175899\n",
       "...         ...         ...          ...     ...              ...\n",
       "193575  8923215        trnD            -     NaN              NaN\n",
       "193576  8923216        trnP            -     NaN              NaN\n",
       "193577  8923217        trnA            -     NaN              NaN\n",
       "193578  8923218        COX1            -     NaN              NaN\n",
       "193579  8923219    16S rRNA            -     NaN              NaN\n",
       "\n",
       "[239237 rows x 5 columns]"
      ]
     },
     "execution_count": 681,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', ncbi_file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_ncbi_df['alias_symbol'] = (\n",
    "        mini_ncbi_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split('|')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_ncbi_df = mini_ncbi_df.explode('alias_symbol')\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 682,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.replace(\" \", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"\", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"-\", np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records for the NCBI data set before any kind of cleaning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 683,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "192410"
      ]
     },
     "execution_count": 683,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_raw_gene_record_count_ncbi = mini_ncbi_df['NCBI_ID'].dropna().nunique()\n",
    "total_raw_gene_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no primary gene symbol?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 684,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 684,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_no_primary_symbol_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].isna()]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_no_primary_symbol_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a LOC placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 685,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147698"
      ]
     },
     "execution_count": 685,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_loc_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('LOC', na=False)&\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "raw_loc_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop sub species records: \"homo sapiens ssp denisova\"[Organism] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 686,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_denisova_records_df = pd.read_csv(\"../input/ncbi_denisova_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_denisova_records_df = ncbi_denisova_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 687,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_denisova_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop sub species records: \"homo sapiens neanderthalensis\"[Organism] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 688,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_neanderthalensis_records_df = pd.read_csv(\"../input/ncbi_neanderthalensis_20250630.txt\", sep=\"\\t\")\n",
    "\n",
    "ncbi_neanderthalensis_records_df = ncbi_neanderthalensis_records_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Aliases\": \"alias_symbol\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 689,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df[~mini_ncbi_df['NCBI_ID'].isin(ncbi_neanderthalensis_records_df['NCBI_ID'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 690,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0        1        A1BG          A1B        5  ENSG00000121410\n",
       "0        1        A1BG          ABG        5  ENSG00000121410\n",
       "0        1        A1BG          GAB        5  ENSG00000121410\n",
       "0        1        A1BG     HYST2477        5  ENSG00000121410\n",
       "1        2         A2M         A2MD        7  ENSG00000175899"
      ]
     },
     "execution_count": 690,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ncbi_df, \"NCBI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 691,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = pd.read_csv(\n",
    "    \"../output/mini_ncbi_df.csv\", index_col=[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 692,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193502</th>\n",
       "      <td>141732005</td>\n",
       "      <td>ADCY2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40064</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193503</th>\n",
       "      <td>141732006</td>\n",
       "      <td>NSG2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41074</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193504</th>\n",
       "      <td>141732007</td>\n",
       "      <td>ST18-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193505</th>\n",
       "      <td>141732008</td>\n",
       "      <td>MICAL2-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>58437</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193506</th>\n",
       "      <td>141732009</td>\n",
       "      <td>TRIM41-AS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40863</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>91377 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0               1        A1BG          A1B        5  ENSG00000121410\n",
       "0               1        A1BG          ABG        5  ENSG00000121410\n",
       "0               1        A1BG          GAB        5  ENSG00000121410\n",
       "0               1        A1BG     HYST2477        5  ENSG00000121410\n",
       "1               2         A2M         A2MD        7  ENSG00000175899\n",
       "...           ...         ...          ...      ...              ...\n",
       "193502  141732005   ADCY2-AS1          NaN    40064              NaN\n",
       "193503  141732006    NSG2-AS1          NaN    41074              NaN\n",
       "193504  141732007    ST18-AS1          NaN    58430              NaN\n",
       "193505  141732008  MICAL2-AS1          NaN    58437              NaN\n",
       "193506  141732009  TRIM41-AS1          NaN    40863              NaN\n",
       "\n",
       "[91377 rows x 5 columns]"
      ]
     },
     "execution_count": 692,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ncbi_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ncbi_df.csv\", index_col=[0])\n",
    "subset_genes_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 693,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3494</td>\n",
       "      <td>IGHA2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>5479</td>\n",
       "      <td>ENSG00000211890</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>54776</td>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>14947</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>AAVS1,LENG3,MBS85,p84,p85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6646</td>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>11177</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>ACACT,ACAT,ACAT-1,ACAT1,SOAT,STAT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8435</td>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>11178</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>ACACT2,ACAT2,ARGP2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9034</td>\n",
       "      <td>CCRL2</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>1612</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>ACKR5,CKRX,CRAM,CRAM-A,CRAM-B,HCR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2032</th>\n",
       "      <td>29903</td>\n",
       "      <td>CCDC106</td>\n",
       "      <td>ZNF581</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>30181</td>\n",
       "      <td>ENSG00000173581</td>\n",
       "      <td>HSU79303,ZNF581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2033</th>\n",
       "      <td>146540</td>\n",
       "      <td>ZNF785</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>26496</td>\n",
       "      <td>ENSG00000197162</td>\n",
       "      <td>ZNF688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2034</th>\n",
       "      <td>57829</td>\n",
       "      <td>ZP4</td>\n",
       "      <td>ZP1</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>15770</td>\n",
       "      <td>ENSG00000116996</td>\n",
       "      <td>ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2036</th>\n",
       "      <td>55663</td>\n",
       "      <td>ZNF446</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>21036</td>\n",
       "      <td>ENSG00000083838</td>\n",
       "      <td>ZKSCAN20,ZSCAN30,ZSCAN52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2037</th>\n",
       "      <td>342357</td>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>25677</td>\n",
       "      <td>ENSG00000155592</td>\n",
       "      <td>ZNF694,ZSCAN31,ZSCAN34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1614 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      NCBI_ID gene_symbol collision source  HGNC_ID          ENSG_ID  \\\n",
       "0        3494       IGHA2       A2M   NCBI     5479  ENSG00000211890   \n",
       "1       54776    PPP1R12C     AAVS1   NCBI    14947  ENSG00000125503   \n",
       "2        6646       SOAT1     ACAT1   NCBI    11177  ENSG00000057252   \n",
       "3        8435       SOAT2     ACAT2   NCBI    11178  ENSG00000167780   \n",
       "4        9034       CCRL2     ACKR5   NCBI     1612  ENSG00000121797   \n",
       "...       ...         ...       ...    ...      ...              ...   \n",
       "2032    29903     CCDC106    ZNF581   NCBI    30181  ENSG00000173581   \n",
       "2033   146540      ZNF785    ZNF688   NCBI    26496  ENSG00000197162   \n",
       "2034    57829         ZP4       ZP1   NCBI    15770  ENSG00000116996   \n",
       "2036    55663      ZNF446   ZSCAN30   NCBI    21036  ENSG00000083838   \n",
       "2037   342357     ZKSCAN2   ZSCAN31   NCBI    25677  ENSG00000155592   \n",
       "\n",
       "                           alias_symbol  \n",
       "0                                   A2M  \n",
       "1             AAVS1,LENG3,MBS85,p84,p85  \n",
       "2     ACACT,ACAT,ACAT-1,ACAT1,SOAT,STAT  \n",
       "3                    ACACT2,ACAT2,ARGP2  \n",
       "4     ACKR5,CKRX,CRAM,CRAM-A,CRAM-B,HCR  \n",
       "...                                 ...  \n",
       "2032                    HSU79303,ZNF581  \n",
       "2033                             ZNF688  \n",
       "2034         ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4  \n",
       "2036           ZKSCAN20,ZSCAN30,ZSCAN52  \n",
       "2037             ZNF694,ZSCAN31,ZSCAN34  \n",
       "\n",
       "[1614 rows x 7 columns]"
      ]
     },
     "execution_count": 693,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ncbi_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ncbi_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records are in the NCBI database?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 694,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44639"
      ]
     },
     "execution_count": 694,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_record_set_ncbi = set(mini_ncbi_df['NCBI_ID'])\n",
    "gene_record_count_ncbi = len(gene_record_set_ncbi)\n",
    "gene_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique primary gene symbols are in the NCBI data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 695,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44541"
      ]
     },
     "execution_count": 695,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primary_symbol_set_ncbi = set(mini_ncbi_df['gene_symbol'])\n",
    "primary_symbol_count_ncbi = len(primary_symbol_set_ncbi)\n",
    "primary_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique alias gene symbols are in the NCBI data set?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 696,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "69227"
      ]
     },
     "execution_count": 696,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alias_symbol_set_ncbi = set(mini_ncbi_df['alias_symbol'])\n",
    "alias_symbol_count_ncbi = len(alias_symbol_set_ncbi)\n",
    "alias_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many unique gene symbols in total are there in the NCBI data set??"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 697,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "112294"
      ]
     },
     "execution_count": 697,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_symbol_count_ncbi = pd.concat([mini_ncbi_df[\"gene_symbol\"], mini_ncbi_df[\"alias_symbol\"]]).dropna().nunique()\n",
    "total_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have no alias symbols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 698,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17380"
      ]
     },
     "execution_count": 698,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_record_count_ncbi = (\n",
    "    mini_ncbi_df[mini_ncbi_df[\"alias_symbol\"].isna()]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    "    .nunique()\n",
    ")\n",
    "no_alias_symbol_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records have a primary gene symbol that is a C#orf?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 699,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5"
      ]
     },
     "execution_count": 699,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_record_set_ncbi = set(mini_ncbi_df[\n",
    "    mini_ncbi_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "])\n",
    "orf_record_count_ncbi = len(orf_record_set_ncbi)\n",
    "orf_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a FAM placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 700,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "371"
      ]
     },
     "execution_count": 700,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fam_record_set_ncbi = set(\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('FAM', na=False)&\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    ")\n",
    "fam_record_count_ncbi = len(fam_record_set_ncbi)\n",
    "fam_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "how many gene records that have a primary gene symbol that is a KIAA placeholder?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 701,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33"
      ]
     },
     "execution_count": 701,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kiaa_record_set_ncbi = set(\n",
    "    mini_ncbi_df[mini_ncbi_df[\"gene_symbol\"].str.startswith('KIAA', na=False)&\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)]\n",
    "    .dropna(subset=[\"NCBI_ID\"])\n",
    "    [\"NCBI_ID\"]\n",
    ")\n",
    "kiaa_record_count_ncbi = len(kiaa_record_set_ncbi)\n",
    "kiaa_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "which gene records share a primary symbol with other gene records?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 702,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "110\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3550</th>\n",
       "      <td>4549</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4762</th>\n",
       "      <td>6052</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10082</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3551</th>\n",
       "      <td>4550</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>MTRNR2</td>\n",
       "      <td>7471</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>6053</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10083</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65892</th>\n",
       "      <td>124901565</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65891</th>\n",
       "      <td>124901564</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65890</th>\n",
       "      <td>124901563</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65889</th>\n",
       "      <td>124901562</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70782</th>\n",
       "      <td>124906586</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65888</th>\n",
       "      <td>124901561</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70114</th>\n",
       "      <td>124905878</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70102</th>\n",
       "      <td>124905866</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70099</th>\n",
       "      <td>124905863</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70108</th>\n",
       "      <td>124905872</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70096</th>\n",
       "      <td>124905860</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70093</th>\n",
       "      <td>124905857</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70111</th>\n",
       "      <td>124905875</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70090</th>\n",
       "      <td>124905854</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70117</th>\n",
       "      <td>124905881</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70120</th>\n",
       "      <td>124905884</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70123</th>\n",
       "      <td>124905887</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70105</th>\n",
       "      <td>124905869</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70126</th>\n",
       "      <td>124905890</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70138</th>\n",
       "      <td>124905902</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70132</th>\n",
       "      <td>124905896</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70135</th>\n",
       "      <td>124905899</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70129</th>\n",
       "      <td>124905893</td>\n",
       "      <td>TRNAD-GUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70103</th>\n",
       "      <td>124905867</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70106</th>\n",
       "      <td>124905870</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70100</th>\n",
       "      <td>124905864</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70139</th>\n",
       "      <td>124905903</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70121</th>\n",
       "      <td>124905885</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70097</th>\n",
       "      <td>124905861</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70136</th>\n",
       "      <td>124905900</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70109</th>\n",
       "      <td>124905873</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70124</th>\n",
       "      <td>124905888</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70133</th>\n",
       "      <td>124905897</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70091</th>\n",
       "      <td>124905855</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70127</th>\n",
       "      <td>124905891</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70112</th>\n",
       "      <td>124905876</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70115</th>\n",
       "      <td>124905879</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70130</th>\n",
       "      <td>124905894</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70094</th>\n",
       "      <td>124905858</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70118</th>\n",
       "      <td>124905882</td>\n",
       "      <td>TRNAE-CUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70144</th>\n",
       "      <td>124905908</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54286</th>\n",
       "      <td>107987368</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69857</th>\n",
       "      <td>124905580</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69860</th>\n",
       "      <td>124905583</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69861</th>\n",
       "      <td>124905584</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69863</th>\n",
       "      <td>124905586</td>\n",
       "      <td>TRNAE-UUC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69858</th>\n",
       "      <td>124905581</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69855</th>\n",
       "      <td>124905578</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69865</th>\n",
       "      <td>124905588</td>\n",
       "      <td>TRNAG-CCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70143</th>\n",
       "      <td>124905907</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70146</th>\n",
       "      <td>124905910</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70083</th>\n",
       "      <td>124905847</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70148</th>\n",
       "      <td>124905912</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70150</th>\n",
       "      <td>124905914</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70085</th>\n",
       "      <td>124905849</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70152</th>\n",
       "      <td>124905916</td>\n",
       "      <td>TRNAG-GCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         NCBI_ID gene_symbol alias_symbol  HGNC_ID ENSG_ID\n",
       "3550        4549        RNR1       MTRNR1     7470     NaN\n",
       "4762        6052        RNR1          NaN    10082     NaN\n",
       "3551        4550        RNR2       MTRNR2     7471     NaN\n",
       "4763        6053        RNR2          NaN    10083     NaN\n",
       "65892  124901565   TRNAA-AGC          NaN        0     NaN\n",
       "65891  124901564   TRNAA-AGC          NaN        0     NaN\n",
       "65890  124901563   TRNAA-AGC          NaN        0     NaN\n",
       "65889  124901562   TRNAA-AGC          NaN        0     NaN\n",
       "70782  124906586   TRNAA-AGC          NaN        0     NaN\n",
       "65888  124901561   TRNAA-AGC          NaN        0     NaN\n",
       "70114  124905878   TRNAD-GUC          NaN        0     NaN\n",
       "70102  124905866   TRNAD-GUC          NaN        0     NaN\n",
       "70099  124905863   TRNAD-GUC          NaN        0     NaN\n",
       "70108  124905872   TRNAD-GUC          NaN        0     NaN\n",
       "70096  124905860   TRNAD-GUC          NaN        0     NaN\n",
       "70093  124905857   TRNAD-GUC          NaN        0     NaN\n",
       "70111  124905875   TRNAD-GUC          NaN        0     NaN\n",
       "70090  124905854   TRNAD-GUC          NaN        0     NaN\n",
       "70117  124905881   TRNAD-GUC          NaN        0     NaN\n",
       "70120  124905884   TRNAD-GUC          NaN        0     NaN\n",
       "70123  124905887   TRNAD-GUC          NaN        0     NaN\n",
       "70105  124905869   TRNAD-GUC          NaN        0     NaN\n",
       "70126  124905890   TRNAD-GUC          NaN        0     NaN\n",
       "70138  124905902   TRNAD-GUC          NaN        0     NaN\n",
       "70132  124905896   TRNAD-GUC          NaN        0     NaN\n",
       "70135  124905899   TRNAD-GUC          NaN        0     NaN\n",
       "70129  124905893   TRNAD-GUC          NaN        0     NaN\n",
       "70103  124905867   TRNAE-CUC          NaN        0     NaN\n",
       "70106  124905870   TRNAE-CUC          NaN        0     NaN\n",
       "70100  124905864   TRNAE-CUC          NaN        0     NaN\n",
       "70139  124905903   TRNAE-CUC          NaN        0     NaN\n",
       "70121  124905885   TRNAE-CUC          NaN        0     NaN\n",
       "70097  124905861   TRNAE-CUC          NaN        0     NaN\n",
       "70136  124905900   TRNAE-CUC          NaN        0     NaN\n",
       "70109  124905873   TRNAE-CUC          NaN        0     NaN\n",
       "70124  124905888   TRNAE-CUC          NaN        0     NaN\n",
       "70133  124905897   TRNAE-CUC          NaN        0     NaN\n",
       "70091  124905855   TRNAE-CUC          NaN        0     NaN\n",
       "70127  124905891   TRNAE-CUC          NaN        0     NaN\n",
       "70112  124905876   TRNAE-CUC          NaN        0     NaN\n",
       "70115  124905879   TRNAE-CUC          NaN        0     NaN\n",
       "70130  124905894   TRNAE-CUC          NaN        0     NaN\n",
       "70094  124905858   TRNAE-CUC          NaN        0     NaN\n",
       "70118  124905882   TRNAE-CUC          NaN        0     NaN\n",
       "70144  124905908   TRNAE-UUC          NaN        0     NaN\n",
       "54286  107987368   TRNAE-UUC          NaN        0     NaN\n",
       "69857  124905580   TRNAE-UUC          NaN        0     NaN\n",
       "69860  124905583   TRNAE-UUC          NaN        0     NaN\n",
       "69861  124905584   TRNAE-UUC          NaN        0     NaN\n",
       "69863  124905586   TRNAE-UUC          NaN        0     NaN\n",
       "69858  124905581   TRNAG-CCC          NaN        0     NaN\n",
       "69855  124905578   TRNAG-CCC          NaN        0     NaN\n",
       "69865  124905588   TRNAG-CCC          NaN        0     NaN\n",
       "70143  124905907   TRNAG-GCC          NaN        0     NaN\n",
       "70146  124905910   TRNAG-GCC          NaN        0     NaN\n",
       "70083  124905847   TRNAG-GCC          NaN        0     NaN\n",
       "70148  124905912   TRNAG-GCC          NaN        0     NaN\n",
       "70150  124905914   TRNAG-GCC          NaN        0     NaN\n",
       "70085  124905849   TRNAG-GCC          NaN        0     NaN\n",
       "70152  124905916   TRNAG-GCC          NaN        0     NaN"
      ]
     },
     "execution_count": 702,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_ncbi_df.groupby([\"gene_symbol\"]).NCBI_ID.nunique().gt(1)\n",
    "\n",
    "num_rows = mini_ncbi_df[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].shape[0]\n",
    "print(num_rows)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_country`:\n",
    "multi_record_same_symbol_df = mini_ncbi_df.loc[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol')\n",
    "multi_record_same_symbol_df.head(60)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many ambiguous symbols result from alias-primary collisions?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 703,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1394"
      ]
     },
     "execution_count": 703,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_collision_ambiguous_symbol_set_ncbi = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"collision\"]\n",
    ")\n",
    "ap_collision_ambiguous_symbol_count_ncbi = len(ap_collision_ambiguous_symbol_set_ncbi)\n",
    "ap_collision_ambiguous_symbol_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many records have at least one alias-primary collision (alias that matches another record's primary gene symbol)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 704,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1614"
      ]
     },
     "execution_count": 704,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ap_record_set_ncbi = set(merged_alias_ap_collision_ncbi_df[\"NCBI_ID\"])\n",
    "ap_record_count_ncbi = len(ap_record_set_ncbi)\n",
    "ap_record_count_ncbi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge 3 sets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 705,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1,DKFZP434D0412,MBS85,p84,p85</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRL2</td>\n",
       "      <td>ACKR5,CKRX,CRAM-A,CRAM-B,HCR</td>\n",
       "      <td>ACKR5</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000121797</td>\n",
       "      <td>1612</td>\n",
       "      <td>9034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>806</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>808</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>810</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2928 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol collision source  \\\n",
       "0      PPP1R12C  AAVS1,DKFZP434D0412,MBS85,p84,p85     AAVS1   HGNC   \n",
       "1         SOAT1                         ACAT,ACAT1     ACAT1   HGNC   \n",
       "2         SOAT2                              ACAT2     ACAT2   HGNC   \n",
       "3         CCRL2       ACKR5,CKRX,CRAM-A,CRAM-B,HCR     ACKR5   HGNC   \n",
       "4       NDUFAB1               ACP,ACP1,FASN2A,SDAP      ACP1   HGNC   \n",
       "..          ...                                ...       ...    ...   \n",
       "806      ZNF121                D19S204,ZHC32,ZNF20     ZNF20   ENSG   \n",
       "807      RNF141                       ZFP26,ZNF230    ZNF230   ENSG   \n",
       "808    ZNF322P1                     ZNF322,ZNF322B    ZNF322   ENSG   \n",
       "809      ZNF106               SH3BP3,ZFP106,ZNF474    ZNF474   ENSG   \n",
       "810        ZFP1                    FLJ34243,ZNF475    ZNF475   ENSG   \n",
       "\n",
       "             ENSG_ID  HGNC_ID NCBI_ID  \n",
       "0    ENSG00000125503    14947   54776  \n",
       "1    ENSG00000057252    11177    6646  \n",
       "2    ENSG00000167780    11178    8435  \n",
       "3    ENSG00000121797     1612    9034  \n",
       "4    ENSG00000004779     7694    4706  \n",
       "..               ...      ...     ...  \n",
       "806  ENSG00000197961    12904    7675  \n",
       "807  ENSG00000110315    21159   50862  \n",
       "808  ENSG00000188801    14003       0  \n",
       "809  ENSG00000103994    12886   64397  \n",
       "810  ENSG00000184517    23328  162239  \n",
       "\n",
       "[2928 rows x 7 columns]"
      ]
     },
     "execution_count": 705,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df = pd.concat(\n",
    "    [\n",
    "        merged_alias_ap_collision_hgnc_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ncbi_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ensg_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\", \"ENSG_ID\", \"HGNC_ID\", \"NCBI_ID\"]\n",
    "        ],\n",
    "    ]\n",
    ")\n",
    "merged_alias_primary_collisions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 706,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>964</th>\n",
       "      <td>NRAS</td>\n",
       "      <td>ALPS4,CMNS,KRAS,N-ras,NCMS,NRAS1,NS6</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>ENSG00000213281</td>\n",
       "      <td>7989</td>\n",
       "      <td>4893</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                          alias_symbol collision source  \\\n",
       "964        NRAS  ALPS4,CMNS,KRAS,N-ras,NCMS,NRAS1,NS6      KRAS   NCBI   \n",
       "\n",
       "             ENSG_ID  HGNC_ID NCBI_ID  \n",
       "964  ENSG00000213281     7989    4893  "
      ]
     },
     "execution_count": 706,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"KRAS\"\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 707,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_primary_collisions_df.to_csv(\n",
    "    \"../output/merged_alias_primary_collisions_df.csv\", index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 708,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACAT1',\n",
       " 'ACAT2',\n",
       " 'ACP1',\n",
       " 'ACTBP2',\n",
       " 'ADA2',\n",
       " 'ADRA1A',\n",
       " 'AGT',\n",
       " 'AIP',\n",
       " 'ALB',\n",
       " 'AMN',\n",
       " 'APC2',\n",
       " 'AR',\n",
       " 'ARC',\n",
       " 'ARG1',\n",
       " 'ARHGAP10',\n",
       " 'ARL1',\n",
       " 'ARSB',\n",
       " 'ASIP',\n",
       " 'ATF1',\n",
       " 'ATR',\n",
       " 'AVP',\n",
       " 'B3GNT8',\n",
       " 'BACH1',\n",
       " 'BAP1',\n",
       " 'BCAM',\n",
       " 'BCRP1',\n",
       " 'BCRP2',\n",
       " 'BDP1',\n",
       " 'BNC1',\n",
       " 'BRAP',\n",
       " 'BRCC3',\n",
       " 'BRI3',\n",
       " 'BST1',\n",
       " 'BTF3',\n",
       " 'C6',\n",
       " 'C7',\n",
       " 'CA11',\n",
       " 'CABP1',\n",
       " 'CAD',\n",
       " 'CAMP',\n",
       " 'CAP1',\n",
       " 'CAP2',\n",
       " 'CAPS',\n",
       " 'CAPS2',\n",
       " 'CARF',\n",
       " 'CAST',\n",
       " 'CBLC',\n",
       " 'CCR10',\n",
       " 'CCR4',\n",
       " 'CDH1',\n",
       " 'CDH20',\n",
       " 'CDH7',\n",
       " 'CDKN1A',\n",
       " 'CDS1',\n",
       " 'CES2',\n",
       " 'CHD5',\n",
       " 'CHL1',\n",
       " 'CHP1',\n",
       " 'CIC',\n",
       " 'CKLF',\n",
       " 'CLC',\n",
       " 'CLCP1',\n",
       " 'CLIP4',\n",
       " 'CNP',\n",
       " 'CNR1',\n",
       " 'CNR2',\n",
       " 'COP1',\n",
       " 'CPA3',\n",
       " 'CPA6',\n",
       " 'CPD',\n",
       " 'CPN1',\n",
       " 'CPN2',\n",
       " 'CPS1',\n",
       " 'CREB3',\n",
       " 'CRIP1',\n",
       " 'CRP',\n",
       " 'CSN2',\n",
       " 'CSN3',\n",
       " 'CSNK2A1',\n",
       " 'CST6',\n",
       " 'CSTP1',\n",
       " 'CTAGE4',\n",
       " 'CTH',\n",
       " 'DAO',\n",
       " 'DAP',\n",
       " 'DAP3',\n",
       " 'DBI',\n",
       " 'DBP',\n",
       " 'DEF6',\n",
       " 'DIO1',\n",
       " 'DLC1',\n",
       " 'DMC1',\n",
       " 'DMP1',\n",
       " 'DOK1',\n",
       " 'DRC3',\n",
       " 'DRG1',\n",
       " 'DRP2',\n",
       " 'DSC1',\n",
       " 'DUSP1',\n",
       " 'DUSP23',\n",
       " 'DUSP26',\n",
       " 'EAF2',\n",
       " 'EBP',\n",
       " 'ECD',\n",
       " 'EIF2A',\n",
       " 'ELF1',\n",
       " 'ELK1',\n",
       " 'ELN',\n",
       " 'EMB',\n",
       " 'EPO',\n",
       " 'F11',\n",
       " 'F3',\n",
       " 'FAAH',\n",
       " 'FABP5',\n",
       " 'FADS6',\n",
       " 'FAH',\n",
       " 'FAS',\n",
       " 'FBP2',\n",
       " 'FBXO31',\n",
       " 'FCRL2',\n",
       " 'FEM1A',\n",
       " 'FEZ1',\n",
       " 'FGF2',\n",
       " 'FLG',\n",
       " 'FLI1',\n",
       " 'FZD3',\n",
       " 'GAB1',\n",
       " 'GAL',\n",
       " 'GAL3ST2',\n",
       " 'GALNT15',\n",
       " 'GALNT17',\n",
       " 'GAR1',\n",
       " 'GCK',\n",
       " 'GCKR',\n",
       " 'GCOM1',\n",
       " 'GDA',\n",
       " 'GEM',\n",
       " 'GIP',\n",
       " 'GK2',\n",
       " 'GLA',\n",
       " 'GLS',\n",
       " 'GLUD1',\n",
       " 'GLUD2',\n",
       " 'GOT1',\n",
       " 'GPI',\n",
       " 'GPT',\n",
       " 'GPX6',\n",
       " 'GRIN1',\n",
       " 'GRIP1',\n",
       " 'GRP',\n",
       " 'GSG1',\n",
       " 'HAP1',\n",
       " 'HAS1',\n",
       " 'HBM',\n",
       " 'HCP5',\n",
       " 'HCRTR2',\n",
       " 'HCST',\n",
       " 'HDC',\n",
       " 'HES1',\n",
       " 'HGF',\n",
       " 'HK1',\n",
       " 'HK2',\n",
       " 'HK3',\n",
       " 'HLA-H',\n",
       " 'HLF',\n",
       " 'HP',\n",
       " 'HPCA',\n",
       " 'HPF1',\n",
       " 'HRG',\n",
       " 'HRH1',\n",
       " 'HSD17B7',\n",
       " 'HTRA3',\n",
       " 'IFNA13',\n",
       " 'IK',\n",
       " 'IL25',\n",
       " 'IMP3',\n",
       " 'IMP4',\n",
       " 'IQCH',\n",
       " 'ISY1',\n",
       " 'KLF14',\n",
       " 'KLHL6',\n",
       " 'KLK7',\n",
       " 'KRTAP5-1',\n",
       " 'KRTAP5-11',\n",
       " 'KRTAP5-2',\n",
       " 'KRTAP5-3',\n",
       " 'KRTAP5-8',\n",
       " 'KRTAP5-9',\n",
       " 'KSR2',\n",
       " 'LAMA3',\n",
       " 'LAT',\n",
       " 'LAT2',\n",
       " 'LBH',\n",
       " 'LCP1',\n",
       " 'LDB1',\n",
       " 'LIG1',\n",
       " 'LIPA',\n",
       " 'LNP1',\n",
       " 'LRP1',\n",
       " 'LRP2BP',\n",
       " 'LRP4',\n",
       " 'MADD',\n",
       " 'MAFA',\n",
       " 'MAK',\n",
       " 'MAL',\n",
       " 'MAP1A',\n",
       " 'MAP2',\n",
       " 'MAS1',\n",
       " 'MAST1',\n",
       " 'MBP',\n",
       " 'ME1',\n",
       " 'ME2',\n",
       " 'ME3',\n",
       " 'MED1',\n",
       " 'MED24',\n",
       " 'MED25',\n",
       " 'MET',\n",
       " 'MGA',\n",
       " 'MGAT1',\n",
       " 'MGAT2',\n",
       " 'MGAT3',\n",
       " 'MIA2',\n",
       " 'MIF',\n",
       " 'MIP',\n",
       " 'MLC1',\n",
       " 'MLN',\n",
       " 'MOS',\n",
       " 'MPDZ',\n",
       " 'MPP1',\n",
       " 'MPP2',\n",
       " 'MPP4',\n",
       " 'MPP7',\n",
       " 'MRPL27',\n",
       " 'MRPS28',\n",
       " 'MRPS35',\n",
       " 'MS4A10',\n",
       " 'MS4A7',\n",
       " 'MS4A8',\n",
       " 'MSL1',\n",
       " 'MST1',\n",
       " 'MTCP1',\n",
       " 'MTF1',\n",
       " 'MYH14',\n",
       " 'MYNN',\n",
       " 'MYO1C',\n",
       " 'MYT1',\n",
       " 'NAF1',\n",
       " 'NANOGP2',\n",
       " 'NANOGP3',\n",
       " 'NANOGP4',\n",
       " 'NAPA',\n",
       " 'NAPB',\n",
       " 'NARF',\n",
       " 'NAT1',\n",
       " 'NAT2',\n",
       " 'NBN',\n",
       " 'NET1',\n",
       " 'NGB',\n",
       " 'NLK',\n",
       " 'NMB',\n",
       " 'NME1',\n",
       " 'NMI',\n",
       " 'NOP9',\n",
       " 'NOS1',\n",
       " 'NOS2',\n",
       " 'NOS3',\n",
       " 'NPR2',\n",
       " 'NPR3',\n",
       " 'NRAP',\n",
       " 'NRF1',\n",
       " 'NRP2',\n",
       " 'NSL1',\n",
       " 'NTN4',\n",
       " 'ODC1',\n",
       " 'OPA1',\n",
       " 'ORC1',\n",
       " 'OSM',\n",
       " 'OSR1',\n",
       " 'OXR1',\n",
       " 'PAK1',\n",
       " 'PAK3',\n",
       " 'PAK5',\n",
       " 'PAM',\n",
       " 'PAN2',\n",
       " 'PAN3',\n",
       " 'PC',\n",
       " 'PCM1',\n",
       " 'PDF',\n",
       " 'PDK1',\n",
       " 'PDP1',\n",
       " 'PDX1',\n",
       " 'PGAM1',\n",
       " 'PGP',\n",
       " 'PGS1',\n",
       " 'PIGF',\n",
       " 'PIM1',\n",
       " 'PISD',\n",
       " 'PKD2',\n",
       " 'PLAC8',\n",
       " 'PMP22',\n",
       " 'PMS2P13',\n",
       " 'PNP',\n",
       " 'POLK',\n",
       " 'POLQ',\n",
       " 'POMP',\n",
       " 'POP1',\n",
       " 'POP4',\n",
       " 'POR',\n",
       " 'PPAT',\n",
       " 'PPOX',\n",
       " 'PRB1',\n",
       " 'PRDX6',\n",
       " 'PREP',\n",
       " 'PRG2',\n",
       " 'PRG3',\n",
       " 'PRR4',\n",
       " 'PRX',\n",
       " 'PSAP',\n",
       " 'PSG1',\n",
       " 'PTH2',\n",
       " 'PTPRO',\n",
       " 'PVALB',\n",
       " 'RAC1',\n",
       " 'RAC3',\n",
       " 'RAD1',\n",
       " 'RAE1',\n",
       " 'RAF1',\n",
       " 'RAI1',\n",
       " 'RASSF4',\n",
       " 'RAX',\n",
       " 'RBM5-AS1',\n",
       " 'RBP1',\n",
       " 'RBP2',\n",
       " 'RBP3',\n",
       " 'REN',\n",
       " 'RFC1',\n",
       " 'RGP1',\n",
       " 'RGR',\n",
       " 'RGS13',\n",
       " 'RHEB',\n",
       " 'RHO',\n",
       " 'RIF1',\n",
       " 'RILP',\n",
       " 'RLF',\n",
       " 'RNASE9',\n",
       " 'ROR1',\n",
       " 'RP1',\n",
       " 'RP2',\n",
       " 'RPA1',\n",
       " 'RPA2',\n",
       " 'RPF1',\n",
       " 'RPL10',\n",
       " 'RPL13',\n",
       " 'RPL17',\n",
       " 'RPL23',\n",
       " 'RPL24',\n",
       " 'RPN1',\n",
       " 'RPN2',\n",
       " 'RPS12',\n",
       " 'RRP1',\n",
       " 'RS1',\n",
       " 'S100A16',\n",
       " 'SAG',\n",
       " 'SAP130',\n",
       " 'SAT2',\n",
       " 'SCAF1',\n",
       " 'SCP2',\n",
       " 'SCYL1',\n",
       " 'SCYL2',\n",
       " 'SCYL3',\n",
       " 'SDS',\n",
       " 'SELP',\n",
       " 'SF1',\n",
       " 'SF3B1',\n",
       " 'SFN',\n",
       " 'SI',\n",
       " 'SIT1',\n",
       " 'SKP2',\n",
       " 'SLA',\n",
       " 'SLIT1',\n",
       " 'SLIT3',\n",
       " 'SLK',\n",
       " 'SMAD1',\n",
       " 'SMAP2',\n",
       " 'SMO',\n",
       " 'SMS',\n",
       " 'SNORA58B',\n",
       " 'SNORD115-48',\n",
       " 'SNURF',\n",
       " 'SNX30',\n",
       " 'SP110',\n",
       " 'SPATA8',\n",
       " 'SPN',\n",
       " 'SPP1',\n",
       " 'SPP2',\n",
       " 'SPR',\n",
       " 'SRA1',\n",
       " 'SRM',\n",
       " 'ST7',\n",
       " 'STAG1',\n",
       " 'STAR',\n",
       " 'STK3',\n",
       " 'STRAP',\n",
       " 'TAF4B',\n",
       " 'TAP1',\n",
       " 'TAP2',\n",
       " 'TEN1',\n",
       " 'TEP1',\n",
       " 'TF',\n",
       " 'THY1-AS1',\n",
       " 'TIFA',\n",
       " 'TKT',\n",
       " 'TNFRSF11B',\n",
       " 'TNNC1',\n",
       " 'TPH2',\n",
       " 'TPO',\n",
       " 'TPP1',\n",
       " 'TPRKB',\n",
       " 'TRAP1',\n",
       " 'TRAV11',\n",
       " 'TSN',\n",
       " 'TTF1',\n",
       " 'UCN2',\n",
       " 'UGT2B11',\n",
       " 'USE1',\n",
       " 'USP10',\n",
       " 'USP16',\n",
       " 'USP21',\n",
       " 'VIP',\n",
       " 'WBP1',\n",
       " 'ZNF20',\n",
       " 'ZNF230',\n",
       " 'ZNF474',\n",
       " 'ZNF475'}"
      ]
     },
     "execution_count": 708,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_ap_collision_ambiguous_symbol_set = (\n",
    "    ap_collision_ambiguous_symbol_set_ncbi\n",
    "    & ap_collision_ambiguous_symbol_set_hgnc\n",
    "    & ap_collision_ambiguous_symbol_set_ensg\n",
    ")\n",
    "common_ap_collision_ambiguous_symbol_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 709,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "434"
      ]
     },
     "execution_count": 709,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_ap_collision_ambiguous_symbol_set)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
