{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 469,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 436,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_ap_collision_df(mini_xxxx_df: pd.DataFrame, source: str) -> pd.DataFrame:\n",
    "    \"\"\"Create a df of alias-primary collision symbols \n",
    "\n",
    "    :param mini_xxxx_df: Processed df of gene records\n",
    "    :param source: Representation of the source of the gene records\n",
    "    :param split_on_character: Character that is used to seperate alias symbols in the mini_xxxx_df\n",
    "    :return: A df of genes that share an alias with another gene\n",
    "    \"\"\"\n",
    "\n",
    "    #Strip \"HGNC\" from HGNC ID\n",
    "    mini_xxxx_df[\"HGNC_ID\"] = mini_xxxx_df[\"HGNC_ID\"].str.lstrip(\"HGNC:\")\n",
    "\n",
    "    #Remove placeholder gene records (those with no primary gene symbol)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\" \", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.replace(\"-\", np.nan)\n",
    "    mini_xxxx_df = mini_xxxx_df.dropna(subset=[\"gene_symbol\"])\n",
    "\n",
    "    #Remove placeholder LOC genes\n",
    "    mini_xxxx_df = mini_xxxx_df[~mini_xxxx_df['gene_symbol'].str.startswith('LOC', na=False)]\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    mini_xxxx_df.to_csv(f'../output/mini_{source.lower()}_df.csv', index=True) \n",
    "\n",
    "    #Make a new df where the alias symbols are merged together for each record\n",
    "    merged_alias_xxxx_df = mini_xxxx_df.copy()\n",
    "    merged_alias_xxxx_df[\"alias_symbol\"] = merged_alias_xxxx_df[\"alias_symbol\"].fillna(\"\").astype(str)\n",
    "    merged_alias_xxxx_df = (\n",
    "    merged_alias_xxxx_df.groupby(\n",
    "        [\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], dropna=False\n",
    "    )[\"alias_symbol\"]\n",
    "    .apply(lambda x: \",\".join(x.dropna())).reset_index())\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    merged_alias_xxxx_df.to_csv(f'../output/merged_alias_{source}_df.csv', index=True) \n",
    "\n",
    "    #Make a set of the primary gene symbols\n",
    "    xxxx_gene_symbol_set = set(mini_xxxx_df[\"gene_symbol\"])\n",
    "\n",
    "    #Drop genes without any aliases\n",
    "    subset_genes_xxxx_df = mini_xxxx_df.copy()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.dropna(subset=[\"alias_symbol\"])\n",
    "\n",
    "    #Remove duplicate instances of primary gene symbol- alias pairs\n",
    "    #(occur because the same primary gene symbol may have multiple different ENSG IDs, \n",
    "    #see gene RFLNB for example)\n",
    "    subset_genes_xxxx_df[\"gene_symbol\"] = subset_genes_xxxx_df[\"gene_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df[\"alias_symbol\"] = subset_genes_xxxx_df[\"alias_symbol\"].str.upper()\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df.drop_duplicates(subset=['gene_symbol', 'alias_symbol'], keep='first')\n",
    "\n",
    "    #Remove instances where the primary gene symbol is an alias for that gene record\n",
    "    #the pair would be the same symbol twice\n",
    "    subset_genes_xxxx_df = subset_genes_xxxx_df[subset_genes_xxxx_df['gene_symbol'] != subset_genes_xxxx_df['alias_symbol']]\n",
    "\n",
    "    #Convert the df into a csv and save\n",
    "    subset_genes_xxxx_df.to_csv(f'../output/subset_genes_{source}_df.csv', index=True) \n",
    "\n",
    "    #Create df with genes that have an alias that can be found as another gene's primary gene symbol\n",
    "    ap_collision_xxxx_df = subset_genes_xxxx_df.copy()\n",
    "    ap_collision_xxxx_df.loc[:, \"alias_symbol\"] = ap_collision_xxxx_df[\"alias_symbol\"].apply(lambda x: {x})\n",
    "    ap_collision_xxxx_df.loc[:, \"collision\"] = ap_collision_xxxx_df[\"alias_symbol\"].apply(\n",
    "        lambda x: x & xxxx_gene_symbol_set\n",
    "    )\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df[ap_collision_xxxx_df[\"collision\"].apply(lambda x: len(x) > 0)]\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n",
    "    ap_collision_xxxx_df['NCBI_ID'] = ap_collision_xxxx_df['NCBI_ID'].fillna(0).astype(int)\n",
    "    ap_collision_xxxx_df = ap_collision_xxxx_df.sort_values(\"collision\")\n",
    "\n",
    "    #Add a source tag for future merging efforts\n",
    "    ap_collision_xxxx_df[\"source\"] = str(source.upper())\n",
    "\n",
    "    #Convert the df into a csv\n",
    "    ap_collision_xxxx_df.to_csv(f'../output/single_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "\n",
    "    #Create a secondary collision df that merges the alias symbols for each record\n",
    "    merged_alias_ap_collision_xxxx_df = ap_collision_xxxx_df.drop(columns=['alias_symbol'])\n",
    "    merged_alias_ap_collision_xxxx_df = pd.merge(merged_alias_ap_collision_xxxx_df, merged_alias_xxxx_df, on=[\"ENSG_ID\", \"gene_symbol\", \"HGNC_ID\"], how=\"left\")\n",
    "    merged_alias_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df[[\"gene_symbol\",\"alias_symbol\",\"ENSG_ID\",\"HGNC_ID\",\"NCBI_ID\",\"collision\",\"source\"]]\n",
    "\n",
    "    #Test to make sure all collisions are in the alias list for the record\n",
    "    test_ap_collision_xxxx_df = merged_alias_ap_collision_xxxx_df.copy()\n",
    "    test_ap_collision_xxxx_df['alias_symbol'] = test_ap_collision_xxxx_df['alias_symbol'].apply(lambda x: x if isinstance(x, list) else [x])\n",
    "    test_ap_collision_xxxx_df['collision_in_alias'] = test_ap_collision_xxxx_df.apply(lambda row: row['collision'] in test_ap_collision_xxxx_df['alias_symbol'].values, axis=1)\n",
    "    true_count = test_ap_collision_xxxx_df['collision_in_alias'].sum()\n",
    "\n",
    "    if true_count == 0:\n",
    "        #Convert the df into a csv and save\n",
    "        merged_alias_ap_collision_xxxx_df.to_csv(f'../output/merged_alias_ap_collision_{source.lower()}_df.csv', index=True)\n",
    "        print(\"All collisions are present in gene alias lists.\")\n",
    "    else:\n",
    "        print(\"Some collisions are not present in gene alias lists.\")\n",
    "\n",
    "    return mini_xxxx_df.head(), subset_genes_xxxx_df.head(), ap_collision_xxxx_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 625,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7481</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:7470</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:52502</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133061</th>\n",
       "      <td>ENSG00000289291</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824</td>\n",
       "      <td>HGNC:10108</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133064</th>\n",
       "      <td>ENSG00000296488</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>133065 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID     HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049     <NA>   HGNC:7481         MTTF       MT-TF\n",
       "1       ENSG00000210049     <NA>   HGNC:7481         TRNF       MT-TF\n",
       "2       ENSG00000211459     <NA>   HGNC:7470          12S     MT-RNR1\n",
       "3       ENSG00000211459     <NA>   HGNC:7470       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459     <NA>   HGNC:7470       MTRNR1     MT-RNR1\n",
       "...                 ...      ...         ...          ...         ...\n",
       "133060  ENSG00000229388     <NA>  HGNC:52502    LINC01715    TAF12-DT\n",
       "133061  ENSG00000289291     <NA>         NaN          NaN         NaN\n",
       "133062  ENSG00000274978    26824  HGNC:10108      RNU11-1       RNU11\n",
       "133063  ENSG00000274978    26824  HGNC:10108          U11       RNU11\n",
       "133064  ENSG00000296488     <NA>         NaN          NaN         NaN\n",
       "\n",
       "[133065 rows x 5 columns]"
      ]
     },
     "execution_count": 625,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = pd.read_csv(\n",
    "    \"../input/ensg_biomart_gene20250625.txt\", sep=\"\\t\",dtype={\"NCBI gene (formerly Entrezgene) ID\": pd.Int64Dtype()}\n",
    ")\n",
    "mini_ensg_df = mini_ensg_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Gene Synonym\": \"alias_symbol\",\n",
    "        \"Gene name\": \"gene_symbol\",\n",
    "        \"Gene stable ID\": \"ENSG_ID\",\n",
    "        \"NCBI gene (formerly Entrezgene) ID\": \"NCBI_ID\",\n",
    "    }\n",
    ")\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 626,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86364"
      ]
     },
     "execution_count": 626,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ensg_df['ENSG_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 627,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38041"
      ]
     },
     "execution_count": 627,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df = mini_ensg_df.replace(\" \", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"\", np.nan)\n",
    "mini_ensg_df = mini_ensg_df.replace(\"-\", np.nan)\n",
    "\n",
    "unique_ensg_df = mini_ensg_df[mini_ensg_df['ENSG_ID'].duplicated(keep=False) == False]\n",
    "\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_ensg_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 628,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "57009"
      ]
     },
     "execution_count": 628,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now count rows in that subset where 'alias_symbol' is missing\n",
    "no_alias_symbol_count = unique_ensg_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 629,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "420"
      ]
     },
     "execution_count": 629,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ensg_df[\n",
    "    mini_ensg_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ensg_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 630,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1155</th>\n",
       "      <td>ENSG00000291362</td>\n",
       "      <td>65265</td>\n",
       "      <td>HGNC:26104</td>\n",
       "      <td>FLJ20989</td>\n",
       "      <td>C8orf33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1207</th>\n",
       "      <td>ENSG00000288239</td>\n",
       "      <td>646457</td>\n",
       "      <td>HGNC:34354</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C19orf67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2838</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>BTG3-7:1</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2839</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>C21ORF14</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2840</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>C21ORF38</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2841</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>CSSG1</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2842</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>EURL</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2843</th>\n",
       "      <td>ENSG00000154642</td>\n",
       "      <td>54149</td>\n",
       "      <td>HGNC:16459</td>\n",
       "      <td>YG81</td>\n",
       "      <td>C21orf91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4632</th>\n",
       "      <td>ENSG00000141428</td>\n",
       "      <td>83608</td>\n",
       "      <td>HGNC:28802</td>\n",
       "      <td>HST3108</td>\n",
       "      <td>C18orf21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4633</th>\n",
       "      <td>ENSG00000141428</td>\n",
       "      <td>83608</td>\n",
       "      <td>HGNC:28802</td>\n",
       "      <td>PNAS-124</td>\n",
       "      <td>C18orf21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4634</th>\n",
       "      <td>ENSG00000141428</td>\n",
       "      <td>83608</td>\n",
       "      <td>HGNC:28802</td>\n",
       "      <td>PNAS-131</td>\n",
       "      <td>C18orf21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6124</th>\n",
       "      <td>ENSG00000188277</td>\n",
       "      <td>643338</td>\n",
       "      <td>HGNC:34489</td>\n",
       "      <td>LOC643338</td>\n",
       "      <td>C15orf62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6486</th>\n",
       "      <td>ENSG00000179008</td>\n",
       "      <td>317761</td>\n",
       "      <td>HGNC:19849</td>\n",
       "      <td>SIX6OS1</td>\n",
       "      <td>C14orf39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6865</th>\n",
       "      <td>ENSG00000179133</td>\n",
       "      <td>256815</td>\n",
       "      <td>HGNC:28716</td>\n",
       "      <td>BA215C7.4</td>\n",
       "      <td>C10orf67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6866</th>\n",
       "      <td>ENSG00000179133</td>\n",
       "      <td>256815</td>\n",
       "      <td>HGNC:28716</td>\n",
       "      <td>C10ORF115</td>\n",
       "      <td>C10orf67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6867</th>\n",
       "      <td>ENSG00000179133</td>\n",
       "      <td>256815</td>\n",
       "      <td>HGNC:28716</td>\n",
       "      <td>LINC01552</td>\n",
       "      <td>C10orf67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6868</th>\n",
       "      <td>ENSG00000179133</td>\n",
       "      <td>256815</td>\n",
       "      <td>HGNC:28716</td>\n",
       "      <td>MGC46732</td>\n",
       "      <td>C10orf67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7075</th>\n",
       "      <td>ENSG00000154274</td>\n",
       "      <td>55286</td>\n",
       "      <td>HGNC:25618</td>\n",
       "      <td>FLJ11017</td>\n",
       "      <td>C4orf19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7364</th>\n",
       "      <td>ENSG00000125975</td>\n",
       "      <td>140873</td>\n",
       "      <td>HGNC:16166</td>\n",
       "      <td>DJ477O4.4</td>\n",
       "      <td>C20orf173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9598</th>\n",
       "      <td>ENSG00000131379</td>\n",
       "      <td>84077</td>\n",
       "      <td>HGNC:25320</td>\n",
       "      <td>DKFZP434N1817</td>\n",
       "      <td>C3orf20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9662</th>\n",
       "      <td>ENSG00000165084</td>\n",
       "      <td>116328</td>\n",
       "      <td>HGNC:30905</td>\n",
       "      <td>VEST-1</td>\n",
       "      <td>C8orf34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9663</th>\n",
       "      <td>ENSG00000165084</td>\n",
       "      <td>116328</td>\n",
       "      <td>HGNC:30905</td>\n",
       "      <td>VEST1</td>\n",
       "      <td>C8orf34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10054</th>\n",
       "      <td>ENSG00000226119</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:55036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C3orf49P1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10874</th>\n",
       "      <td>ENSG00000177576</td>\n",
       "      <td>497661</td>\n",
       "      <td>HGNC:31690</td>\n",
       "      <td>FLJ23458</td>\n",
       "      <td>C18orf32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11350</th>\n",
       "      <td>ENSG00000160298</td>\n",
       "      <td>54058</td>\n",
       "      <td>HGNC:1300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C21orf58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11590</th>\n",
       "      <td>ENSG00000185753</td>\n",
       "      <td>159013</td>\n",
       "      <td>HGNC:28589</td>\n",
       "      <td>MGC39350</td>\n",
       "      <td>CXorf38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11725</th>\n",
       "      <td>ENSG00000279020</td>\n",
       "      <td>147276</td>\n",
       "      <td>HGNC:26447</td>\n",
       "      <td>FLJ31338</td>\n",
       "      <td>C18orf15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11726</th>\n",
       "      <td>ENSG00000279020</td>\n",
       "      <td>147276</td>\n",
       "      <td>HGNC:26447</td>\n",
       "      <td>HST3231</td>\n",
       "      <td>C18orf15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12208</th>\n",
       "      <td>ENSG00000242259</td>\n",
       "      <td>128977</td>\n",
       "      <td>HGNC:27012</td>\n",
       "      <td>MGC74441</td>\n",
       "      <td>C22orf39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12209</th>\n",
       "      <td>ENSG00000242259</td>\n",
       "      <td>128977</td>\n",
       "      <td>HGNC:27012</td>\n",
       "      <td>PANTS</td>\n",
       "      <td>C22orf39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12386</th>\n",
       "      <td>ENSG00000149609</td>\n",
       "      <td>128864</td>\n",
       "      <td>HGNC:16137</td>\n",
       "      <td>BCLT</td>\n",
       "      <td>C20orf144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12387</th>\n",
       "      <td>ENSG00000149609</td>\n",
       "      <td>128864</td>\n",
       "      <td>HGNC:16137</td>\n",
       "      <td>DJ63M2.6</td>\n",
       "      <td>C20orf144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13467</th>\n",
       "      <td>ENSG00000128346</td>\n",
       "      <td>84645</td>\n",
       "      <td>HGNC:18589</td>\n",
       "      <td>EVG1</td>\n",
       "      <td>C22orf23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13468</th>\n",
       "      <td>ENSG00000128346</td>\n",
       "      <td>84645</td>\n",
       "      <td>HGNC:18589</td>\n",
       "      <td>FLJ32787</td>\n",
       "      <td>C22orf23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13469</th>\n",
       "      <td>ENSG00000128346</td>\n",
       "      <td>84645</td>\n",
       "      <td>HGNC:18589</td>\n",
       "      <td>LOC84645</td>\n",
       "      <td>C22orf23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13693</th>\n",
       "      <td>ENSG00000235699</td>\n",
       "      <td>100133053</td>\n",
       "      <td>HGNC:42787</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CXorf51B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13906</th>\n",
       "      <td>ENSG00000291474</td>\n",
       "      <td>122455339</td>\n",
       "      <td>HGNC:56305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C8orf90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14968</th>\n",
       "      <td>ENSG00000291484</td>\n",
       "      <td>388960</td>\n",
       "      <td>HGNC:34349</td>\n",
       "      <td>COG5373</td>\n",
       "      <td>C2orf78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14969</th>\n",
       "      <td>ENSG00000291484</td>\n",
       "      <td>388960</td>\n",
       "      <td>HGNC:34349</td>\n",
       "      <td>FLJ43987</td>\n",
       "      <td>C2orf78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14970</th>\n",
       "      <td>ENSG00000291484</td>\n",
       "      <td>388960</td>\n",
       "      <td>HGNC:34349</td>\n",
       "      <td>HCG1989538</td>\n",
       "      <td>C2orf78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15129</th>\n",
       "      <td>ENSG00000285312</td>\n",
       "      <td>347744</td>\n",
       "      <td>HGNC:20881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C6orf52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15693</th>\n",
       "      <td>ENSG00000215131</td>\n",
       "      <td>646174</td>\n",
       "      <td>HGNC:34455</td>\n",
       "      <td>LOC646174</td>\n",
       "      <td>C16orf90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15891</th>\n",
       "      <td>ENSG00000100802</td>\n",
       "      <td>60686</td>\n",
       "      <td>HGNC:20162</td>\n",
       "      <td>FLJ12154</td>\n",
       "      <td>C14orf93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15892</th>\n",
       "      <td>ENSG00000100802</td>\n",
       "      <td>60686</td>\n",
       "      <td>HGNC:20162</td>\n",
       "      <td>RTFC</td>\n",
       "      <td>C14orf93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16199</th>\n",
       "      <td>ENSG00000179933</td>\n",
       "      <td>55017</td>\n",
       "      <td>HGNC:20270</td>\n",
       "      <td>FLJ20671</td>\n",
       "      <td>C14orf119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16648</th>\n",
       "      <td>ENSG00000206043</td>\n",
       "      <td>644041</td>\n",
       "      <td>HGNC:40037</td>\n",
       "      <td>DKFZP781G0119</td>\n",
       "      <td>C18orf63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17011</th>\n",
       "      <td>ENSG00000142686</td>\n",
       "      <td>127703</td>\n",
       "      <td>HGNC:26800</td>\n",
       "      <td>FLJ38984</td>\n",
       "      <td>C1orf216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18131</th>\n",
       "      <td>ENSG00000151470</td>\n",
       "      <td>132321</td>\n",
       "      <td>HGNC:27025</td>\n",
       "      <td>FLJ33703</td>\n",
       "      <td>C4orf33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18177</th>\n",
       "      <td>ENSG00000196476</td>\n",
       "      <td>140680</td>\n",
       "      <td>HGNC:16227</td>\n",
       "      <td>DJ1103G7.2</td>\n",
       "      <td>C20orf96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18986</th>\n",
       "      <td>ENSG00000220745</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>HGNC:56319</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C11orf98P2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19915</th>\n",
       "      <td>ENSG00000241224</td>\n",
       "      <td>401081</td>\n",
       "      <td>HGNC:53432</td>\n",
       "      <td>FLJ22763</td>\n",
       "      <td>C3orf85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20344</th>\n",
       "      <td>ENSG00000162598</td>\n",
       "      <td>127795</td>\n",
       "      <td>HGNC:28547</td>\n",
       "      <td>CREF</td>\n",
       "      <td>C1orf87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20345</th>\n",
       "      <td>ENSG00000162598</td>\n",
       "      <td>127795</td>\n",
       "      <td>HGNC:28547</td>\n",
       "      <td>MGC34837</td>\n",
       "      <td>C1orf87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21423</th>\n",
       "      <td>ENSG00000120055</td>\n",
       "      <td>79946</td>\n",
       "      <td>HGNC:25880</td>\n",
       "      <td>FLJ14280</td>\n",
       "      <td>C10orf95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22100</th>\n",
       "      <td>ENSG00000260456</td>\n",
       "      <td>100506581</td>\n",
       "      <td>HGNC:40033</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C16orf95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22108</th>\n",
       "      <td>ENSG00000204165</td>\n",
       "      <td>158830</td>\n",
       "      <td>HGNC:33713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CXorf65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22622</th>\n",
       "      <td>ENSG00000182307</td>\n",
       "      <td>65265</td>\n",
       "      <td>HGNC:26104</td>\n",
       "      <td>FLJ20989</td>\n",
       "      <td>C8orf33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23455</th>\n",
       "      <td>ENSG00000100249</td>\n",
       "      <td>25770</td>\n",
       "      <td>HGNC:26931</td>\n",
       "      <td>BK747E2.1</td>\n",
       "      <td>C22orf31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23456</th>\n",
       "      <td>ENSG00000100249</td>\n",
       "      <td>25770</td>\n",
       "      <td>HGNC:26931</td>\n",
       "      <td>HS747E2A</td>\n",
       "      <td>C22orf31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23804</th>\n",
       "      <td>ENSG00000222018</td>\n",
       "      <td>101928147</td>\n",
       "      <td>HGNC:39602</td>\n",
       "      <td>FAM243A</td>\n",
       "      <td>C21orf140</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID    NCBI_ID     HGNC_ID   alias_symbol gene_symbol\n",
       "1155   ENSG00000291362      65265  HGNC:26104       FLJ20989     C8orf33\n",
       "1207   ENSG00000288239     646457  HGNC:34354            NaN    C19orf67\n",
       "2838   ENSG00000154642      54149  HGNC:16459       BTG3-7:1    C21orf91\n",
       "2839   ENSG00000154642      54149  HGNC:16459       C21ORF14    C21orf91\n",
       "2840   ENSG00000154642      54149  HGNC:16459       C21ORF38    C21orf91\n",
       "2841   ENSG00000154642      54149  HGNC:16459          CSSG1    C21orf91\n",
       "2842   ENSG00000154642      54149  HGNC:16459           EURL    C21orf91\n",
       "2843   ENSG00000154642      54149  HGNC:16459           YG81    C21orf91\n",
       "4632   ENSG00000141428      83608  HGNC:28802        HST3108    C18orf21\n",
       "4633   ENSG00000141428      83608  HGNC:28802       PNAS-124    C18orf21\n",
       "4634   ENSG00000141428      83608  HGNC:28802       PNAS-131    C18orf21\n",
       "6124   ENSG00000188277     643338  HGNC:34489      LOC643338    C15orf62\n",
       "6486   ENSG00000179008     317761  HGNC:19849        SIX6OS1    C14orf39\n",
       "6865   ENSG00000179133     256815  HGNC:28716      BA215C7.4    C10orf67\n",
       "6866   ENSG00000179133     256815  HGNC:28716      C10ORF115    C10orf67\n",
       "6867   ENSG00000179133     256815  HGNC:28716      LINC01552    C10orf67\n",
       "6868   ENSG00000179133     256815  HGNC:28716       MGC46732    C10orf67\n",
       "7075   ENSG00000154274      55286  HGNC:25618       FLJ11017     C4orf19\n",
       "7364   ENSG00000125975     140873  HGNC:16166      DJ477O4.4   C20orf173\n",
       "9598   ENSG00000131379      84077  HGNC:25320  DKFZP434N1817     C3orf20\n",
       "9662   ENSG00000165084     116328  HGNC:30905         VEST-1     C8orf34\n",
       "9663   ENSG00000165084     116328  HGNC:30905          VEST1     C8orf34\n",
       "10054  ENSG00000226119       <NA>  HGNC:55036            NaN   C3orf49P1\n",
       "10874  ENSG00000177576     497661  HGNC:31690       FLJ23458    C18orf32\n",
       "11350  ENSG00000160298      54058   HGNC:1300            NaN    C21orf58\n",
       "11590  ENSG00000185753     159013  HGNC:28589       MGC39350     CXorf38\n",
       "11725  ENSG00000279020     147276  HGNC:26447       FLJ31338    C18orf15\n",
       "11726  ENSG00000279020     147276  HGNC:26447        HST3231    C18orf15\n",
       "12208  ENSG00000242259     128977  HGNC:27012       MGC74441    C22orf39\n",
       "12209  ENSG00000242259     128977  HGNC:27012          PANTS    C22orf39\n",
       "12386  ENSG00000149609     128864  HGNC:16137           BCLT   C20orf144\n",
       "12387  ENSG00000149609     128864  HGNC:16137       DJ63M2.6   C20orf144\n",
       "13467  ENSG00000128346      84645  HGNC:18589           EVG1    C22orf23\n",
       "13468  ENSG00000128346      84645  HGNC:18589       FLJ32787    C22orf23\n",
       "13469  ENSG00000128346      84645  HGNC:18589       LOC84645    C22orf23\n",
       "13693  ENSG00000235699  100133053  HGNC:42787            NaN    CXorf51B\n",
       "13906  ENSG00000291474  122455339  HGNC:56305            NaN     C8orf90\n",
       "14968  ENSG00000291484     388960  HGNC:34349        COG5373     C2orf78\n",
       "14969  ENSG00000291484     388960  HGNC:34349       FLJ43987     C2orf78\n",
       "14970  ENSG00000291484     388960  HGNC:34349     HCG1989538     C2orf78\n",
       "15129  ENSG00000285312     347744  HGNC:20881            NaN     C6orf52\n",
       "15693  ENSG00000215131     646174  HGNC:34455      LOC646174    C16orf90\n",
       "15891  ENSG00000100802      60686  HGNC:20162       FLJ12154    C14orf93\n",
       "15892  ENSG00000100802      60686  HGNC:20162           RTFC    C14orf93\n",
       "16199  ENSG00000179933      55017  HGNC:20270       FLJ20671   C14orf119\n",
       "16648  ENSG00000206043     644041  HGNC:40037  DKFZP781G0119    C18orf63\n",
       "17011  ENSG00000142686     127703  HGNC:26800       FLJ38984    C1orf216\n",
       "18131  ENSG00000151470     132321  HGNC:27025       FLJ33703     C4orf33\n",
       "18177  ENSG00000196476     140680  HGNC:16227     DJ1103G7.2    C20orf96\n",
       "18986  ENSG00000220745       <NA>  HGNC:56319            NaN  C11orf98P2\n",
       "19915  ENSG00000241224     401081  HGNC:53432       FLJ22763     C3orf85\n",
       "20344  ENSG00000162598     127795  HGNC:28547           CREF     C1orf87\n",
       "20345  ENSG00000162598     127795  HGNC:28547       MGC34837     C1orf87\n",
       "21423  ENSG00000120055      79946  HGNC:25880       FLJ14280    C10orf95\n",
       "22100  ENSG00000260456  100506581  HGNC:40033            NaN    C16orf95\n",
       "22108  ENSG00000204165     158830  HGNC:33713            NaN     CXorf65\n",
       "22622  ENSG00000182307      65265  HGNC:26104       FLJ20989     C8orf33\n",
       "23455  ENSG00000100249      25770  HGNC:26931      BK747E2.1    C22orf31\n",
       "23456  ENSG00000100249      25770  HGNC:26931       HS747E2A    C22orf31\n",
       "23804  ENSG00000222018  101928147  HGNC:39602        FAM243A   C21orf140"
      ]
     },
     "execution_count": 630,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows.head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 631,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 631,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_ensg_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 632,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "738"
      ]
     },
     "execution_count": 632,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_ensg_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 633,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 633,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_ensg_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique ENSG ID\n",
    "multi_primary = mini_ensg_df.groupby([\"gene_symbol\"]).ENSG_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_ensg_df.loc[mini_ensg_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').iloc[6000:6060]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 634,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/775146428.py:65: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(           ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol\n",
       " 0  ENSG00000210049     <NA>    7481         MTTF       MT-TF\n",
       " 1  ENSG00000210049     <NA>    7481         TRNF       MT-TF\n",
       " 2  ENSG00000211459     <NA>    7470          12S     MT-RNR1\n",
       " 3  ENSG00000211459     <NA>    7470       MOTS-C     MT-RNR1\n",
       " 4  ENSG00000211459     <NA>    7470       MTRNR1     MT-RNR1,\n",
       "            ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol\n",
       " 0  ENSG00000210049     <NA>    7481         MTTF       MT-TF\n",
       " 1  ENSG00000210049     <NA>    7481         TRNF       MT-TF\n",
       " 2  ENSG00000211459     <NA>    7470          12S     MT-RNR1\n",
       " 3  ENSG00000211459     <NA>    7470       MOTS-C     MT-RNR1\n",
       " 4  ENSG00000211459     <NA>    7470       MTRNR1     MT-RNR1,\n",
       "                ENSG_ID  NCBI_ID HGNC_ID alias_symbol gene_symbol collision  \\\n",
       " 24761  ENSG00000283293   125050   10037          7SK       RN7SK       7SK   \n",
       " 37365  ENSG00000057252     6646   11177        ACAT1       SOAT1     ACAT1   \n",
       " 17666  ENSG00000167780     8435   11178        ACAT2       SOAT2     ACAT2   \n",
       " 35063  ENSG00000004779     4706    7694         ACP1     NDUFAB1      ACP1   \n",
       " 47647  ENSG00000220267        0     141       ACTBP2      ACTBP8    ACTBP2   \n",
       " \n",
       "       source  \n",
       " 24761   ENSG  \n",
       " 37365   ENSG  \n",
       " 17666   ENSG  \n",
       " 35063   ENSG  \n",
       " 47647   ENSG  )"
      ]
     },
     "execution_count": 634,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ensg_df, \"ENSG\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 635,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52502.0</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95024 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049      NaN   7481.0         MTTF       MT-TF\n",
       "1       ENSG00000210049      NaN   7481.0         TRNF       MT-TF\n",
       "2       ENSG00000211459      NaN   7470.0          12S     MT-RNR1\n",
       "3       ENSG00000211459      NaN   7470.0       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459      NaN   7470.0       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989  85028.0  30062.0    LINC00100      SNHG12\n",
       "133059  ENSG00000197989  85028.0  30062.0     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388      NaN  52502.0    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978  26824.0  10108.0      RNU11-1       RNU11\n",
       "133063  ENSG00000274978  26824.0  10108.0          U11       RNU11\n",
       "\n",
       "[95024 rows x 5 columns]"
      ]
     },
     "execution_count": 635,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df= pd.read_csv(\n",
    "    \"../output/mini_ensg_df.csv\", index_col=[0])\n",
    "mini_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 636,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>MTTF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7481.0</td>\n",
       "      <td>TRNF</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>12S</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MOTS-C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7470.0</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133058</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>LINC00100</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133059</th>\n",
       "      <td>ENSG00000197989</td>\n",
       "      <td>85028.0</td>\n",
       "      <td>30062.0</td>\n",
       "      <td>PNAS-123</td>\n",
       "      <td>SNHG12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133060</th>\n",
       "      <td>ENSG00000229388</td>\n",
       "      <td>NaN</td>\n",
       "      <td>52502.0</td>\n",
       "      <td>LINC01715</td>\n",
       "      <td>TAF12-DT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133062</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>RNU11-1</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133063</th>\n",
       "      <td>ENSG00000274978</td>\n",
       "      <td>26824.0</td>\n",
       "      <td>10108.0</td>\n",
       "      <td>U11</td>\n",
       "      <td>RNU11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>57423 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "0       ENSG00000210049      NaN   7481.0         MTTF       MT-TF\n",
       "1       ENSG00000210049      NaN   7481.0         TRNF       MT-TF\n",
       "2       ENSG00000211459      NaN   7470.0          12S     MT-RNR1\n",
       "3       ENSG00000211459      NaN   7470.0       MOTS-C     MT-RNR1\n",
       "4       ENSG00000211459      NaN   7470.0       MTRNR1     MT-RNR1\n",
       "...                 ...      ...      ...          ...         ...\n",
       "133058  ENSG00000197989  85028.0  30062.0    LINC00100      SNHG12\n",
       "133059  ENSG00000197989  85028.0  30062.0     PNAS-123      SNHG12\n",
       "133060  ENSG00000229388      NaN  52502.0    LINC01715    TAF12-DT\n",
       "133062  ENSG00000274978  26824.0  10108.0      RNU11-1       RNU11\n",
       "133063  ENSG00000274978  26824.0  10108.0          U11       RNU11\n",
       "\n",
       "[57423 rows x 5 columns]"
      ]
     },
     "execution_count": 636,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ensg_df.csv\", index_col=[0])\n",
    "subset_genes_ensg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 637,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45334</th>\n",
       "      <td>ENSG00000232629</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68189</th>\n",
       "      <td>ENSG00000230675</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73816</th>\n",
       "      <td>ENSG00000228813</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74541</th>\n",
       "      <td>ENSG00000224305</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76050</th>\n",
       "      <td>ENSG00000228254</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76102</th>\n",
       "      <td>ENSG00000229493</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77670</th>\n",
       "      <td>ENSG00000196610</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78251</th>\n",
       "      <td>ENSG00000226165</td>\n",
       "      <td>3120.0</td>\n",
       "      <td>4945.0</td>\n",
       "      <td>HLA-DXB</td>\n",
       "      <td>HLA-DQB2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               ENSG_ID  NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "45334  ENSG00000232629   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "68189  ENSG00000230675   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "73816  ENSG00000228813   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "74541  ENSG00000224305   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "76050  ENSG00000228254   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "76102  ENSG00000229493   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "77670  ENSG00000196610   3120.0   4945.0      HLA-DXB    HLA-DQB2\n",
       "78251  ENSG00000226165   3120.0   4945.0      HLA-DXB    HLA-DQB2"
      ]
     },
     "execution_count": 637,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ensg_df.loc[mini_ensg_df[\"gene_symbol\"] == \"HLA-DQB2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 638,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4202</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4203</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>LNCAMPC</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4204</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4205</th>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>31696.0</td>\n",
       "      <td>RNF165</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              ENSG_ID   NCBI_ID  HGNC_ID alias_symbol gene_symbol\n",
       "4202  ENSG00000141622  494470.0  31696.0        ARKL2       ARK2C\n",
       "4203  ENSG00000141622  494470.0  31696.0      LNCAMPC       ARK2C\n",
       "4204  ENSG00000141622  494470.0  31696.0     RNF111L2       ARK2C\n",
       "4205  ENSG00000141622  494470.0  31696.0       RNF165       ARK2C"
      ]
     },
     "execution_count": 638,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ensg_df.loc[\n",
    "    subset_genes_ensg_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 639,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RN7SK</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG00000283293</td>\n",
       "      <td>10037.0</td>\n",
       "      <td>125050</td>\n",
       "      <td>7SK</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1,SOAT,STAT</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177.0</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178.0</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>ACP,ACP1,FASN2A,SDAP</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694.0</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTBP8</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG00000220267</td>\n",
       "      <td>141.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ACTBP2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904.0</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159.0</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ENSG00000188801</td>\n",
       "      <td>14003.0</td>\n",
       "      <td>0</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886.0</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328.0</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>690 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol          alias_symbol          ENSG_ID  HGNC_ID  NCBI_ID  \\\n",
       "0         RN7SK                   7SK  ENSG00000283293  10037.0   125050   \n",
       "1         SOAT1  ACAT,ACAT1,SOAT,STAT  ENSG00000057252  11177.0     6646   \n",
       "2         SOAT2                 ACAT2  ENSG00000167780  11178.0     8435   \n",
       "3       NDUFAB1  ACP,ACP1,FASN2A,SDAP  ENSG00000004779   7694.0     4706   \n",
       "4        ACTBP8                ACTBP2  ENSG00000220267    141.0        0   \n",
       "..          ...                   ...              ...      ...      ...   \n",
       "685      ZNF121   D19S204,ZHC32,ZNF20  ENSG00000197961  12904.0     7675   \n",
       "686      RNF141          ZFP26,ZNF230  ENSG00000110315  21159.0    50862   \n",
       "687    ZNF322P1        ZNF322,ZNF322B  ENSG00000188801  14003.0        0   \n",
       "688      ZNF106  SH3BP3,ZFP106,ZNF474  ENSG00000103994  12886.0    64397   \n",
       "689        ZFP1       FLJ34243,ZNF475  ENSG00000184517  23328.0   162239   \n",
       "\n",
       "    collision source  \n",
       "0         7SK   ENSG  \n",
       "1       ACAT1   ENSG  \n",
       "2       ACAT2   ENSG  \n",
       "3        ACP1   ENSG  \n",
       "4      ACTBP2   ENSG  \n",
       "..        ...    ...  \n",
       "685     ZNF20   ENSG  \n",
       "686    ZNF230   ENSG  \n",
       "687    ZNF322   ENSG  \n",
       "688    ZNF474   ENSG  \n",
       "689    ZNF475   ENSG  \n",
       "\n",
       "[690 rows x 7 columns]"
      ]
     },
     "execution_count": 639,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ensg_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ensg_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ensg_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 640,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensg_gene_symbol_set = set(mini_ensg_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 641,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "41164"
      ]
     },
     "execution_count": 641,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ensembl_gene_symbols = len(ensg_gene_symbol_set)\n",
    "total_number_ensembl_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 642,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "618"
      ]
     },
     "execution_count": 642,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_set = set(merged_alias_ap_collision_ensg_df[\"collision\"])\n",
    "len(ensg_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 643,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "674"
      ]
     },
     "execution_count": 643,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensg_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_ensg_df[\"gene_symbol\"])\n",
    "len(ensg_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Why is the alias-gene collision set not the same length as the set of primary symbols with collisions ?\n",
    "2. Why is the length of the alias-gene collison set shorter?\n",
    " - A priamry gene symbol with an alias-gene collision has an alias that matches a different gene's primary gene symbol.\n",
    " - Multiple genes can share a single alias (alias-alias collision)\n",
    " - If that shared alias is an alias-gene collision, then there will be more unique gene symbols in the set of primary symbols with collisions than the set of alias-gene collisions. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HGNC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set up table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 644,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGNC:24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>HGNC:29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>HGNC:24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          HGNC_ID  alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       "0          HGNC:5           NaN        1  ENSG00000121410        A1BG\n",
       "1      HGNC:37133      FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       "2      HGNC:24086           ACF    29974  ENSG00000148584        A1CF\n",
       "3      HGNC:24086           ASP    29974  ENSG00000148584        A1CF\n",
       "4      HGNC:24086         ACF64    29974  ENSG00000148584        A1CF\n",
       "...           ...           ...      ...              ...         ...\n",
       "67578  HGNC:29027      KIAA0399    23140  ENSG00000074755       ZZEF1\n",
       "67579  HGNC:29027          ZZZ4    23140  ENSG00000074755       ZZEF1\n",
       "67580  HGNC:29027      FLJ10821    23140  ENSG00000074755       ZZEF1\n",
       "67581  HGNC:24523  DKFZP564I052    26009  ENSG00000036549        ZZZ3\n",
       "67582  HGNC:24523         ATAC1    26009  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 644,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_path = \"../input/hgnc_biomart_gene20240626.txt\"\n",
    "\n",
    "mini_hgnc_df = pd.read_csv(\n",
    "    file_path, sep=\"\\t\"\n",
    ")\n",
    "\n",
    "# Rename columns\n",
    "mini_hgnc_df = mini_hgnc_df.rename(\n",
    "    columns={\n",
    "        \"HGNC ID\": \"HGNC_ID\",\n",
    "        \"Approved symbol\": \"gene_symbol\",\n",
    "        \"Alias symbols\": \"alias_symbol\",\n",
    "        \"Ensembl gene ID\": \"ENSG_ID\",\n",
    "    }\n",
    ")\n",
    "\n",
    "#structure and labeling in HGNC download files changed between 2024 amd 2025\n",
    "if \"Alias symbol\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbol\": \"alias_symbol\"})\n",
    "elif \"Alias symbols\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"Alias symbols\": \"alias_symbol\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"alias_symbol\"] = pd.NA\n",
    "\n",
    "if \"NCBI gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI gene ID\": \"NCBI_ID\"})\n",
    "elif \"NCBI Gene ID\" in mini_hgnc_df.columns:\n",
    "    mini_hgnc_df = mini_hgnc_df.rename(columns={\"NCBI Gene ID\": \"NCBI_ID\"})\n",
    "else:\n",
    "    mini_hgnc_df[\"NCBI_ID\"] = pd.NA   \n",
    "\n",
    "mini_hgnc_df[\"NCBI_ID\"] = mini_hgnc_df[\"NCBI_ID\"].astype(pd.Int64Dtype())\n",
    "\n",
    "# Extract date from filename and check if it is before June 25, 2025\n",
    "match = re.search(r'(\\d{8})', file_path)\n",
    "file_date = datetime.strptime(match.group(1), \"%Y%m%d\") if match else None\n",
    "cutoff_date = datetime.strptime(\"20250625\", \"%Y%m%d\")\n",
    "\n",
    "# Apply list-splitting logic for newer files\n",
    "if file_date and file_date >= cutoff_date:\n",
    "    mini_hgnc_df['alias_symbol'] = (\n",
    "        mini_hgnc_df['alias_symbol']\n",
    "        .fillna('')\n",
    "        .str.split(',')\n",
    "        .apply(lambda x: [a.strip() for a in x if a.strip()])\n",
    "    )\n",
    "    mini_hgnc_df = mini_hgnc_df.explode('alias_symbol')\n",
    "\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 645,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 645,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_hgnc_df['HGNC_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 646,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 646,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df = mini_hgnc_df.replace(\" \", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"\", np.nan)\n",
    "mini_hgnc_df = mini_hgnc_df.replace(\"-\", np.nan)\n",
    "unique_hcng_df = mini_hgnc_df[mini_hgnc_df['HGNC_ID'].duplicated(keep=False) == False]\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_hgnc_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 647,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22999"
      ]
     },
     "execution_count": 647,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_count = unique_hcng_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 648,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "470"
      ]
     },
     "execution_count": 648,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_hgnc_df[\n",
    "    mini_hgnc_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_hgnc_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 649,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 649,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_hgnc_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 650,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "494"
      ]
     },
     "execution_count": 650,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_hgnc_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 651,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46"
      ]
     },
     "execution_count": 651,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_hgnc_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 652,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6727</th>\n",
       "      <td>HGNC:1624</td>\n",
       "      <td>CTRCT8</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        HGNC_ID alias_symbol  NCBI_ID ENSG_ID gene_symbol\n",
       "6727  HGNC:1624       CTRCT8     <NA>     NaN         CCV"
      ]
     },
     "execution_count": 652,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mask = mini_hgnc_df['gene_symbol'].str.contains('CCV', na=False) \n",
    "\n",
    "# Use the boolean mask to filter the DataFrame\n",
    "filtered_df = mini_hgnc_df[mask]\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 653,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/775146428.py:65: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(  HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 0       5          NaN        1  ENSG00000121410        A1BG\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF,\n",
       "   HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol\n",
       " 1   37133     FLJ23569   503538  ENSG00000268895    A1BG-AS1\n",
       " 2   24086          ACF    29974  ENSG00000148584        A1CF\n",
       " 3   24086          ASP    29974  ENSG00000148584        A1CF\n",
       " 4   24086        ACF64    29974  ENSG00000148584        A1CF\n",
       " 5   24086        ACF65    29974  ENSG00000148584        A1CF,\n",
       "       HGNC_ID alias_symbol  NCBI_ID          ENSG_ID gene_symbol collision  \\\n",
       " 42104   14947        AAVS1    54776  ENSG00000125503    PPP1R12C     AAVS1   \n",
       " 56627   11177        ACAT1     6646  ENSG00000057252       SOAT1     ACAT1   \n",
       " 56628   11178        ACAT2     8435  ENSG00000167780       SOAT2     ACAT2   \n",
       " 17689    4319         ACLS     2737  ENSG00000106571        GLI3      ACLS   \n",
       " 35323    7694         ACP1     4706  ENSG00000004779     NDUFAB1      ACP1   \n",
       " \n",
       "       source  \n",
       " 42104   HGNC  \n",
       " 56627   HGNC  \n",
       " 56628   HGNC  \n",
       " 17689   HGNC  \n",
       " 35323   HGNC  )"
      ]
     },
     "execution_count": 653,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_hgnc_df, \"HGNC\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 654,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67583 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "0            5           NaN       1.0  ENSG00000121410        A1BG\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[67583 rows x 5 columns]"
      ]
     },
     "execution_count": 654,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_hgnc_df= pd.read_csv(\n",
    "    \"../output/mini_hgnc_df.csv\", index_col=[0])\n",
    "mini_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 655,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [HGNC_ID, alias_symbol, NCBI_ID, ENSG_ID, gene_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 655,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_hgnc_df.groupby([\"gene_symbol\"]).HGNC_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_primary`:\n",
    "mini_hgnc_df.loc[mini_hgnc_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 656,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37133</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>503538.0</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24086</td>\n",
       "      <td>ASP</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF64</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>24086</td>\n",
       "      <td>ACF65</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>ENSG00000148584</td>\n",
       "      <td>A1CF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67578</th>\n",
       "      <td>29027</td>\n",
       "      <td>KIAA0399</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67579</th>\n",
       "      <td>29027</td>\n",
       "      <td>ZZZ4</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67580</th>\n",
       "      <td>29027</td>\n",
       "      <td>FLJ10821</td>\n",
       "      <td>23140.0</td>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>ZZEF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67581</th>\n",
       "      <td>24523</td>\n",
       "      <td>DKFZP564I052</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67582</th>\n",
       "      <td>24523</td>\n",
       "      <td>ATAC1</td>\n",
       "      <td>26009.0</td>\n",
       "      <td>ENSG00000036549</td>\n",
       "      <td>ZZZ3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44542 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       HGNC_ID  alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "1        37133      FLJ23569  503538.0  ENSG00000268895    A1BG-AS1\n",
       "2        24086           ACF   29974.0  ENSG00000148584        A1CF\n",
       "3        24086           ASP   29974.0  ENSG00000148584        A1CF\n",
       "4        24086         ACF64   29974.0  ENSG00000148584        A1CF\n",
       "5        24086         ACF65   29974.0  ENSG00000148584        A1CF\n",
       "...        ...           ...       ...              ...         ...\n",
       "67578    29027      KIAA0399   23140.0  ENSG00000074755       ZZEF1\n",
       "67579    29027          ZZZ4   23140.0  ENSG00000074755       ZZEF1\n",
       "67580    29027      FLJ10821   23140.0  ENSG00000074755       ZZEF1\n",
       "67581    24523  DKFZP564I052   26009.0  ENSG00000036549        ZZZ3\n",
       "67582    24523         ATAC1   26009.0  ENSG00000036549        ZZZ3\n",
       "\n",
       "[44542 rows x 5 columns]"
      ]
     },
     "execution_count": 656,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df = pd.read_csv(\n",
    "    \"../output/subset_genes_hgnc_df.csv\", index_col=[0])\n",
    "subset_genes_hgnc_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 657,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2708</th>\n",
       "      <td>31696</td>\n",
       "      <td>ARKL2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709</th>\n",
       "      <td>31696</td>\n",
       "      <td>RNF111L2</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711</th>\n",
       "      <td>31696</td>\n",
       "      <td>LNCAMPC</td>\n",
       "      <td>494470.0</td>\n",
       "      <td>ENSG00000141622</td>\n",
       "      <td>ARK2C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC_ID alias_symbol   NCBI_ID          ENSG_ID gene_symbol\n",
       "2708    31696        ARKL2  494470.0  ENSG00000141622       ARK2C\n",
       "2709    31696     RNF111L2  494470.0  ENSG00000141622       ARK2C\n",
       "2711    31696      LNCAMPC  494470.0  ENSG00000141622       ARK2C"
      ]
     },
     "execution_count": 657,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_hgnc_df.loc[\n",
    "    subset_genes_hgnc_df[\"gene_symbol\"] == \"ARK2C\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 658,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ENSG00000057252</td>\n",
       "      <td>11177</td>\n",
       "      <td>6646</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000167780</td>\n",
       "      <td>11178</td>\n",
       "      <td>8435</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ENSG00000106571</td>\n",
       "      <td>4319</td>\n",
       "      <td>2737</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ENSG00000004779</td>\n",
       "      <td>7694</td>\n",
       "      <td>4706</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>ZHC32,ZNF20</td>\n",
       "      <td>ENSG00000197961</td>\n",
       "      <td>12904</td>\n",
       "      <td>7675</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ENSG00000110315</td>\n",
       "      <td>21159</td>\n",
       "      <td>50862</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>ZNF474,SH3BP3</td>\n",
       "      <td>ENSG00000103994</td>\n",
       "      <td>12886</td>\n",
       "      <td>64397</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ENSG00000184517</td>\n",
       "      <td>23328</td>\n",
       "      <td>162239</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>ZNF208</td>\n",
       "      <td>PMIDP,ZNF95</td>\n",
       "      <td>ENSG00000160321</td>\n",
       "      <td>12999</td>\n",
       "      <td>7757</td>\n",
       "      <td>ZNF95</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>653 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol          ENSG_ID  HGNC_ID  \\\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1  ENSG00000125503    14947   \n",
       "1         SOAT1                         ACAT,ACAT1  ENSG00000057252    11177   \n",
       "2         SOAT2                              ACAT2  ENSG00000167780    11178   \n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV  ENSG00000106571     4319   \n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1  ENSG00000004779     7694   \n",
       "..          ...                                ...              ...      ...   \n",
       "648      ZNF121                        ZHC32,ZNF20  ENSG00000197961    12904   \n",
       "649      RNF141                       ZFP26,ZNF230  ENSG00000110315    21159   \n",
       "650      ZNF106                      ZNF474,SH3BP3  ENSG00000103994    12886   \n",
       "651        ZFP1                    FLJ34243,ZNF475  ENSG00000184517    23328   \n",
       "652      ZNF208                        PMIDP,ZNF95  ENSG00000160321    12999   \n",
       "\n",
       "     NCBI_ID collision source  \n",
       "0      54776     AAVS1   HGNC  \n",
       "1       6646     ACAT1   HGNC  \n",
       "2       8435     ACAT2   HGNC  \n",
       "3       2737      ACLS   HGNC  \n",
       "4       4706      ACP1   HGNC  \n",
       "..       ...       ...    ...  \n",
       "648     7675     ZNF20   HGNC  \n",
       "649    50862    ZNF230   HGNC  \n",
       "650    64397    ZNF474   HGNC  \n",
       "651   162239    ZNF475   HGNC  \n",
       "652     7757     ZNF95   HGNC  \n",
       "\n",
       "[653 rows x 7 columns]"
      ]
     },
     "execution_count": 658,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_hgnc_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_hgnc_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_hgnc_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of the primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 659,
   "metadata": {},
   "outputs": [],
   "source": [
    "hgnc_gene_symbol_set = set(mini_hgnc_df[\"gene_symbol\"])\n",
    "# all_gene_symbols_set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 660,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45646"
      ]
     },
     "execution_count": 660,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_hgnc_gene_symbols = len(hgnc_gene_symbol_set)\n",
    "total_number_hgnc_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 661,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "576"
      ]
     },
     "execution_count": 661,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_set = set(merged_alias_ap_collision_hgnc_df[\"collision\"])\n",
    "len(hgnc_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 662,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "637"
      ]
     },
     "execution_count": 662,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hgnc_alias_primary_collision_primary_symbol_set = set(merged_alias_ap_collision_hgnc_df[\"gene_symbol\"])\n",
    "len(hgnc_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NCBI Info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 663,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_path = \"../input/Homo_sapiens.gene_info20250625\"\n",
    "\n",
    "mini_ncbi_df = pd.read_csv(file_path, sep=\"\\t\")\n",
    "# Drop all columns besides ENSG_ID, gene_symbol, and alias_symbol\n",
    "mini_ncbi_df = mini_ncbi_df[\n",
    "[\"GeneID\", \"Symbol\", \"Synonyms\", \"dbXrefs\"]\n",
    "]\n",
    "mini_ncbi_df = mini_ncbi_df.rename(\n",
    "    columns={\"GeneID\": \"NCBI_ID\", \"Symbol\": \"gene_symbol\", \"Synonyms\": \"alias_symbol\"})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split dbXrefs into individual columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 664,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.assign(\n",
    "    MIM=np.nan,\n",
    "    HGNC_ID=np.nan,\n",
    "    ENSG_ID=np.nan,\n",
    "    AllianceGenome=np.nan,\n",
    "    MIRbase=np.nan,\n",
    "    IMGTgene_db=np.nan,\n",
    "    dash=np.nan,\n",
    "    unknown=np.nan,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 665,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:11: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '138670' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:14: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:17: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'ensg00000121410' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:20: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'hgnc:5' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:26: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'cd1d' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:28: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '-' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"dash\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/2507567125.py:23: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'mi0000060' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "193580\n"
     ]
    }
   ],
   "source": [
    "index_pos = 0\n",
    "\n",
    "print(len(mini_ncbi_df))\n",
    "while index_pos < len(mini_ncbi_df):\n",
    "    xrefs = mini_ncbi_df[\"dbXrefs\"][index_pos].split(\"|\")\n",
    "\n",
    "    for xref in xrefs:\n",
    "        xref = xref.lower()\n",
    "        if xref.startswith(\"mim:\"):\n",
    "            xref = xref.replace(\"mim:\", \"\")\n",
    "            mini_ncbi_df[\"MIM\"][index_pos] = xref\n",
    "        elif xref.startswith(\"hgnc:hgnc:\"):\n",
    "            xref = xref.replace(\"hgnc:hgnc:\", \"\")\n",
    "            mini_ncbi_df[\"HGNC_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"ensembl:\"):\n",
    "            xref = xref.replace(\"ensembl:\", \"\")\n",
    "            mini_ncbi_df[\"ENSG_ID\"][index_pos] = xref\n",
    "        elif xref.startswith(\"alliancegenome:\"):\n",
    "            xref = xref.replace(\"alliancegenome:\", \"\")\n",
    "            mini_ncbi_df[\"AllianceGenome\"][index_pos] = xref\n",
    "        elif xref.startswith(\"mirbase\"):\n",
    "            xref = xref.replace(\"mirbase:\", \"\")\n",
    "            mini_ncbi_df[\"MIRbase\"][index_pos] = xref\n",
    "        elif xref.startswith(\"imgt/gene-db:\"):\n",
    "            xref = xref.replace(\"imgt/gene-db:\", \"\")\n",
    "            mini_ncbi_df[\"IMGTgene_db\"][index_pos] = xref\n",
    "        elif xref.startswith(\"-\"):\n",
    "            mini_ncbi_df[\"dash\"][index_pos] = xref\n",
    "        else:\n",
    "            mini_ncbi_df[\"unknown\"][index_pos] = xref\n",
    "\n",
    "    index_pos += 1\n",
    "    pass\n",
    "\n",
    "print(index_pos)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 666,
   "metadata": {},
   "outputs": [],
   "source": [
    "mini_ncbi_df[\"ENSG_ID\"] = mini_ncbi_df[\"ENSG_ID\"].str.replace(\"ensg\", \"ENSG\", 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 667,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B|ABG|GAB|HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD|CPAMD5|FWP007|S863-7</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>NAT1</td>\n",
       "      <td>AAC1|MNAT|NAT-1|NATI</td>\n",
       "      <td>7645</td>\n",
       "      <td>ENSG00000171428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>AAC2|NAT-2|PNAT</td>\n",
       "      <td>7646</td>\n",
       "      <td>ENSG00000156006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>NATP</td>\n",
       "      <td>AACP|NATP1</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193580 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol               alias_symbol HGNC_ID  \\\n",
       "0             1        A1BG       A1B|ABG|GAB|HYST2477       5   \n",
       "1             2         A2M  A2MD|CPAMD5|FWP007|S863-7       7   \n",
       "2             9        NAT1       AAC1|MNAT|NAT-1|NATI    7645   \n",
       "3            10        NAT2            AAC2|NAT-2|PNAT    7646   \n",
       "4            11        NATP                 AACP|NATP1      15   \n",
       "...         ...         ...                        ...     ...   \n",
       "193575  8923215        trnD                          -     NaN   \n",
       "193576  8923216        trnP                          -     NaN   \n",
       "193577  8923217        trnA                          -     NaN   \n",
       "193578  8923218        COX1                          -     NaN   \n",
       "193579  8923219    16S rRNA                          -     NaN   \n",
       "\n",
       "                ENSG_ID  \n",
       "0       ENSG00000121410  \n",
       "1       ENSG00000175899  \n",
       "2       ENSG00000171428  \n",
       "3       ENSG00000156006  \n",
       "4                   NaN  \n",
       "...                 ...  \n",
       "193575              NaN  \n",
       "193576              NaN  \n",
       "193577              NaN  \n",
       "193578              NaN  \n",
       "193579              NaN  \n",
       "\n",
       "[193580 rows x 5 columns]"
      ]
     },
     "execution_count": 667,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.drop(\n",
    "    [\n",
    "        \"AllianceGenome\",\n",
    "        \"MIRbase\",\n",
    "        \"IMGTgene_db\",\n",
    "        \"dash\",\n",
    "        \"unknown\",\n",
    "        \"dbXrefs\",\n",
    "        \"MIM\",\n",
    "    ],\n",
    "    axis=1,\n",
    ")\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 668,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "193580"
      ]
     },
     "execution_count": 668,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(mini_ncbi_df['NCBI_ID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 669,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 669,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df = mini_ncbi_df.replace(\" \", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"\", np.nan)\n",
    "mini_ncbi_df = mini_ncbi_df.replace(\"-\", np.nan)\n",
    "\n",
    "unique_ncbi_df = mini_ncbi_df[mini_ncbi_df['NCBI_ID'].duplicated(keep=False) == False]\n",
    "\n",
    "# Count rows with missing gene symbols before dropping\n",
    "no_symbol_count = mini_ncbi_df[\"gene_symbol\"].isna().sum()\n",
    "no_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 670,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "165885"
      ]
     },
     "execution_count": 670,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_alias_symbol_count = unique_ncbi_df[\"alias_symbol\"].isna().sum()\n",
    "no_alias_symbol_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 671,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "248"
      ]
     },
     "execution_count": 671,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orf_rows = mini_ncbi_df[\n",
    "    mini_ncbi_df['gene_symbol']\n",
    "    .str.contains(r'^C.*ORF', case=False, na=False) &\n",
    "    ~mini_ncbi_df['gene_symbol']\n",
    "    .str.contains('-', na=False)\n",
    "]\n",
    "len(orf_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 672,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "147700"
      ]
     },
     "execution_count": 672,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_loc = mini_ncbi_df['gene_symbol'].str.startswith('LOC', na=False).sum()\n",
    "count_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 673,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "396"
      ]
     },
     "execution_count": 673,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_FAM = mini_ncbi_df['gene_symbol'].str.startswith('FAM', na=False).sum()\n",
    "count_FAM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 674,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 674,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_KIAA = mini_ncbi_df['gene_symbol'].str.startswith('KIAA', na=False).sum()\n",
    "count_KIAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 675,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>193532</th>\n",
       "      <td>6775087</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193573</th>\n",
       "      <td>8923213</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193548</th>\n",
       "      <td>8923188</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3522</th>\n",
       "      <td>4508</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>ATPase6|MTATP6</td>\n",
       "      <td>7414</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193519</th>\n",
       "      <td>6775074</td>\n",
       "      <td>ATP6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3523</th>\n",
       "      <td>4509</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>ATPase8|MTATP8</td>\n",
       "      <td>7415</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193522</th>\n",
       "      <td>6775077</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193571</th>\n",
       "      <td>8923211</td>\n",
       "      <td>ATP8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193528</th>\n",
       "      <td>6775083</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3525</th>\n",
       "      <td>4512</td>\n",
       "      <td>COX1</td>\n",
       "      <td>COI|MTCO1</td>\n",
       "      <td>7419</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193562</th>\n",
       "      <td>8923202</td>\n",
       "      <td>COX2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193524</th>\n",
       "      <td>6775079</td>\n",
       "      <td>COX2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3526</th>\n",
       "      <td>4513</td>\n",
       "      <td>COX2</td>\n",
       "      <td>COII|MTCO2</td>\n",
       "      <td>7421</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193547</th>\n",
       "      <td>8923187</td>\n",
       "      <td>COX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3527</th>\n",
       "      <td>4514</td>\n",
       "      <td>COX3</td>\n",
       "      <td>COIII|MTCO3</td>\n",
       "      <td>7422</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193518</th>\n",
       "      <td>6775073</td>\n",
       "      <td>COX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3529</th>\n",
       "      <td>4519</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>MTCYB</td>\n",
       "      <td>7427</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193510</th>\n",
       "      <td>6775065</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193565</th>\n",
       "      <td>8923205</td>\n",
       "      <td>CYTB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52286</th>\n",
       "      <td>107648861</td>\n",
       "      <td>DEL11P13</td>\n",
       "      <td>C11DELp13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36284</th>\n",
       "      <td>100528024</td>\n",
       "      <td>DEL11P13</td>\n",
       "      <td>C11DELp13|WAGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31435</th>\n",
       "      <td>100240737</td>\n",
       "      <td>DEL1P36</td>\n",
       "      <td>C1DELp36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63076</th>\n",
       "      <td>123670537</td>\n",
       "      <td>DEL1P36</td>\n",
       "      <td>C1DELp36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2459</th>\n",
       "      <td>3045</td>\n",
       "      <td>HBD</td>\n",
       "      <td>HBK</td>\n",
       "      <td>4829</td>\n",
       "      <td>ENSG00000223609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30652</th>\n",
       "      <td>100187828</td>\n",
       "      <td>HBD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35568</th>\n",
       "      <td>100505381</td>\n",
       "      <td>MMD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19942</th>\n",
       "      <td>221938</td>\n",
       "      <td>MMD2</td>\n",
       "      <td>PAQR10</td>\n",
       "      <td>30133</td>\n",
       "      <td>ENSG00000136297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3538</th>\n",
       "      <td>4535</td>\n",
       "      <td>ND1</td>\n",
       "      <td>MTND1</td>\n",
       "      <td>7455</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193569</th>\n",
       "      <td>8923209</td>\n",
       "      <td>ND1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193543</th>\n",
       "      <td>6775098</td>\n",
       "      <td>ND1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193539</th>\n",
       "      <td>6775094</td>\n",
       "      <td>ND2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3539</th>\n",
       "      <td>4536</td>\n",
       "      <td>ND2</td>\n",
       "      <td>MTND2</td>\n",
       "      <td>7456</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193572</th>\n",
       "      <td>8923212</td>\n",
       "      <td>ND2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193566</th>\n",
       "      <td>8923206</td>\n",
       "      <td>ND3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3540</th>\n",
       "      <td>4537</td>\n",
       "      <td>ND3</td>\n",
       "      <td>MTND3</td>\n",
       "      <td>7458</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193521</th>\n",
       "      <td>6775076</td>\n",
       "      <td>ND3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193517</th>\n",
       "      <td>6775072</td>\n",
       "      <td>ND4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3541</th>\n",
       "      <td>4538</td>\n",
       "      <td>ND4</td>\n",
       "      <td>MTND4</td>\n",
       "      <td>7459</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193557</th>\n",
       "      <td>8923197</td>\n",
       "      <td>ND4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193516</th>\n",
       "      <td>6775071</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193561</th>\n",
       "      <td>8923201</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3542</th>\n",
       "      <td>4539</td>\n",
       "      <td>ND4L</td>\n",
       "      <td>MTND4L</td>\n",
       "      <td>7460</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193511</th>\n",
       "      <td>6775066</td>\n",
       "      <td>ND5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3543</th>\n",
       "      <td>4540</td>\n",
       "      <td>ND5</td>\n",
       "      <td>MTND5</td>\n",
       "      <td>7461</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193570</th>\n",
       "      <td>8923210</td>\n",
       "      <td>ND5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193508</th>\n",
       "      <td>6775063</td>\n",
       "      <td>ND6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3544</th>\n",
       "      <td>4541</td>\n",
       "      <td>ND6</td>\n",
       "      <td>MTND6</td>\n",
       "      <td>7462</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193558</th>\n",
       "      <td>8923198</td>\n",
       "      <td>ND6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3550</th>\n",
       "      <td>4549</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>MTRNR1</td>\n",
       "      <td>7470</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4762</th>\n",
       "      <td>6052</td>\n",
       "      <td>RNR1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10082</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3551</th>\n",
       "      <td>4550</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>MTRNR2</td>\n",
       "      <td>7471</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>6053</td>\n",
       "      <td>RNR2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10083</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5560</th>\n",
       "      <td>7006</td>\n",
       "      <td>TEC</td>\n",
       "      <td>PSCTK4</td>\n",
       "      <td>11719</td>\n",
       "      <td>ENSG00000135605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28617</th>\n",
       "      <td>100124696</td>\n",
       "      <td>TEC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65888</th>\n",
       "      <td>124901561</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65889</th>\n",
       "      <td>124901562</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70782</th>\n",
       "      <td>124906586</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65891</th>\n",
       "      <td>124901564</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65892</th>\n",
       "      <td>124901565</td>\n",
       "      <td>TRNAA-AGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol    alias_symbol HGNC_ID          ENSG_ID\n",
       "193532    6775087    12S rRNA             NaN     NaN              NaN\n",
       "193573    8923213    12S rRNA             NaN     NaN              NaN\n",
       "193548    8923188        ATP6             NaN     NaN              NaN\n",
       "3522         4508        ATP6  ATPase6|MTATP6    7414              NaN\n",
       "193519    6775074        ATP6             NaN     NaN              NaN\n",
       "3523         4509        ATP8  ATPase8|MTATP8    7415              NaN\n",
       "193522    6775077        ATP8             NaN     NaN              NaN\n",
       "193571    8923211        ATP8             NaN     NaN              NaN\n",
       "193528    6775083        COX1             NaN     NaN              NaN\n",
       "193578    8923218        COX1             NaN     NaN              NaN\n",
       "3525         4512        COX1       COI|MTCO1    7419              NaN\n",
       "193562    8923202        COX2             NaN     NaN              NaN\n",
       "193524    6775079        COX2             NaN     NaN              NaN\n",
       "3526         4513        COX2      COII|MTCO2    7421              NaN\n",
       "193547    8923187        COX3             NaN     NaN              NaN\n",
       "3527         4514        COX3     COIII|MTCO3    7422              NaN\n",
       "193518    6775073        COX3             NaN     NaN              NaN\n",
       "3529         4519        CYTB           MTCYB    7427              NaN\n",
       "193510    6775065        CYTB             NaN     NaN              NaN\n",
       "193565    8923205        CYTB             NaN     NaN              NaN\n",
       "52286   107648861    DEL11P13       C11DELp13     NaN              NaN\n",
       "36284   100528024    DEL11P13  C11DELp13|WAGR     NaN              NaN\n",
       "31435   100240737     DEL1P36        C1DELp36     NaN              NaN\n",
       "63076   123670537     DEL1P36        C1DELp36     NaN              NaN\n",
       "2459         3045         HBD             HBK    4829  ENSG00000223609\n",
       "30652   100187828         HBD             NaN     NaN              NaN\n",
       "35568   100505381        MMD2             NaN     NaN              NaN\n",
       "19942      221938        MMD2          PAQR10   30133  ENSG00000136297\n",
       "3538         4535         ND1           MTND1    7455              NaN\n",
       "193569    8923209         ND1             NaN     NaN              NaN\n",
       "193543    6775098         ND1             NaN     NaN              NaN\n",
       "193539    6775094         ND2             NaN     NaN              NaN\n",
       "3539         4536         ND2           MTND2    7456              NaN\n",
       "193572    8923212         ND2             NaN     NaN              NaN\n",
       "193566    8923206         ND3             NaN     NaN              NaN\n",
       "3540         4537         ND3           MTND3    7458              NaN\n",
       "193521    6775076         ND3             NaN     NaN              NaN\n",
       "193517    6775072         ND4             NaN     NaN              NaN\n",
       "3541         4538         ND4           MTND4    7459              NaN\n",
       "193557    8923197         ND4             NaN     NaN              NaN\n",
       "193516    6775071        ND4L             NaN     NaN              NaN\n",
       "193561    8923201        ND4L             NaN     NaN              NaN\n",
       "3542         4539        ND4L          MTND4L    7460              NaN\n",
       "193511    6775066         ND5             NaN     NaN              NaN\n",
       "3543         4540         ND5           MTND5    7461              NaN\n",
       "193570    8923210         ND5             NaN     NaN              NaN\n",
       "193508    6775063         ND6             NaN     NaN              NaN\n",
       "3544         4541         ND6           MTND6    7462              NaN\n",
       "193558    8923198         ND6             NaN     NaN              NaN\n",
       "3550         4549        RNR1          MTRNR1    7470              NaN\n",
       "4762         6052        RNR1             NaN   10082              NaN\n",
       "3551         4550        RNR2          MTRNR2    7471              NaN\n",
       "4763         6053        RNR2             NaN   10083              NaN\n",
       "5560         7006         TEC          PSCTK4   11719  ENSG00000135605\n",
       "28617   100124696         TEC             NaN     NaN              NaN\n",
       "65888   124901561   TRNAA-AGC             NaN     NaN              NaN\n",
       "65889   124901562   TRNAA-AGC             NaN     NaN              NaN\n",
       "70782   124906586   TRNAA-AGC             NaN     NaN              NaN\n",
       "65891   124901564   TRNAA-AGC             NaN     NaN              NaN\n",
       "65892   124901565   TRNAA-AGC             NaN     NaN              NaN"
      ]
     },
     "execution_count": 675,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# groupby name and return a boolean of whether each has more than 1 unique Country\n",
    "multi_primary = mini_ncbi_df.groupby([\"gene_symbol\"]).NCBI_ID.nunique().gt(1)\n",
    "\n",
    "# use loc to only see those values that have `True` in `multi_country`:\n",
    "mini_ncbi_df.loc[mini_ncbi_df.gene_symbol.isin(multi_primary[multi_primary].index)].sort_values(by='gene_symbol').head(60)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 676,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>240508 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B       5  ENSG00000121410\n",
       "0             1        A1BG          ABG       5  ENSG00000121410\n",
       "0             1        A1BG          GAB       5  ENSG00000121410\n",
       "0             1        A1BG     HYST2477       5  ENSG00000121410\n",
       "1             2         A2M         A2MD       7  ENSG00000175899\n",
       "...         ...         ...          ...     ...              ...\n",
       "193575  8923215        trnD          NaN     NaN              NaN\n",
       "193576  8923216        trnP          NaN     NaN              NaN\n",
       "193577  8923217        trnA          NaN     NaN              NaN\n",
       "193578  8923218        COX1          NaN     NaN              NaN\n",
       "193579  8923219    16S rRNA          NaN     NaN              NaN\n",
       "\n",
       "[240508 rows x 5 columns]"
      ]
     },
     "execution_count": 676,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df['alias_symbol'] = mini_ncbi_df['alias_symbol'].str.split('|')\n",
    "mini_ncbi_df = mini_ncbi_df.explode('alias_symbol')\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 677,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All collisions are present in gene alias lists.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/vt/znzp_c6s02q6kjcmqfk0cb500000gq/T/ipykernel_36019/775146428.py:65: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.\n",
      "  ap_collision_xxxx_df = ap_collision_xxxx_df.applymap(lambda x: ', '.join(map(str, x)) if isinstance(x, set) else x)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(   NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "    NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID\n",
       " 0        1        A1BG          A1B       5  ENSG00000121410\n",
       " 0        1        A1BG          ABG       5  ENSG00000121410\n",
       " 0        1        A1BG          GAB       5  ENSG00000121410\n",
       " 0        1        A1BG     HYST2477       5  ENSG00000121410\n",
       " 1        2         A2M         A2MD       7  ENSG00000175899,\n",
       "        NCBI_ID gene_symbol alias_symbol HGNC_ID          ENSG_ID collision  \\\n",
       " 2809      3494       IGHA2          A2M    5479  ENSG00000211890       A2M   \n",
       " 24213   404744   NPSR1-AS1         AAA1   22128  ENSG00000197085      AAA1   \n",
       " 2307      2875         GPT         AAT1    4552  ENSG00000167701      AAT1   \n",
       " 16372    89876      CFAP91         AAT1   24010  ENSG00000183833      AAT1   \n",
       " 12222    54776    PPP1R12C        AAVS1   14947  ENSG00000125503     AAVS1   \n",
       " \n",
       "       source  \n",
       " 2809    NCBI  \n",
       " 24213   NCBI  \n",
       " 2307    NCBI  \n",
       " 16372   NCBI  \n",
       " 12222   NCBI  )"
      ]
     },
     "execution_count": 677,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_ap_collision_df(mini_ncbi_df, \"NCBI\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 678,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193575</th>\n",
       "      <td>8923215</td>\n",
       "      <td>trnD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193576</th>\n",
       "      <td>8923216</td>\n",
       "      <td>trnP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193577</th>\n",
       "      <td>8923217</td>\n",
       "      <td>trnA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193578</th>\n",
       "      <td>8923218</td>\n",
       "      <td>COX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193579</th>\n",
       "      <td>8923219</td>\n",
       "      <td>16S rRNA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92719 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0             1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0             1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0             1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0             1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1             2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...         ...         ...          ...      ...              ...\n",
       "193575  8923215        trnD          NaN      NaN              NaN\n",
       "193576  8923216        trnP          NaN      NaN              NaN\n",
       "193577  8923217        trnA          NaN      NaN              NaN\n",
       "193578  8923218        COX1          NaN      NaN              NaN\n",
       "193579  8923219    16S rRNA          NaN      NaN              NaN\n",
       "\n",
       "[92719 rows x 5 columns]"
      ]
     },
     "execution_count": 678,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df= pd.read_csv(\n",
    "    \"../output/mini_ncbi_df.csv\", index_col=[0])\n",
    "mini_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 679,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [NCBI_ID, gene_symbol, alias_symbol, HGNC_ID, ENSG_ID]\n",
       "Index: []"
      ]
     },
     "execution_count": 679,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini_ncbi_df[mini_ncbi_df['gene_symbol'].str.startswith('LOC')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 680,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ENSG_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>ABG</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>GAB</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>HYST2477</td>\n",
       "      <td>5.0</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>A2MD</td>\n",
       "      <td>7.0</td>\n",
       "      <td>ENSG00000175899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193317</th>\n",
       "      <td>139281660</td>\n",
       "      <td>IFT70A-AS1</td>\n",
       "      <td>AGPS</td>\n",
       "      <td>58181.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193317</th>\n",
       "      <td>139281660</td>\n",
       "      <td>IFT70A-AS1</td>\n",
       "      <td>PDE11A</td>\n",
       "      <td>58181.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193324</th>\n",
       "      <td>139281667</td>\n",
       "      <td>BCAT1-DT</td>\n",
       "      <td>LNC-BCAT1</td>\n",
       "      <td>54396.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193398</th>\n",
       "      <td>139440214</td>\n",
       "      <td>LNCOB1</td>\n",
       "      <td>LNC-OB1</td>\n",
       "      <td>56209.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193447</th>\n",
       "      <td>141445969</td>\n",
       "      <td>LINC03145</td>\n",
       "      <td>CRART16</td>\n",
       "      <td>56020.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74401 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          NCBI_ID gene_symbol alias_symbol  HGNC_ID          ENSG_ID\n",
       "0               1        A1BG          A1B      5.0  ENSG00000121410\n",
       "0               1        A1BG          ABG      5.0  ENSG00000121410\n",
       "0               1        A1BG          GAB      5.0  ENSG00000121410\n",
       "0               1        A1BG     HYST2477      5.0  ENSG00000121410\n",
       "1               2         A2M         A2MD      7.0  ENSG00000175899\n",
       "...           ...         ...          ...      ...              ...\n",
       "193317  139281660  IFT70A-AS1         AGPS  58181.0              NaN\n",
       "193317  139281660  IFT70A-AS1       PDE11A  58181.0              NaN\n",
       "193324  139281667    BCAT1-DT    LNC-BCAT1  54396.0              NaN\n",
       "193398  139440214      LNCOB1      LNC-OB1  56209.0              NaN\n",
       "193447  141445969   LINC03145      CRART16  56020.0              NaN\n",
       "\n",
       "[74401 rows x 5 columns]"
      ]
     },
     "execution_count": 680,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_genes_ncbi_df = pd.read_csv(\n",
    "    \"../output/subset_genes_ncbi_df.csv\", index_col=[0])\n",
    "subset_genes_ncbi_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 681,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>ENSG_ID</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>NCBI_ID</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IGHA2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>ENSG00000211890</td>\n",
       "      <td>5479.0</td>\n",
       "      <td>3494</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NPSR1-AS1</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>ENSG00000197085</td>\n",
       "      <td>22128.0</td>\n",
       "      <td>404744</td>\n",
       "      <td>AAA1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GPT</td>\n",
       "      <td>AAT1,ALT,ALT1,GPT1,SGPT</td>\n",
       "      <td>ENSG00000167701</td>\n",
       "      <td>4552.0</td>\n",
       "      <td>2875</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFAP91</td>\n",
       "      <td>AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...</td>\n",
       "      <td>ENSG00000183833</td>\n",
       "      <td>24010.0</td>\n",
       "      <td>89876</td>\n",
       "      <td>AAT1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>AAVS1,LENG3,MBS85,p84,p85</td>\n",
       "      <td>ENSG00000125503</td>\n",
       "      <td>14947.0</td>\n",
       "      <td>54776</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2159</th>\n",
       "      <td>ZNF785</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>ENSG00000197162</td>\n",
       "      <td>26496.0</td>\n",
       "      <td>146540</td>\n",
       "      <td>ZNF688</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2160</th>\n",
       "      <td>ZP4</td>\n",
       "      <td>ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4</td>\n",
       "      <td>ENSG00000116996</td>\n",
       "      <td>15770.0</td>\n",
       "      <td>57829</td>\n",
       "      <td>ZP1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2161</th>\n",
       "      <td>LMBR1</td>\n",
       "      <td>ACHP,C7orf2,DIF14,LSS,PPD2,THYP,TPT,TPTPS,ZRS</td>\n",
       "      <td>ENSG00000105983</td>\n",
       "      <td>13243.0</td>\n",
       "      <td>64327</td>\n",
       "      <td>ZRS</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2162</th>\n",
       "      <td>ZNF446</td>\n",
       "      <td>ZKSCAN20,ZSCAN30,ZSCAN52</td>\n",
       "      <td>ENSG00000083838</td>\n",
       "      <td>21036.0</td>\n",
       "      <td>55663</td>\n",
       "      <td>ZSCAN30</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2163</th>\n",
       "      <td>ZKSCAN2</td>\n",
       "      <td>ZNF694,ZSCAN31,ZSCAN34</td>\n",
       "      <td>ENSG00000155592</td>\n",
       "      <td>25677.0</td>\n",
       "      <td>342357</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2164 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene_symbol                                       alias_symbol  \\\n",
       "0          IGHA2                                                A2M   \n",
       "1      NPSR1-AS1                                               AAA1   \n",
       "2            GPT                            AAT1,ALT,ALT1,GPT1,SGPT   \n",
       "3         CFAP91  AAT1,AAT1alpha,C3orf15,CaM-IP2,MAATS1,SPATA26,...   \n",
       "4       PPP1R12C                          AAVS1,LENG3,MBS85,p84,p85   \n",
       "...          ...                                                ...   \n",
       "2159      ZNF785                                             ZNF688   \n",
       "2160         ZP4                         ZBP,ZP1,ZP1B,ZPB,ZPB2,Zp-4   \n",
       "2161       LMBR1      ACHP,C7orf2,DIF14,LSS,PPD2,THYP,TPT,TPTPS,ZRS   \n",
       "2162      ZNF446                           ZKSCAN20,ZSCAN30,ZSCAN52   \n",
       "2163     ZKSCAN2                             ZNF694,ZSCAN31,ZSCAN34   \n",
       "\n",
       "              ENSG_ID  HGNC_ID  NCBI_ID collision source  \n",
       "0     ENSG00000211890   5479.0     3494       A2M   NCBI  \n",
       "1     ENSG00000197085  22128.0   404744      AAA1   NCBI  \n",
       "2     ENSG00000167701   4552.0     2875      AAT1   NCBI  \n",
       "3     ENSG00000183833  24010.0    89876      AAT1   NCBI  \n",
       "4     ENSG00000125503  14947.0    54776     AAVS1   NCBI  \n",
       "...               ...      ...      ...       ...    ...  \n",
       "2159  ENSG00000197162  26496.0   146540    ZNF688   NCBI  \n",
       "2160  ENSG00000116996  15770.0    57829       ZP1   NCBI  \n",
       "2161  ENSG00000105983  13243.0    64327       ZRS   NCBI  \n",
       "2162  ENSG00000083838  21036.0    55663   ZSCAN30   NCBI  \n",
       "2163  ENSG00000155592  25677.0   342357   ZSCAN31   NCBI  \n",
       "\n",
       "[2164 rows x 7 columns]"
      ]
     },
     "execution_count": 681,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_ap_collision_ncbi_df = pd.read_csv(\n",
    "    \"../output/merged_alias_ap_collision_ncbi_df.csv\", index_col=[0])\n",
    "merged_alias_ap_collision_ncbi_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make a set of primary gene symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 682,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_gene_symbol_set = set(mini_ncbi_df[\"gene_symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 683,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45727"
      ]
     },
     "execution_count": 683,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_number_ncbi_gene_symbols = len(ncbi_gene_symbol_set)\n",
    "total_number_ncbi_gene_symbols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How many collisions are there? How many records are involved in one?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 684,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1612"
      ]
     },
     "execution_count": 684,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"collision\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: make the records set based on ID not symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 685,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1728"
      ]
     },
     "execution_count": 685,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncbi_alias_primary_collision_primary_symbol_set = set(\n",
    "    merged_alias_ap_collision_ncbi_df[\"gene_symbol\"]\n",
    ")\n",
    "len(ncbi_alias_primary_collision_primary_symbol_set)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge 3 sets together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 686,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PPP1R12C</td>\n",
       "      <td>DKFZP434D0412,p84,MBS85,p85,AAVS1</td>\n",
       "      <td>AAVS1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SOAT1</td>\n",
       "      <td>ACAT,ACAT1</td>\n",
       "      <td>ACAT1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SOAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GLI3</td>\n",
       "      <td>PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV</td>\n",
       "      <td>ACLS</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NDUFAB1</td>\n",
       "      <td>SDAP,FASN2A,ACP,ACP1</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>ZNF121</td>\n",
       "      <td>D19S204,ZHC32,ZNF20</td>\n",
       "      <td>ZNF20</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>RNF141</td>\n",
       "      <td>ZFP26,ZNF230</td>\n",
       "      <td>ZNF230</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>ZNF322P1</td>\n",
       "      <td>ZNF322,ZNF322B</td>\n",
       "      <td>ZNF322</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>ZNF106</td>\n",
       "      <td>SH3BP3,ZFP106,ZNF474</td>\n",
       "      <td>ZNF474</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>ZFP1</td>\n",
       "      <td>FLJ34243,ZNF475</td>\n",
       "      <td>ZNF475</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3507 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                       alias_symbol collision source\n",
       "0      PPP1R12C  DKFZP434D0412,p84,MBS85,p85,AAVS1     AAVS1   HGNC\n",
       "1         SOAT1                         ACAT,ACAT1     ACAT1   HGNC\n",
       "2         SOAT2                              ACAT2     ACAT2   HGNC\n",
       "3          GLI3   PAP-A,PAPA,PAPA1,PAPB,ACLS,PPDIV      ACLS   HGNC\n",
       "4       NDUFAB1               SDAP,FASN2A,ACP,ACP1      ACP1   HGNC\n",
       "..          ...                                ...       ...    ...\n",
       "685      ZNF121                D19S204,ZHC32,ZNF20     ZNF20   ENSG\n",
       "686      RNF141                       ZFP26,ZNF230    ZNF230   ENSG\n",
       "687    ZNF322P1                     ZNF322,ZNF322B    ZNF322   ENSG\n",
       "688      ZNF106               SH3BP3,ZFP106,ZNF474    ZNF474   ENSG\n",
       "689        ZFP1                    FLJ34243,ZNF475    ZNF475   ENSG\n",
       "\n",
       "[3507 rows x 4 columns]"
      ]
     },
     "execution_count": 686,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df = pd.concat(\n",
    "    [\n",
    "        merged_alias_ap_collision_hgnc_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ncbi_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "        merged_alias_ap_collision_ensg_df[\n",
    "            [\"gene_symbol\", \"alias_symbol\", \"collision\", \"source\"]\n",
    "        ],\n",
    "    ]\n",
    ")\n",
    "merged_alias_primary_collisions_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 687,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>MGC45871,RefilinB,Cfm1</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>RFLNB</td>\n",
       "      <td>CFM1,FAM101B</td>\n",
       "      <td>CFM1</td>\n",
       "      <td>NCBI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol            alias_symbol collision source\n",
       "84        RFLNB  MGC45871,RefilinB,Cfm1      CFM1   HGNC\n",
       "212       RFLNB            CFM1,FAM101B      CFM1   NCBI"
      ]
     },
     "execution_count": 687,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_alias_primary_collisions_df.loc[\n",
    "    merged_alias_primary_collisions_df[\"collision\"] == \"CFM1\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 688,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DANCR</td>\n",
       "      <td>ANCR,AGU2,lncRNA-ANCR</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>UBE3A</td>\n",
       "      <td>AS,ANCR,E6-AP,FLJ26981</td>\n",
       "      <td>ANCR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SLC25A23</td>\n",
       "      <td>FLJ30339,MGC2615,APC2</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ANAPC2</td>\n",
       "      <td>APC2,KIAA1406</td>\n",
       "      <td>APC2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AREG</td>\n",
       "      <td>AR,CRDGF</td>\n",
       "      <td>AR</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>661</th>\n",
       "      <td>RNU4-8P</td>\n",
       "      <td>RNU4P2,U4,U4/14</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>RNU4-6P</td>\n",
       "      <td>RNU4P6,U4,U4/8</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>663</th>\n",
       "      <td>RNU4-5P</td>\n",
       "      <td>RNU4P5,U4,U4/6</td>\n",
       "      <td>U4</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>664</th>\n",
       "      <td>RNU6-1</td>\n",
       "      <td>RNU6A,U6,U6-1</td>\n",
       "      <td>U6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>665</th>\n",
       "      <td>RNU6-50P</td>\n",
       "      <td>RNU6-50,RNU6P1,U6</td>\n",
       "      <td>U6</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1056 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol            alias_symbol collision source\n",
       "13        DANCR   ANCR,AGU2,lncRNA-ANCR      ANCR   HGNC\n",
       "14        UBE3A  AS,ANCR,E6-AP,FLJ26981      ANCR   HGNC\n",
       "15     SLC25A23   FLJ30339,MGC2615,APC2      APC2   HGNC\n",
       "16       ANAPC2           APC2,KIAA1406      APC2   HGNC\n",
       "19         AREG                AR,CRDGF        AR   HGNC\n",
       "..          ...                     ...       ...    ...\n",
       "661     RNU4-8P         RNU4P2,U4,U4/14        U4   ENSG\n",
       "662     RNU4-6P          RNU4P6,U4,U4/8        U4   ENSG\n",
       "663     RNU4-5P          RNU4P5,U4,U4/6        U4   ENSG\n",
       "664      RNU6-1           RNU6A,U6,U6-1        U6   ENSG\n",
       "665    RNU6-50P       RNU6-50,RNU6P1,U6        U6   ENSG\n",
       "\n",
       "[1056 rows x 4 columns]"
      ]
     },
     "execution_count": 688,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['collision','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 689,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>alias_symbol</th>\n",
       "      <th>collision</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>ATF1,ATF2,HFB1-GDNF</td>\n",
       "      <td>ATF1</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GDNF</td>\n",
       "      <td>ATF1,ATF2,HFB1-GDNF</td>\n",
       "      <td>ATF2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>TMPRSS4</td>\n",
       "      <td>TMPRSS3,MT-SP2,CAP2</td>\n",
       "      <td>CAP2</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>ACKR2</td>\n",
       "      <td>CCR10,D6,CCR9</td>\n",
       "      <td>CCR10</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>ACKR2</td>\n",
       "      <td>CCR10,D6,CCR9</td>\n",
       "      <td>CCR9</td>\n",
       "      <td>HGNC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>580</th>\n",
       "      <td>SLIT3</td>\n",
       "      <td>MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2</td>\n",
       "      <td>SLIT1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>581</th>\n",
       "      <td>SLIT3</td>\n",
       "      <td>MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2</td>\n",
       "      <td>SLIT2</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>SPNS1</td>\n",
       "      <td>HSPIN1,LAT,NRS,PP2030,SLC63A1,SPIN1,SPINL</td>\n",
       "      <td>SPIN1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>612</th>\n",
       "      <td>UXT</td>\n",
       "      <td>ART-27,SKP2,STAP1</td>\n",
       "      <td>STAP1</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>TMPRSS4</td>\n",
       "      <td>CAP2,MT-SP2,TMPRSS3</td>\n",
       "      <td>TMPRSS3</td>\n",
       "      <td>ENSG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>648 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene_symbol                               alias_symbol collision source\n",
       "28         GDNF                        ATF1,ATF2,HFB1-GDNF      ATF1   HGNC\n",
       "29         GDNF                        ATF1,ATF2,HFB1-GDNF      ATF2   HGNC\n",
       "62      TMPRSS4                        TMPRSS3,MT-SP2,CAP2      CAP2   HGNC\n",
       "72        ACKR2                              CCR10,D6,CCR9     CCR10   HGNC\n",
       "74        ACKR2                              CCR10,D6,CCR9      CCR9   HGNC\n",
       "..          ...                                        ...       ...    ...\n",
       "580       SLIT3             MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2     SLIT1   ENSG\n",
       "581       SLIT3             MEGF5,SLIL2,SLIT-3,SLIT1,SLIT2     SLIT2   ENSG\n",
       "599       SPNS1  HSPIN1,LAT,NRS,PP2030,SLC63A1,SPIN1,SPINL     SPIN1   ENSG\n",
       "612         UXT                          ART-27,SKP2,STAP1     STAP1   ENSG\n",
       "631     TMPRSS4                        CAP2,MT-SP2,TMPRSS3   TMPRSS3   ENSG\n",
       "\n",
       "[648 rows x 4 columns]"
      ]
     },
     "execution_count": 689,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicate_rows = merged_alias_primary_collisions_df[merged_alias_primary_collisions_df.duplicated(subset=['gene_symbol','source'], keep=False)]\n",
    "duplicate_rows"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 690,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_alias_primary_collisions_df.to_csv(\n",
    "    \"../output/merged_alias_primary_collisions_df.csv\", index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 691,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ABCD1',\n",
       " 'ACD',\n",
       " 'ACKR2',\n",
       " 'ACOD1',\n",
       " 'ACTBP8',\n",
       " 'ADRA1D',\n",
       " 'AGXT',\n",
       " 'AIFM2',\n",
       " 'AKR1B1',\n",
       " 'AKR1B10',\n",
       " 'ALPK3',\n",
       " 'AMH',\n",
       " 'ANAPC2',\n",
       " 'ANKRD37',\n",
       " 'ANTXR1',\n",
       " 'AOC1',\n",
       " 'APEX1',\n",
       " 'AREG',\n",
       " 'ARHGAP21',\n",
       " 'ARHGEF7',\n",
       " 'ARID4A',\n",
       " 'ART4',\n",
       " 'ARTN',\n",
       " 'ASIC2',\n",
       " 'AURKAIP1',\n",
       " 'AZIN2',\n",
       " 'BANF1P1',\n",
       " 'BCAT2',\n",
       " 'BRIP1',\n",
       " 'BTF3P11',\n",
       " 'BTN3A3',\n",
       " 'C1D',\n",
       " 'C1QTNF1',\n",
       " 'C6ORF89',\n",
       " 'CACNA1A',\n",
       " 'CADPS',\n",
       " 'CADPS2',\n",
       " 'CAPN5',\n",
       " 'CARD16',\n",
       " 'CCL13',\n",
       " 'CCL14',\n",
       " 'CCL15',\n",
       " 'CCM2',\n",
       " 'CD200R1',\n",
       " 'CDH19',\n",
       " 'CDH20',\n",
       " 'CDPF1',\n",
       " 'CELSR1',\n",
       " 'CES1',\n",
       " 'CFAP73',\n",
       " 'CFH',\n",
       " 'CHAF1B',\n",
       " 'CHAMP1',\n",
       " 'CHD6',\n",
       " 'CHEK2',\n",
       " 'CHORDC1',\n",
       " 'CLASP1',\n",
       " 'CLCF1',\n",
       " 'CNGB1',\n",
       " 'CNKSR2',\n",
       " 'CNOT6',\n",
       " 'CNRIP1',\n",
       " 'CNTN1',\n",
       " 'COASY',\n",
       " 'COPS2',\n",
       " 'COPS3',\n",
       " 'CORIN',\n",
       " 'COX7A2L',\n",
       " 'CPA4',\n",
       " 'CPAMD8',\n",
       " 'CPNE1',\n",
       " 'CPNE2',\n",
       " 'CPPED1',\n",
       " 'CREB3L4',\n",
       " 'CSNK2A2',\n",
       " 'CSTB',\n",
       " 'CTDSP2',\n",
       " 'CXCL10',\n",
       " 'CXXC1',\n",
       " 'CYCSP5',\n",
       " 'CYP11B1',\n",
       " 'CYP11B2',\n",
       " 'CYP21A2',\n",
       " 'CYP2A6',\n",
       " 'DCAF5',\n",
       " 'DCBLD2',\n",
       " 'DDOST',\n",
       " 'DDR2',\n",
       " 'DDX11',\n",
       " 'DDX18',\n",
       " 'DEAF1',\n",
       " 'DEFA6',\n",
       " 'DEPDC1B',\n",
       " 'DFFB',\n",
       " 'DHX8',\n",
       " 'DIDO1',\n",
       " 'DLC1',\n",
       " 'DLEC1',\n",
       " 'DLGAP3',\n",
       " 'DMAP1',\n",
       " 'DMTF1',\n",
       " 'DNPEP',\n",
       " 'DPAGT1',\n",
       " 'DPYSL2',\n",
       " 'DSC3',\n",
       " 'DUSP12',\n",
       " 'DUSP28',\n",
       " 'DYNLL1',\n",
       " 'E2F1',\n",
       " 'EIF2S1',\n",
       " 'EIF4G2',\n",
       " 'ELOF1',\n",
       " 'EPAS1',\n",
       " 'EPS8L1',\n",
       " 'EPX',\n",
       " 'ERC2',\n",
       " 'ERCC4',\n",
       " 'ETFA',\n",
       " 'F3',\n",
       " 'FA2H',\n",
       " 'FABP5P1',\n",
       " 'FADS1',\n",
       " 'FANCA',\n",
       " 'FASN',\n",
       " 'FBF1',\n",
       " 'FBXO5',\n",
       " 'FCRLA',\n",
       " 'FCRLB',\n",
       " 'FEM1C',\n",
       " 'FGF13',\n",
       " 'FGFR1',\n",
       " 'FLII',\n",
       " 'FOXM1',\n",
       " 'FYN',\n",
       " 'FZD9',\n",
       " 'FZR1',\n",
       " 'GAL3ST3',\n",
       " 'GALK2',\n",
       " 'GALNT15',\n",
       " 'GALNT18',\n",
       " 'GALNTL6',\n",
       " 'GALP',\n",
       " 'GARS1',\n",
       " 'GATD3',\n",
       " 'GC',\n",
       " 'GDF15',\n",
       " 'GDNF',\n",
       " 'GET1',\n",
       " 'GJD3',\n",
       " 'GKN1',\n",
       " 'GLB1',\n",
       " 'GLS2',\n",
       " 'GMNN',\n",
       " 'GNAI2',\n",
       " 'GNPDA1',\n",
       " 'GOLT1A',\n",
       " 'GOLT1B',\n",
       " 'GPI',\n",
       " 'GPKOW',\n",
       " 'GPNMB',\n",
       " 'GPRIN1',\n",
       " 'GPX7',\n",
       " 'GRID1',\n",
       " 'GRID2',\n",
       " 'GTPBP4',\n",
       " 'GUCY2C',\n",
       " 'HCRT',\n",
       " 'HCRTR1',\n",
       " 'HECA',\n",
       " 'HFE',\n",
       " 'HID1',\n",
       " 'HNF1A-AS1',\n",
       " 'HOMER2',\n",
       " 'HOOK1',\n",
       " 'HOOK2',\n",
       " 'HOOK3',\n",
       " 'HRNR',\n",
       " 'HSD17B4',\n",
       " 'HSD17B7P2',\n",
       " 'IAPP',\n",
       " 'IFNA1',\n",
       " 'IGF2BP3',\n",
       " 'IL22',\n",
       " 'IL6',\n",
       " 'ILF3',\n",
       " 'INSL3',\n",
       " 'INSL6',\n",
       " 'IPO4',\n",
       " 'ITM2C',\n",
       " 'KANSL1',\n",
       " 'KCNA4',\n",
       " 'KCNA5',\n",
       " 'KCNH4',\n",
       " 'KCNH8',\n",
       " 'KCNN4',\n",
       " 'KCTD11',\n",
       " 'KHSRP',\n",
       " 'KIF20B',\n",
       " 'KIF2A',\n",
       " 'KLF5',\n",
       " 'KLHL7',\n",
       " 'KLK1',\n",
       " 'KLK6',\n",
       " 'KLRG1',\n",
       " 'KRTAP5-11',\n",
       " 'KRTAP5-2',\n",
       " 'KRTAP5-3',\n",
       " 'KRTAP5-5',\n",
       " 'KRTAP5-7',\n",
       " 'KRTAP5-8',\n",
       " 'KRTAP5-9',\n",
       " 'LAMA4',\n",
       " 'LDB2',\n",
       " 'LGALS12',\n",
       " 'LGALS13',\n",
       " 'LGALSL',\n",
       " 'LNPK',\n",
       " 'LNX1',\n",
       " 'LONP1',\n",
       " 'LRIF1',\n",
       " 'LRIG1',\n",
       " 'LRP12',\n",
       " 'LRP5',\n",
       " 'LTB4R2',\n",
       " 'LZTS1',\n",
       " 'MAFIP',\n",
       " 'MAGI1',\n",
       " 'MAP4K2',\n",
       " 'MAP4K5',\n",
       " 'MAPRE2',\n",
       " 'MB',\n",
       " 'MBD1',\n",
       " 'MBD4',\n",
       " 'MBL2',\n",
       " 'MED12',\n",
       " 'MED22',\n",
       " 'MED9',\n",
       " 'MELTF',\n",
       " 'METAP1',\n",
       " 'METAP2',\n",
       " 'MGAM',\n",
       " 'MIPEP',\n",
       " 'MLF2',\n",
       " 'MMAB',\n",
       " 'MMACHC',\n",
       " 'MOCOS',\n",
       " 'MOGAT1',\n",
       " 'MOGAT2',\n",
       " 'MOGAT3',\n",
       " 'MRLN',\n",
       " 'MRPL13',\n",
       " 'MRPL41',\n",
       " 'MRPS12',\n",
       " 'MRPS28',\n",
       " 'MRPS35',\n",
       " 'MRTFA',\n",
       " 'MS4A12',\n",
       " 'MS4A4A',\n",
       " 'MS4A7',\n",
       " 'MTCH1',\n",
       " 'MYCBP2',\n",
       " 'MYDGF',\n",
       " 'MYH7B',\n",
       " 'MYL1',\n",
       " 'MYO1C',\n",
       " 'MYO1E',\n",
       " 'MYT1',\n",
       " 'MYZAP',\n",
       " 'NANOGP2',\n",
       " 'NANOGP4',\n",
       " 'NANOGP7',\n",
       " 'NANOS1',\n",
       " 'NANOS2',\n",
       " 'NANOS3',\n",
       " 'NAPSA',\n",
       " 'NAPSB',\n",
       " 'NAT8',\n",
       " 'NCOA2',\n",
       " 'NCOA3',\n",
       " 'NDRG1',\n",
       " 'NDUFAB1',\n",
       " 'NECTIN4',\n",
       " 'NEDD4',\n",
       " 'NELL2',\n",
       " 'NFE2L1',\n",
       " 'NKX2-1',\n",
       " 'NLRP2B',\n",
       " 'NLRP3',\n",
       " 'NLRP4',\n",
       " 'NLRP6',\n",
       " 'NOL3',\n",
       " 'NOL6',\n",
       " 'NPPC',\n",
       " 'NPRL2',\n",
       " 'NPRL3',\n",
       " 'NR5A1',\n",
       " 'NRG1',\n",
       " 'NT5C3A',\n",
       " 'NTN1',\n",
       " 'NUDT19',\n",
       " 'OSER1',\n",
       " 'PAEP',\n",
       " 'PAK6',\n",
       " 'PAPPA-AS1',\n",
       " 'PARD3',\n",
       " 'PCDHA4',\n",
       " 'PCDHA6',\n",
       " 'PCDHB11',\n",
       " 'PCDHB16',\n",
       " 'PCDHGA5',\n",
       " 'PCDHGB4',\n",
       " 'PDHX',\n",
       " 'PDPK1',\n",
       " 'PEDS1',\n",
       " 'PGAM4',\n",
       " 'PGAP1',\n",
       " 'PGF',\n",
       " 'PGPEP1',\n",
       " 'PIGU',\n",
       " 'PITRM1',\n",
       " 'PKMYT1',\n",
       " 'PKN1',\n",
       " 'PLPP6',\n",
       " 'PLPPR3',\n",
       " 'PM20D1',\n",
       " 'PMEL',\n",
       " 'PMEPA1',\n",
       " 'PMP22',\n",
       " 'PMPCB',\n",
       " 'PMS2CL',\n",
       " 'PNKD',\n",
       " 'PODXL',\n",
       " 'POLK',\n",
       " 'POLR1A',\n",
       " 'POLR1B',\n",
       " 'POLR1G',\n",
       " 'PPIAP10',\n",
       " 'PPP1R9B',\n",
       " 'PPY',\n",
       " 'PRDX5',\n",
       " 'PRDX6',\n",
       " 'PRG2',\n",
       " 'PRICKLE1',\n",
       " 'PRKD2',\n",
       " 'PRKRA',\n",
       " 'PRPF38B',\n",
       " 'PRSS8',\n",
       " 'PSG2',\n",
       " 'PSMA7',\n",
       " 'PSMD1',\n",
       " 'PSMD2',\n",
       " 'PTEN',\n",
       " 'PTENP1',\n",
       " 'PTPMT1',\n",
       " 'PTPN18',\n",
       " 'PTPRU',\n",
       " 'PTRH2',\n",
       " 'PXDN',\n",
       " 'PXMP2',\n",
       " 'PYDC1',\n",
       " 'RAB43',\n",
       " 'RAI1',\n",
       " 'RBL1',\n",
       " 'RDH14',\n",
       " 'REXO2',\n",
       " 'RGL4',\n",
       " 'RGPD1',\n",
       " 'RGS18',\n",
       " 'RHEBP1',\n",
       " 'RHOD',\n",
       " 'RIMBP2',\n",
       " 'RIN2',\n",
       " 'RMRP',\n",
       " 'RNASE1',\n",
       " 'RNASE10',\n",
       " 'RNASE12',\n",
       " 'RNASE3',\n",
       " 'RNASE6',\n",
       " 'RNASE7',\n",
       " 'RNASEH1P1',\n",
       " 'RNF138',\n",
       " 'RNF141',\n",
       " 'RNF2',\n",
       " 'RNF4',\n",
       " 'RNF7',\n",
       " 'RORA',\n",
       " 'RPL15',\n",
       " 'RPL17',\n",
       " 'RPL23',\n",
       " 'RRP1B',\n",
       " 'RSC1A1',\n",
       " 'RSL24D1',\n",
       " 'S100A9',\n",
       " 'S100G',\n",
       " 'SBDS',\n",
       " 'SCGB1D1',\n",
       " 'SEC14L2',\n",
       " 'SEC14L3',\n",
       " 'SELENOP',\n",
       " 'SEM1',\n",
       " 'SEMA3F-AS1',\n",
       " 'SEPSECS',\n",
       " 'SERPINA2',\n",
       " 'SERPINB8',\n",
       " 'SF3B2',\n",
       " 'SF3B3',\n",
       " 'SFI1',\n",
       " 'SGPP2',\n",
       " 'SH2D1A',\n",
       " 'SH3D19',\n",
       " 'SLC19A1',\n",
       " 'SLC25A1',\n",
       " 'SLC25A15',\n",
       " 'SLC25A16',\n",
       " 'SLC25A21',\n",
       " 'SLC25A23',\n",
       " 'SLC35A1',\n",
       " 'SLC38A1',\n",
       " 'SLC38A2',\n",
       " 'SLC6A20',\n",
       " 'SLC7A8',\n",
       " 'SLIT1',\n",
       " 'SLIT3',\n",
       " 'SLTM',\n",
       " 'SLURP1',\n",
       " 'SMIM29',\n",
       " 'SMOC2',\n",
       " 'SMOX',\n",
       " 'SND1',\n",
       " 'SNORA58',\n",
       " 'SNORD115-45',\n",
       " 'SNRPB2',\n",
       " 'SNX33',\n",
       " 'SOAT1',\n",
       " 'SOAT2',\n",
       " 'SOS1',\n",
       " 'SOX9',\n",
       " 'SP6',\n",
       " 'SPATA17',\n",
       " 'SPATS1',\n",
       " 'SPNS1',\n",
       " 'SPPL2A',\n",
       " 'SPPL2B',\n",
       " 'SRFBP1',\n",
       " 'SRMS',\n",
       " 'STK24',\n",
       " 'STK4',\n",
       " 'SYCP2',\n",
       " 'TACR1',\n",
       " 'TADA2A',\n",
       " 'TDO2',\n",
       " 'TENM1',\n",
       " 'TENT4A',\n",
       " 'TGFBRAP1',\n",
       " 'THPO',\n",
       " 'TIMP1',\n",
       " 'TINAGL1',\n",
       " 'TIRAP',\n",
       " 'TMEM184B-AS1',\n",
       " 'TMPRSS4',\n",
       " 'TNIP1',\n",
       " 'TNNI3',\n",
       " 'TNPO1',\n",
       " 'TOX4',\n",
       " 'TP53RK',\n",
       " 'TPGS1',\n",
       " 'TRAPPC8',\n",
       " 'TRAV1-1',\n",
       " 'TSPO',\n",
       " 'TTC5',\n",
       " 'UBASH3A',\n",
       " 'UBE2Z',\n",
       " 'UCMA',\n",
       " 'UGT2B4',\n",
       " 'USP2-AS1',\n",
       " 'USP21',\n",
       " 'USP25',\n",
       " 'USP32',\n",
       " 'UTS2',\n",
       " 'UXT',\n",
       " 'VSIG2',\n",
       " 'XPO1',\n",
       " 'ZBTB32',\n",
       " 'ZFP1',\n",
       " 'ZNF106',\n",
       " 'ZNF121',\n",
       " 'ZNF160',\n",
       " 'ZNF83'}"
      ]
     },
     "execution_count": 691,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_ap_collisions = (\n",
    "    ncbi_alias_primary_collision_primary_symbol_set\n",
    "    & hgnc_alias_primary_collision_primary_symbol_set\n",
    "    & ensg_alias_primary_collision_primary_symbol_set\n",
    ")\n",
    "common_ap_collisions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 692,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "488"
      ]
     },
     "execution_count": 692,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_ap_collisions)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('gene-normalization-kln2mzEG')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "63db6d8944633f0f1ac66f1e84c479e7ccc3c7c835ee283fbd6bddfaad96182b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
